diff --git a/src/include/storage/mysql_table_entry.hpp b/src/include/storage/mysql_table_entry.hpp
index 99e4599..7dc21f1 100644
--- a/src/include/storage/mysql_table_entry.hpp
+++ b/src/include/storage/mysql_table_entry.hpp
@@ -41,6 +41,8 @@ public:
 	MySQLTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, MySQLTableInfo &info);
 
 public:
+	const ColumnList &GetColumns() const override;
+
 	unique_ptr<BaseStatistics> GetStatistics(ClientContext &context, column_t column_id) override;
 
 	TableFunction GetScanFunction(ClientContext &context, unique_ptr<FunctionData> &bind_data) override;
@@ -49,6 +51,9 @@ public:
 
 	void BindUpdateConstraints(Binder &binder, LogicalGet &get, LogicalProjection &proj, LogicalUpdate &update,
 	                           ClientContext &context) override;
+
+protected:
+	ColumnList columns;
 };
 
 } // namespace duckdb
diff --git a/src/storage/mysql_table_entry.cpp b/src/storage/mysql_table_entry.cpp
index a7f9424..7754616 100644
--- a/src/storage/mysql_table_entry.cpp
+++ b/src/storage/mysql_table_entry.cpp
@@ -73,15 +73,19 @@ static bool TableIsInternal(const SchemaCatalogEntry &schema, const string &name
 }
 
 MySQLTableEntry::MySQLTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, CreateTableInfo &info)
-    : TableCatalogEntry(catalog, schema, info) {
+    : TableCatalogEntry(catalog, schema, info), columns(std::move(info.columns)) {
 	this->internal = TableIsInternal(schema, name.GetIdentifierName());
 }
 
 MySQLTableEntry::MySQLTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, MySQLTableInfo &info)
-    : TableCatalogEntry(catalog, schema, *info.create_info) {
+    : TableCatalogEntry(catalog, schema, *info.create_info), columns(std::move(info.create_info->columns)) {
 	this->internal = TableIsInternal(schema, name.GetIdentifierName());
 }
 
+const ColumnList &MySQLTableEntry::GetColumns() const {
+	return columns;
+}
+
 unique_ptr<BaseStatistics> MySQLTableEntry::GetStatistics(ClientContext &context, column_t column_id) {
 	return nullptr;
 }
