diff --git a/src/mysql_sql_writer.cpp b/src/mysql_sql_writer.cpp
index 0ed02cf..8d213d4 100644
--- a/src/mysql_sql_writer.cpp
+++ b/src/mysql_sql_writer.cpp
@@ -23,7 +23,7 @@
 #include "duckdb/parser/tableref/expressionlistref.hpp"
 #include "duckdb/parser/tableref/joinref.hpp"
 #include "duckdb/parser/tableref/subqueryref.hpp"
-#include "duckdb/common/extra_type_info.hpp"
+#include "duckdb/common/logical_type_info.hpp"
 
 #include "mysql_utils.hpp"
 
