diff --git a/src/storage/mysql_catalog.cpp b/src/storage/mysql_catalog.cpp
index 75e12fe..380f597 100644
--- a/src/storage/mysql_catalog.cpp
+++ b/src/storage/mysql_catalog.cpp
@@ -460,7 +460,7 @@ optional_ptr<CatalogEntry> MySQLCatalog::CreateSchema(CatalogTransaction transac
 	if (info.on_conflict == OnCreateConflict::REPLACE_ON_CONFLICT) {
 		DropInfo try_drop;
 		try_drop.type = CatalogType::SCHEMA_ENTRY;
-		try_drop.name = info.schema;
+		try_drop.NameMutable() = info.schema;
 		try_drop.if_not_found = OnEntryNotFound::RETURN_NULL;
 		try_drop.cascade = false;
 		schemas.DropEntry(transaction.GetContext(), try_drop);
diff --git a/src/storage/mysql_catalog_set.cpp b/src/storage/mysql_catalog_set.cpp
index 1aaf5d8..b69a09e 100644
--- a/src/storage/mysql_catalog_set.cpp
+++ b/src/storage/mysql_catalog_set.cpp
@@ -33,7 +33,7 @@ void MySQLCatalogSet::DropEntry(ClientContext &context, DropInfo &info) {
 	if (info.if_not_found == OnEntryNotFound::RETURN_NULL) {
 		drop_query += " IF EXISTS ";
 	}
-	drop_query += MySQLUtils::WriteIdentifier(info.name.GetIdentifierName());
+	drop_query += MySQLUtils::WriteIdentifier(info.Name().GetIdentifierName());
 	if (info.type != CatalogType::SCHEMA_ENTRY) {
 		if (info.cascade) {
 			drop_query += " CASCADE";
@@ -43,7 +43,7 @@ void MySQLCatalogSet::DropEntry(ClientContext &context, DropInfo &info) {
 	transaction.Query(drop_query);
 
 	// erase the entry from the catalog set
-	EraseEntryInternal(info.name.GetIdentifierName());
+	EraseEntryInternal(info.Name().GetIdentifierName());
 }
 
 void MySQLCatalogSet::EraseEntryInternal(const string &name) {
diff --git a/src/storage/mysql_index_set.cpp b/src/storage/mysql_index_set.cpp
index a1352a6..df0c898 100644
--- a/src/storage/mysql_index_set.cpp
+++ b/src/storage/mysql_index_set.cpp
@@ -11,22 +11,22 @@ MySQLIndexSet::MySQLIndexSet(MySQLSchemaEntry &schema) : MySQLInSchemaSet(schema
 }
 
 void MySQLIndexSet::DropEntry(ClientContext &context, DropInfo &info) {
-	auto entry = GetEntry(context, info.name.GetIdentifierName());
+	auto entry = GetEntry(context, info.Name().GetIdentifierName());
 	if (!entry) {
 		if (info.if_not_found == OnEntryNotFound::RETURN_NULL) {
 			return;
 		}
-		throw CatalogException("Failed to DROP INDEX \"%s\": entry not found", info.name);
+		throw CatalogException("Failed to DROP INDEX \"%s\": entry not found", info.Name());
 	}
 	auto &mysql_index = entry->Cast<MySQLIndexEntry>();
 	string drop_query = "DROP INDEX ";
-	drop_query += MySQLUtils::WriteIdentifier(info.name.GetIdentifierName());
+	drop_query += MySQLUtils::WriteIdentifier(info.Name().GetIdentifierName());
 	drop_query += " ON ";
 	drop_query += MySQLUtils::WriteIdentifier(mysql_index.table_name);
 	auto &transaction = MySQLTransaction::Get(context, catalog);
 	transaction.Query(drop_query);
 
-	EraseEntryInternal(info.name.GetIdentifierName());
+	EraseEntryInternal(info.Name().GetIdentifierName());
 }
 
 void MySQLIndexSet::LoadEntries(ClientContext &context) {
diff --git a/src/storage/mysql_schema_entry.cpp b/src/storage/mysql_schema_entry.cpp
index 77e2e8c..69cbff8 100644
--- a/src/storage/mysql_schema_entry.cpp
+++ b/src/storage/mysql_schema_entry.cpp
@@ -28,7 +28,7 @@ MySQLTransaction &GetMySQLTransaction(CatalogTransaction transaction) {
 void MySQLSchemaEntry::TryDropEntry(ClientContext &context, CatalogType catalog_type, const string &name) {
 	DropInfo info;
 	info.type = catalog_type;
-	info.name = Identifier(name);
+	info.NameMutable() = Identifier(name);
 	info.cascade = false;
 	info.if_not_found = OnEntryNotFound::RETURN_NULL;
 	DropEntry(context, info);
diff --git a/src/storage/mysql_table_set.cpp b/src/storage/mysql_table_set.cpp
index e6be4a9..3791e15 100644
--- a/src/storage/mysql_table_set.cpp
+++ b/src/storage/mysql_table_set.cpp
@@ -235,7 +235,7 @@ optional_ptr<CatalogEntry> MySQLTableSet::CreateTable(ClientContext &context, Bo
 void MySQLTableSet::AlterTable(ClientContext &context, RenameTableInfo &info) {
 	auto &transaction = MySQLTransaction::Get(context, catalog);
 	string sql = "ALTER TABLE ";
-	sql += MySQLUtils::WriteIdentifier(info.name.GetIdentifierName());
+	sql += MySQLUtils::WriteIdentifier(info.Name().GetIdentifierName());
 	sql += " RENAME TO ";
 	sql += MySQLUtils::WriteIdentifier(info.new_table_name.GetIdentifierName());
 	transaction.Query(sql);
@@ -244,7 +244,7 @@ void MySQLTableSet::AlterTable(ClientContext &context, RenameTableInfo &info) {
 void MySQLTableSet::AlterTable(ClientContext &context, RenameColumnInfo &info) {
 	auto &transaction = MySQLTransaction::Get(context, catalog);
 	string sql = "ALTER TABLE ";
-	sql += MySQLUtils::WriteIdentifier(info.name.GetIdentifierName());
+	sql += MySQLUtils::WriteIdentifier(info.Name().GetIdentifierName());
 	sql += " RENAME COLUMN  ";
 	sql += MySQLUtils::WriteIdentifier(info.old_name.GetIdentifierName());
 	sql += " TO ";
@@ -256,7 +256,7 @@ void MySQLTableSet::AlterTable(ClientContext &context, RenameColumnInfo &info) {
 void MySQLTableSet::AlterTable(ClientContext &context, AddColumnInfo &info) {
 	auto &transaction = MySQLTransaction::Get(context, catalog);
 	string sql = "ALTER TABLE ";
-	sql += MySQLUtils::WriteIdentifier(info.name.GetIdentifierName());
+	sql += MySQLUtils::WriteIdentifier(info.Name().GetIdentifierName());
 	sql += " ADD COLUMN  ";
 	if (info.if_column_not_exists) {
 		sql += "IF NOT EXISTS ";
@@ -270,7 +270,7 @@ void MySQLTableSet::AlterTable(ClientContext &context, AddColumnInfo &info) {
 void MySQLTableSet::AlterTable(ClientContext &context, RemoveColumnInfo &info) {
 	auto &transaction = MySQLTransaction::Get(context, catalog);
 	string sql = "ALTER TABLE ";
-	sql += MySQLUtils::WriteIdentifier(info.name.GetIdentifierName());
+	sql += MySQLUtils::WriteIdentifier(info.Name().GetIdentifierName());
 	sql += " DROP COLUMN  ";
 	if (info.if_column_exists) {
 		throw NotImplementedException("DROP COLUMN IF EXISTS not supported in MySQL");
