--- a/src/storage/mysql_predicate_analyzer.cpp
+++ b/src/storage/mysql_predicate_analyzer.cpp
@@ -1,4 +1,8 @@
 #include "storage/mysql_predicate_analyzer.hpp"
+#include "duckdb/planner/expression/bound_conjunction_expression.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_comparison_expression.hpp"
+#include "duckdb/planner/filter/expression_filter.hpp"
 #include "mysql_utils.hpp"
 #include "duckdb/planner/expression/bound_operator_expression.hpp"
 #include "duckdb/planner/filter/conjunction_filter.hpp"
