--- a/src/mysql_scanner.cpp
+++ b/src/mysql_scanner.cpp
@@ -40,7 +40,7 @@ struct MySQLGlobalState : public GlobalTableFunctionState {
 };
 
 static unique_ptr<FunctionData> MySQLBind(ClientContext &context, TableFunctionBindInput &input,
-                                          vector<LogicalType> &return_types, vector<string> &names) {
+                                          vector<LogicalType> &return_types, vector<Identifier> &names) {
 	throw InternalException("MySQLBind");
 }
 
@@ -179,7 +179,7 @@ MySQLScanFunction::MySQLScanFunction()
 // MySQL Query
 //===--------------------------------------------------------------------===//
 static unique_ptr<FunctionData> MySQLQueryBind(ClientContext &context, TableFunctionBindInput &input,
-                                               vector<LogicalType> &return_types, vector<string> &names) {
+                                               vector<LogicalType> &return_types, vector<Identifier> &names) {
 	if (input.inputs[0].IsNull() || input.inputs[1].IsNull()) {
 		throw BinderException("Parameters to mysql_query cannot be NULL");
 	}
@@ -215,7 +215,7 @@ static unique_ptr<FunctionData> MySQLQueryBind(ClientContext &context, TableFunc
 		auto stmt = conn.Prepare(sql);
 		if (stmt->Fields().size() > 0) {
 			for (auto &field : stmt->Fields()) {
-				names.push_back(field.name);
+				names.emplace_back(field.name);
 				return_types.push_back(field.duckdb_type);
 			}
 		} else {
--- a/src/storage/mysql_clear_cache.cpp
+++ b/src/storage/mysql_clear_cache.cpp
@@ -15,7 +15,7 @@ struct ClearCacheFunctionData : public TableFunctionData {
 };
 
 static unique_ptr<FunctionData> ClearCacheBind(ClientContext &context, TableFunctionBindInput &input,
-                                               vector<LogicalType> &return_types, vector<string> &names) {
+                                               vector<LogicalType> &return_types, vector<Identifier> &names) {
 
 	auto result = make_uniq<ClearCacheFunctionData>();
 	return_types.push_back(LogicalType::BOOLEAN);
