diff --git a/src/storage/mysql_clear_cache.cpp b/src/storage/mysql_clear_cache.cpp
index bf288c0..2be4e46 100644
--- a/src/storage/mysql_clear_cache.cpp
+++ b/src/storage/mysql_clear_cache.cpp
@@ -24,7 +24,7 @@ static unique_ptr<FunctionData> ClearCacheBind(ClientContext &context, TableFunc
 static void ClearMySQLCaches(ClientContext &context) {
 	auto databases = DatabaseManager::Get(context).GetDatabases(context);
 	for (auto &db_ref : databases) {
-		auto &db = db_ref.get();
+		auto &db = *db_ref;
 		auto &catalog = db.GetCatalog();
 		if (catalog.GetCatalogType() != "mysql") {
 			continue;
