diff --git a/src/include/storage/mysql_optimizer.hpp b/src/include/storage/mysql_optimizer.hpp
index 7a44c5b..2b27c72 100644
--- a/src/include/storage/mysql_optimizer.hpp
+++ b/src/include/storage/mysql_optimizer.hpp
@@ -9,6 +9,7 @@
 #pragma once

 #include "duckdb/main/config.hpp"
+#include "duckdb/optimizer/optimizer_extension.hpp"

 namespace duckdb {
 class MySQLOptimizer {
diff --git a/src/mysql_extension.cpp b/src/mysql_extension.cpp
index ec3c7e3..3e69d23 100644
--- a/src/mysql_extension.cpp
+++ b/src/mysql_extension.cpp
@@ -108,7 +108,7 @@ static void LoadInternal(ExtensionLoader &loader) {
 	auto &db = loader.GetDatabaseInstance();
 
 	auto &config = DBConfig::GetConfig(db);
-	config.storage_extensions["mysql_scanner"] = make_uniq<MySQLStorageExtension>();
+	StorageExtension::Register(config, "mysql_scanner", make_shared_ptr<MySQLStorageExtension>());
 
 	config.AddExtensionOption("mysql_experimental_filter_pushdown",
 	                          "Whether or not to use filter pushdown (currently experimental)", LogicalType::BOOLEAN,
@@ -129,7 +129,7 @@ static void LoadInternal(ExtensionLoader &loader) {
 
 	OptimizerExtension mysql_optimizer;
 	mysql_optimizer.optimize_function = MySQLOptimizer::Optimize;
-	config.optimizer_extensions.push_back(std::move(mysql_optimizer));
+	OptimizerExtension::Register(config, std::move(mysql_optimizer));
 }
 
 void MysqlScannerExtension::Load(ExtensionLoader &loader) {
diff --git a/src/storage/mysql_schema_entry.cpp b/src/storage/mysql_schema_entry.cpp
index 889ff75..d82c444 100644
--- a/src/storage/mysql_schema_entry.cpp
+++ b/src/storage/mysql_schema_entry.cpp
@@ -10,6 +10,7 @@
 #include "duckdb/parser/parsed_data/alter_info.hpp"
 #include "duckdb/parser/parsed_data/alter_table_info.hpp"
 #include "duckdb/parser/parsed_expression_iterator.hpp"
+#include "duckdb/parser/expression/columnref_expression.hpp"
 
 namespace duckdb {
 
