diff --git a/src/include/storage/mysql_execute_query.hpp b/src/include/storage/mysql_execute_query.hpp
index ea653c1..f9e02cd 100644
--- a/src/include/storage/mysql_execute_query.hpp
+++ b/src/include/storage/mysql_execute_query.hpp
@@ -46,7 +46,7 @@ public:
 	}
 
 	string GetName() const override;
-	string ParamsToString() const override;
+	InsertionOrderPreservingMap<string> ParamsToString() const override;
 };
 
 } // namespace duckdb
diff --git a/src/include/storage/mysql_insert.hpp b/src/include/storage/mysql_insert.hpp
index ee067ac..3c3213d 100644
--- a/src/include/storage/mysql_insert.hpp
+++ b/src/include/storage/mysql_insert.hpp
@@ -53,7 +53,7 @@ public:
 	}
 
 	string GetName() const override;
-	string ParamsToString() const override;
+	InsertionOrderPreservingMap<string> ParamsToString() const override;
 };
 
 } // namespace duckdb
diff --git a/src/storage/mysql_execute_query.cpp b/src/storage/mysql_execute_query.cpp
index 906cab4..c0e5b6c 100644
--- a/src/storage/mysql_execute_query.cpp
+++ b/src/storage/mysql_execute_query.cpp
@@ -71,8 +71,10 @@ string MySQLExecuteQuery::GetName() const {
 	return op_name;
 }
 
-string MySQLExecuteQuery::ParamsToString() const {
-	return table.name;
+InsertionOrderPreservingMap<string> MySQLExecuteQuery::ParamsToString() const {
+	InsertionOrderPreservingMap<string> result;
+	result["Table Name"] = table.name;
+	return result;
 }
 
 //===--------------------------------------------------------------------===//
diff --git a/src/storage/mysql_insert.cpp b/src/storage/mysql_insert.cpp
index c12756f..1db6df7 100644
--- a/src/storage/mysql_insert.cpp
+++ b/src/storage/mysql_insert.cpp
@@ -256,8 +256,10 @@ string MySQLInsert::GetName() const {
 	return table ? "MYSQL_INSERT" : "MYSQL_CREATE_TABLE_AS";
 }
 
-string MySQLInsert::ParamsToString() const {
-	return table ? table->name : info->Base().table;
+InsertionOrderPreservingMap<string> MySQLInsert::ParamsToString() const {
+	InsertionOrderPreservingMap<string> result;
+	result["Table Name"] = table ? table->name : info->Base().table;
+	return result;
 }
 
 //===--------------------------------------------------------------------===//
