diff --git a/src/include/storage/mysql_optimizer.hpp b/src/include/storage/mysql_optimizer.hpp
index 3970851..7a44c5b 100644
--- a/src/include/storage/mysql_optimizer.hpp
+++ b/src/include/storage/mysql_optimizer.hpp
@@ -13,7 +13,7 @@
 namespace duckdb {
 class MySQLOptimizer {
 public:
-	static void Optimize(ClientContext &context, OptimizerExtensionInfo *info, unique_ptr<LogicalOperator> &plan);
+	static void Optimize(OptimizerExtensionInput &input, unique_ptr<LogicalOperator> &plan);
 };
 
 } // namespace duckdb
diff --git a/src/storage/mysql_optimizer.cpp b/src/storage/mysql_optimizer.cpp
index 3f3b1c4..2985a0e 100644
--- a/src/storage/mysql_optimizer.cpp
+++ b/src/storage/mysql_optimizer.cpp
@@ -56,7 +56,7 @@ void OptimizeMySQLScan(unique_ptr<LogicalOperator> &op) {
 	}
 }
 
-void MySQLOptimizer::Optimize(ClientContext &context, OptimizerExtensionInfo *info, unique_ptr<LogicalOperator> &plan) {
+void MySQLOptimizer::Optimize(OptimizerExtensionInput &input, unique_ptr<LogicalOperator> &plan) {
 	OptimizeMySQLScan(plan);
 }
 
