diff --git a/src/include/storage/unity_catalog.hpp b/src/include/storage/unity_catalog.hpp
index 0c6f6bc..cd000ae 100644
--- a/src/include/storage/unity_catalog.hpp
+++ b/src/include/storage/unity_catalog.hpp
@@ -35,7 +35,7 @@ public:
 class UnityCatalog : public Catalog {
 public:
 	explicit UnityCatalog(AttachedDatabase &db_p, const string &internal_name, AttachOptions &attach_options,
-	                      UCCredentials credentials, const string &default_schema,
+	                      UCCredentials credentials, const Identifier &default_schema,
 	                      string catalog_name = "unity_catalog");
 	~UnityCatalog();
 
@@ -81,7 +81,7 @@ public:
 	                                            unique_ptr<LogicalOperator> plan) override;
 
 	DatabaseSize GetDatabaseSize(ClientContext &context) override;
-	string GetDefaultSchema() const override;
+	Identifier GetDefaultSchema() const override;
 	void OnDetach(ClientContext &context) override;
 
 	//! Whether or not this is an in-memory UC database
@@ -95,7 +95,7 @@ private:
 
 private:
 	UCSchemaSet schemas;
-	string default_schema;
+	Identifier default_schema;
 };
 
 } // namespace duckdb
diff --git a/src/storage/unity_catalog.cpp b/src/storage/unity_catalog.cpp
index de4e211..53f44ee 100644
--- a/src/storage/unity_catalog.cpp
+++ b/src/storage/unity_catalog.cpp
@@ -15,7 +15,7 @@
 namespace duckdb {
 
 UnityCatalog::UnityCatalog(AttachedDatabase &db_p, const string &internal_name, AttachOptions &attach_options,
-                           UCCredentials credentials, const string &default_schema, string catalog_name_p)
+                           UCCredentials credentials, const Identifier &default_schema, string catalog_name_p)
     : Catalog(db_p), internal_name(internal_name), access_mode(attach_options.access_mode),
       credentials(std::move(credentials)), catalog_name(std::move(catalog_name_p)), schemas(*this),
       default_schema(default_schema) {
@@ -56,7 +56,7 @@ optional_ptr<SchemaCatalogEntry> UnityCatalog::LookupSchema(CatalogTransaction t
 			    "specify a DEFAULT_SCHEMA on ATTACH: `ATTACH '..' (TYPE unity_catalog, DEFAULT_SCHEMA 'my_schema')`",
 			    GetName());
 		}
-		return GetSchema(transaction, Identifier(default_schema), if_not_found);
+		return GetSchema(transaction, default_schema, if_not_found);
 	}
 	auto entry = schemas.GetEntry(transaction.GetContext(), schema_lookup);
 	if (!entry && if_not_found != OnEntryNotFound::RETURN_NULL) {
@@ -73,7 +73,7 @@ string UnityCatalog::GetDBPath() {
 	return internal_name;
 }
 
-string UnityCatalog::GetDefaultSchema() const {
+Identifier UnityCatalog::GetDefaultSchema() const {
 	return default_schema;
 }
 
diff --git a/src/unity_catalog_extension.cpp b/src/unity_catalog_extension.cpp
index cacce07..0cb5485 100644
--- a/src/unity_catalog_extension.cpp
+++ b/src/unity_catalog_extension.cpp
@@ -80,7 +80,7 @@ static unique_ptr<Catalog> UnityCatalogAttach(optional_ptr<StorageExtensionInfo>
 
 	// check if we have a secret provided
 	string secret_name;
-	string default_schema;
+	Identifier default_schema;
 	for (auto &entry : info.options) {
 		auto lower_name = StringUtil::Lower(entry.first);
 		if (lower_name == "type" || lower_name == "read_only") {
@@ -88,7 +88,7 @@ static unique_ptr<Catalog> UnityCatalogAttach(optional_ptr<StorageExtensionInfo>
 		} else if (lower_name == "secret") {
 			secret_name = entry.second.ToString();
 		} else if (lower_name == "default_schema") {
-			default_schema = entry.second.ToString();
+			default_schema = Identifier(entry.second.ToString());
 		} else {
 			throw BinderException("Unrecognized option for UC attach: %s", entry.first);
 		}
@@ -136,7 +136,7 @@ static unique_ptr<Catalog> UnityCatalogAttach(optional_ptr<StorageExtensionInfo>
 		//! No explicit default schema provided, ask the catalog:
 		// Fixme: default namespace endpoint not available in OSS unity catalog, hence we throw
 		try {
-			default_schema = UCAPI::GetDefaultSchema(context, credentials);
+			default_schema = Identifier(UCAPI::GetDefaultSchema(context, credentials));
 		} catch (Exception &e) {
 			DUCKDB_LOG_ERROR(context, "Failed to fetch default schema: %s", e.what());
 		}
