diff --git a/src/storage/uc_table_set.cpp b/src/storage/uc_table_set.cpp
index ccd4229..a8bdd46 100644
--- a/src/storage/uc_table_set.cpp
+++ b/src/storage/uc_table_set.cpp
@@ -27,12 +27,12 @@ static idx_t ParseDeltaVersionFromAtClause(const BoundAtClause &at_clause) {
 	if (at_clause.Unit() != "version") {
 		throw InvalidConfigurationException("Delta tables only support at_clause with unit 'version'");
 	}
-	Value version_value = at_clause.GetValue();
-	if (!version_value.DefaultTryCastAs(LogicalType::UBIGINT, false)) {
+	auto version_value = at_clause.GetValue().DefaultTryCastAs(LogicalType::UBIGINT);
+	if (!version_value) {
 		throw InvalidInputException("Failed to parse version number '%s' into a valid version",
 		                            at_clause.GetValue().ToString().c_str());
 	}
-	return version_value.GetValue<idx_t>();
+	return version_value->GetValue<idx_t>();
 }
 
 UCTableSet::UCTableSet(UCSchemaEntry &schema) : catalog(schema.ParentCatalog().Cast<UnityCatalog>()), schema(schema) {
