diff --git a/src/include/storage/unity_catalog.hpp b/src/include/storage/unity_catalog.hpp
index cd000ae..bf8d275 100644
--- a/src/include/storage/unity_catalog.hpp
+++ b/src/include/storage/unity_catalog.hpp
@@ -81,7 +81,7 @@ public:
 	                                            unique_ptr<LogicalOperator> plan) override;
 
 	DatabaseSize GetDatabaseSize(ClientContext &context) override;
-	Identifier GetDefaultSchema() const override;
+	optional<Identifier> GetDefaultSchema() const override;
 	void OnDetach(ClientContext &context) override;
 
 	//! Whether or not this is an in-memory UC database
diff --git a/src/storage/unity_catalog.cpp b/src/storage/unity_catalog.cpp
index 53f44ee..48c093e 100644
--- a/src/storage/unity_catalog.cpp
+++ b/src/storage/unity_catalog.cpp
@@ -73,7 +73,11 @@ string UnityCatalog::GetDBPath() {
 	return internal_name;
 }
 
-Identifier UnityCatalog::GetDefaultSchema() const {
+optional<Identifier> UnityCatalog::GetDefaultSchema() const {
+	if (default_schema.empty()) {
+		//! auto-detection failed and no DEFAULT_SCHEMA was configured on ATTACH
+		return nullopt;
+	}
 	return default_schema;
 }
 
