diff --git a/src/include/storage/uc_table_entry.hpp b/src/include/storage/uc_table_entry.hpp
index 28a2acd..d573f8b 100644
--- a/src/include/storage/uc_table_entry.hpp
+++ b/src/include/storage/uc_table_entry.hpp
@@ -40,6 +40,8 @@ public:
 	UCTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, TableInformation &table, CreateTableInfo &info);
 
 public:
+	const ColumnList &GetColumns() const override;
+
 	unique_ptr<BaseStatistics> GetStatistics(ClientContext &context, column_t column_id) override;
 
 	TableFunction GetScanFunction(ClientContext &context, unique_ptr<FunctionData> &bind_data) override;
@@ -54,6 +56,9 @@ public:
 	void BindUpdateConstraints(Binder &binder, LogicalGet &get, LogicalProjection &proj, LogicalUpdate &update,
 	                           ClientContext &context) override;
 
+protected:
+	ColumnList columns;
+
 public:
 	TableInformation &table;
 };
diff --git a/src/storage/uc_table_entry.cpp b/src/storage/uc_table_entry.cpp
index efcb338..4d674e7 100644
--- a/src/storage/uc_table_entry.cpp
+++ b/src/storage/uc_table_entry.cpp
@@ -11,10 +11,14 @@
 namespace duckdb {
 
 UCTableEntry::UCTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, TableInformation &table, CreateTableInfo &info)
-    : TableCatalogEntry(catalog, schema, info), table(table) {
+    : TableCatalogEntry(catalog, schema, info), columns(std::move(info.columns)), table(table) {
 	this->internal = false;
 }
 
+const ColumnList &UCTableEntry::GetColumns() const {
+	return columns;
+}
+
 unique_ptr<BaseStatistics> UCTableEntry::GetStatistics(ClientContext &context, column_t column_id) {
 	return nullptr;
 }
