diff --git a/test/sql/aws_secret.test b/test/sql/aws_secret.test
index b6c8698..9752d03 100644
--- a/test/sql/aws_secret.test
+++ b/test/sql/aws_secret.test
@@ -30,7 +30,7 @@ CREATE PERSISTENT SECRET aws_all_s3_params_set (
 query IIIIIII
 SELECT * from duckdb_secrets(redact=false);
 ----
-aws_all_s3_params_set	s3	credential_chain	1	local_file	[s3://, s3n://, s3a://]	name=aws_all_s3_params_set;type=s3;provider=credential_chain;serializable=true;scope=s3://,s3n://,s3a://;endpoint=endpoint_override;key_id=key_override;region=region_override;secret=secret_override;session_token=session_override;url_style=style_override;use_ssl=true
+aws_all_s3_params_set	s3	credential_chain	1	local_file	['s3://', 's3n://', 's3a://']	name=aws_all_s3_params_set;type=s3;provider=credential_chain;serializable=true;scope=s3://,s3n://,s3a://;endpoint=endpoint_override;key_id=key_override;region=region_override;secret=secret_override;session_token=session_override;url_style=style_override;use_ssl=true
 
 restart
 
@@ -40,9 +40,9 @@ set secret_directory='__TEST_DIR__/aws_secret'
 query IIIIIII
 SELECT * from duckdb_secrets(redact=false);
 ----
-aws_all_s3_params_set	s3	credential_chain	1	local_file	[s3://, s3n://, s3a://]	name=aws_all_s3_params_set;type=s3;provider=credential_chain;serializable=true;scope=s3://,s3n://,s3a://;endpoint=endpoint_override;key_id=key_override;region=region_override;secret=secret_override;session_token=session_override;url_style=style_override;use_ssl=true
+aws_all_s3_params_set	s3	credential_chain	1	local_file	['s3://', 's3n://', 's3a://']	name=aws_all_s3_params_set;type=s3;provider=credential_chain;serializable=true;scope=s3://,s3n://,s3a://;endpoint=endpoint_override;key_id=key_override;region=region_override;secret=secret_override;session_token=session_override;url_style=style_override;use_ssl=true
 
 query IIIIIII
 SELECT * from duckdb_secrets();
 ----
-aws_all_s3_params_set	s3	credential_chain	1	local_file	[s3://, s3n://, s3a://]	name=aws_all_s3_params_set;type=s3;provider=credential_chain;serializable=true;scope=s3://,s3n://,s3a://;endpoint=endpoint_override;key_id=key_override;region=region_override;secret=redacted;session_token=redacted;url_style=style_override;use_ssl=true
\ No newline at end of file
+aws_all_s3_params_set	s3	credential_chain	1	local_file	['s3://', 's3n://', 's3a://']	name=aws_all_s3_params_set;type=s3;provider=credential_chain;serializable=true;scope=s3://,s3n://,s3a://;endpoint=endpoint_override;key_id=key_override;region=region_override;secret=redacted;session_token=redacted;url_style=style_override;use_ssl=true
\ No newline at end of file
