diff --git a/test/sql/aws_errors.test b/test/sql/aws_errors.test
index 50fb7d7..292724e 100644
--- a/test/sql/aws_errors.test
+++ b/test/sql/aws_errors.test
@@ -2,7 +2,7 @@
 # description: test aws extension
 # group: [aws]
 
-require no_extension_autoloading
+require no_extension_autoloading "EXPECTED: Test relies on autoloading not to be there"
 
 # Before we load the extension, this will fail
 statement error
@@ -22,4 +22,4 @@ httpfs extension is required for load_aws_credentials
 require httpfs
 
 statement ok
-CALL load_aws_credentials();
\ No newline at end of file
+CALL load_aws_credentials();
