diff --git a/test/sql/aws_errors.test b/test/sql/aws_errors.test
index a4ca207..84d86ce 100644
--- a/test/sql/aws_errors.test
+++ b/test/sql/aws_errors.test
@@ -2,6 +2,8 @@
 # description: test aws extension
 # group: [aws]
 
+require no_extension_autoloading
+
 # Before we load the extension, this will fail
 statement error
 CALL load_aws_credentials();
@@ -20,4 +22,4 @@ httpfs extension is required for load_aws_credentials
 require httpfs
 
 statement ok
-CALL load_aws_credentials();
\ No newline at end of file
+CALL load_aws_credentials();
