diff --git a/src/spatial/modules/geos/geos_module.cpp b/src/spatial/modules/geos/geos_module.cpp
index e11c11b..b4a38bb 100644
--- a/src/spatial/modules/geos/geos_module.cpp
+++ b/src/spatial/modules/geos/geos_module.cpp
@@ -2675,7 +2675,7 @@ struct ST_Union_Agg {
 		}
 	};
 
-	static idx_t StateSize(const BoundAggregateFunction &) {
+	static idx_t StateSize(AggregateStateInput &) {
 		return sizeof(State);
 	}
 
@@ -2700,12 +2700,14 @@ struct ST_Union_Agg {
 		return GeosSerde::Deserialize(context, arena, ptr, size);
 	}
 
-	static void Initialize(const BoundAggregateFunction &, data_ptr_t state_mem) {
-		const auto state_ptr = new (state_mem) State();
-		auto &state = *state_ptr;
-		state.context = GEOS_init_r();
-		GEOSContext_setErrorMessageHandler_r(
-		    state.context, [](const char *message, void *) { throw InvalidInputException(message); }, nullptr);
+	static void Initialize(AggregateStateInput &, data_ptr_t *states, idx_t count) {
+		for (idx_t i = 0; i < count; i++) {
+			const auto state_ptr = new (states[i]) State();
+			auto &state = *state_ptr;
+			state.context = GEOS_init_r();
+			GEOSContext_setErrorMessageHandler_r(
+			    state.context, [](const char *message, void *) { throw InvalidInputException(message); }, nullptr);
+		}
 	}
 
 	static void Update(Vector inputs[], AggregateInputData &aggr, idx_t, Vector &state_vec, idx_t count) {
@@ -2905,12 +2907,14 @@ struct GEOSCoverageAggFunction {
 		return GeosSerde::Deserialize(context, arena, ptr, size);
 	}
 
-	static void Initialize(const BoundAggregateFunction &, data_ptr_t state_mem) {
-		const auto state_ptr = new (state_mem) State();
-		auto &state = *state_ptr;
-		state.context = GEOS_init_r();
-		GEOSContext_setErrorMessageHandler_r(
-		    state.context, [](const char *message, void *) { throw InvalidInputException(message); }, nullptr);
+	static void Initialize(AggregateStateInput &, data_ptr_t *states, idx_t count) {
+		for (idx_t i = 0; i < count; i++) {
+			const auto state_ptr = new (states[i]) State();
+			auto &state = *state_ptr;
+			state.context = GEOS_init_r();
+			GEOSContext_setErrorMessageHandler_r(
+			    state.context, [](const char *message, void *) { throw InvalidInputException(message); }, nullptr);
+		}
 	}
 
 	static void Absorb(Vector &state_vec, Vector &combined, AggregateInputData &aggr_input_data, idx_t count) {
@@ -2974,7 +2978,7 @@ struct GEOSCoverageAggFunction {
 		}
 	}
 
-	static idx_t StateSize(const BoundAggregateFunction &) {
+	static idx_t StateSize(AggregateStateInput &) {
 		return sizeof(State);
 	}
 
diff --git a/src/spatial/modules/mvt/mvt_module.cpp b/src/spatial/modules/mvt/mvt_module.cpp
index 27f0c9f..5ce359f 100644
--- a/src/spatial/modules/mvt/mvt_module.cpp
+++ b/src/spatial/modules/mvt/mvt_module.cpp
@@ -1010,12 +1010,14 @@ struct ST_AsMVT {
 		MVTLayer layer;
 	};
 
-	static idx_t StateSize(const BoundAggregateFunction &) {
+	static idx_t StateSize(AggregateStateInput &) {
 		return sizeof(State);
 	}
 
-	static void Initialize(const BoundAggregateFunction &, data_ptr_t state_mem) {
-		new (state_mem) State();
+	static void Initialize(AggregateStateInput &, data_ptr_t *states, idx_t count) {
+		for (idx_t i = 0; i < count; i++) {
+			new (states[i]) State();
+		}
 	}
 
 	//------------------------------------------------------------------------------------------------------------------
