diff --git a/src/spatial/modules/main/spatial_functions_scalar.cpp b/src/spatial/modules/main/spatial_functions_scalar.cpp
index a8145b3..7fbc617 100644
--- a/src/spatial/modules/main/spatial_functions_scalar.cpp
+++ b/src/spatial/modules/main/spatial_functions_scalar.cpp
@@ -9481,7 +9481,7 @@ bool ST_DWithinHelper::TryGetConstDistance(const unique_ptr<FunctionData> &bind_
 void RegisterSpatialScalarFunctions(ExtensionLoader &loader) {
 	ST_Affine::Register(loader);
 	ST_Area::Register(loader);
-	ST_AsGeoJSON::Register(loader);
+	// ST_AsGeoJSON::Register(loader);
 	ST_AsText::Register(loader);
 	// ST_AsWKB::Register(loader);
 	ST_AsHEXWKB::Register(loader);
@@ -9508,7 +9508,7 @@ void RegisterSpatialScalarFunctions(ExtensionLoader &loader) {
 	ST_Force4D::Register(loader);
 	ST_GeometryType::Register(loader);
 	ST_GeomFromHEXWKB::Register(loader);
-	ST_GeomFromGeoJSON::Register(loader);
+	// ST_GeomFromGeoJSON::Register(loader);
 	ST_GeomFromText::Register(loader);
 	ST_GeomFromWKB::Register(loader);
 	ST_HasZ::Register(loader);
diff --git a/test/sql/geometry/st_asgeojson.test b/test/sql/geometry/st_asgeojson.test
index 3f486ab..20ac4e2 100644
--- a/test/sql/geometry/st_asgeojson.test
+++ b/test/sql/geometry/st_asgeojson.test
@@ -3,6 +3,8 @@
 
 require spatial
 
+require json
+
 # Geometry
 statement ok
 CREATE TABLE types (geom GEOMETRY);
@@ -114,5 +116,5 @@ SELECT ST_AsGeoJSON('LINESTRING ZM (1 2 3 4, 4 5 6 7)');
 statement error
 select st_geomfromgeojson('null');
 ----
-Could not parse GeoJSON input: Not a valid JSON object, (null)
+Invalid GeoJSON: geometry must be an object
 
