diff --git a/src/spatial/modules/shapefile/shapefile_module.cpp b/src/spatial/modules/shapefile/shapefile_module.cpp
index f12d1eb..6ef8fb5 100644
--- a/src/spatial/modules/shapefile/shapefile_module.cpp
+++ b/src/spatial/modules/shapefile/shapefile_module.cpp
@@ -966,8 +966,8 @@ struct Shapefile_Meta {
 			auto str = string_t(shape_type_map[i].shp_name);
 			varchar_data[i] = str.IsInlined() ? str : StringVector::AddString(varchar_vector, str);
 		}
-		auto shape_type_enum = LogicalType::ENUM("SHAPE_TYPE", varchar_vector, shape_type_count);
-		shape_type_enum.SetAlias("SHAPE_TYPE");
+		auto shape_type_enum =
+		    LogicalType::ENUM("SHAPE_TYPE", varchar_vector, shape_type_count).WithAlias("SHAPE_TYPE");
 
 		return_types.push_back(LogicalType::VARCHAR);
 		return_types.push_back(shape_type_enum);
diff --git a/src/spatial/modules/wkb/wkb_module.cpp b/src/spatial/modules/wkb/wkb_module.cpp
index 43e7d1a..5395fca 100644
--- a/src/spatial/modules/wkb/wkb_module.cpp
+++ b/src/spatial/modules/wkb/wkb_module.cpp
@@ -11,9 +11,7 @@ namespace {
 struct WKBTypes {
 
 	static LogicalType WKB_BLOB() {
-		auto blob_type = LogicalType(LogicalTypeId::BLOB);
-		blob_type.SetAlias("WKB_BLOB");
-		return blob_type;
+		return LogicalType(LogicalTypeId::BLOB).WithAlias("WKB_BLOB");
 	}
 
 	static bool ToWKBCast(Vector &source, Vector &result, idx_t count, CastParameters &) {
diff --git a/src/spatial/spatial_types.cpp b/src/spatial/spatial_types.cpp
index d98b266..e4ee52f 100644
--- a/src/spatial/spatial_types.cpp
+++ b/src/spatial/spatial_types.cpp
@@ -7,70 +7,59 @@
 namespace duckdb {
 
 LogicalType GeoTypes::POINT_2D() {
-	auto type = LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}});
-	type.SetAlias("POINT_2D");
-	return type;
+	return LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}}).WithAlias("POINT_2D");
 }
 
 LogicalType GeoTypes::POINT_3D() {
-	auto type =
-	    LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}, {"z", LogicalType::DOUBLE}});
-	type.SetAlias("POINT_3D");
-	return type;
+	return LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}, {"z", LogicalType::DOUBLE}})
+	    .WithAlias("POINT_3D");
 }
 
 LogicalType GeoTypes::POINT_4D() {
-	auto type = LogicalType::STRUCT({{"x", LogicalType::DOUBLE},
-	                                 {"y", LogicalType::DOUBLE},
-	                                 {"z", LogicalType::DOUBLE},
-	                                 {"m", LogicalType::DOUBLE}});
-	type.SetAlias("POINT_4D");
-	return type;
+	return LogicalType::STRUCT({{"x", LogicalType::DOUBLE},
+	                            {"y", LogicalType::DOUBLE},
+	                            {"z", LogicalType::DOUBLE},
+	                            {"m", LogicalType::DOUBLE}})
+	    .WithAlias("POINT_4D");
 }
 
 LogicalType GeoTypes::BOX_2D() {
-	auto type = LogicalType::STRUCT({{"min_x", LogicalType::DOUBLE},
-	                                 {"min_y", LogicalType::DOUBLE},
-	                                 {"max_x", LogicalType::DOUBLE},
-	                                 {"max_y", LogicalType::DOUBLE}});
-	type.SetAlias("BOX_2D");
-	return type;
+	return LogicalType::STRUCT({{"min_x", LogicalType::DOUBLE},
+	                            {"min_y", LogicalType::DOUBLE},
+	                            {"max_x", LogicalType::DOUBLE},
+	                            {"max_y", LogicalType::DOUBLE}})
+	    .WithAlias("BOX_2D");
 }
 
 LogicalType GeoTypes::BOX_2DF() {
-	auto type = LogicalType::STRUCT({{"min_x", LogicalType::FLOAT},
-	                                 {"min_y", LogicalType::FLOAT},
-	                                 {"max_x", LogicalType::FLOAT},
-	                                 {"max_y", LogicalType::FLOAT}});
-	type.SetAlias("BOX_2DF");
-	return type;
+	return LogicalType::STRUCT({{"min_x", LogicalType::FLOAT},
+	                            {"min_y", LogicalType::FLOAT},
+	                            {"max_x", LogicalType::FLOAT},
+	                            {"max_y", LogicalType::FLOAT}})
+	    .WithAlias("BOX_2DF");
 }
 
 LogicalType GeoTypes::LINESTRING_2D() {
-	auto type = LogicalType::LIST(LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}}));
-	type.SetAlias("LINESTRING_2D");
-	return type;
+	return LogicalType::LIST(LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}}))
+	    .WithAlias("LINESTRING_2D");
 }
 
 LogicalType GeoTypes::LINESTRING_3D() {
-	auto type = LogicalType::LIST(
-	    LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}, {"z", LogicalType::DOUBLE}}));
-	type.SetAlias("LINESTRING_3D");
-	return type;
+	return LogicalType::LIST(LogicalType::STRUCT(
+	                             {{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}, {"z", LogicalType::DOUBLE}}))
+	    .WithAlias("LINESTRING_3D");
 }
 
 LogicalType GeoTypes::POLYGON_2D() {
-	auto type = LogicalType::LIST(
-	    LogicalType::LIST(LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}})));
-	type.SetAlias("POLYGON_2D");
-	return type;
+	return LogicalType::LIST(
+	           LogicalType::LIST(LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}})))
+	    .WithAlias("POLYGON_2D");
 }
 
 LogicalType GeoTypes::POLYGON_3D() {
-	auto type = LogicalType::LIST(LogicalType::LIST(
-	    LogicalType::STRUCT({{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}, {"z", LogicalType::DOUBLE}})));
-	type.SetAlias("POLYGON_3D");
-	return type;
+	return LogicalType::LIST(LogicalType::LIST(LogicalType::STRUCT(
+	                             {{"x", LogicalType::DOUBLE}, {"y", LogicalType::DOUBLE}, {"z", LogicalType::DOUBLE}})))
+	    .WithAlias("POLYGON_3D");
 }
 
 LogicalType GeoTypes::CreateEnumType(const string &name, const vector<string> &members) {
@@ -80,9 +69,7 @@ LogicalType GeoTypes::CreateEnumType(const string &name, const vector<string> &m
 		auto str = string_t(members[i]);
 		varchar_data[i] = str.IsInlined() ? str : StringVector::AddString(varchar_vector, str);
 	}
-	auto enum_type = LogicalType::ENUM(name, varchar_vector, members.size());
-	enum_type.SetAlias(name);
-	return enum_type;
+	return LogicalType::ENUM(name, varchar_vector, members.size()).WithAlias(name);
 }
 
 static unique_ptr<FunctionData> PropagateTypesInternal(ClientContext &context, BoundSimpleFunction &bound_function,
