diff --git a/test/sql/index/rtree_persistence_load.test b/test/sql/index/rtree_persistence_load.test
index 4da1f16..ebfaaa3 100644
--- a/test/sql/index/rtree_persistence_load.test
+++ b/test/sql/index/rtree_persistence_load.test
@@ -32,7 +32,7 @@ SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
 statement error
 DELETE FROM t1 WHERE rowid < 1000;
 ----
-Missing Extension Error: Cannot bind index 't1', unknown index type 'RTREE'. You need to load the extension that provides this index type before table 't1' can be modified.
+Missing Extension Error: Cannot bind index "t1", unknown index type 'RTREE'. You need to load the extension that provides this index type before table "t1" can be modified.
 
 restart
 
