diff --git a/src/spatial/index/rtree/rtree_index.cpp b/src/spatial/index/rtree/rtree_index.cpp
index 7469bcd..7393f1c 100644
--- a/src/spatial/index/rtree/rtree_index.cpp
+++ b/src/spatial/index/rtree/rtree_index.cpp
@@ -280,7 +280,7 @@ IndexStorageInfo RTreeIndex::SerializeToWAL(const case_insensitive_map_t<Value>
 	return info;
 }
 
-idx_t RTreeIndex::GetInMemorySize(IndexLock &state) {
+idx_t RTreeIndex::GetInMemorySize(IndexLock &state) const {
 	const auto &leaf_alloc = tree->GetLeafAllocator();
 	const auto &node_alloc = tree->GetNodeAllocator();
 	return leaf_alloc.GetInMemorySize() + node_alloc.GetInMemorySize();
diff --git a/src/spatial/index/rtree/rtree_index.hpp b/src/spatial/index/rtree/rtree_index.hpp
index 8af3ba3..dc2335d 100644
--- a/src/spatial/index/rtree/rtree_index.hpp
+++ b/src/spatial/index/rtree/rtree_index.hpp
@@ -59,7 +59,7 @@ public:
 	//! Serializes RTree memory to the WAL and returns the index storage information.
 	IndexStorageInfo SerializeToWAL(const case_insensitive_map_t<Value> &options) override;
 
-	idx_t GetInMemorySize(IndexLock &state) override;
+	idx_t GetInMemorySize(IndexLock &state) const override;
 
 	//! Merge another index into this index. The lock obtained from InitializeLock must be held, and the other
 	//! index must also be locked during the merge
