diff --git a/src/spatial/index/rtree/rtree_index_create_physical.cpp b/src/spatial/index/rtree/rtree_index_create_physical.cpp
--- a/src/spatial/index/rtree/rtree_index_create_physical.cpp
+++ b/src/spatial/index/rtree/rtree_index_create_physical.cpp
@@ -299,7 +299,7 @@
 	duck_index.initial_index_size = gstate.rtree->Cast<BoundIndex>().GetInMemorySize();
 
 	// Finally add it to storage
-	storage.AddIndex(std::move(gstate.rtree));
+	storage.AddIndex(std::move(gstate.rtree), duck_index.GetIndexId());
 }
 
 class RTreeIndexConstructionEvent final : public BasePipelineEvent {
diff --git a/src/spatial/index/rtree/rtree_index_pragmas.cpp b/src/spatial/index/rtree/rtree_index_pragmas.cpp
--- a/src/spatial/index/rtree/rtree_index_pragmas.cpp
+++ b/src/spatial/index/rtree/rtree_index_pragmas.cpp
@@ -86,7 +86,7 @@
 			if (entry->GetBindState() != IndexBindState::BOUND || RTreeIndex::TYPE_NAME != entry->GetIndexType()) {
 				continue;
 			}
-			if (entry->GetName() == index_entry.name) {
+			if (entry->GetId() == index_entry.Cast<DuckIndexEntry>().GetIndexId()) {
 				found_index = true;
 				break;
 			}
@@ -130,7 +130,7 @@
 		if (entry->GetBindState() != IndexBindState::BOUND || RTreeIndex::TYPE_NAME != entry->GetIndexType()) {
 			continue;
 		}
-		if (entry->GetName() == index_entry.name) {
+		if (entry->GetId() == index_entry.Cast<DuckIndexEntry>().GetIndexId()) {
 			return entry;
 		}
 	};
diff --git a/src/spatial/index/rtree/rtree_index_scan.cpp b/src/spatial/index/rtree/rtree_index_scan.cpp
--- a/src/spatial/index/rtree/rtree_index_scan.cpp
+++ b/src/spatial/index/rtree/rtree_index_scan.cpp
@@ -220,6 +220,9 @@
 	// Now also lookup the index by name
 	const auto index_name = deserializer.ReadProperty<string>(103, "index_name");
+	auto &catalog_index =
+	    Catalog::GetEntry<IndexCatalogEntry>(context, Identifier(catalog), Identifier(schema), Identifier(index_name))
+	        .Cast<DuckIndexEntry>();
 	RTreeBounds bbox;
 	deserializer.ReadObject(104, "bbox", [&](Deserializer &ser) {
 		bbox.min.x = ser.ReadProperty<float>(10, "min_x");
 		bbox.min.y = ser.ReadProperty<float>(11, "min_y");
@@ -239,7 +242,7 @@
 		    RTreeIndex::TYPE_NAME != index_entry->GetIndexType()) {
 			continue;
 		}
-		if (index_entry->GetName() == index_name) {
+		if (index_entry->GetId() == catalog_index.GetIndexId()) {
 			result = make_uniq<RTreeIndexScanBindData>(duck_table, index_entry, Identifier(index_name), bbox);
 			break;
 		}
