diff --git a/src/spatial/index/rtree/rtree_index_create_physical.cpp b/src/spatial/index/rtree/rtree_index_create_physical.cpp
index 08abd6f..e42559b 100644
--- a/src/spatial/index/rtree/rtree_index_create_physical.cpp
+++ b/src/spatial/index/rtree/rtree_index_create_physical.cpp
@@ -300,7 +300,7 @@ static void AddIndexToCatalog(ClientContext &context, CreateRTreeIndexGlobalStat
 	duck_index.initial_index_size = gstate.rtree->Cast<BoundIndex>().GetInMemorySize();
 
 	// Finally add it to storage
-	storage.AddIndex(std::move(gstate.rtree));
+	storage.AddIndex(std::move(gstate.rtree), duck_index.oid);
 }
 
 class RTreeIndexConstructionEvent final : public BasePipelineEvent {
