diff --git a/src/spatial/operators/spatial_join_physical.cpp b/src/spatial/operators/spatial_join_physical.cpp
--- a/src/spatial/operators/spatial_join_physical.cpp
+++ b/src/spatial/operators/spatial_join_physical.cpp
@@ -426,7 +426,7 @@
 
 	// Always make sure we have a consistent order of the output columns, regardless if we have projection maps or not
 
-	const auto &lop = op.Cast<LogicalJoin>();
+	const auto &lop = op.Cast<LogicalSpatialJoin>();
 
 	// Probe-side
 	const auto &probe_side_input_types = children[0].get().types;
