diff --git a/src/spatial/index/rtree/rtree_index_plan_scan.cpp b/src/spatial/index/rtree/rtree_index_plan_scan.cpp
index 2ee2ed1..687f129 100644
--- a/src/spatial/index/rtree/rtree_index_plan_scan.cpp
+++ b/src/spatial/index/rtree/rtree_index_plan_scan.cpp
@@ -43,14 +43,14 @@ public:
 		if (expr.GetExpressionType() == ExpressionType::BOUND_COLUMN_REF) {
 			auto &bound_colref = expr.Cast<BoundColumnRefExpression>();
 			// bound column ref: rewrite to fit in the current set of bound column ids
-			bound_colref.binding.table_index = get.table_index;
+			bound_colref.BindingMutable().table_index = get.table_index;
 			auto &column_ids = index.GetColumnIds();
 			auto &get_column_ids = get.GetColumnIds();
-			column_t referenced_column = column_ids[bound_colref.binding.column_index];
+			column_t referenced_column = column_ids[bound_colref.Binding().column_index];
 			// search for the referenced column in the set of column_ids
 			for (idx_t i = 0; i < get_column_ids.size(); i++) {
 				if (get_column_ids[i].GetPrimaryIndex() == referenced_column) {
-					bound_colref.binding.column_index = ProjectionIndex(i);
+					bound_colref.BindingMutable().column_index = ProjectionIndex(i);
 					return;
 				}
 			}
diff --git a/src/spatial/modules/gdal/gdal_module.cpp b/src/spatial/modules/gdal/gdal_module.cpp
index fd7970c..56f081f 100644
--- a/src/spatial/modules/gdal/gdal_module.cpp
+++ b/src/spatial/modules/gdal/gdal_module.cpp
@@ -791,13 +791,13 @@ auto Pushdown(ClientContext &context, LogicalGet &get, FunctionData *bind_data,
 				    multi_geom_expr = true;
 				    return;
 			    }
-			    if (ref.binding.table_index != get.table_index) {
+			    if (ref.Binding().table_index != get.table_index) {
 				    // Not from the same table
 				    return;
 			    }
 
 			    const auto &col_ids = get.GetColumnIds();
-			    const auto &col = col_ids[ref.binding.column_index];
+			    const auto &col = col_ids[ref.Binding().column_index];
 
 			    if (!col.HasPrimaryIndex()) {
 				    return;
