diff --git a/src/spatial/operators/spatial_join_physical.cpp b/src/spatial/operators/spatial_join_physical.cpp
index c330385..3c707b6 100644
--- a/src/spatial/operators/spatial_join_physical.cpp
+++ b/src/spatial/operators/spatial_join_physical.cpp
@@ -441,7 +441,7 @@ PhysicalSpatialJoin::PhysicalSpatialJoin(PhysicalPlan &physical_plan, LogicalOpe
 	unordered_map<idx_t, idx_t> conditions_in_layout;
 	// TODO: Loop over multiple conds
 	if (build_side_key->GetExpressionClass() == ExpressionClass::BOUND_REF) {
-		conditions_in_layout.emplace(build_side_key->Cast<BoundReferenceExpression>().index, 0); // TODO: i, not 0
+		conditions_in_layout.emplace(build_side_key->Cast<BoundReferenceExpression>().Index(), 0); // TODO: i, not 0
 	}
 	// TODO Add rest too
 	build_side_key_types.push_back(build_side_key->GetReturnType());
