diff --git a/src/spatial/index/rtree/rtree_index.cpp b/src/spatial/index/rtree/rtree_index.cpp
index 83fd983..c688289 100644
--- a/src/spatial/index/rtree/rtree_index.cpp
+++ b/src/spatial/index/rtree/rtree_index.cpp
@@ -78,7 +78,8 @@ RTreeIndex::RTreeIndex(const string &name, IndexConstraintType index_constraint_
                        const vector<unique_ptr<Expression>> &unbound_expressions, AttachedDatabase &db,
                        const case_insensitive_map_t<Value> &options, ClientContext &context,
                        const IndexStorageInfo &info, idx_t estimated_cardinality)
-    : BoundIndex(name, TYPE_NAME, index_constraint_type, column_ids, table_io_manager, unbound_expressions, db) {
+    : BoundIndex(Identifier(name), TYPE_NAME, index_constraint_type, column_ids, table_io_manager, unbound_expressions,
+                 db) {
 
 	if (index_constraint_type != IndexConstraintType::NONE) {
 		throw NotImplementedException("RTree indexes do not support unique or primary key constraints");
@@ -111,7 +112,7 @@ RTreeIndex::RTreeIndex(const string &name, IndexConstraintType index_constraint_
 	// Construct the key expression executor
 	auto &source_type = unbound_expressions[0]->GetReturnType();
 	auto &catalog = Catalog::GetSystemCatalog(context);
-	auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, DEFAULT_SCHEMA, "ST_Extent_Approx");
+	auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, Identifier::DefaultSchema(), "ST_Extent_Approx");
 	const auto &func = entry.functions.GetFunctionByArguments(context, {source_type});
 	auto child_expr = make_uniq<BoundReferenceExpression>(source_type, 0);
 
diff --git a/src/spatial/index/rtree/rtree_index_create_logical.cpp b/src/spatial/index/rtree/rtree_index_create_logical.cpp
index 9539b16..747748c 100644
--- a/src/spatial/index/rtree/rtree_index_create_logical.cpp
+++ b/src/spatial/index/rtree/rtree_index_create_logical.cpp
@@ -51,7 +51,7 @@ static PhysicalOperator &CreateNullFilter(PhysicalPlanGenerator &generator, cons
 
 	// Filter IS_NOT_EMPTY on the GEOMETRY column
 	auto &catalog = Catalog::GetSystemCatalog(context);
-	auto &is_empty_entry = catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, DEFAULT_SCHEMA, "ST_IsEmpty")
+	auto &is_empty_entry = catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, Identifier::DefaultSchema(), "ST_IsEmpty")
 	                           .Cast<ScalarFunctionCatalogEntry>();
 
 	auto is_empty_func = is_empty_entry.functions.GetFunctionByArguments(context, {LogicalType::GEOMETRY()});
@@ -76,7 +76,7 @@ static PhysicalOperator &CreateBoundingBoxProjection(PhysicalPlanGenerator &plan
 
 	// Get the bounding box function
 	auto &bbox_func_entry =
-	    catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, DEFAULT_SCHEMA, "ST_Extent_Approx")
+	    catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, Identifier::DefaultSchema(), "ST_Extent_Approx")
 	        .Cast<ScalarFunctionCatalogEntry>();
 	const auto &bbox_func = bbox_func_entry.functions.GetFunctionByArguments(context, {LogicalType::GEOMETRY()});
 
@@ -102,7 +102,7 @@ static PhysicalOperator &CreateOrderByMinX(PhysicalPlanGenerator &planner, const
 
 	// Get the centroid value function
 	auto &centroid_func_entry =
-	    catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, DEFAULT_SCHEMA, "st_centroid")
+	    catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, Identifier::DefaultSchema(), "st_centroid")
 	        .Cast<ScalarFunctionCatalogEntry>();
 	const auto &centroid_func = centroid_func_entry.functions.GetFunctionByArguments(context, {GeoTypes::BOX_2DF()});
 	vector<unique_ptr<Expression>> centroid_func_args;
@@ -114,7 +114,7 @@ static PhysicalOperator &CreateOrderByMinX(PhysicalPlanGenerator &planner, const
 	auto centroid_expr = centroid_func.Bind(context, std::move(centroid_func_args));
 
 	// Get the xmin value function
-	auto &xmin_func_entry = catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, DEFAULT_SCHEMA, "st_xmin")
+	auto &xmin_func_entry = catalog.GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, Identifier::DefaultSchema(), "st_xmin")
 	                            .Cast<ScalarFunctionCatalogEntry>();
 	const auto &xmin_func = xmin_func_entry.functions.GetFunctionByArguments(context, {GeoTypes::POINT_2D()});
 	vector<unique_ptr<Expression>> xmin_func_args;
diff --git a/src/spatial/index/rtree/rtree_index_create_physical.cpp b/src/spatial/index/rtree/rtree_index_create_physical.cpp
index b4d5601..f3d776c 100644
--- a/src/spatial/index/rtree/rtree_index_create_physical.cpp
+++ b/src/spatial/index/rtree/rtree_index_create_physical.cpp
@@ -79,7 +79,7 @@ unique_ptr<GlobalSinkState> PhysicalCreateRTreeIndex::GetGlobalSinkState(ClientC
 	auto &constraint_type = info->constraint_type;
 	auto &db = storage.db;
 	gstate->rtree =
-	    make_uniq<RTreeIndex>(info->index_name, constraint_type, storage_ids, table_manager, unbound_expressions, db,
+	    make_uniq<RTreeIndex>(info->index_name.GetIdentifierName(), constraint_type, storage_ids, table_manager, unbound_expressions, db,
 	                          info->options, context, IndexStorageInfo(), estimated_cardinality);
 
 	gstate->max_node_capacity = gstate->rtree->tree->GetConfig().max_node_capacity;
diff --git a/src/spatial/index/rtree/rtree_index_plan_scan.cpp b/src/spatial/index/rtree/rtree_index_plan_scan.cpp
index 687f129..4aaa1b0 100644
--- a/src/spatial/index/rtree/rtree_index_plan_scan.cpp
+++ b/src/spatial/index/rtree/rtree_index_plan_scan.cpp
@@ -94,7 +94,7 @@ public:
 
 	static bool IsSpatialPredicate(const BoundScalarFunction &function, const unordered_set<string> &predicates) {
 
-		if (predicates.find(function.GetName()) == predicates.end()) {
+		if (predicates.find(function.GetName().GetIdentifierName()) == predicates.end()) {
 			return false;
 		}
 		if (function.GetArguments().size() < 2) {
@@ -120,7 +120,7 @@ public:
 
 		// make a new box expression
 		auto &catalog = Catalog::GetSystemCatalog(context);
-		auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, DEFAULT_SCHEMA, "ST_Extent_Approx");
+		auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, Identifier::DefaultSchema(), "ST_Extent_Approx");
 		const auto &func = entry.functions.GetFunctionByArguments(context, {LogicalType::GEOMETRY()});
 
 		vector<unique_ptr<Expression>> children;
diff --git a/src/spatial/index/rtree/rtree_index_pragmas.cpp b/src/spatial/index/rtree/rtree_index_pragmas.cpp
index 44790d8..d21bec8 100644
--- a/src/spatial/index/rtree/rtree_index_pragmas.cpp
+++ b/src/spatial/index/rtree/rtree_index_pragmas.cpp
@@ -72,8 +72,8 @@ static void RTreeIndexInfoExecute(ClientContext &context, TableFunctionInput &da
 	idx_t row = 0;
 	while (data.offset < data.entries.size() && row < STANDARD_VECTOR_SIZE) {
 		auto &index_entry = data.entries[data.offset++].get();
-		auto &table_entry = index_entry.schema.catalog.GetEntry<TableCatalogEntry>(context, index_entry.GetSchemaName(),
-		                                                                           index_entry.GetTableName());
+		auto &table_entry = index_entry.schema.catalog.GetEntry<TableCatalogEntry>(
+		    context, index_entry.GetSchemaName(), index_entry.GetTableName());
 		auto &storage = table_entry.GetStorage();
 		RTreeIndex *rtree_index = nullptr;
 
@@ -97,9 +97,9 @@ static void RTreeIndexInfoExecute(ClientContext &context, TableFunctionInput &da
 		idx_t col = 0;
 
 		output.data[col++].SetValue(row, Value(index_entry.catalog.GetName()));
-		output.data[col++].SetValue(row, Value(index_entry.schema.name));
-		output.data[col++].SetValue(row, Value(index_entry.name));
-		output.data[col++].SetValue(row, Value(table_entry.name));
+		output.data[col++].SetValue(row, Value(index_entry.schema.name.GetIdentifierName()));
+		output.data[col++].SetValue(row, Value(index_entry.name.GetIdentifierName()));
+		output.data[col++].SetValue(row, Value(table_entry.name.GetIdentifierName()));
 
 		row++;
 	}
@@ -113,7 +113,8 @@ static optional_ptr<RTreeIndex> TryGetIndex(ClientContext &context, const string
 	Binder::BindSchemaOrCatalog(context, qname.catalog, qname.schema);
 	auto &index_entry = Catalog::GetEntry(context, CatalogType::INDEX_ENTRY, qname.catalog, qname.schema, qname.name)
 	                        .Cast<IndexCatalogEntry>();
-	auto &table_entry = Catalog::GetEntry(context, CatalogType::TABLE_ENTRY, qname.catalog, index_entry.GetSchemaName(),
+	auto &table_entry = Catalog::GetEntry(context, CatalogType::TABLE_ENTRY, qname.catalog,
+	                                      index_entry.GetSchemaName(),
 	                                      index_entry.GetTableName())
 	                        .Cast<TableCatalogEntry>();
 
diff --git a/src/spatial/index/rtree/rtree_index_scan.cpp b/src/spatial/index/rtree/rtree_index_scan.cpp
index a1bfbe3..ca12c7d 100644
--- a/src/spatial/index/rtree/rtree_index_scan.cpp
+++ b/src/spatial/index/rtree/rtree_index_scan.cpp
@@ -175,8 +175,8 @@ static InsertionOrderPreservingMap<string> RTreeIndexScanToString(TableFunctionT
 	D_ASSERT(input.bind_data);
 	InsertionOrderPreservingMap<string> result;
 	auto &bind_data = input.bind_data->Cast<RTreeIndexScanBindData>();
-	result["Table"] = bind_data.table.name;
-	result["Index"] = bind_data.index.GetIndexName();
+	result["Table"] = bind_data.table.name.GetIdentifierName();
+	result["Index"] = bind_data.index.GetIndexName().GetIdentifierName();
 	return result;
 }
 
@@ -205,7 +205,7 @@ static unique_ptr<FunctionData> RTreeScanDeserialize(Deserializer &deserializer,
 	const auto catalog = deserializer.ReadProperty<string>(100, "catalog");
 	const auto schema = deserializer.ReadProperty<string>(101, "schema");
 	const auto table = deserializer.ReadProperty<string>(102, "table");
-	auto &catalog_entry = Catalog::GetEntry<TableCatalogEntry>(context, catalog, schema, table);
+	auto &catalog_entry = Catalog::GetEntry<TableCatalogEntry>(context, Identifier(catalog), Identifier(schema), Identifier(table));
 	if (catalog_entry.type != CatalogType::TABLE_ENTRY) {
 		throw SerializationException("Cant find table for %s.%s", schema, table);
 	}
diff --git a/src/spatial/modules/mvt/mvt_module.cpp b/src/spatial/modules/mvt/mvt_module.cpp
index f0c19f8..93f63fc 100644
--- a/src/spatial/modules/mvt/mvt_module.cpp
+++ b/src/spatial/modules/mvt/mvt_module.cpp
@@ -982,7 +982,7 @@ struct ST_AsMVT {
 					                            "DOUBLE, INTEGER, BIGINT, BOOLEAN",
 					                            name.c_str(), type_name.c_str());
 				}
-				result->tag_names.push_back(name);
+				result->tag_names.emplace_back(name);
 			}
 		}
 
diff --git a/src/spatial/modules/proj/proj_module.cpp b/src/spatial/modules/proj/proj_module.cpp
index 8262a8c..61e1b0f 100644
--- a/src/spatial/modules/proj/proj_module.cpp
+++ b/src/spatial/modules/proj/proj_module.cpp
@@ -1579,14 +1579,14 @@ public:
 	}
 
 public:
-	unique_ptr<CatalogEntry> CreateDefaultEntry(ClientContext &context, const string &entry_name) override {
+	unique_ptr<CatalogEntry> CreateDefaultEntry(ClientContext &context, const Identifier &entry_name) override {
 
 		if (schema.name != DEFAULT_SCHEMA) {
 			return nullptr;
 		}
 
 		// Try to split name by ":"
-		auto parts = StringUtil::Split(entry_name, ":");
+		auto parts = StringUtil::Split(entry_name.GetIdentifierName(), ":");
 		if (parts.size() != 2) {
 			return nullptr;
 		}
@@ -1624,20 +1624,20 @@ public:
 
 		proj_destroy(crs);
 
-		auto info = CreateCoordinateSystemInfo(entry_name, auth_name, auth_code, projjson_text, wkt_text);
+		auto info = CreateCoordinateSystemInfo(entry_name.GetIdentifierName(), auth_name, auth_code, projjson_text, wkt_text);
 		info.on_conflict = OnCreateConflict::IGNORE_ON_CONFLICT;
 
 		auto result = make_uniq<CoordinateSystemCatalogEntry>(catalog, schema, info);
 		return std::move(result);
 	}
 
-	vector<string> GetDefaultEntries() override {
+	vector<Identifier> GetDefaultEntries() override {
 
 		if (schema.name != DEFAULT_SCHEMA) {
 			return {};
 		}
 
-		vector<string> entries;
+		vector<Identifier> entries;
 
 		auto scan_authority = [&](const char *auth) {
 			int ncrs = 0;
@@ -1652,7 +1652,7 @@ public:
 						continue;
 					}
 
-					entries.push_back(StringUtil::Format("%s:%s", auth_name, auth_code));
+					entries.emplace_back(StringUtil::Format("%s:%s", auth_name, auth_code));
 				}
 			}
 
@@ -1672,7 +1672,7 @@ public:
 		auto &db = loader.GetDatabaseInstance();
 		auto system_transaction = CatalogTransaction::GetSystemTransaction(db);
 		auto &catalog = Catalog::GetSystemCatalog(db);
-		auto &schema = catalog.GetSchema(system_transaction, DEFAULT_SCHEMA);
+		auto &schema = catalog.GetSchema(system_transaction, Identifier::DefaultSchema());
 		auto &duck_schema = schema.Cast<DuckSchemaEntry>();
 
 		auto &set = duck_schema.GetCatalogSet(CatalogType::COORDINATE_SYSTEM_ENTRY);
diff --git a/src/spatial/operators/spatial_join_optimizer.cpp b/src/spatial/operators/spatial_join_optimizer.cpp
index cabbe23..6f17e74 100644
--- a/src/spatial/operators/spatial_join_optimizer.cpp
+++ b/src/spatial/operators/spatial_join_optimizer.cpp
@@ -56,7 +56,7 @@ static bool HasInversePredicate(const string &func_name) {
 static unique_ptr<Expression> GetInversePredicate(ClientContext &context, unique_ptr<Expression> expr) {
 	auto &func = expr->Cast<BoundFunctionExpression>();
 
-	const auto it = spatial_predicate_inverse_map.find(func.Function().GetName());
+	const auto it = spatial_predicate_inverse_map.find(func.Function().GetName().GetIdentifierName());
 	D_ASSERT(it != spatial_predicate_inverse_map.end());
 
 	// Swap the arguments
@@ -69,7 +69,7 @@ static unique_ptr<Expression> GetInversePredicate(ClientContext &context, unique
 
 	// Get the function from the catalog
 	auto &catalog = Catalog::GetSystemCatalog(context);
-	auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, DEFAULT_SCHEMA, it->second);
+	auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, Identifier::DefaultSchema(), Identifier(it->second));
 	const auto &inverse_func =
 	    entry.functions.GetFunctionByArguments(context, {func.GetChildren()[0]->GetReturnType(), func.GetChildren()[1]->GetReturnType()});
 
@@ -104,7 +104,7 @@ static bool IsSpatialJoinPredicate(const unique_ptr<Expression> &expr, const uno
 	}
 
 	// The function must be a recognized spatial predicate
-	if (spatial_predicate_map.count(func.Function().GetName()) == 0) {
+	if (spatial_predicate_map.count(func.Function().GetName().GetIdentifierName()) == 0) {
 		return false;
 	}
 
@@ -117,7 +117,7 @@ static bool IsSpatialJoinPredicate(const unique_ptr<Expression> &expr, const uno
 	}
 
 	if (left_side == JoinSide::RIGHT) {
-		if (!HasInversePredicate(func.Function().GetName())) {
+		if (!HasInversePredicate(func.Function().GetName().GetIdentifierName())) {
 			return false;
 		}
 		needs_flipping = true;
diff --git a/src/spatial/operators/spatial_join_physical.cpp b/src/spatial/operators/spatial_join_physical.cpp
index 3c707b6..7026b6f 100644
--- a/src/spatial/operators/spatial_join_physical.cpp
+++ b/src/spatial/operators/spatial_join_physical.cpp
@@ -386,7 +386,7 @@ private:
 
 static unique_ptr<Expression> GetBBOXExpression(ClientContext &context, const LogicalType &geom_type) {
 	auto &catalog = Catalog::GetSystemCatalog(context);
-	auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, DEFAULT_SCHEMA, "ST_Extent_Approx");
+	auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, Identifier::DefaultSchema(), "ST_Extent_Approx");
 	const auto &func = entry.functions.GetFunctionByArguments(context, {geom_type});
 
 	auto child_expr = make_uniq<BoundReferenceExpression>(geom_type, 0);
@@ -491,7 +491,7 @@ InsertionOrderPreservingMap<string> PhysicalSpatialJoin::ParamsToString() const
 	// TODO: Add condition to the result (GetName is wrong)
 	auto result = PhysicalOperator::ParamsToString();
 	result["Join Type"] = EnumUtil::ToString(join_type);
-	result["Conditions"] = condition->GetName();
+	result["Conditions"] = condition->GetName().GetIdentifierName();
 	SetEstimatedCardinality(result, estimated_cardinality);
 	return result;
 }
@@ -546,7 +546,8 @@ public:
 		auto &geom_type = op.build_side_key->GetReturnType();
 
 		auto &catalog = Catalog::GetSystemCatalog(context);
-		auto &entry = catalog.GetEntry<ScalarFunctionCatalogEntry>(context, DEFAULT_SCHEMA, "ST_IsEmpty");
+		auto &entry =
+	    catalog.GetEntry<ScalarFunctionCatalogEntry>(context, Identifier::DefaultSchema(), "ST_IsEmpty");
 		const auto &func = entry.functions.GetFunctionByArguments(context, {geom_type});
 
 		vector<unique_ptr<Expression>> children;
diff --git a/src/spatial/util/function_builder.cpp b/src/spatial/util/function_builder.cpp
index d4c6e43..6b92b27 100644
--- a/src/spatial/util/function_builder.cpp
+++ b/src/spatial/util/function_builder.cpp
@@ -64,8 +64,8 @@ void FunctionBuilder::Register(ExtensionLoader &loader, const char *name, Scalar
 	// Also add the parameter names. We need to access the catalog entry for this.
 	auto &catalog = Catalog::GetSystemCatalog(db);
 	auto transaction = CatalogTransaction::GetSystemTransaction(db);
-	auto &schema = catalog.GetSchema(transaction, DEFAULT_SCHEMA);
-	auto catalog_entry = schema.GetEntry(transaction, CatalogType::SCALAR_FUNCTION_ENTRY, name);
+	auto &schema = catalog.GetSchema(transaction, Identifier::DefaultSchema());
+	auto catalog_entry = schema.GetEntry(transaction, CatalogType::SCALAR_FUNCTION_ENTRY, Identifier(name));
 	if (!catalog_entry) {
 		// This should not happen, we just registered the function
 		throw InternalException("Function with name \"%s\" not found in FunctionBuilder::AddScalar", name);
@@ -108,8 +108,8 @@ void FunctionBuilder::Register(ExtensionLoader &loader, const char *name, Aggreg
 	// Also add the parameter names. We need to access the catalog entry for this.
 	auto &catalog = Catalog::GetSystemCatalog(db);
 	auto transaction = CatalogTransaction::GetSystemTransaction(db);
-	auto &schema = catalog.GetSchema(transaction, DEFAULT_SCHEMA);
-	auto catalog_entry = schema.GetEntry(transaction, CatalogType::AGGREGATE_FUNCTION_ENTRY, name);
+	auto &schema = catalog.GetSchema(transaction, Identifier::DefaultSchema());
+	auto catalog_entry = schema.GetEntry(transaction, CatalogType::AGGREGATE_FUNCTION_ENTRY, Identifier(name));
 	if (!catalog_entry) {
 		// This should not happen, we just registered the function
 		throw InternalException("Function with name \"%s\" not found in FunctionBuilder::AddAggregate", name);
@@ -182,8 +182,8 @@ void FunctionBuilder::AddTableFunctionDocs(ExtensionLoader &loader, const char *
 	auto &db = loader.GetDatabaseInstance();
 	auto &catalog = Catalog::GetSystemCatalog(db);
 	auto transaction = CatalogTransaction::GetSystemTransaction(db);
-	auto &schema = catalog.GetSchema(transaction, DEFAULT_SCHEMA);
-	auto catalog_entry = schema.GetEntry(transaction, CatalogType::TABLE_FUNCTION_ENTRY, name);
+	auto &schema = catalog.GetSchema(transaction, Identifier::DefaultSchema());
+	auto catalog_entry = schema.GetEntry(transaction, CatalogType::TABLE_FUNCTION_ENTRY, Identifier(name));
 	if (!catalog_entry) {
 		// This should not happen, we just registered the function
 		throw InternalException("Function with name \"%s\" not found in FunctionBuilder::AddScalar", name);
diff --git a/src/spatial/util/function_builder.hpp b/src/spatial/util/function_builder.hpp
index bdebe71..5057774 100644
--- a/src/spatial/util/function_builder.hpp
+++ b/src/spatial/util/function_builder.hpp
@@ -68,7 +68,7 @@ private:
 };
 
 inline void ScalarFunctionVariantBuilder::AddParameter(const char *name, const LogicalType &type) {
-	function.GetSignature().AddParameter(name, type);
+	function.GetSignature().AddParameter(Identifier(name), type);
 	description.parameter_names.emplace_back(name);
 	description.parameter_types.emplace_back(type);
 }
