diff --git a/src/spatial/modules/geos/geos_module.cpp b/src/spatial/modules/geos/geos_module.cpp
index dd74e23..4a248ad 100644
--- a/src/spatial/modules/geos/geos_module.cpp
+++ b/src/spatial/modules/geos/geos_module.cpp
@@ -2625,7 +2625,7 @@ struct ST_Union_Agg {
 		}
 	}
 
-	static void Finalize(Vector &state_vec, AggregateInputData &aggr_input_data, Vector &result, idx_t count,
+	static void Finalize(Vector &state_vec, AggregateFinalizeInputData &aggr_input_data, Vector &result, idx_t count,
 	                     idx_t offset) {
 
 		UnifiedVectorFormat state_format;
@@ -2815,7 +2815,7 @@ struct GEOSCoverageAggFunction {
 	}
 
 	template <class OP>
-	static void Finalize(Vector &state_vec, AggregateInputData &aggr_input_data, Vector &result, idx_t count,
+	static void Finalize(Vector &state_vec, AggregateFinalizeInputData &aggr_input_data, Vector &result, idx_t count,
 	                     idx_t offset) {
 
 		UnifiedVectorFormat state_format;
diff --git a/src/spatial/modules/mvt/mvt_module.cpp b/src/spatial/modules/mvt/mvt_module.cpp
index 93f63fc..53ce70f 100644
--- a/src/spatial/modules/mvt/mvt_module.cpp
+++ b/src/spatial/modules/mvt/mvt_module.cpp
@@ -1178,7 +1178,7 @@ struct ST_AsMVT {
 	//------------------------------------------------------------------------------------------------------------------
 	// Finalize
 	//------------------------------------------------------------------------------------------------------------------
-	static void Finalize(Vector &state_vec, AggregateInputData &aggr, Vector &result, idx_t count, idx_t offset) {
+	static void Finalize(Vector &state_vec, AggregateFinalizeInputData &aggr, Vector &result, idx_t count, idx_t offset) {
 		const auto &bdata = aggr.bind_data->Cast<BindData>();
 
 		UnifiedVectorFormat state_format;
