diff --git a/test/sql/index/rtree_basic.test b/test/sql/index/rtree_basic.test
index 67360a8..e26f09d 100644
--- a/test/sql/index/rtree_basic.test
+++ b/test/sql/index/rtree_basic.test
@@ -1,5 +1,8 @@
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE t1 (geom GEOMETRY);
 
diff --git a/test/sql/index/rtree_basic_data.test_slow b/test/sql/index/rtree_basic_data.test_slow
index 0f21a6c..b8cfd34 100644
--- a/test/sql/index/rtree_basic_data.test_slow
+++ b/test/sql/index/rtree_basic_data.test_slow
@@ -2,6 +2,9 @@ require parquet
 
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE t1 as SELECT st_point(pickup_longitude, pickup_latitude) as geom, trip_distance, fare_amount
 FROM read_parquet('{WORKING_DIRECTORY}/test/data/nyc_taxi/yellow_tripdata_2010-01-limit1mil.parquet');
diff --git a/test/sql/index/rtree_basic_points.test b/test/sql/index/rtree_basic_points.test
index 77c9f74..f11e611 100644
--- a/test/sql/index/rtree_basic_points.test
+++ b/test/sql/index/rtree_basic_points.test
@@ -1,5 +1,8 @@
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE t1 AS SELECT point::GEOMETRY as geom
 FROM st_generatepoints({min_x: 0, min_y: 0, max_x: 1000, max_y: 1000}::BOX_2D, 100_00, 1337);
diff --git a/test/sql/index/rtree_multi.test b/test/sql/index/rtree_multi.test
index c16557d..f7c815a 100644
--- a/test/sql/index/rtree_multi.test
+++ b/test/sql/index/rtree_multi.test
@@ -3,6 +3,9 @@
 
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE t1 (id INT, g1 GEOMETRY, g2 GEOMETRY);
 
diff --git a/test/sql/index/rtree_multiple_index.test b/test/sql/index/rtree_multiple_index.test
index d330399..fe4356c 100644
--- a/test/sql/index/rtree_multiple_index.test
+++ b/test/sql/index/rtree_multiple_index.test
@@ -1,6 +1,9 @@
 require spatial
 
 # Create a table with 10_000 random points, where the table also has an ART primary key index
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE IF NOT EXISTS t1 (
             id          LONG PRIMARY KEY,
diff --git a/test/sql/index/rtree_persistence.test_slow b/test/sql/index/rtree_persistence.test_slow
index f2b2e2d..fef5d9b 100644
--- a/test/sql/index/rtree_persistence.test_slow
+++ b/test/sql/index/rtree_persistence.test_slow
@@ -3,6 +3,9 @@ require spatial
 # Create a persistent database
 load {TEST_DIR}/rtree_persistence_test.db
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE t1 AS SELECT point::GEOMETRY as geom
 FROM st_generatepoints({min_x: 0, min_y: 0, max_x: 10000, max_y: 10000}::BOX_2D, 1_000_000, 1337);
@@ -22,6 +25,9 @@ physical_plan	<REGEX>:.*RTREE_INDEX_SCAN.*
 
 restart
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 # We should still have the index
 query II
 EXPLAIN SELECT count(*) FROM t1 WHERE ST_Within(geom, ST_MakeEnvelope(450, 450, 650, 650));
diff --git a/test/sql/index/rtree_persistence_load.test b/test/sql/index/rtree_persistence_load.test
index 1cc9788..4da1f16 100644
--- a/test/sql/index/rtree_persistence_load.test
+++ b/test/sql/index/rtree_persistence_load.test
@@ -3,6 +3,9 @@ require spatial
 # Create a persistent database
 load {TEST_DIR}/rtree_persistence_test.db
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE t1 AS SELECT point::GEOMETRY as geom
 FROM st_generatepoints({min_x: 0, min_y: 0, max_x: 10000, max_y: 10000}::BOX_2D, 100_000, 1337);
@@ -22,6 +25,9 @@ physical_plan	<REGEX>:.*RTREE_INDEX_SCAN.*
 
 restart no_extension_load
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 # Try to modify the table
 statement error
 DELETE FROM t1 WHERE rowid < 1000;
@@ -30,6 +36,9 @@ Missing Extension Error: Cannot bind index 't1', unknown index type 'RTREE'. You
 
 restart
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 # We should still have the index
 query II
 EXPLAIN SELECT count(*) FROM t1 WHERE ST_Within(geom, ST_MakeEnvelope(450, 450, 650, 650));
diff --git a/test/sql/index/rtree_persistence_wal.test b/test/sql/index/rtree_persistence_wal.test
index 542809f..18bed8b 100644
--- a/test/sql/index/rtree_persistence_wal.test
+++ b/test/sql/index/rtree_persistence_wal.test
@@ -3,6 +3,9 @@ require spatial
 # Create a persistent database
 load {TEST_DIR}/rtree_persistence_wal_test.db
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 PRAGMA disable_checkpoint_on_shutdown;
 
@@ -25,6 +28,9 @@ physical_plan	<REGEX>:.*RTREE_INDEX_SCAN.*
 
 restart
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 # We should still have the index
 query II
 EXPLAIN SELECT count(*) FROM t1 WHERE ST_Within(geom, ST_MakeEnvelope(450, 450, 650, 650));
diff --git a/test/sql/index/rtree_pushdown.test b/test/sql/index/rtree_pushdown.test
index 5e0f7ee..bfbf618 100644
--- a/test/sql/index/rtree_pushdown.test
+++ b/test/sql/index/rtree_pushdown.test
@@ -1,5 +1,8 @@
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE t1 (geom GEOMETRY, id INT);
 
diff --git a/test/sql/join/spatial_join_extra_pred.test b/test/sql/join/spatial_join_extra_pred.test
index 4c8c254..374ff1b 100644
--- a/test/sql/join/spatial_join_extra_pred.test
+++ b/test/sql/join/spatial_join_extra_pred.test
@@ -1,5 +1,8 @@
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE t1 AS
 SELECT
diff --git a/test/sql/join/spatial_join_outer.test_slow b/test/sql/join/spatial_join_outer.test_slow
index 67a9e2c..4439685 100644
--- a/test/sql/join/spatial_join_outer.test_slow
+++ b/test/sql/join/spatial_join_outer.test_slow
@@ -2,6 +2,9 @@ require spatial
 
 # Create a bunch of points on a grid
 # between 0 and 500 in both x and y directions
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE lhs AS
 SELECT
diff --git a/test/sql/join/spatial_join_predicates.test b/test/sql/join/spatial_join_predicates.test
index 4f9ad8f..a38c129 100644
--- a/test/sql/join/spatial_join_predicates.test
+++ b/test/sql/join/spatial_join_predicates.test
@@ -1,6 +1,9 @@
 require spatial
 
 # Set up test tables
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 CREATE TABLE lhs AS
 SELECT
diff --git a/test/sql/join/spatial_join_test.test_slow b/test/sql/join/spatial_join_test.test_slow
index 58e41c3..f1f3283 100644
--- a/test/sql/join/spatial_join_test.test_slow
+++ b/test/sql/join/spatial_join_test.test_slow
@@ -1,5 +1,8 @@
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 create table t1(g GEOMETRY);
 
diff --git a/test/sql/join/spatial_join_test_2.test_slow b/test/sql/join/spatial_join_test_2.test_slow
index 720eda1..c01c550 100644
--- a/test/sql/join/spatial_join_test_2.test_slow
+++ b/test/sql/join/spatial_join_test_2.test_slow
@@ -1,5 +1,8 @@
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 SELECT setseed(0.5);
 
diff --git a/test/sql/join/spatial_join_test_3.test_slow b/test/sql/join/spatial_join_test_3.test_slow
index 35b820b..4d92e31 100644
--- a/test/sql/join/spatial_join_test_3.test_slow
+++ b/test/sql/join/spatial_join_test_3.test_slow
@@ -1,5 +1,8 @@
 require spatial
 
+statement ok
+SET profiling_renderer_settings = MAP {'operator_casing': 'upper'};
+
 statement ok
 SELECT setseed(0.5);
 
