diff --git a/spatial/include/spatial/core/util/managed_collection.hpp b/spatial/include/spatial/core/util/managed_collection.hpp
index 3535aff..dbf21f1 100644
--- a/spatial/include/spatial/core/util/managed_collection.hpp
+++ b/spatial/include/spatial/core/util/managed_collection.hpp
@@ -95,7 +95,8 @@ void ManagedCollection<T>::InitializeAppend(ManagedCollectionAppendState &state,
 		state.block = &blocks.back();
 		state.block->item_count = 0;
 		state.block->item_capacity = block_capacity;
-		state.handle = manager.Allocate(MemoryTag::EXTENSION, block_size, true, &state.block->handle);
+		state.handle = manager.Allocate(MemoryTag::EXTENSION, block_size, true);
+		state.block->handle = state.handle.GetBlockHandle();
 	}
 }
 
@@ -108,7 +109,8 @@ void ManagedCollection<T>::Append(ManagedCollectionAppendState &state, const T *
 			state.block = &blocks.back();
 			state.block->item_count = 0;
 			state.block->item_capacity = block_capacity;
-			state.handle = manager.Allocate(MemoryTag::EXTENSION, block_size, true, &state.block->handle);
+			state.handle = manager.Allocate(MemoryTag::EXTENSION, block_size, true);
+			state.block->handle = state.handle.GetBlockHandle();
 		}
 
 		// Compute how much we can copy before the block is full or we run out of elements
