diff --git a/spatial/src/spatial/gdal/functions/st_write.cpp b/spatial/src/spatial/gdal/functions/st_write.cpp
index 266a52a..15ebcf4 100644
--- a/spatial/src/spatial/gdal/functions/st_write.cpp
+++ b/spatial/src/spatial/gdal/functions/st_write.cpp
@@ -55,14 +55,9 @@ struct GlobalState : public GlobalFunctionData {
 //===--------------------------------------------------------------------===//
 // Bind
 //===--------------------------------------------------------------------===//
-// The parameters are const in duckdb > 0.9.1, ifdef so we can build for both versions for now.
-#if DUCKDB_PATCH_VERSION == 1
-static unique_ptr<FunctionData> Bind(ClientContext &context, CopyInfo &info, vector<string> &names,
-                                     vector<LogicalType> &sql_types) {
-#else
 static unique_ptr<FunctionData> Bind(ClientContext &context, const CopyInfo &info, const vector<string> &names,
                                      const vector<LogicalType> &sql_types) {
-#endif
+
 	GdalFileHandler::SetLocalClientContext(context);
 
 	auto bind_data = make_uniq<BindData>(info.file_path, sql_types, names);
