diff --git a/spatial/src/spatial/gdal/file_handler.cpp b/spatial/src/spatial/gdal/file_handler.cpp
index cb50144..3fe1abf 100644
--- a/spatial/src/spatial/gdal/file_handler.cpp
+++ b/spatial/src/spatial/gdal/file_handler.cpp
@@ -385,10 +385,8 @@ string GDALClientContextState::GetPrefix(const string &value) const {
 }
 
 GDALClientContextState &GDALClientContextState::GetOrCreate(ClientContext &context) {
-	if (!context.registered_state["gdal"]) {
-		context.registered_state["gdal"] = make_uniq<GDALClientContextState>(context);
-	}
-	return *dynamic_cast<GDALClientContextState *>(context.registered_state["gdal"].get());
+	auto gdal_state = context.registered_state->GetOrCreate<GDALClientContextState>("gdal", context);
+	return *gdal_state;
 }
 
 } // namespace gdal
