diff --git a/spatial/src/spatial/gdal/functions/st_write.cpp b/spatial/src/spatial/gdal/functions/st_write.cpp
index a9fb04a..667bac4 100644
--- a/spatial/src/spatial/gdal/functions/st_write.cpp
+++ b/spatial/src/spatial/gdal/functions/st_write.cpp
@@ -104,7 +104,8 @@ static unique_ptr<FunctionData> Bind(ClientContext &context, CopyInfo &info, vec
 
 	if (bind_data->layer_name.empty()) {
 		// Default to the base name of the file
-		bind_data->layer_name = FileSystem::ExtractBaseName(bind_data->file_path);
+		auto& fs = FileSystem::GetFileSystem(context);
+		bind_data->layer_name = fs.ExtractBaseName(bind_data->file_path);
 	}
 
 	return std::move(bind_data);
@@ -413,4 +414,4 @@ void GdalCopyFunction::Register(ClientContext &context) {
 
 } // namespace gdal
 
-} // namespace spatial
\ No newline at end of file
+} // namespace spatial
