diff --git a/src/spatial/index/rtree/rtree_index.cpp b/src/spatial/index/rtree/rtree_index.cpp
index cc81600..b1f2b03 100644
--- a/src/spatial/index/rtree/rtree_index.cpp
+++ b/src/spatial/index/rtree/rtree_index.cpp
@@ -231,11 +231,11 @@ void RTreeIndex::Delete(IndexLock &lock, DataChunk &input, Vector &rowid_vec) {
 	}
 }
 
-IndexStorageInfo RTreeIndex::SerializeToDisk(QueryContext context, const case_insensitive_map_t<Value> &options) {
+unique_ptr<IndexStorageInfo> RTreeIndex::SerializeToDisk(QueryContext context, const case_insensitive_map_t<Value> &options) {
 
-	IndexStorageInfo info;
-	info.name = name;
-	info.root = tree->GetRoot().pointer.Get();
+	auto info = make_uniq<IndexStorageInfo>();
+	info->name = name;
+	info->root = tree->GetRoot().pointer.Get();
 
 	auto &leaf_allocator = tree->GetLeafAllocator();
 	auto &node_allocator = tree->GetNodeAllocator();
@@ -250,17 +250,17 @@ IndexStorageInfo RTreeIndex::SerializeToDisk(QueryContext context, const case_in
 	node_allocator.SerializeBuffers(partial_block_manager);
 	partial_block_manager.FlushPartialBlocks();
 
-	info.allocator_infos.push_back(leaf_allocator.GetInfo());
-	info.allocator_infos.push_back(node_allocator.GetInfo());
+	info->allocator_infos.push_back(leaf_allocator.GetInfo());
+	info->allocator_infos.push_back(node_allocator.GetInfo());
 
 	return info;
 }
 
-IndexStorageInfo RTreeIndex::SerializeToWAL(const case_insensitive_map_t<Value> &options) {
+unique_ptr<IndexStorageInfo> RTreeIndex::SerializeToWAL(const case_insensitive_map_t<Value> &options) {
 
-	IndexStorageInfo info;
-	info.name = name;
-	info.root = tree->GetRoot().pointer.Get();
+	auto info = make_uniq<IndexStorageInfo>();
+	info->name = name;
+	info->root = tree->GetRoot().pointer.Get();
 
 	auto &leaf_allocator = tree->GetLeafAllocator();
 	auto &node_allocator = tree->GetNodeAllocator();
@@ -268,11 +268,11 @@ IndexStorageInfo RTreeIndex::SerializeToWAL(const case_insensitive_map_t<Value>
 	leaf_allocator.RemoveEmptyBuffers();
 	node_allocator.RemoveEmptyBuffers();
 
-	info.buffers.push_back(leaf_allocator.InitSerializationToWAL());
-	info.buffers.push_back(node_allocator.InitSerializationToWAL());
+	info->buffers.push_back(leaf_allocator.InitSerializationToWAL());
+	info->buffers.push_back(node_allocator.InitSerializationToWAL());
 
-	info.allocator_infos.push_back(leaf_allocator.GetInfo());
-	info.allocator_infos.push_back(node_allocator.GetInfo());
+	info->allocator_infos.push_back(leaf_allocator.GetInfo());
+	info->allocator_infos.push_back(node_allocator.GetInfo());
 
 	return info;
 }
diff --git a/src/spatial/index/rtree/rtree_index.hpp b/src/spatial/index/rtree/rtree_index.hpp
index 202a6c7..8cee03f 100644
--- a/src/spatial/index/rtree/rtree_index.hpp
+++ b/src/spatial/index/rtree/rtree_index.hpp
@@ -48,9 +48,9 @@ public:
 	ErrorData Insert(IndexLock &lock, DataChunk &data, Vector &row_ids) override;
 
 	//! Serializes RTree memory to disk and returns the index storage information.
-	IndexStorageInfo SerializeToDisk(QueryContext context, const case_insensitive_map_t<Value> &options) override;
+	unique_ptr<IndexStorageInfo> SerializeToDisk(QueryContext context, const case_insensitive_map_t<Value> &options) override;
 	//! Serializes RTree memory to the WAL and returns the index storage information.
-	IndexStorageInfo SerializeToWAL(const case_insensitive_map_t<Value> &options) override;
+	unique_ptr<IndexStorageInfo> SerializeToWAL(const case_insensitive_map_t<Value> &options) override;
 
 	idx_t GetInMemorySize(IndexLock &state) override;
 
