diff --git a/src/spatial/operators/spatial_join_optimizer.cpp b/src/spatial/operators/spatial_join_optimizer.cpp
index e46d5ad..7c08e82 100644
--- a/src/spatial/operators/spatial_join_optimizer.cpp
+++ b/src/spatial/operators/spatial_join_optimizer.cpp
@@ -167,7 +167,6 @@ static bool TrySwapComparisonJoin(OptimizerExtensionInput &input, unique_ptr<Log
 	spatial_join->types = std::move(cmp_join.types);
 	spatial_join->left_projection_map = std::move(cmp_join.left_projection_map);
 	spatial_join->right_projection_map = std::move(cmp_join.right_projection_map);
-	spatial_join->join_stats = std::move(cmp_join.join_stats);
 	spatial_join->mark_index = cmp_join.mark_index;
 	spatial_join->has_estimated_cardinality = cmp_join.has_estimated_cardinality;
 	spatial_join->estimated_cardinality = cmp_join.estimated_cardinality;
@@ -274,7 +273,6 @@ static void TrySwapAnyJoin(OptimizerExtensionInput &input, unique_ptr<LogicalOpe
 	spatial_join->types = std::move(any_join.types);
 	spatial_join->left_projection_map = std::move(any_join.left_projection_map);
 	spatial_join->right_projection_map = std::move(any_join.right_projection_map);
-	spatial_join->join_stats = std::move(any_join.join_stats);
 	spatial_join->mark_index = any_join.mark_index;
 	spatial_join->has_estimated_cardinality = any_join.has_estimated_cardinality;
 	spatial_join->estimated_cardinality = any_join.estimated_cardinality;
