diff --git a/spatial/src/spatial/core/io/shapefile/read_shapefile_meta.cpp b/spatial/src/spatial/core/io/shapefile/read_shapefile_meta.cpp
index da3fe1e..aa919d2 100644
--- a/spatial/src/spatial/core/io/shapefile/read_shapefile_meta.cpp
+++ b/spatial/src/spatial/core/io/shapefile/read_shapefile_meta.cpp
@@ -47,8 +47,9 @@ static ShapeTypeEntry shape_type_map[] = {
 static unique_ptr<FunctionData> ShapeFileMetaBind(ClientContext &context, TableFunctionBindInput &input,
                                                   vector<LogicalType> &return_types, vector<string> &names) {
 	auto result = make_uniq<ShapeFileMetaBindData>();
-	auto files = MultiFileReader::GetFileList(context, input.inputs[0], "ShapeFiles", FileGlobOptions::ALLOW_EMPTY);
-	for (auto &file : files) {
+	auto file_list = MultiFileReader().GetFileList(context, input.inputs[0], "ShapeFiles", FileGlobOptions::ALLOW_EMPTY);
+
+	for (auto &file : file_list->GetAllFiles()) {
 		if (StringUtil::EndsWith(StringUtil::Lower(file), ".shp")) {
 			result->files.push_back(file);
 		}
diff --git a/spatial/src/spatial/gdal/functions/st_read_meta.cpp b/spatial/src/spatial/gdal/functions/st_read_meta.cpp
index 2293072..bcfa747 100644
--- a/spatial/src/spatial/gdal/functions/st_read_meta.cpp
+++ b/spatial/src/spatial/gdal/functions/st_read_meta.cpp
@@ -61,7 +61,7 @@ static unique_ptr<FunctionData> Bind(ClientContext &context, TableFunctionBindIn
 	auto result = make_uniq<GDALMetadataBindData>();
 
 	result->file_names =
-	    MultiFileReader::GetFileList(context, input.inputs[0], "gdal metadata", FileGlobOptions::ALLOW_EMPTY);
+	    MultiFileReader().GetFileList(context, input.inputs[0], "gdal metadata", FileGlobOptions::ALLOW_EMPTY)->ToStringVector();
 
 	names.push_back("file_name");
 	return_types.push_back(LogicalType::VARCHAR);
