diff --git a/test/sql/geos/st_maximuminscribedcircle.test b/test/sql/geos/st_maximuminscribedcircle.test
index 386afe2..0a651b7 100644
--- a/test/sql/geos/st_maximuminscribedcircle.test
+++ b/test/sql/geos/st_maximuminscribedcircle.test
@@ -1,11 +1,11 @@
 require spatial
 
 query I
-SELECT st_maximuminscribedcircle('POLYGON((0 0, 1 0, 1 1, 0 1, 0 0))'::GEOMETRY);
+SELECT st_maximuminscribedcircle('POLYGON((0 0, 1 0, 1 1, 0 1, 0 0))'::GEOMETRY).radius;
 ----
-{'center': 'POINT (0.5 0.5)', 'nearest': 'POINT (0.5 0)', 'radius': 0.5}
+0.5
 
 statement error
 SELECT st_maximuminscribedcircle('POLYGON EMPTY'::GEOMETRY);
 ----
-IllegalArgumentException: Empty input geometry is not supported
+Invalid Input Error: IllegalArgumentException: Empty input is not supported
