diff --git a/src/spatial/operators/spatial_join_physical.cpp b/src/spatial/operators/spatial_join_physical.cpp
index 3038ed8..900894e 100644
--- a/src/spatial/operators/spatial_join_physical.cpp
+++ b/src/spatial/operators/spatial_join_physical.cpp
@@ -414,7 +414,7 @@ PhysicalSpatialJoin::PhysicalSpatialJoin(LogicalOperator &op, PhysicalOperator &
 	// Initialize the layout
 	// TODO: Align?
 	layout = make_shared_ptr<TupleDataLayout>();
-	layout->Initialize(std::move(layout_types), false);
+	layout->Initialize(std::move(layout_types), TupleDataValidityType::CAN_HAVE_NULL_VALUES);
 
 	// For right/outer joins, this is where the build side match column goes
 	if (PropagatesBuildSide(join_type)) {
@@ -1083,4 +1083,4 @@ ProgressData PhysicalSpatialJoin::GetProgress(ClientContext &context, GlobalSour
 	return res;
 }
 
-} // namespace duckdb
\ No newline at end of file
+} // namespace duckdb
