diff --git a/test/sql/storage/attach_on_conflict.test b/test/sql/storage/attach_on_conflict.test
index c2b515f..f3dfffb 100644
--- a/test/sql/storage/attach_on_conflict.test
+++ b/test/sql/storage/attach_on_conflict.test
@@ -18,10 +18,22 @@ INSERT INTO s.tbl VALUES (1)
 ----
 UNIQUE constraint failed
 
-statement error
+# ON CONFLICT is planned as a MERGE INTO - the conflicting row is skipped
+statement ok
 INSERT OR IGNORE INTO s.tbl VALUES (1)
+
+query I
+SELECT count(*) FROM s.tbl
+----
+3
+
+statement ok
+INSERT OR IGNORE INTO s.tbl VALUES (4)
+
+query I
+SELECT count(*) FROM s.tbl
 ----
-does not support
+4
 
 # INSERT OR IGNORE in a table without primary key constraints
 statement ok
