diff --git a/src/sqlite_scanner.cpp b/src/sqlite_scanner.cpp
index 91b7180..a002078 100644
--- a/src/sqlite_scanner.cpp
+++ b/src/sqlite_scanner.cpp
@@ -270,7 +270,7 @@ static void SqliteScan(ClientContext &context, TableFunctionInput &data, DataChu
 				auto &out_vec = output.data[col_idx];
 				auto sqlite_column_type = stmt.GetType(col_idx);
 				if (sqlite_column_type == SQLITE_NULL) {
-					auto &mask = FlatVector::Validity(out_vec);
+					auto &mask = FlatVector::ValidityMutable(out_vec);
 					mask.Set(out_idx, false);
 					continue;
 				}
@@ -279,26 +279,26 @@ static void SqliteScan(ClientContext &context, TableFunctionInput &data, DataChu
 				switch (out_vec.GetType().id()) {
 				case LogicalTypeId::BIGINT:
 					stmt.CheckTypeMatches(bind_data, val, sqlite_column_type, SQLITE_INTEGER, col_idx);
-					FlatVector::GetData<int64_t>(out_vec)[out_idx] = sqlite3_value_int64(val);
+					FlatVector::GetDataMutable<int64_t>(out_vec)[out_idx] = sqlite3_value_int64(val);
 					break;
 				case LogicalTypeId::DOUBLE:
 					stmt.CheckTypeIsFloatOrInteger(val, sqlite_column_type, col_idx);
-					FlatVector::GetData<double>(out_vec)[out_idx] = sqlite3_value_double(val);
+					FlatVector::GetDataMutable<double>(out_vec)[out_idx] = sqlite3_value_double(val);
 					break;
 				case LogicalTypeId::VARCHAR:
 					stmt.CheckTypeMatches(bind_data, val, sqlite_column_type, SQLITE_TEXT, col_idx);
-					FlatVector::GetData<string_t>(out_vec)[out_idx] = StringVector::AddString(
+					FlatVector::GetDataMutable<string_t>(out_vec)[out_idx] = StringVector::AddString(
 					    out_vec, (const char *)sqlite3_value_text(val), sqlite3_value_bytes(val));
 					break;
 				case LogicalTypeId::DATE:
 					if (sqlite_column_type == SQLITE_INTEGER) {
 						// unix timestamp
-						FlatVector::GetData<date_t>(out_vec)[out_idx] =
+						FlatVector::GetDataMutable<date_t>(out_vec)[out_idx] =
 						    Timestamp::GetDate(ConvertTimestampInteger(val));
 					} else if (sqlite_column_type == SQLITE_FLOAT) {
-						FlatVector::GetData<date_t>(out_vec)[out_idx] = Timestamp::GetDate(ConvertTimestampFloat(val));
+						FlatVector::GetDataMutable<date_t>(out_vec)[out_idx] = Timestamp::GetDate(ConvertTimestampFloat(val));
 					} else if (sqlite_column_type == SQLITE_TEXT) {
-						FlatVector::GetData<date_t>(out_vec)[out_idx] =
+						FlatVector::GetDataMutable<date_t>(out_vec)[out_idx] =
 						    Date::FromCString((const char *)sqlite3_value_text(val), sqlite3_value_bytes(val));
 					} else {
 						throw NotImplementedException("Unimplemented SQLite type for column of type DATE\n* SET "
@@ -317,12 +317,12 @@ static void SqliteScan(ClientContext &context, TableFunctionInput &data, DataChu
 					// timestamps
 					if (sqlite_column_type == SQLITE_INTEGER) {
 						// unix timestamp
-						FlatVector::GetData<timestamp_t>(out_vec)[out_idx] = ConvertTimestampInteger(val);
+						FlatVector::GetDataMutable<timestamp_t>(out_vec)[out_idx] = ConvertTimestampInteger(val);
 					} else if (sqlite_column_type == SQLITE_FLOAT) {
-						FlatVector::GetData<timestamp_t>(out_vec)[out_idx] = ConvertTimestampFloat(val);
+						FlatVector::GetDataMutable<timestamp_t>(out_vec)[out_idx] = ConvertTimestampFloat(val);
 					} else if (sqlite_column_type == SQLITE_TEXT) {
 						// ISO-8601
-						FlatVector::GetData<timestamp_t>(out_vec)[out_idx] =
+						FlatVector::GetDataMutable<timestamp_t>(out_vec)[out_idx] =
 						    Timestamp::FromCString((const char *)sqlite3_value_text(val), sqlite3_value_bytes(val));
 					} else {
 						throw NotImplementedException("Unimplemented SQLite type for column of type TIMESTAMP\n* SET "
@@ -331,7 +331,7 @@ static void SqliteScan(ClientContext &context, TableFunctionInput &data, DataChu
 					}
 					break;
 				case LogicalTypeId::BLOB:
-					FlatVector::GetData<string_t>(out_vec)[out_idx] = StringVector::AddStringOrBlob(
+					FlatVector::GetDataMutable<string_t>(out_vec)[out_idx] = StringVector::AddStringOrBlob(
 					    out_vec, (const char *)sqlite3_value_blob(val), sqlite3_value_bytes(val));
 					break;
 				default:
