diff --git a/src/storage/sqlite_delete.cpp b/src/storage/sqlite_delete.cpp
index a35686e..582b3ce 100644
--- a/src/storage/sqlite_delete.cpp
+++ b/src/storage/sqlite_delete.cpp
@@ -68,7 +68,7 @@ SourceResultType SQLiteDelete::GetDataInternal(ExecutionContext &context, DataCh
 	if (!insert_gstate.delete_done) {
 		if (!insert_gstate.statement.IsOpen()) {
 			auto &transaction = SQLiteTransaction::Get(context.client, insert_gstate.table.catalog);
-			insert_gstate.statement = transaction.GetDB().Prepare(GetDeleteSQL(insert_gstate.table.name));
+			insert_gstate.statement = transaction.GetDB().Prepare(GetDeleteSQL(insert_gstate.table.name.GetName()));
 		}
 		for (auto row_id : insert_gstate.rowids) {
 			insert_gstate.statement.Bind<int64_t>(0, row_id);
@@ -94,7 +94,7 @@ string SQLiteDelete::GetName() const {
 
 InsertionOrderPreservingMap<string> SQLiteDelete::ParamsToString() const {
 	InsertionOrderPreservingMap<string> result;
-	result["Table Name"] = table.name;
+	result["Table Name"] = table.name.GetName();
 	return result;
 }
 
diff --git a/src/storage/sqlite_index_entry.cpp b/src/storage/sqlite_index_entry.cpp
index 8f92089..d9cce68 100644
--- a/src/storage/sqlite_index_entry.cpp
+++ b/src/storage/sqlite_index_entry.cpp
@@ -9,7 +9,7 @@ SQLiteIndexEntry::SQLiteIndexEntry(Catalog &catalog, SchemaCatalogEntry &schema,
 }
 
 string SQLiteIndexEntry::GetSchemaName() const {
-	return schema.name;
+	return schema.name.GetName();
 }
 
 string SQLiteIndexEntry::GetTableName() const {
diff --git a/src/storage/sqlite_insert.cpp b/src/storage/sqlite_insert.cpp
index 44502ef..1108f95 100644
--- a/src/storage/sqlite_insert.cpp
+++ b/src/storage/sqlite_insert.cpp
@@ -40,7 +40,7 @@ public:
 
 string GetInsertSQL(const SQLiteInsert &insert, SQLiteTableEntry *entry) {
 	string result;
-	result = "INSERT INTO " + KeywordHelper::WriteOptionallyQuoted(entry->name);
+	result = "INSERT INTO " + KeywordHelper::WriteOptionallyQuoted(entry->name.GetName());
 	auto &columns = entry->GetColumns();
 	idx_t column_count;
 	if (!insert.column_index_map.empty()) {
@@ -136,7 +136,7 @@ string SQLiteInsert::GetName() const {
 
 InsertionOrderPreservingMap<string> SQLiteInsert::ParamsToString() const {
 	InsertionOrderPreservingMap<string> result;
-	result["Table Name"] = table ? table->name : info->Base().table;
+	result["Table Name"] = table ? table->name.GetName() : info->Base().table.GetName();
 	return result;
 }
 
diff --git a/src/storage/sqlite_schema_entry.cpp b/src/storage/sqlite_schema_entry.cpp
index e43f0f9..ce8adef 100644
--- a/src/storage/sqlite_schema_entry.cpp
+++ b/src/storage/sqlite_schema_entry.cpp
@@ -37,7 +37,7 @@ string GetCreateTableSQL(CreateTableInfo &info) {
 	if (info.on_conflict == OnCreateConflict::IGNORE_ON_CONFLICT) {
 		ss << "IF NOT EXISTS ";
 	}
-	ss << KeywordHelper::WriteOptionallyQuoted(info.table);
+	ss << KeywordHelper::WriteOptionallyQuoted(info.table.GetName());
 	ss << TableCatalogEntry::ColumnsToSQL(info.columns, info.constraints);
 	ss << ";";
 	return ss.str();
@@ -58,11 +58,11 @@ optional_ptr<CatalogEntry> SQLiteSchemaEntry::CreateTable(CatalogTransaction tra
 	auto table_name = base_info.table;
 	if (base_info.on_conflict == OnCreateConflict::REPLACE_ON_CONFLICT) {
 		// CREATE OR REPLACE - drop any existing entries first (if any)
-		TryDropEntry(transaction.GetContext(), CatalogType::TABLE_ENTRY, table_name);
+		TryDropEntry(transaction.GetContext(), CatalogType::TABLE_ENTRY, table_name.GetName());
 	}
 
 	sqlite_transaction.GetDB().Execute(GetCreateTableSQL(base_info));
-	return GetEntry(transaction, CatalogType::TABLE_ENTRY, table_name);
+	return GetEntry(transaction, CatalogType::TABLE_ENTRY, table_name.GetName());
 }
 
 optional_ptr<CatalogEntry> SQLiteSchemaEntry::CreateFunction(CatalogTransaction transaction, CreateFunctionInfo &info) {
@@ -89,9 +89,9 @@ string GetCreateIndexSQL(CreateIndexInfo &info, TableCatalogEntry &tbl) {
 	if (info.on_conflict == OnCreateConflict::IGNORE_ON_CONFLICT) {
 		sql += " IF NOT EXISTS ";
 	}
-	sql += KeywordHelper::WriteOptionallyQuoted(info.index_name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.index_name.GetName());
 	sql += " ON ";
-	sql += KeywordHelper::WriteOptionallyQuoted(tbl.name);
+	sql += KeywordHelper::WriteOptionallyQuoted(tbl.name.GetName());
 	sql += "(";
 	for (idx_t i = 0; i < info.parsed_expressions.size(); i++) {
 		if (i > 0) {
@@ -117,7 +117,7 @@ string GetCreateViewSQL(CreateViewInfo &info) {
 	if (info.on_conflict == OnCreateConflict::IGNORE_ON_CONFLICT) {
 		sql += "IF NOT EXISTS ";
 	}
-	sql += KeywordHelper::WriteOptionallyQuoted(info.view_name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.view_name.GetName());
 	sql += " ";
 	if (!info.aliases.empty()) {
 		sql += "(";
@@ -142,11 +142,11 @@ optional_ptr<CatalogEntry> SQLiteSchemaEntry::CreateView(CatalogTransaction tran
 	}
 	if (info.on_conflict == OnCreateConflict::REPLACE_ON_CONFLICT) {
 		// CREATE OR REPLACE - drop any existing entries first (if any)
-		TryDropEntry(transaction.GetContext(), CatalogType::VIEW_ENTRY, info.view_name);
+		TryDropEntry(transaction.GetContext(), CatalogType::VIEW_ENTRY, info.view_name.GetName());
 	}
 	auto &sqlite_transaction = GetSQLiteTransaction(transaction);
 	sqlite_transaction.GetDB().Execute(GetCreateViewSQL(info));
-	return GetEntry(transaction, CatalogType::VIEW_ENTRY, info.view_name);
+	return GetEntry(transaction, CatalogType::VIEW_ENTRY, info.view_name.GetName());
 }
 
 optional_ptr<CatalogEntry> SQLiteSchemaEntry::CreateSequence(CatalogTransaction transaction, CreateSequenceInfo &info) {
@@ -179,30 +179,30 @@ optional_ptr<CatalogEntry> SQLiteSchemaEntry::CreateType(CatalogTransaction tran
 
 void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, RenameTableInfo &info) {
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.name.GetName());
 	sql += " RENAME TO ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.new_table_name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.new_table_name.GetName());
 	sqlite_transaction.GetDB().Execute(sql);
 }
 
 void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, RenameColumnInfo &info) {
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.name.GetName());
 	sql += " RENAME COLUMN  ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.old_name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.old_name.GetName());
 	sql += " TO ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.new_name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.new_name.GetName());
 	sqlite_transaction.GetDB().Execute(sql);
 }
 
 void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, AddColumnInfo &info) {
 	if (info.if_column_not_exists) {
-		if (sqlite_transaction.GetDB().ColumnExists(info.name, info.new_column.GetName())) {
+		if (sqlite_transaction.GetDB().ColumnExists(info.name.GetName(), info.new_column.GetName())) {
 			return;
 		}
 	}
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.name.GetName());
 	sql += " ADD COLUMN  ";
 	sql += KeywordHelper::WriteOptionallyQuoted(info.new_column.Name());
 	sql += " ";
@@ -212,14 +212,14 @@ void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, AddCol
 
 void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, RemoveColumnInfo &info) {
 	if (info.if_column_exists) {
-		if (!sqlite_transaction.GetDB().ColumnExists(info.name, info.removed_column)) {
+		if (!sqlite_transaction.GetDB().ColumnExists(info.name.GetName(), info.removed_column.GetName())) {
 			return;
 		}
 	}
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.name.GetName());
 	sql += " DROP COLUMN  ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.removed_column);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.removed_column.GetName());
 	sqlite_transaction.GetDB().Execute(sql);
 }
 
@@ -247,7 +247,7 @@ void SQLiteSchemaEntry::Alter(CatalogTransaction catalog_transaction, AlterInfo
 		                      "support RENAME TABLE, RENAME COLUMN, "
 		                      "ADD COLUMN and DROP COLUMN");
 	}
-	transaction.ClearTableEntry(info.name);
+	transaction.ClearTableEntry(info.name.GetName());
 }
 
 void SQLiteSchemaEntry::Scan(ClientContext &context, CatalogType type,
@@ -294,7 +294,7 @@ void SQLiteSchemaEntry::DropEntry(ClientContext &context, DropInfo &info) {
 		throw InternalException("Failed to drop entry \"%s\" - could not find entry", info.name);
 	}
 	auto &transaction = SQLiteTransaction::Get(context, catalog);
-	transaction.DropEntry(info.type, info.name, info.cascade);
+	transaction.DropEntry(info.type, info.name.GetName(), info.cascade);
 }
 
 optional_ptr<CatalogEntry> SQLiteSchemaEntry::LookupEntry(CatalogTransaction transaction,
diff --git a/src/storage/sqlite_table_entry.cpp b/src/storage/sqlite_table_entry.cpp
index f7c0316..e338167 100644
--- a/src/storage/sqlite_table_entry.cpp
+++ b/src/storage/sqlite_table_entry.cpp
@@ -28,13 +28,13 @@ TableFunction SQLiteTableEntry::GetScanFunction(ClientContext &context, unique_p
 	}
 	auto &sqlite_catalog = catalog.Cast<SQLiteCatalog>();
 	result->file_name = sqlite_catalog.path;
-	result->table_name = name;
+	result->table_name = name.GetName();
 	result->all_varchar = all_varchar;
 
 	auto &transaction = Transaction::Get(context, catalog).Cast<SQLiteTransaction>();
 	auto &db = transaction.GetDB();
 
-	if (!db.GetRowIdInfo(name, result->row_id_info)) {
+	if (!db.GetRowIdInfo(name.GetName(), result->row_id_info)) {
 		result->rows_per_group = optional_idx();
 	}
 
@@ -71,14 +71,14 @@ TableStorageInfo SQLiteTableEntry::GetStorageInfo(ClientContext &context) {
 	TableStorageInfo result;
 
 	RowIdInfo info;
-	if (!db.GetRowIdInfo(name, info)) {
+	if (!db.GetRowIdInfo(name.GetName(), info)) {
 		// probably
 		result.cardinality = 10000;
 	} else {
 		result.cardinality = info.max_rowid.GetIndex() - info.min_rowid.GetIndex();
 	}
 
-	result.index_info = db.GetIndexInfo(name);
+	result.index_info = db.GetIndexInfo(name.GetName());
 	return result;
 }
 
diff --git a/src/storage/sqlite_update.cpp b/src/storage/sqlite_update.cpp
index 29ab390..99a87ff 100644
--- a/src/storage/sqlite_update.cpp
+++ b/src/storage/sqlite_update.cpp
@@ -29,7 +29,7 @@ public:
 
 string GetUpdateSQL(SQLiteTableEntry &table, const vector<PhysicalIndex> &index) {
 	string result;
-	result = "UPDATE " + KeywordHelper::WriteOptionallyQuoted(table.name);
+	result = "UPDATE " + KeywordHelper::WriteOptionallyQuoted(table.name.GetName());
 	result += " SET ";
 	for (idx_t i = 0; i < index.size(); i++) {
 		if (i > 0) {
@@ -99,7 +99,7 @@ string SQLiteUpdate::GetName() const {
 
 InsertionOrderPreservingMap<string> SQLiteUpdate::ParamsToString() const {
 	InsertionOrderPreservingMap<string> result;
-	result["Table Name"] = table.name;
+	result["Table Name"] = table.name.GetName();
 	return result;
 }
 
