diff --git a/src/include/storage/sqlite_table_entry.hpp b/src/include/storage/sqlite_table_entry.hpp
index b08319b..812381d 100644
--- a/src/include/storage/sqlite_table_entry.hpp
+++ b/src/include/storage/sqlite_table_entry.hpp
@@ -19,6 +19,8 @@ public:
 	bool all_varchar;
 
 public:
+	const ColumnList &GetColumns() const override;
+
 	unique_ptr<BaseStatistics> GetStatistics(ClientContext &context, column_t column_id) override;
 
 	TableFunction GetScanFunction(ClientContext &context, unique_ptr<FunctionData> &bind_data) override;
@@ -27,6 +29,9 @@ public:
 
 	void BindUpdateConstraints(Binder &binder, LogicalGet &get, LogicalProjection &proj, LogicalUpdate &update,
 	                           ClientContext &context) override;
+
+protected:
+	ColumnList columns;
 };
 
 } // namespace duckdb
diff --git a/src/storage/sqlite_table_entry.cpp b/src/storage/sqlite_table_entry.cpp
index 3256ffe..7d88105 100644
--- a/src/storage/sqlite_table_entry.cpp
+++ b/src/storage/sqlite_table_entry.cpp
@@ -2,6 +2,7 @@
 #include "storage/sqlite_table_entry.hpp"
 #include "storage/sqlite_transaction.hpp"
 #include "duckdb/storage/statistics/base_statistics.hpp"
+#include "duckdb/parser/parsed_data/create_table_info.hpp"
 #include "sqlite_scanner.hpp"
 #include "duckdb/storage/table_storage_info.hpp"
 
@@ -9,7 +10,11 @@ namespace duckdb {
 
 SQLiteTableEntry::SQLiteTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, CreateTableInfo &info,
                                    bool all_varchar)
-    : TableCatalogEntry(catalog, schema, info), all_varchar(all_varchar) {
+    : TableCatalogEntry(catalog, schema, info), all_varchar(all_varchar), columns(std::move(info.columns)) {
+}
+
+const ColumnList &SQLiteTableEntry::GetColumns() const {
+	return columns;
 }
 
 unique_ptr<BaseStatistics> SQLiteTableEntry::GetStatistics(ClientContext &context, column_t column_id) {
