diff --git a/test/sql/storage/attach_update.test b/test/sql/storage/attach_update.test
index b7db911..09b4fd7 100644
--- a/test/sql/storage/attach_update.test
+++ b/test/sql/storage/attach_update.test
@@ -50,10 +50,8 @@ SELECT rowid, * FROM s1.test
 4	NULL
 
 # update with DEFAULT clause
-statement error
+statement ok
 UPDATE s1.test SET i = DEFAULT WHERE i = 4
-----
-not yet supported
 
 # multi column update in different orders
 statement ok
