diff --git a/src/storage/sqlite_schema_entry.cpp b/src/storage/sqlite_schema_entry.cpp
index 5693fdd..8c1e028 100644
--- a/src/storage/sqlite_schema_entry.cpp
+++ b/src/storage/sqlite_schema_entry.cpp
@@ -46,7 +46,7 @@ string GetCreateTableSQL(CreateTableInfo &info) {
 void SQLiteSchemaEntry::TryDropEntry(ClientContext &context, CatalogType catalog_type, const string &name) {
 	DropInfo info;
 	info.type = catalog_type;
-	info.name = Identifier(name);
+	info.NameMutable() = Identifier(name);
 	info.cascade = false;
 	info.if_not_found = OnEntryNotFound::RETURN_NULL;
 	DropEntry(context, info);
@@ -179,7 +179,7 @@ optional_ptr<CatalogEntry> SQLiteSchemaEntry::CreateType(CatalogTransaction tran
 
 void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, RenameTableInfo &info) {
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name.GetIdentifierName());
+	sql += KeywordHelper::WriteOptionallyQuoted(info.Name().GetIdentifierName());
 	sql += " RENAME TO ";
 	sql += KeywordHelper::WriteOptionallyQuoted(info.new_table_name.GetIdentifierName());
 	sqlite_transaction.GetDB().Execute(sql);
@@ -187,7 +187,7 @@ void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, Rename
 
 void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, RenameColumnInfo &info) {
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name.GetIdentifierName());
+	sql += KeywordHelper::WriteOptionallyQuoted(info.Name().GetIdentifierName());
 	sql += " RENAME COLUMN  ";
 	sql += KeywordHelper::WriteOptionallyQuoted(info.old_name.GetIdentifierName());
 	sql += " TO ";
@@ -197,12 +197,12 @@ void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, Rename
 
 void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, AddColumnInfo &info) {
 	if (info.if_column_not_exists) {
-		if (sqlite_transaction.GetDB().ColumnExists(info.name.GetIdentifierName(), info.new_column.GetName().GetIdentifierName())) {
+		if (sqlite_transaction.GetDB().ColumnExists(info.Name().GetIdentifierName(), info.new_column.GetName().GetIdentifierName())) {
 			return;
 		}
 	}
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name.GetIdentifierName());
+	sql += KeywordHelper::WriteOptionallyQuoted(info.Name().GetIdentifierName());
 	sql += " ADD COLUMN  ";
 	sql += KeywordHelper::WriteOptionallyQuoted(info.new_column.Name().GetIdentifierName());
 	sql += " ";
@@ -212,12 +212,12 @@ void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, AddCol
 
 void SQLiteSchemaEntry::AlterTable(SQLiteTransaction &sqlite_transaction, RemoveColumnInfo &info) {
 	if (info.if_column_exists) {
-		if (!sqlite_transaction.GetDB().ColumnExists(info.name.GetIdentifierName(), info.removed_column.GetIdentifierName())) {
+		if (!sqlite_transaction.GetDB().ColumnExists(info.Name().GetIdentifierName(), info.removed_column.GetIdentifierName())) {
 			return;
 		}
 	}
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name.GetIdentifierName());
+	sql += KeywordHelper::WriteOptionallyQuoted(info.Name().GetIdentifierName());
 	sql += " DROP COLUMN  ";
 	sql += KeywordHelper::WriteOptionallyQuoted(info.removed_column.GetIdentifierName());
 	sqlite_transaction.GetDB().Execute(sql);
@@ -247,7 +247,7 @@ void SQLiteSchemaEntry::Alter(CatalogTransaction catalog_transaction, AlterInfo
 		                      "support RENAME TABLE, RENAME COLUMN, "
 		                      "ADD COLUMN and DROP COLUMN");
 	}
-	transaction.ClearTableEntry(info.name.GetIdentifierName());
+	transaction.ClearTableEntry(info.Name().GetIdentifierName());
 }
 
 void SQLiteSchemaEntry::Scan(ClientContext &context, CatalogType type,
@@ -286,15 +286,15 @@ void SQLiteSchemaEntry::DropEntry(ClientContext &context, DropInfo &info) {
 		throw BinderException("SQLite databases do not support dropping entries of type \"%s\"",
 		                      CatalogTypeToString(type));
 	}
-	auto table = GetEntry(GetCatalogTransaction(context), info.type, info.name);
+	auto table = GetEntry(GetCatalogTransaction(context), info.type, info.Name());
 	if (!table) {
 		if (info.if_not_found == OnEntryNotFound::RETURN_NULL) {
 			return;
 		}
-		throw InternalException("Failed to drop entry \"%s\" - could not find entry", info.name);
+		throw InternalException("Failed to drop entry \"%s\" - could not find entry", info.Name());
 	}
 	auto &transaction = SQLiteTransaction::Get(context, catalog);
-	transaction.DropEntry(info.type, info.name.GetIdentifierName(), info.cascade);
+	transaction.DropEntry(info.type, info.Name().GetIdentifierName(), info.cascade);
 }
 
 optional_ptr<CatalogEntry> SQLiteSchemaEntry::LookupEntry(CatalogTransaction transaction,
