diff --git a/test/sql/storage/attach_concurrent_clients.test b/test/sql/storage/attach_concurrent_clients.test
index 42cea50..cce5237 100644
--- a/test/sql/storage/attach_concurrent_clients.test
+++ b/test/sql/storage/attach_concurrent_clients.test
@@ -4,6 +4,8 @@
 
 require sqlite_scanner
 
+require nowindows
+
 statement ok
 ATTACH '__TEST_DIR__/attach_concurrentclients.db' AS s1 (TYPE SQLITE)
 
