--- a/src/sqlite_scanner.cpp
+++ b/src/sqlite_scanner.cpp
@@ -46,7 +46,7 @@ struct SqliteGlobalState : public GlobalTableFunctionState {
 };
 
 static unique_ptr<FunctionData> SqliteBind(ClientContext &context, TableFunctionBindInput &input,
-                                           vector<LogicalType> &return_types, vector<string> &names) {
+                                           vector<LogicalType> &return_types, vector<Identifier> &names) {
 
 	auto result = make_uniq<SqliteBindData>();
 	result->file_name = input.inputs[0].GetValue<string>();
@@ -80,7 +80,7 @@ static unique_ptr<FunctionData> SqliteBind(ClientContext &context, TableFunction
 		result->rows_per_group = optional_idx();
 	}
 
-	result->names = names;
+	result->names = IdentifiersToStrings(names);
 	result->types = return_types;
 
 	return std::move(result);
@@ -397,7 +397,7 @@ struct AttachFunctionData : public TableFunctionData {
 };
 
 static unique_ptr<FunctionData> AttachBind(ClientContext &context, TableFunctionBindInput &input,
-                                           vector<LogicalType> &return_types, vector<string> &names) {
+                                           vector<LogicalType> &return_types, vector<Identifier> &names) {
 
 	auto result = make_uniq<AttachFunctionData>();
 	result->file_name = input.inputs[0].GetValue<string>();
--- a/src/storage/sqlite_query.cpp
+++ b/src/storage/sqlite_query.cpp
@@ -14,7 +14,7 @@
 namespace duckdb {
 
 static unique_ptr<FunctionData> SQLiteQueryBind(ClientContext &context, TableFunctionBindInput &input,
-                                                vector<LogicalType> &return_types, vector<string> &names) {
+                                                vector<LogicalType> &return_types, vector<Identifier> &names) {
 	auto result = make_uniq<SqliteBindData>();
 
 	if (input.inputs[0].IsNull() || input.inputs[1].IsNull()) {
