diff --git a/test/sql/storage/attach_describe.test b/test/sql/storage/attach_describe.test
index d356bef..b92ebd6 100644
--- a/test/sql/storage/attach_describe.test
+++ b/test/sql/storage/attach_describe.test
@@ -13,7 +13,7 @@ CREATE TABLE s1.test(i BIGINT PRIMARY KEY, j BIGINT DEFAULT 42);
 query IIIIII
 DESCRIBE s1.test
 ----
-i	BIGINT	YES	NULL	NULL	NULL
+i	BIGINT	YES	PRI	NULL	NULL
 j	BIGINT	YES	NULL	42	NULL
 
 query ITTTTT
