diff --git a/test/sql/storage/attach_detach.test b/test/sql/storage/attach_detach.test
index 930dd45..febb1d7 100644
--- a/test/sql/storage/attach_detach.test
+++ b/test/sql/storage/attach_detach.test
@@ -18,6 +18,7 @@ DETACH s1
 
 statement error
 SELECT * FROM s1.test
+----
 
 
 statement ok
diff --git a/test/sql/storage/attach_drop.test b/test/sql/storage/attach_drop.test
index 0cf6aa5..c9cbf4a 100644
--- a/test/sql/storage/attach_drop.test
+++ b/test/sql/storage/attach_drop.test
@@ -19,6 +19,7 @@ DROP TABLE simple.test
 # verify the drop was successful
 statement error
 SELECT * FROM simple.test;
+----
 
 statement error
 DROP TABLE simple.testx
@@ -26,4 +27,4 @@ DROP TABLE simple.testx
 Table with name testx does not exist
 
 statement ok
-DROP TABLE IF EXISTS simple.testx
\ No newline at end of file
+DROP TABLE IF EXISTS simple.testx
