diff --git a/src/storage/sqlite_schema_entry.cpp b/src/storage/sqlite_schema_entry.cpp
index 3926af1..a8202ec 100644
--- a/src/storage/sqlite_schema_entry.cpp
+++ b/src/storage/sqlite_schema_entry.cpp
@@ -81,11 +81,11 @@ void UnqualifyColumnReferences(ParsedExpression &expr) {
 string GetCreateIndexSQL(CreateIndexInfo &info, TableCatalogEntry &tbl) {
 	string sql;
 	sql = "CREATE";
-	if (info.index_constraint_type == IndexConstraintType::UNIQUE) {
+	if (info.constraint_type == IndexConstraintType::UNIQUE) {
 		sql += " UNIQUE";
 	}
 	sql += " INDEX ";
-	sql += KeywordHelper::WriteOptionallyQuoted(info.name);
+	sql += KeywordHelper::WriteOptionallyQuoted(info.index_name);
 	sql += " ON ";
 	sql += KeywordHelper::WriteOptionallyQuoted(tbl.name);
 	sql += "(";
diff --git a/src/storage/sqlite_transaction.cpp b/src/storage/sqlite_transaction.cpp
index 99feb85..a655228 100644
--- a/src/storage/sqlite_transaction.cpp
+++ b/src/storage/sqlite_transaction.cpp
@@ -78,7 +78,7 @@ optional_ptr<CatalogEntry> SQLiteTransaction::GetCatalogEntry(const string &entr
 	}
 	case CatalogType::INDEX_ENTRY: {
 		CreateIndexInfo info;
-		info.name = entry_name;
+		info.index_name = entry_name;
 
 		string table_name;
 		string sql;
