diff --git a/src/sqlite_storage.cpp b/src/sqlite_storage.cpp
index d65bf18..ad72003 100644
--- a/src/sqlite_storage.cpp
+++ b/src/sqlite_storage.cpp
@@ -12,8 +12,9 @@
 
 namespace duckdb {
 
-static unique_ptr<Catalog> SQLiteAttach(StorageExtensionInfo *storage_info, AttachedDatabase &db, const string &name,
-                                        AttachInfo &info, AccessMode access_mode) {
+static unique_ptr<Catalog> SQLiteAttach(StorageExtensionInfo *storage_info, ClientContext &context,
+                                        AttachedDatabase &db, const string &name, AttachInfo &info,
+                                        AccessMode access_mode) {
 	return make_uniq<SQLiteCatalog>(db, info.path, access_mode);
 }
 
