diff --git a/src/include/storage/sqlite_delete.hpp b/src/include/storage/sqlite_delete.hpp
index 18e9b96..3c07df3 100644
--- a/src/include/storage/sqlite_delete.hpp
+++ b/src/include/storage/sqlite_delete.hpp
@@ -42,7 +42,7 @@ public:
 	}
 
 	string GetName() const override;
-	string ParamsToString() const override;
+	InsertionOrderPreservingMap<string> ParamsToString() const override;
 };
 
 } // namespace duckdb
diff --git a/src/include/storage/sqlite_insert.hpp b/src/include/storage/sqlite_insert.hpp
index d75f7ee..65ec357 100644
--- a/src/include/storage/sqlite_insert.hpp
+++ b/src/include/storage/sqlite_insert.hpp
@@ -51,7 +51,7 @@ public:
 	}
 
 	string GetName() const override;
-	string ParamsToString() const override;
+	InsertionOrderPreservingMap<string> ParamsToString() const override;
 };
 
 } // namespace duckdb
diff --git a/src/include/storage/sqlite_update.hpp b/src/include/storage/sqlite_update.hpp
index 88ab6e1..0061620 100644
--- a/src/include/storage/sqlite_update.hpp
+++ b/src/include/storage/sqlite_update.hpp
@@ -44,7 +44,7 @@ public:
 	}
 
 	string GetName() const override;
-	string ParamsToString() const override;
+	InsertionOrderPreservingMap<string> ParamsToString() const override;
 };
 
 } // namespace duckdb
diff --git a/src/storage/sqlite_delete.cpp b/src/storage/sqlite_delete.cpp
index 5510de1..201e70b 100644
--- a/src/storage/sqlite_delete.cpp
+++ b/src/storage/sqlite_delete.cpp
@@ -78,8 +78,10 @@ string SQLiteDelete::GetName() const {
 	return "DELETE";
 }
 
-string SQLiteDelete::ParamsToString() const {
-	return table.name;
+InsertionOrderPreservingMap<string> SQLiteDelete::ParamsToString() const {
+	InsertionOrderPreservingMap<string> result;
+	result["Table Name"] = table.name;
+	return result;
 }
 
 //===--------------------------------------------------------------------===//
diff --git a/src/storage/sqlite_insert.cpp b/src/storage/sqlite_insert.cpp
index 94de131..d25aa9f 100644
--- a/src/storage/sqlite_insert.cpp
+++ b/src/storage/sqlite_insert.cpp
@@ -132,8 +132,10 @@ string SQLiteInsert::GetName() const {
 	return table ? "INSERT" : "CREATE_TABLE_AS";
 }
 
-string SQLiteInsert::ParamsToString() const {
-	return table ? table->name : info->Base().table;
+InsertionOrderPreservingMap<string> SQLiteInsert::ParamsToString() const {
+	InsertionOrderPreservingMap<string> result;
+	result["Table Name"] = table ? table->name : info->Base().table;
+	return result;
 }
 
 //===--------------------------------------------------------------------===//
diff --git a/src/storage/sqlite_update.cpp b/src/storage/sqlite_update.cpp
index 8ccc03c..90a3934 100644
--- a/src/storage/sqlite_update.cpp
+++ b/src/storage/sqlite_update.cpp
@@ -94,8 +94,10 @@ string SQLiteUpdate::GetName() const {
 	return "UPDATE";
 }
 
-string SQLiteUpdate::ParamsToString() const {
-	return table.name;
+InsertionOrderPreservingMap<string> SQLiteUpdate::ParamsToString() const {
+	InsertionOrderPreservingMap<string> result;
+	result["Table Name"] = table.name;
+	return result;
 }
 
 //===--------------------------------------------------------------------===//
