diff --git a/src/storage/sqlite_index.cpp b/src/storage/sqlite_index.cpp
index c5f9d49..ebac185 100644
--- a/src/storage/sqlite_index.cpp
+++ b/src/storage/sqlite_index.cpp
@@ -38,7 +38,7 @@ public:
 		return make_uniq<SQLiteCreateIndex>(std::move(info), table);
 	}
 
-	void Serialize(FieldWriter &writer) const override {
+	void Serialize(Serializer &serializer) const override {
 		throw InternalException("Cannot serialize SQLite Create index");
 	}
 
