diff --git a/src/include/sqlite_scanner.hpp b/src/include/sqlite_scanner.hpp
index c6230a5..f79006e 100644
--- a/src/include/sqlite_scanner.hpp
+++ b/src/include/sqlite_scanner.hpp
@@ -13,6 +13,7 @@
 
 namespace duckdb {
 class SQLiteDB;
+class TableCatalogEntry;
 
 struct SqliteBindData : public TableFunctionData {
 	string file_name;
@@ -26,6 +27,8 @@ struct SqliteBindData : public TableFunctionData {
 
 	optional_idx rows_per_group = 122880;
 	SQLiteDB *global_db;
+
+	optional_ptr<TableCatalogEntry> table;
 };
 
 class SqliteScanFunction : public TableFunction {
diff --git a/src/sqlite_scanner.cpp b/src/sqlite_scanner.cpp
index 391d821..b4ac571 100644
--- a/src/sqlite_scanner.cpp
+++ b/src/sqlite_scanner.cpp
@@ -283,6 +283,13 @@ static string SqliteToString(const FunctionData *bind_data_p) {
 	return StringUtil::Format("%s:%s", bind_data.file_name, bind_data.table_name);
 }
 
+BindInfo SqliteBindInfo(const optional_ptr<FunctionData> bind_data_p) {
+	BindInfo info(ScanType::EXTERNAL);
+	auto &bind_data = bind_data_p->Cast<SqliteBindData>();
+	info.table = bind_data.table;
+	return info;
+}
+
 /*
 static unique_ptr<BaseStatistics>
 SqliteStatistics(ClientContext &context, const FunctionData *bind_data_p,
@@ -300,6 +307,7 @@ SqliteScanFunction::SqliteScanFunction()
                     SqliteInitGlobalState, SqliteInitLocalState) {
 	cardinality = SqliteCardinality;
 	to_string = SqliteToString;
+	get_bind_info = SqliteBindInfo;
 	projection_pushdown = true;
 }
 
diff --git a/src/storage/sqlite_table_entry.cpp b/src/storage/sqlite_table_entry.cpp
index 0a9f850..fd17c2d 100644
--- a/src/storage/sqlite_table_entry.cpp
+++ b/src/storage/sqlite_table_entry.cpp
@@ -44,6 +44,7 @@ TableFunction SQLiteTableEntry::GetScanFunction(ClientContext &context, unique_p
 		result->global_db = &db;
 		result->rows_per_group = optional_idx();
 	}
+	result->table = this;
 
 	bind_data = std::move(result);
 	return static_cast<TableFunction>(SqliteScanFunction());
