diff --git a/src/sqlite_scanner.cpp b/src/sqlite_scanner.cpp
index e5b50c3..919e808 100644
--- a/src/sqlite_scanner.cpp
+++ b/src/sqlite_scanner.cpp
@@ -315,10 +315,13 @@ static void SqliteScan(ClientContext &context, TableFunctionInput &data, DataChu
 	}
 }
 
-static string SqliteToString(const FunctionData *bind_data_p) {
-	D_ASSERT(bind_data_p);
-	auto &bind_data = bind_data_p->Cast<SqliteBindData>();
-	return StringUtil::Format("%s:%s", bind_data.file_name, bind_data.table_name);
+static InsertionOrderPreservingMap<string> SqliteToString(TableFunctionToStringInput &input) {
+	D_ASSERT(input.bind_data);
+	InsertionOrderPreservingMap<string> result;
+	auto &bind_data = input.bind_data->Cast<SqliteBindData>();
+        result["Table"] = bind_data.table_name;
+        result["File"] = bind_data.file_name;
+	return result;
 }
 
 /*
