diff --git a/src/include/storage/postgres_catalog.hpp b/src/include/storage/postgres_catalog.hpp
index 4a021d6..6e1e348 100644
--- a/src/include/storage/postgres_catalog.hpp
+++ b/src/include/storage/postgres_catalog.hpp
@@ -47,7 +47,7 @@ public:
 	string GetCatalogType() override {
 		return "postgres";
 	}
-	string GetDefaultSchema() const override {
+	Identifier GetDefaultSchema() const override {
 		return default_schema.empty() ? "public" : default_schema;
 	}
 
@@ -147,7 +147,7 @@ private:
 	PostgresVersion version;
 	PostgresSchemaSet schemas;
 	shared_ptr<PostgresConnectionPool> connection_pool;
-	string default_schema;
+	Identifier default_schema;
 	SecretStorageTable secret_storage_table;
 
 	PostgresAwsRdsTokenConfig rds_token_config;
diff --git a/src/storage/postgres_catalog.cpp b/src/storage/postgres_catalog.cpp
index b99b41e..b489414 100644
--- a/src/storage/postgres_catalog.cpp
+++ b/src/storage/postgres_catalog.cpp
@@ -77,7 +77,7 @@ PostgresCatalog::PostgresCatalog(ClientContext &ctx, AttachedDatabase &db_p, str
     : Catalog(db_p), attach_path(std::move(attach_path_p)), access_mode(access_mode), isolation_level(isolation_level),
       text_protocol_mode(text_protocol_mode), schemas(*this, schemas_to_load),
       connection_pool(make_shared_ptr<PostgresConnectionPool>(*this, ctx)),
-      default_schema(schemas_to_load.size() > 0 ? schemas_to_load[0] : std::string()),
+      default_schema(schemas_to_load.size() > 0 ? Identifier(schemas_to_load[0]) : Identifier()),
       secret_storage_table(std::move(secret_storage_table_p)) {
 	auto secret_entry = GetSecretEntry(ctx, secret_name);
 	this->rds_token_config = PostgresAws::ExtractTokenConfigFromSecret(secret_entry);
