diff --git a/src/storage/postgres_catalog.cpp b/src/storage/postgres_catalog.cpp
index f0aa317..ee492d6 100644
--- a/src/storage/postgres_catalog.cpp
+++ b/src/storage/postgres_catalog.cpp
@@ -24,11 +24,11 @@ namespace duckdb {
 
 unique_ptr<TableRef> PostgresCatalog::RemoteExecute(ClientContext &context, const string &sql) {
 	vector<unique_ptr<ParsedExpression>> args;
-	args.push_back(make_uniq<ConstantExpression>(Value(GetName())));
-	args.push_back(make_uniq<ConstantExpression>(Value(sql)));
+	args.push_back(ConstantExpression::FromValue(Value(GetName())));
+	args.push_back(ConstantExpression::FromValue(Value(sql)));
 	args.push_back(make_uniq<ComparisonExpression>(ExpressionType::COMPARE_EQUAL,
 	                                               make_uniq<ColumnRefExpression>("suppress_dml_output"),
-	                                               make_uniq<ConstantExpression>(Value::BOOLEAN(true))));
+	                                               ConstantExpression::FromValue(Value::BOOLEAN(true))));
 	auto func_ref = make_uniq<TableFunctionRef>();
 	func_ref->function = make_uniq<FunctionExpression>("postgres_query", std::move(args));
 	return func_ref;
diff --git a/src/storage/postgres_table_set.cpp b/src/storage/postgres_table_set.cpp
index f3cc4a6..2ff3f37 100644
--- a/src/storage/postgres_table_set.cpp
+++ b/src/storage/postgres_table_set.cpp
@@ -486,7 +486,7 @@ void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &t
 			                      "only constant DEFAULT expressions are supported");
 		}
 		const ConstantExpression &default_const_expr = default_expr.Cast<ConstantExpression>();
-		std::string sql_str = default_const_expr.GetValue().ToSQLString();
+		std::string sql_str = default_const_expr.GetLiteral().ToValue().ToSQLString();
 		sql += " DEFAULT ";
 		sql += sql_str;
 	}
