diff --git a/database-connector/src/include/dbconnector/optimizer/aggregate_optimizer.hpp b/database-connector/src/include/dbconnector/optimizer/aggregate_optimizer.hpp
--- a/database-connector/src/include/dbconnector/optimizer/aggregate_optimizer.hpp
+++ b/database-connector/src/include/dbconnector/optimizer/aggregate_optimizer.hpp
@@ -22,7 +22,8 @@
 		should_push_aggregate_t should_push_aggregate = nullptr;
 	};
 
-	static Config CreateConfig(duckdb::ClientContext &ctx, const std::string &enabled_option, char identifier_quote,
+	static Config CreateConfig(duckdb::ClientContext &ctx, const duckdb::Identifier &enabled_option,
+	                           char identifier_quote,
 	                           query::QuoteEscapeStyle escape_style, std::string table_scan_name,
 	                           should_push_aggregate_t should_push_aggregate = nullptr);
 
diff --git a/database-connector/src/include/dbconnector/optimizer/order_by_and_limit_optimizer.hpp b/database-connector/src/include/dbconnector/optimizer/order_by_and_limit_optimizer.hpp
--- a/database-connector/src/include/dbconnector/optimizer/order_by_and_limit_optimizer.hpp
+++ b/database-connector/src/include/dbconnector/optimizer/order_by_and_limit_optimizer.hpp
@@ -19,7 +19,8 @@
 		std::string table_scan_name;
 	};
 
-	static Config CreateConfig(duckdb::ClientContext &ctx, const std::string &enabled_option, char identifier_quote,
+	static Config CreateConfig(duckdb::ClientContext &ctx, const duckdb::Identifier &enabled_option,
+	                           char identifier_quote,
 	                           query::QuoteEscapeStyle escape_style, std::string table_scan_name);
 
 	static void Optimize(const Config &config, duckdb::OptimizerExtensionInput &input,
diff --git a/database-connector/src/optimizer/aggregate_optimizer.cpp b/database-connector/src/optimizer/aggregate_optimizer.cpp
--- a/database-connector/src/optimizer/aggregate_optimizer.cpp
+++ b/database-connector/src/optimizer/aggregate_optimizer.cpp
@@ -25,7 +25,7 @@
 
 using namespace duckdb;
 
-AggregateOptimizer::Config AggregateOptimizer::CreateConfig(ClientContext &ctx, const std::string &enabled_option,
+AggregateOptimizer::Config AggregateOptimizer::CreateConfig(ClientContext &ctx, const Identifier &enabled_option,
                                                             char identifier_quote, query::QuoteEscapeStyle escape_style,
                                                             std::string table_scan_name,
                                                             should_push_aggregate_t should_push_aggregate) {
diff --git a/database-connector/src/optimizer/order_by_and_limit_optimizer.cpp b/database-connector/src/optimizer/order_by_and_limit_optimizer.cpp
--- a/database-connector/src/optimizer/order_by_and_limit_optimizer.cpp
+++ b/database-connector/src/optimizer/order_by_and_limit_optimizer.cpp
@@ -19,7 +19,7 @@
 using namespace duckdb;
 
 OrderByAndLimitOptimizer::Config
-OrderByAndLimitOptimizer::CreateConfig(ClientContext &ctx, const std::string &enabled_option, char identifier_quote,
+OrderByAndLimitOptimizer::CreateConfig(ClientContext &ctx, const Identifier &enabled_option, char identifier_quote,
                                        query::QuoteEscapeStyle escape_style, std::string table_scan_name) {
 	Config res;
 
