diff --git a/src/include/storage/postgres_catalog.hpp b/src/include/storage/postgres_catalog.hpp
index 6e1e348..ab520cd 100644
--- a/src/include/storage/postgres_catalog.hpp
+++ b/src/include/storage/postgres_catalog.hpp
@@ -47,8 +47,8 @@ public:
 	string GetCatalogType() override {
 		return "postgres";
 	}
-	Identifier GetDefaultSchema() const override {
-		return default_schema.empty() ? "public" : default_schema;
+	optional<Identifier> GetDefaultSchema() const override {
+		return default_schema.empty() ? Identifier("public") : default_schema;
 	}
 
 	string GetConnectionString();
