--- a/src/postgres_binary_copy.cpp
+++ b/src/postgres_binary_copy.cpp
@@ -98,7 +98,7 @@
 
 struct PostgresBinaryReadBindData : public TableFunctionData {
 	string file_path;
-	vector<string> names;
+	vector<Identifier> names;
 	vector<LogicalType> types;
 	vector<PostgresType> postgres_types;
 	idx_t buffer_size = PostgresBinaryFileReader::DEFAULT_BUFFER_SIZE;
@@ -125,7 +125,7 @@
 
 unique_ptr<FunctionData> PostgresBinaryCopyFunction::PostgresBinaryReadBind(ClientContext &context,
                                                                             CopyFromFunctionBindInput &info,
-                                                                            vector<string> &expected_names,
+                                                                            vector<Identifier> &expected_names,
                                                                             vector<LogicalType> &expected_types) {
 	auto result = make_uniq<PostgresBinaryReadBindData>();
 	result->file_path = info.info.file_path;
@@ -178,7 +178,7 @@
 		result->postgres_types.push_back(PostgresUtils::CreateEmptyPostgresType(col_type));
 	}
 
-	result->names = IdentifiersToStrings(names);
+	result->names = names;
 	result->types = return_types;
 
 	if (input.named_parameters.count("buffer_size")) {
--- a/src/include/postgres_binary_copy.hpp
+++ b/src/include/postgres_binary_copy.hpp
@@ -35,7 +35,7 @@
 	                                        GlobalFunctionData &gstate);
 
 	static unique_ptr<FunctionData> PostgresBinaryReadBind(ClientContext &context, CopyFromFunctionBindInput &info,
-	                                                       vector<string> &expected_names,
+	                                                       vector<Identifier> &expected_names,
 	                                                       vector<LogicalType> &expected_types);
 };
 
