diff --git a/src/include/postgres_binary_copy.hpp b/src/include/postgres_binary_copy.hpp
index f69e4fa..baecddf 100644
--- a/src/include/postgres_binary_copy.hpp
+++ b/src/include/postgres_binary_copy.hpp
@@ -19,7 +19,7 @@ public:
 	PostgresBinaryCopyFunction();
 
 	static unique_ptr<FunctionData> PostgresBinaryWriteBind(ClientContext &context, CopyFunctionBindInput &input,
-	                                                        const vector<string> &names,
+	                                                        const vector<Identifier> &names,
 	                                                        const vector<LogicalType> &sql_types);
 
 	static unique_ptr<GlobalFunctionData>
diff --git a/src/include/storage/postgres_index_entry.hpp b/src/include/storage/postgres_index_entry.hpp
index c800f23..b468421 100644
--- a/src/include/storage/postgres_index_entry.hpp
+++ b/src/include/storage/postgres_index_entry.hpp
@@ -19,8 +19,8 @@ public:
 	string table_name;
 
 public:
-	string GetSchemaName() const override;
-	string GetTableName() const override;
+	Identifier GetSchemaName() const override;
+	Identifier GetTableName() const override;
 };
 
 } // namespace duckdb
diff --git a/src/include/storage/postgres_table_entry.hpp b/src/include/storage/postgres_table_entry.hpp
index 529c234..1c23ff2 100644
--- a/src/include/storage/postgres_table_entry.hpp
+++ b/src/include/storage/postgres_table_entry.hpp
@@ -20,16 +20,16 @@ struct PostgresTableInfo {
 		create_info->columns.SetAllowDuplicates(true);
 	}
 	PostgresTableInfo(const string &schema, const string &table) {
-		create_info = make_uniq<CreateTableInfo>(string(), schema, table);
+		create_info = make_uniq<CreateTableInfo>(Identifier(), Identifier(schema), Identifier(table));
 		create_info->columns.SetAllowDuplicates(true);
 	}
 	PostgresTableInfo(const SchemaCatalogEntry &schema, const string &table) {
-		create_info = make_uniq<CreateTableInfo>((SchemaCatalogEntry &)schema, table);
+		create_info = make_uniq<CreateTableInfo>((SchemaCatalogEntry &)schema, Identifier(table));
 		create_info->columns.SetAllowDuplicates(true);
 	}
 
 	const string &GetTableName() const {
-		return create_info->table;
+		return create_info->table.GetIdentifierName();
 	}
 
 	unique_ptr<CreateTableInfo> create_info;
diff --git a/src/postgres_binary_copy.cpp b/src/postgres_binary_copy.cpp
index bce7d22..faf96b8 100644
--- a/src/postgres_binary_copy.cpp
+++ b/src/postgres_binary_copy.cpp
@@ -62,7 +62,7 @@ struct PostgresBinaryWriteBindData : public TableFunctionData {};
 
 unique_ptr<FunctionData> PostgresBinaryCopyFunction::PostgresBinaryWriteBind(ClientContext &context,
                                                                              CopyFunctionBindInput &input,
-                                                                             const vector<string> &names,
+                                                                             const vector<Identifier> &names,
                                                                              const vector<LogicalType> &sql_types) {
 	return make_uniq<PostgresBinaryWriteBindData>();
 }
diff --git a/src/postgres_execute.cpp b/src/postgres_execute.cpp
index ea9f6bb..46caebb 100644
--- a/src/postgres_execute.cpp
+++ b/src/postgres_execute.cpp
@@ -28,7 +28,7 @@ static duckdb::unique_ptr<FunctionData> PGExecuteBind(ClientContext &context, Ta
 	// look up the database to query
 	auto db_name = input.inputs[0].GetValue<string>();
 	auto &db_manager = DatabaseManager::Get(context);
-	auto db = db_manager.GetDatabase(context, db_name);
+	auto db = db_manager.GetDatabase(context, Identifier(db_name));
 	if (!db) {
 		throw BinderException("Failed to find attached database \"%s\" referenced in postgres_query", db_name);
 	}
diff --git a/src/postgres_filter_pushdown.cpp b/src/postgres_filter_pushdown.cpp
index 7f126a8..018bdbc 100644
--- a/src/postgres_filter_pushdown.cpp
+++ b/src/postgres_filter_pushdown.cpp
@@ -103,7 +103,7 @@ string PostgresFilterPushdown::TransformExpressionSubject(const string &column_n
 		if (struct_type.id() != LogicalTypeId::STRUCT || StructType::IsUnnamed(struct_type)) {
 			return string();
 		}
-		auto child_name = KeywordHelper::WriteQuoted(StructType::GetChildName(struct_type, child_idx), '\"');
+		auto child_name = KeywordHelper::WriteQuoted(StructType::GetChildName(struct_type, child_idx).GetIdentifierName(), '\"');
 		return "(" + parent_name + ")." + child_name;
 	}
 	default:
diff --git a/src/postgres_query.cpp b/src/postgres_query.cpp
index 700db2f..6921438 100644
--- a/src/postgres_query.cpp
+++ b/src/postgres_query.cpp
@@ -20,7 +20,7 @@ static unique_ptr<FunctionData> PGQueryBind(ClientContext &context, TableFunctio
 	// look up the database to query
 	auto db_name = input.inputs[0].GetValue<string>();
 	auto &db_manager = DatabaseManager::Get(context);
-	auto db = db_manager.GetDatabase(context, db_name);
+	auto db = db_manager.GetDatabase(context, Identifier(db_name));
 	if (!db) {
 		throw BinderException("Failed to find attached database \"%s\" referenced in postgres_query", db_name);
 	}
diff --git a/src/postgres_scanner.cpp b/src/postgres_scanner.cpp
index 829f455..6d0d36a 100644
--- a/src/postgres_scanner.cpp
+++ b/src/postgres_scanner.cpp
@@ -189,7 +189,7 @@ static unique_ptr<FunctionData> PostgresBind(ClientContext &context, TableFuncti
 
 	bind_data->postgres_types = info->postgres_types;
 	for (auto &col : info->create_info->columns.Logical()) {
-		names.push_back(col.GetName());
+		names.emplace_back(col.GetName().GetIdentifierName());
 		return_types.push_back(col.GetType());
 	}
 	bind_data->names = info->postgres_names;
diff --git a/src/postgres_utils.cpp b/src/postgres_utils.cpp
index 646696b..94f434b 100644
--- a/src/postgres_utils.cpp
+++ b/src/postgres_utils.cpp
@@ -199,7 +199,7 @@ LogicalType PostgresUtils::TypeToLogicalType(optional_ptr<PostgresTransaction> t
 			throw InternalException("Context is destroyed!?");
 		}
 		auto entry = schema->GetEntry(CatalogTransaction(schema->ParentCatalog(), *context), CatalogType::TYPE_ENTRY,
-		                              pgtypename);
+		                              Identifier(pgtypename));
 		if (!entry) {
 			// unsupported so fallback to varchar
 			postgres_type.info = PostgresTypeAnnotation::CAST_TO_VARCHAR;
@@ -240,7 +240,7 @@ LogicalType PostgresUtils::ToPostgresType(const LogicalType &input) {
 		for (idx_t c = 0; c < StructType::GetChildCount(input); c++) {
 			auto &name = StructType::GetChildName(input, c);
 			auto &type = StructType::GetChildType(input, c);
-			new_types.push_back(make_pair(name, ToPostgresType(type)));
+			new_types.emplace_back(make_pair(name, ToPostgresType(type)));
 		}
 		auto result = LogicalType::STRUCT(std::move(new_types));
 		result.SetAlias(input.GetAlias());
diff --git a/src/storage/postgres_catalog.cpp b/src/storage/postgres_catalog.cpp
index 866dcab..12942bb 100644
--- a/src/storage/postgres_catalog.cpp
+++ b/src/storage/postgres_catalog.cpp
@@ -148,7 +148,7 @@ void PostgresCatalog::Initialize(bool load_builtin) {
 
 optional_ptr<CatalogEntry> PostgresCatalog::CreateSchema(CatalogTransaction transaction, CreateSchemaInfo &info) {
 	auto &postgres_transaction = PostgresTransaction::Get(transaction.GetContext(), *this);
-	auto entry = schemas.GetEntry(transaction.GetContext(), postgres_transaction, info.schema);
+	auto entry = schemas.GetEntry(transaction.GetContext(), postgres_transaction, info.schema.GetIdentifierName());
 	if (entry) {
 		switch (info.on_conflict) {
 		case OnCreateConflict::REPLACE_ON_CONFLICT: {
diff --git a/src/storage/postgres_catalog_set.cpp b/src/storage/postgres_catalog_set.cpp
index f0218f0..3eb08c8 100644
--- a/src/storage/postgres_catalog_set.cpp
+++ b/src/storage/postgres_catalog_set.cpp
@@ -65,9 +65,9 @@ void PostgresCatalogSet::DropEntry(PostgresTransaction &transaction, DropInfo &i
 		drop_query += " IF EXISTS ";
 	}
 	if (!info.schema.empty()) {
-		drop_query += KeywordHelper::WriteQuoted(info.schema, '"') + ".";
+		drop_query += KeywordHelper::WriteQuoted(info.schema.GetIdentifierName(), '"') + ".";
 	}
-	drop_query += KeywordHelper::WriteQuoted(info.name, '"');
+	drop_query += KeywordHelper::WriteQuoted(info.name.GetIdentifierName(), '"');
 	if (info.cascade) {
 		drop_query += "CASCADE";
 	}
@@ -75,7 +75,7 @@ void PostgresCatalogSet::DropEntry(PostgresTransaction &transaction, DropInfo &i
 
 	// erase the entry from the catalog set
 	lock_guard<mutex> l(entry_lock);
-	entries.erase(info.name);
+	entries.erase(info.name.GetIdentifierName());
 }
 
 void PostgresCatalogSet::Scan(ClientContext &context, PostgresTransaction &transaction, const std::function<void(CatalogEntry &)> &callback) {
diff --git a/src/storage/postgres_delete.cpp b/src/storage/postgres_delete.cpp
index 318daf9..47702ca 100644
--- a/src/storage/postgres_delete.cpp
+++ b/src/storage/postgres_delete.cpp
@@ -20,8 +20,8 @@ PostgresDelete::PostgresDelete(PhysicalPlan &physical_plan, LogicalOperator &op,
 string GetDeleteSQL(const PostgresTableEntry &table, const string &ctid_list) {
 	string result;
 	result = "DELETE FROM ";
-	result += KeywordHelper::WriteQuoted(table.schema.name, '"') + ".";
-	result += PostgresUtils::QuotePostgresIdentifier(table.name);
+	result += KeywordHelper::WriteQuoted(table.schema.name.GetIdentifierName(), '"') + ".";
+	result += PostgresUtils::QuotePostgresIdentifier(table.name.GetIdentifierName());
 	result += " WHERE ctid IN (" + ctid_list + ")";
 	return result;
 }
@@ -114,7 +114,7 @@ string PostgresDelete::GetName() const {
 
 InsertionOrderPreservingMap<string> PostgresDelete::ParamsToString() const {
 	InsertionOrderPreservingMap<string> result;
-	result["Table Name"] = table.name;
+	result["Table Name"] = table.name.GetIdentifierName();
 	return result;
 }
 
diff --git a/src/storage/postgres_index.cpp b/src/storage/postgres_index.cpp
index eae13de..745e743 100644
--- a/src/storage/postgres_index.cpp
+++ b/src/storage/postgres_index.cpp
@@ -90,7 +90,7 @@ unique_ptr<LogicalOperator> PostgresCatalog::BindCreateIndex(Binder &binder, Cre
 	}
 
 	auto &get = plan->Cast<LogicalGet>();
-	index_binder.InitCreateIndexInfo(get, *create_index_info, table.schema.name);
+	index_binder.InitCreateIndexInfo(get, *create_index_info, table.schema.name.GetIdentifierName());
 
 	return make_uniq<LogicalPostgresCreateIndex>(std::move(create_index_info), table);
 }
diff --git a/src/storage/postgres_index_entry.cpp b/src/storage/postgres_index_entry.cpp
index a27ab3e..0f42ef3 100644
--- a/src/storage/postgres_index_entry.cpp
+++ b/src/storage/postgres_index_entry.cpp
@@ -8,12 +8,12 @@ PostgresIndexEntry::PostgresIndexEntry(Catalog &catalog, SchemaCatalogEntry &sch
     : IndexCatalogEntry(catalog, schema, info), table_name(std::move(table_name_p)) {
 }
 
-string PostgresIndexEntry::GetSchemaName() const {
+Identifier PostgresIndexEntry::GetSchemaName() const {
 	return schema.name;
 }
 
-string PostgresIndexEntry::GetTableName() const {
-	return table_name;
+Identifier PostgresIndexEntry::GetTableName() const {
+	return Identifier(table_name);
 }
 
 } // namespace duckdb
diff --git a/src/storage/postgres_index_set.cpp b/src/storage/postgres_index_set.cpp
index f183538..b91dccb 100644
--- a/src/storage/postgres_index_set.cpp
+++ b/src/storage/postgres_index_set.cpp
@@ -37,8 +37,8 @@ void PostgresIndexSet::LoadEntries(ClientContext &context, PostgresTransaction &
 		auto index_name = result.GetString(row, 2);
 		CreateIndexInfo info;
 		info.schema = schema.name;
-		info.table = table_name;
-		info.index_name = index_name;
+		info.table = Identifier(table_name);
+		info.index_name = Identifier(index_name);
 		auto index_entry = make_shared_ptr<PostgresIndexEntry>(catalog, schema, info, table_name);
 		CreateEntry(transaction, std::move(index_entry));
 	}
@@ -62,10 +62,10 @@ string PGGetCreateIndexSQL(CreateIndexInfo &info, TableCatalogEntry &tbl) {
 		sql += " UNIQUE";
 	}
 	sql += " INDEX ";
-	sql += PostgresUtils::QuotePostgresIdentifier(info.index_name);
+	sql += PostgresUtils::QuotePostgresIdentifier(info.index_name.GetIdentifierName());
 	sql += " ON ";
-	sql += PostgresUtils::QuotePostgresIdentifier(tbl.schema.name) + ".";
-	sql += PostgresUtils::QuotePostgresIdentifier(tbl.name);
+	sql += PostgresUtils::QuotePostgresIdentifier(tbl.schema.name.GetIdentifierName()) + ".";
+	sql += PostgresUtils::QuotePostgresIdentifier(tbl.name.GetIdentifierName());
 	sql += "(";
 	for (idx_t i = 0; i < info.parsed_expressions.size(); i++) {
 		if (i > 0) {
@@ -81,7 +81,7 @@ string PGGetCreateIndexSQL(CreateIndexInfo &info, TableCatalogEntry &tbl) {
 optional_ptr<CatalogEntry> PostgresIndexSet::CreateIndex(PostgresTransaction &transaction, CreateIndexInfo &info,
                                                          TableCatalogEntry &table) {
 	transaction.Query(PGGetCreateIndexSQL(info, table));
-	auto index_entry = make_shared_ptr<PostgresIndexEntry>(schema.ParentCatalog(), schema, info, table.name);
+	auto index_entry = make_shared_ptr<PostgresIndexEntry>(schema.ParentCatalog(), schema, info, table.name.GetIdentifierName());
 	return CreateEntry(transaction, std::move(index_entry));
 }
 
diff --git a/src/storage/postgres_insert.cpp b/src/storage/postgres_insert.cpp
index 385351d..c5626ff 100644
--- a/src/storage/postgres_insert.cpp
+++ b/src/storage/postgres_insert.cpp
@@ -72,7 +72,7 @@ vector<string> GetInsertColumns(const PostgresInsert &insert, PostgresTableEntry
 		}
 		for (idx_t c = 0; c < column_count; c++) {
 			auto &col = columns.GetColumn(column_indexes[c]);
-			column_names.push_back(col.GetName());
+			column_names.emplace_back(col.GetName().GetIdentifierName());
 		}
 	}
 	return column_names;
@@ -95,7 +95,8 @@ unique_ptr<GlobalSinkState> PostgresInsert::GetGlobalSinkState(ClientContext &co
 	auto &insert_column_names = result->insert_column_names;
 	if (!insert_columns.empty()) {
 		for (auto &str : insert_columns) {
-			auto index = insert_table->GetColumnIndex(str, true);
+			Identifier str_id(str);
+		auto index = insert_table->GetColumnIndex(str_id, true);
 			if (!index.IsValid()) {
 				insert_column_names.push_back(str);
 			} else {
@@ -115,8 +116,8 @@ SinkResultType PostgresInsert::Sink(ExecutionContext &context, DataChunk &chunk,
 	auto &connection = transaction.GetConnection();
 	if (!gstate.copy_is_active) {
 		// copy hasn't started yet
-		connection.BeginCopyTo(context.client, gstate.copy_state, gstate.format, gstate.table.schema.name,
-		                       gstate.table.name, gstate.insert_column_names);
+		connection.BeginCopyTo(context.client, gstate.copy_state, gstate.format, gstate.table.schema.name.GetIdentifierName(),
+		                       gstate.table.name.GetIdentifierName(), gstate.insert_column_names);
 		gstate.copy_is_active = true;
 	}
 	connection.CopyChunk(context.client, gstate.copy_state, chunk, gstate.varchar_chunk);
@@ -166,7 +167,7 @@ string PostgresInsert::GetName() const {
 
 InsertionOrderPreservingMap<string> PostgresInsert::ParamsToString() const {
 	InsertionOrderPreservingMap<string> result;
-	result["Table Name"] = table ? table->name : info->Base().table;
+	result["Table Name"] = table ? table->name.GetIdentifierName() : info->Base().table.GetIdentifierName();
 	return result;
 }
 
@@ -219,7 +220,7 @@ bool PostgresCatalog::IsPostgresScan(const string &name) {
 void PostgresCatalog::MaterializePostgresScans(PhysicalOperator &op) {
 	if (op.type == PhysicalOperatorType::TABLE_SCAN) {
 		auto &table_scan = op.Cast<PhysicalTableScan>();
-		if (PostgresCatalog::IsPostgresScan(table_scan.function.name)) {
+		if (PostgresCatalog::IsPostgresScan(table_scan.function.name.GetIdentifierName())) {
 			auto &bind_data = table_scan.bind_data->Cast<PostgresBindData>();
 			bind_data.requires_materialization = true;
 			bind_data.max_threads = 1;
diff --git a/src/storage/postgres_optimizer.cpp b/src/storage/postgres_optimizer.cpp
index 57cac81..565dfa5 100644
--- a/src/storage/postgres_optimizer.cpp
+++ b/src/storage/postgres_optimizer.cpp
@@ -28,7 +28,7 @@ static void OptimizePostgresScanLimitPushdown(unique_ptr<LogicalOperator> &op) {
 		}
 
 		auto &get = child.get().Cast<LogicalGet>();
-		if (!PostgresCatalog::IsPostgresScan(get.function.name)) {
+		if (!PostgresCatalog::IsPostgresScan(get.function.name.GetIdentifierName())) {
 			OptimizePostgresScanLimitPushdown(op->children[0]);
 			return;
 		}
@@ -84,7 +84,7 @@ void GatherPostgresScans(LogicalOperator &op, PostgresOperators &result) {
 	if (op.type == LogicalOperatorType::LOGICAL_GET) {
 		auto &get = op.Cast<LogicalGet>();
 		auto &table_scan = get.function;
-		if (!PostgresCatalog::IsPostgresScan(table_scan.name)) {
+		if (!PostgresCatalog::IsPostgresScan(table_scan.name.GetIdentifierName())) {
 			// not a postgres scan - skip
 			return;
 		}
diff --git a/src/storage/postgres_schema_entry.cpp b/src/storage/postgres_schema_entry.cpp
index 5c1b9de..e93e600 100644
--- a/src/storage/postgres_schema_entry.cpp
+++ b/src/storage/postgres_schema_entry.cpp
@@ -42,7 +42,7 @@ PostgresTransaction &GetPostgresTransaction(CatalogTransaction transaction) {
 void PostgresSchemaEntry::TryDropEntry(ClientContext &context, CatalogType catalog_type, const string &name) {
 	DropInfo info;
 	info.type = catalog_type;
-	info.name = name;
+	info.name = Identifier(name);
 	info.cascade = false;
 	info.if_not_found = OnEntryNotFound::RETURN_NULL;
 	DropEntry(context, info);
@@ -55,7 +55,7 @@ optional_ptr<CatalogEntry> PostgresSchemaEntry::CreateTable(CatalogTransaction t
 	auto table_name = base_info.table;
 	if (base_info.on_conflict == OnCreateConflict::REPLACE_ON_CONFLICT) {
 		// CREATE OR REPLACE - drop any existing entries first (if any)
-		TryDropEntry(transaction.GetContext(), CatalogType::TABLE_ENTRY, table_name);
+		TryDropEntry(transaction.GetContext(), CatalogType::TABLE_ENTRY, table_name.GetIdentifierName());
 	}
 	return tables.CreateTable(postgres_transaction, info);
 }
@@ -74,8 +74,8 @@ optional_ptr<CatalogEntry> PostgresSchemaEntry::CreateIndex(CatalogTransaction t
 string PGGetCreateViewSQL(PostgresSchemaEntry &schema, CreateViewInfo &info) {
 	string sql;
 	sql = "CREATE VIEW ";
-	sql += PostgresUtils::QuotePostgresIdentifier(schema.name) + ".";
-	sql += PostgresUtils::QuotePostgresIdentifier(info.view_name);
+	sql += PostgresUtils::QuotePostgresIdentifier(schema.name.GetIdentifierName()) + ".";
+	sql += PostgresUtils::QuotePostgresIdentifier(info.view_name.GetIdentifierName());
 	sql += " ";
 	if (!info.aliases.empty()) {
 		sql += "(";
@@ -84,7 +84,7 @@ string PGGetCreateViewSQL(PostgresSchemaEntry &schema, CreateViewInfo &info) {
 				sql += ", ";
 			}
 			auto &alias = info.aliases[i];
-			sql += PostgresUtils::QuotePostgresIdentifier(alias);
+			sql += PostgresUtils::QuotePostgresIdentifier(alias.GetIdentifierName());
 		}
 		sql += ") ";
 	}
@@ -105,12 +105,12 @@ optional_ptr<CatalogEntry> PostgresSchemaEntry::CreateView(CatalogTransaction tr
 				return current_entry;
 			}
 			// CREATE OR REPLACE - drop any existing entries first (if any)
-			TryDropEntry(transaction.GetContext(), CatalogType::VIEW_ENTRY, info.view_name);
+			TryDropEntry(transaction.GetContext(), CatalogType::VIEW_ENTRY, info.view_name.GetIdentifierName());
 		}
 	}
 	auto &postgres_transaction = GetPostgresTransaction(transaction);
 	postgres_transaction.Query(PGGetCreateViewSQL(*this, info));
-	return tables.ReloadEntry(postgres_transaction, info.view_name);
+	return tables.ReloadEntry(postgres_transaction, info.view_name.GetIdentifierName());
 }
 
 optional_ptr<CatalogEntry> PostgresSchemaEntry::CreateType(CatalogTransaction transaction, CreateTypeInfo &info) {
@@ -118,7 +118,7 @@ optional_ptr<CatalogEntry> PostgresSchemaEntry::CreateType(CatalogTransaction tr
 	auto type_name = info.name;
 	if (info.on_conflict == OnCreateConflict::REPLACE_ON_CONFLICT) {
 		// CREATE OR REPLACE - drop any existing entries first (if any)
-		TryDropEntry(transaction.GetContext(), CatalogType::TYPE_ENTRY, info.name);
+		TryDropEntry(transaction.GetContext(), CatalogType::TYPE_ENTRY, info.name.GetIdentifierName());
 	}
 	return types.CreateType(postgres_transaction, info);
 }
diff --git a/src/storage/postgres_schema_set.cpp b/src/storage/postgres_schema_set.cpp
index e6ef9db..89ad3b0 100644
--- a/src/storage/postgres_schema_set.cpp
+++ b/src/storage/postgres_schema_set.cpp
@@ -72,7 +72,7 @@ void PostgresSchemaSet::LoadEntries(ClientContext &context, PostgresTransaction
 		auto oid = result->GetInt64(row, 0);
 		auto schema_name = result->GetString(row, 1);
 		CreateSchemaInfo info;
-		info.schema = schema_name;
+		info.schema = Identifier(schema_name);
 		info.internal = PostgresSchemaEntry::SchemaIsInternal(schema_name);
 		auto schema = make_shared_ptr<PostgresSchemaEntry>(catalog, info, std::move(tables[row]), std::move(enums[row]),
 		                                                   std::move(composite_types[row]), std::move(indexes[row]));
@@ -85,10 +85,10 @@ optional_ptr<CatalogEntry> PostgresSchemaSet::CreateSchema(PostgresTransaction &
 	if (info.on_conflict == OnCreateConflict::IGNORE_ON_CONFLICT) {
 		create_sql += " IF NOT EXISTS";
 	}
-	create_sql += KeywordHelper::WriteQuoted(info.schema, '"');
+	create_sql += KeywordHelper::WriteQuoted(info.schema.GetIdentifierName(), '"');
 	transaction.Query(create_sql);
 	auto info_copy = info.Copy();
-	info.internal = PostgresSchemaEntry::SchemaIsInternal(info_copy->schema);
+	info.internal = PostgresSchemaEntry::SchemaIsInternal(info_copy->schema.GetIdentifierName());
 	auto schema_entry = make_shared_ptr<PostgresSchemaEntry>(catalog, info_copy->Cast<CreateSchemaInfo>());
 	return CreateEntry(transaction, std::move(schema_entry));
 }
diff --git a/src/storage/postgres_table_entry.cpp b/src/storage/postgres_table_entry.cpp
index 0ce84c8..2470503 100644
--- a/src/storage/postgres_table_entry.cpp
+++ b/src/storage/postgres_table_entry.cpp
@@ -16,7 +16,7 @@ PostgresTableEntry::PostgresTableEntry(Catalog &catalog, SchemaCatalogEntry &sch
 			col.TypeMutable() = PostgresUtils::RemoveAlias(col.GetType());
 		}
 		postgres_types.push_back(PostgresUtils::CreateEmptyPostgresType(col.GetType()));
-		postgres_names.push_back(col.GetName());
+		postgres_names.emplace_back(col.GetName().GetIdentifierName());
 	}
 	approx_num_pages = 0;
 }
@@ -42,8 +42,8 @@ TableFunction PostgresTableEntry::GetScanFunction(ClientContext &context, unique
 
 	auto result = make_uniq<PostgresBindData>(context);
 
-	result->schema_name = schema.name;
-	result->table_name = name;
+	result->schema_name = schema.name.GetIdentifierName();
+	result->table_name = name.GetIdentifierName();
 	result->dsn = transaction.GetDSN();
 	result->attach_path = pg_catalog.attach_path;
 	result->SetCatalog(pg_catalog);
diff --git a/src/storage/postgres_table_set.cpp b/src/storage/postgres_table_set.cpp
index 624e137..9fe9627 100644
--- a/src/storage/postgres_table_set.cpp
+++ b/src/storage/postgres_table_set.cpp
@@ -67,7 +67,7 @@ void PostgresTableSet::AddColumn(optional_ptr<PostgresTransaction> transaction,
 	auto column_type = PostgresUtils::TypeToLogicalType(transaction, schema, type_info, postgres_type);
 	table_info.postgres_types.push_back(std::move(postgres_type));
 	table_info.postgres_names.push_back(column_name);
-	ColumnDefinition column(std::move(column_name), std::move(column_type));
+	ColumnDefinition column(Identifier(std::move(column_name)), std::move(column_type));
 	if (!default_value.empty()) {
 		auto expressions = Parser::ParseExpressionList(default_value);
 		if (expressions.empty()) {
@@ -101,10 +101,10 @@ void PostgresTableSet::AddConstraint(PostgresResult &result, idx_t row, Postgres
 		if (index <= 0 || index > create_info.columns.LogicalColumnCount()) {
 			return;
 		}
-		columns.push_back(create_info.columns.GetColumn(LogicalIndex(index - 1)).Name());
+		columns.emplace_back(create_info.columns.GetColumn(LogicalIndex(index - 1)).Name());
 	}
 
-	create_info.constraints.push_back(make_uniq<UniqueConstraint>(std::move(columns), constraint_type == "p"));
+	create_info.constraints.push_back(make_uniq<UniqueConstraint>(StringsToIdentifiers(columns), constraint_type == "p"));
 }
 
 void PostgresTableSet::AddColumnOrConstraint(optional_ptr<PostgresTransaction> transaction,
@@ -148,7 +148,7 @@ void PostgresTableSet::LoadEntries(ClientContext &context, PostgresTransaction &
 		CreateEntries(transaction, table_result->GetResult(), table_result->start, table_result->end);
 		table_result.reset();
 	} else {
-		auto query = GetInitializeQuery(schema.name);
+		auto query = GetInitializeQuery(schema.name.GetIdentifierName());
 
 		auto result = transaction.Query(query);
 		auto rows = result->Count();
@@ -159,7 +159,7 @@ void PostgresTableSet::LoadEntries(ClientContext &context, PostgresTransaction &
 
 unique_ptr<PostgresTableInfo> PostgresTableSet::GetTableInfo(PostgresTransaction &transaction,
                                                              PostgresSchemaEntry &schema, const string &table_name) {
-	auto query = PostgresTableSet::GetInitializeQuery(schema.name, table_name);
+	auto query = PostgresTableSet::GetInitializeQuery(schema.name.GetIdentifierName(), table_name);
 	auto result = transaction.Query(query);
 	auto rows = result->Count();
 	if (rows == 0) {
@@ -217,7 +217,7 @@ string PostgresColumnsToSQL(const ColumnList &columns, const vector<unique_ptr<C
 			not_null_columns.insert(not_null.index);
 		} else if (constraint->type == ConstraintType::UNIQUE) {
 			auto &pk = constraint->Cast<UniqueConstraint>();
-			vector<string> constraint_columns = pk.columns;
+			auto constraint_columns = IdentifiersToStrings(pk.columns);
 			if (pk.index.index != DConstants::INVALID_INDEX) {
 				// no columns specified: single column constraint
 				if (pk.is_primary_key) {
@@ -231,7 +231,7 @@ string PostgresColumnsToSQL(const ColumnList &columns, const vector<unique_ptr<C
 				if (pk.is_primary_key) {
 					// multi key pk column: insert set of columns into multi_key_pks
 					for (auto &col : pk.columns) {
-						multi_key_pks.insert(col);
+						multi_key_pks.insert(col.GetIdentifierName());
 					}
 				}
 				string base = pk.is_primary_key ? "PRIMARY KEY(" : "UNIQUE(";
@@ -239,7 +239,7 @@ string PostgresColumnsToSQL(const ColumnList &columns, const vector<unique_ptr<C
 					if (i > 0) {
 						base += ", ";
 					}
-					base += KeywordHelper::WriteQuoted(pk.columns[i], '"');
+					base += KeywordHelper::WriteQuoted(pk.columns[i].GetIdentifierName(), '"');
 				}
 				extra_constraints.push_back(base + ")");
 			}
@@ -258,11 +258,11 @@ string PostgresColumnsToSQL(const ColumnList &columns, const vector<unique_ptr<C
 		if (column.Oid() > 0) {
 			ss << ", ";
 		}
-		ss << KeywordHelper::WriteQuoted(column.Name(), '"') << " ";
+		ss << KeywordHelper::WriteQuoted(column.Name().GetIdentifierName(), '"') << " ";
 		ss << PostgresUtils::TypeToString(column.Type());
 		bool not_null = not_null_columns.find(column.Logical()) != not_null_columns.end();
 		bool is_single_key_pk = pk_columns.find(column.Logical()) != pk_columns.end();
-		bool is_multi_key_pk = multi_key_pks.find(column.Name()) != multi_key_pks.end();
+		bool is_multi_key_pk = multi_key_pks.find(column.Name().GetIdentifierName()) != multi_key_pks.end();
 		bool is_unique = unique_columns.find(column.Logical()) != unique_columns.end();
 		if (not_null && !is_single_key_pk && !is_multi_key_pk) {
 			// NOT NULL but not a primary key column
@@ -304,10 +304,10 @@ string GetPostgresCreateTable(CreateTableInfo &info) {
 		ss << "IF NOT EXISTS ";
 	}
 	if (!info.schema.empty()) {
-		ss << KeywordHelper::WriteQuoted(info.schema, '"');
+		ss << KeywordHelper::WriteQuoted(info.schema.GetIdentifierName(), '"');
 		ss << ".";
 	}
-	ss << KeywordHelper::WriteQuoted(info.table, '"');
+	ss << KeywordHelper::WriteQuoted(info.table.GetIdentifierName(), '"');
 	ss << PostgresColumnsToSQL(info.columns, info.constraints);
 	ss << ";";
 	return ss.str();
@@ -322,8 +322,8 @@ optional_ptr<CatalogEntry> PostgresTableSet::CreateTable(PostgresTransaction &tr
 
 string PostgresTableSet::GetAlterTablePrefix(const string &name, optional_ptr<CatalogEntry> entry) {
 	string sql = "ALTER TABLE ";
-	sql += KeywordHelper::WriteQuoted(schema.name, '"') + ".";
-	sql += KeywordHelper::WriteQuoted(entry ? entry->name : name, '"');
+	sql += KeywordHelper::WriteQuoted(schema.name.GetIdentifierName(), '"') + ".";
+	sql += KeywordHelper::WriteQuoted(entry ? entry->name.GetIdentifierName() : name, '"');
 	return sql;
 }
 
@@ -333,7 +333,8 @@ string PostgresTableSet::GetAlterTableColumnName(const string &name, optional_pt
 	}
 	auto &table = entry->Cast<PostgresTableEntry>();
 	string column_name = name;
-	auto column_index = table.GetColumnIndex(column_name, true);
+	Identifier column_name_id(column_name);
+		auto column_index = table.GetColumnIndex(column_name_id, true);
 	if (!column_index.IsValid()) {
 		return name;
 	}
@@ -346,44 +347,44 @@ string PostgresTableSet::GetAlterTablePrefix(ClientContext &context, PostgresTra
 }
 
 void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &transaction, RenameTableInfo &info) {
-	string sql = GetAlterTablePrefix(context, transaction, info.name);
+	string sql = GetAlterTablePrefix(context, transaction, info.name.GetIdentifierName());
 	sql += " RENAME TO ";
-	sql += KeywordHelper::WriteQuoted(info.new_table_name, '"');
+	sql += KeywordHelper::WriteQuoted(info.new_table_name.GetIdentifierName(), '"');
 	transaction.Query(sql);
 }
 
 void PostgresTableSet::AlterTable(ClientContext &context,PostgresTransaction &transaction, RenameColumnInfo &info) {
-	auto entry = GetEntry(context, transaction, info.name);
-	string sql = GetAlterTablePrefix(info.name, entry);
+	auto entry = GetEntry(context, transaction, info.name.GetIdentifierName());
+	string sql = GetAlterTablePrefix(info.name.GetIdentifierName(), entry);
 	sql += " RENAME COLUMN  ";
-	string column_name = GetAlterTableColumnName(info.old_name, entry);
+	string column_name = GetAlterTableColumnName(info.old_name.GetIdentifierName(), entry);
 	sql += KeywordHelper::WriteQuoted(column_name, '"');
 	sql += " TO ";
-	sql += KeywordHelper::WriteQuoted(info.new_name, '"');
+	sql += KeywordHelper::WriteQuoted(info.new_name.GetIdentifierName(), '"');
 
 	transaction.Query(sql);
 }
 
 void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &transaction, AddColumnInfo &info) {
-	string sql = GetAlterTablePrefix(context, transaction, info.name);
+	string sql = GetAlterTablePrefix(context, transaction, info.name.GetIdentifierName());
 	sql += " ADD COLUMN  ";
 	if (info.if_column_not_exists) {
 		sql += "IF NOT EXISTS ";
 	}
-	sql += KeywordHelper::WriteQuoted(info.new_column.Name(), '"');
+	sql += KeywordHelper::WriteQuoted(info.new_column.Name().GetIdentifierName(), '"');
 	sql += " ";
 	sql += info.new_column.Type().ToString();
 	transaction.Query(sql);
 }
 
 void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &transaction, RemoveColumnInfo &info) {
-	auto entry = GetEntry(context, transaction, info.name);
-	string sql = GetAlterTablePrefix(info.name, entry);
+	auto entry = GetEntry(context, transaction, info.name.GetIdentifierName());
+	string sql = GetAlterTablePrefix(info.name.GetIdentifierName(), entry);
 	sql += " DROP COLUMN  ";
 	if (info.if_column_exists) {
 		sql += "IF EXISTS ";
 	}
-	string column_name = GetAlterTableColumnName(info.removed_column, entry);
+	string column_name = GetAlterTableColumnName(info.removed_column.GetIdentifierName(), entry);
 	sql += KeywordHelper::WriteQuoted(column_name, '"');
 	transaction.Query(sql);
 }
diff --git a/src/storage/postgres_type_set.cpp b/src/storage/postgres_type_set.cpp
index f2db80c..5a8679c 100644
--- a/src/storage/postgres_type_set.cpp
+++ b/src/storage/postgres_type_set.cpp
@@ -49,7 +49,7 @@ void PostgresTypeSet::CreateEnum(PostgresTransaction &transaction, PostgresResul
 	PostgresType postgres_type;
 	CreateTypeInfo info;
 	postgres_type.oid = result.GetInt64(start_row, 1);
-	info.name = result.GetString(start_row, 2);
+	info.name = Identifier(result.GetString(start_row, 2));
 	// construct the enum
 	idx_t enum_count = end_row - start_row;
 	Vector duckdb_levels(LogicalType::VARCHAR, enum_count);
@@ -57,7 +57,7 @@ void PostgresTypeSet::CreateEnum(PostgresTransaction &transaction, PostgresResul
 		duckdb_levels.SetValue(enum_idx, result.GetString(start_row + enum_idx, 3));
 	}
 	info.type = LogicalType::ENUM(duckdb_levels, enum_count);
-	info.type.SetAlias(info.name);
+	info.type.SetAlias(info.name.GetIdentifierName());
 	auto type_entry = make_shared_ptr<PostgresTypeEntry>(catalog, schema, info, postgres_type);
 	CreateEntry(transaction, std::move(type_entry));
 }
@@ -107,7 +107,7 @@ void PostgresTypeSet::CreateCompositeType(PostgresTransaction &transaction, Post
 	PostgresType postgres_type;
 	CreateTypeInfo info;
 	postgres_type.oid = result.GetInt64(start_row, 1);
-	info.name = result.GetString(start_row, 2);
+	info.name = Identifier(result.GetString(start_row, 2));
 
 	child_list_t<LogicalType> child_types;
 	for (idx_t row = start_row; row < end_row; row++) {
@@ -115,12 +115,12 @@ void PostgresTypeSet::CreateCompositeType(PostgresTransaction &transaction, Post
 		PostgresTypeData type_data;
 		type_data.type_name = result.GetString(row, 4);
 		PostgresType child_type;
-		child_types.push_back(
+		child_types.emplace_back(
 		    make_pair(type_name, PostgresUtils::TypeToLogicalType(&transaction, &schema, type_data, child_type)));
 		postgres_type.children.push_back(std::move(child_type));
 	}
 	info.type = LogicalType::STRUCT(std::move(child_types));
-	info.type.SetAlias(info.name);
+	info.type.SetAlias(info.name.GetIdentifierName());
 	auto type_entry = make_shared_ptr<PostgresTypeEntry>(catalog, schema, info, postgres_type);
 	CreateEntry(transaction, std::move(type_entry));
 }
@@ -157,7 +157,7 @@ void PostgresTypeSet::LoadEntries(ClientContext &context, PostgresTransaction &t
 
 string GetCreateTypeSQL(CreateTypeInfo &info) {
 	string sql = "CREATE TYPE ";
-	sql += KeywordHelper::WriteQuoted(info.name, '"');
+	sql += KeywordHelper::WriteQuoted(info.name.GetIdentifierName(), '"');
 	sql += " AS ";
 	switch (info.type.id()) {
 	case LogicalTypeId::ENUM: {
@@ -180,7 +180,7 @@ string GetCreateTypeSQL(CreateTypeInfo &info) {
 			if (c > 0) {
 				sql += ", ";
 			}
-			sql += KeywordHelper::WriteQuoted(StructType::GetChildName(info.type, c), '"');
+			sql += KeywordHelper::WriteQuoted(StructType::GetChildName(info.type, c).GetIdentifierName(), '"');
 			sql += " ";
 			sql += PostgresUtils::TypeToString(StructType::GetChildType(info.type, c));
 		}
@@ -198,7 +198,7 @@ optional_ptr<CatalogEntry> PostgresTypeSet::CreateType(PostgresTransaction &tran
 
 	auto create_sql = GetCreateTypeSQL(info);
 	conn.Execute(transaction.GetContext(), create_sql);
-	info.type.SetAlias(info.name);
+	info.type.SetAlias(info.name.GetIdentifierName());
 	auto pg_type = PostgresUtils::CreateEmptyPostgresType(info.type);
 	auto type_entry = make_shared_ptr<PostgresTypeEntry>(catalog, schema, info, pg_type);
 	return CreateEntry(transaction, std::move(type_entry));
diff --git a/src/storage/postgres_update.cpp b/src/storage/postgres_update.cpp
index 3b7d94b..d4ca9f7 100644
--- a/src/storage/postgres_update.cpp
+++ b/src/storage/postgres_update.cpp
@@ -60,8 +60,8 @@ string CreateUpdateTable(const string &name, PostgresTableEntry &table, const ve
 string GetUpdateSQL(const string &name, PostgresTableEntry &table, const vector<PhysicalIndex> &index) {
 	string result;
 	result = "UPDATE ";
-	result += KeywordHelper::WriteQuoted(table.schema.name, '"') + ".";
-	result += KeywordHelper::WriteQuoted(table.name, '"');
+	result += KeywordHelper::WriteQuoted(table.schema.name.GetIdentifierName(), '"') + ".";
+	result += KeywordHelper::WriteQuoted(table.name.GetIdentifierName(), '"');
 	result += " SET ";
 	for (idx_t i = 0; i < index.size(); i++) {
 		if (i > 0) {
@@ -76,7 +76,7 @@ string GetUpdateSQL(const string &name, PostgresTableEntry &table, const vector<
 	}
 	result += " FROM " + PostgresUtils::QuotePostgresIdentifier(name);
 	result += " WHERE ";
-	result += KeywordHelper::WriteQuoted(table.name, '"');
+	result += KeywordHelper::WriteQuoted(table.name.GetIdentifierName(), '"');
 	result += ".ctid=__page_id_string::TID";
 	return result;
 }
@@ -195,7 +195,7 @@ string PostgresUpdate::GetName() const {
 
 InsertionOrderPreservingMap<string> PostgresUpdate::ParamsToString() const {
 	InsertionOrderPreservingMap<string> result;
-	result["Table Name"] = table.name;
+	result["Table Name"] = table.name.GetIdentifierName();
 	return result;
 }
 
