diff --git a/src/storage/postgres_catalog.cpp b/src/storage/postgres_catalog.cpp
index 64c6305..6e9bd15 100644
--- a/src/storage/postgres_catalog.cpp
+++ b/src/storage/postgres_catalog.cpp
@@ -181,7 +181,7 @@ optional_ptr<CatalogEntry> PostgresCatalog::CreateSchema(CatalogTransaction tran
 		case OnCreateConflict::REPLACE_ON_CONFLICT: {
 			DropInfo try_drop;
 			try_drop.type = CatalogType::SCHEMA_ENTRY;
-			try_drop.name = info.schema;
+			try_drop.NameMutable() = info.schema;
 			try_drop.if_not_found = OnEntryNotFound::RETURN_NULL;
 			try_drop.cascade = false;
 			schemas.DropEntry(postgres_transaction, try_drop);
diff --git a/src/storage/postgres_catalog_set.cpp b/src/storage/postgres_catalog_set.cpp
index 2243adb..a6fb68b 100644
--- a/src/storage/postgres_catalog_set.cpp
+++ b/src/storage/postgres_catalog_set.cpp
@@ -66,10 +66,10 @@ void PostgresCatalogSet::DropEntry(PostgresTransaction &transaction, DropInfo &i
 	if (info.if_not_found == OnEntryNotFound::RETURN_NULL) {
 		drop_query += " IF EXISTS ";
 	}
-	if (!info.schema.empty()) {
-		drop_query += PostgresUtils::WriteIdentifier(info.schema.GetIdentifierName()) + ".";
+	if (!info.Schema().empty()) {
+		drop_query += PostgresUtils::WriteIdentifier(info.Schema().GetIdentifierName()) + ".";
 	}
-	drop_query += PostgresUtils::WriteIdentifier(info.name.GetIdentifierName());
+	drop_query += PostgresUtils::WriteIdentifier(info.Name().GetIdentifierName());
 	if (info.cascade) {
 		drop_query += "CASCADE";
 	}
@@ -77,7 +77,7 @@ void PostgresCatalogSet::DropEntry(PostgresTransaction &transaction, DropInfo &i
 
 	// erase the entry from the catalog set
 	lock_guard<mutex> l(entry_lock);
-	entries.erase(info.name.GetIdentifierName());
+	entries.erase(info.Name().GetIdentifierName());
 }
 
 void PostgresCatalogSet::Scan(ClientContext &context, PostgresTransaction &transaction,
diff --git a/src/storage/postgres_index.cpp b/src/storage/postgres_index.cpp
index cf26c37..0ec4a05 100644
--- a/src/storage/postgres_index.cpp
+++ b/src/storage/postgres_index.cpp
@@ -36,8 +36,8 @@ SourceResultType PostgresCreateIndex::GetDataInternal(ExecutionContext &context,
 		case OnCreateConflict::REPLACE_ON_CONFLICT: {
 			DropInfo drop_info;
 			drop_info.type = CatalogType::INDEX_ENTRY;
-			drop_info.schema = info->schema;
-			drop_info.name = info->index_name;
+			drop_info.SchemaMutable() = info->schema;
+			drop_info.NameMutable() = info->index_name;
 			schema.DropEntry(context.client, drop_info);
 			break;
 		}
diff --git a/src/storage/postgres_schema_entry.cpp b/src/storage/postgres_schema_entry.cpp
index f320e6f..1f73fcf 100644
--- a/src/storage/postgres_schema_entry.cpp
+++ b/src/storage/postgres_schema_entry.cpp
@@ -42,7 +42,7 @@ PostgresTransaction &GetPostgresTransaction(CatalogTransaction transaction) {
 void PostgresSchemaEntry::TryDropEntry(ClientContext &context, CatalogType catalog_type, const string &name) {
 	DropInfo info;
 	info.type = catalog_type;
-	info.name = Identifier(name);
+	info.NameMutable() = Identifier(name);
 	info.cascade = false;
 	info.if_not_found = OnEntryNotFound::RETURN_NULL;
 	DropEntry(context, info);
@@ -182,7 +182,7 @@ void PostgresSchemaEntry::Scan(CatalogType type, const std::function<void(Catalo
 }
 
 void PostgresSchemaEntry::DropEntry(ClientContext &context, DropInfo &info) {
-	info.schema = name;
+	info.SchemaMutable() = name;
 	auto &postgres_transaction = PostgresTransaction::Get(context, catalog);
 	GetCatalogSet(info.type).DropEntry(postgres_transaction, info);
 }
diff --git a/src/storage/postgres_table_set.cpp b/src/storage/postgres_table_set.cpp
index 663e6d5..795209e 100644
--- a/src/storage/postgres_table_set.cpp
+++ b/src/storage/postgres_table_set.cpp
@@ -375,15 +375,15 @@ string PostgresTableSet::GetAlterTablePrefix(ClientContext &context, PostgresTra
 }
 
 void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &transaction, RenameTableInfo &info) {
-	string sql = GetAlterTablePrefix(context, transaction, info.name.GetIdentifierName());
+	string sql = GetAlterTablePrefix(context, transaction, info.Name().GetIdentifierName());
 	sql += " RENAME TO ";
 	sql += PostgresUtils::WriteIdentifier(info.new_table_name.GetIdentifierName());
 	transaction.Query(sql);
 }
 
 void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &transaction, RenameColumnInfo &info) {
-	auto entry = GetEntry(context, transaction, info.name.GetIdentifierName());
-	string sql = GetAlterTablePrefix(info.name.GetIdentifierName(), entry);
+	auto entry = GetEntry(context, transaction, info.Name().GetIdentifierName());
+	string sql = GetAlterTablePrefix(info.Name().GetIdentifierName(), entry);
 	sql += " RENAME COLUMN  ";
 	string column_name = GetAlterTableColumnName(info.old_name.GetIdentifierName(), entry);
 	sql += PostgresUtils::WriteIdentifier(column_name);
@@ -394,7 +394,7 @@ void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &t
 }
 
 void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &transaction, AddColumnInfo &info) {
-	string sql = GetAlterTablePrefix(context, transaction, info.name.GetIdentifierName());
+	string sql = GetAlterTablePrefix(context, transaction, info.Name().GetIdentifierName());
 	sql += " ADD COLUMN  ";
 	if (info.if_column_not_exists) {
 		sql += "IF NOT EXISTS ";
@@ -419,8 +419,8 @@ void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &t
 }
 
 void PostgresTableSet::AlterTable(ClientContext &context, PostgresTransaction &transaction, RemoveColumnInfo &info) {
-	auto entry = GetEntry(context, transaction, info.name.GetIdentifierName());
-	string sql = GetAlterTablePrefix(info.name.GetIdentifierName(), entry);
+	auto entry = GetEntry(context, transaction, info.Name().GetIdentifierName());
+	string sql = GetAlterTablePrefix(info.Name().GetIdentifierName(), entry);
 	sql += " DROP COLUMN  ";
 	if (info.if_column_exists) {
 		sql += "IF EXISTS ";
