diff --git a/test/sql/storage/attach_detach.test b/test/sql/storage/attach_detach.test
index 7b7d19f..ba8168f 100644
--- a/test/sql/storage/attach_detach.test
+++ b/test/sql/storage/attach_detach.test
@@ -18,6 +18,7 @@ DETACH s1
 
 statement error
 SELECT * FROM s1.test
+----
 
 statement ok
 ATTACH 'dbname=postgresscanner' AS s1 (TYPE POSTGRES)
diff --git a/test/sql/storage/attach_drop.test b/test/sql/storage/attach_drop.test
index 8f1090c..90f6952 100644
--- a/test/sql/storage/attach_drop.test
+++ b/test/sql/storage/attach_drop.test
@@ -27,6 +27,7 @@ DROP TABLE simple.test
 # verify the drop was successful
 statement error
 SELECT * FROM simple.test;
+----
 
 statement error
 DROP TABLE simple.testx
@@ -34,4 +35,4 @@ DROP TABLE simple.testx
 Table with name testx does not exist
 
 statement ok
-DROP TABLE IF EXISTS simple.testx
\ No newline at end of file
+DROP TABLE IF EXISTS simple.testx
diff --git a/test/sql/storage/attach_views.test b/test/sql/storage/attach_views.test
index e57b45d..0e12b54 100644
--- a/test/sql/storage/attach_views.test
+++ b/test/sql/storage/attach_views.test
@@ -34,9 +34,11 @@ cannot copy to view
 # FIXME - error message here is not very descriptive
 statement error
 UPDATE v1 SET i=84
+----
 
 statement error
 DELETE FROM v1
+----
 
 statement error
 INSERT INTO v1 VALUES (1, 1);
