diff --git a/src/postgres_storage.cpp b/src/postgres_storage.cpp
index d569f2c..8837840 100644
--- a/src/postgres_storage.cpp
+++ b/src/postgres_storage.cpp
@@ -7,8 +7,9 @@
 
 namespace duckdb {
 
-static unique_ptr<Catalog> PostgresAttach(StorageExtensionInfo *storage_info, AttachedDatabase &db, const string &name,
-                                          AttachInfo &info, AccessMode access_mode) {
+static unique_ptr<Catalog> PostgresAttach(StorageExtensionInfo *storage_info, ClientContext &context,
+                                          AttachedDatabase &db, const string &name, AttachInfo &info,
+                                          AccessMode access_mode) {
 	return make_uniq<PostgresCatalog>(db, info.path, access_mode);
 }
 
