diff --git a/src/include/storage/postgres_delete.hpp b/src/include/storage/postgres_delete.hpp
index 61fa2ec..28fb86e 100644
--- a/src/include/storage/postgres_delete.hpp
+++ b/src/include/storage/postgres_delete.hpp
@@ -44,7 +44,7 @@ public:
 	}
 
 	string GetName() const override;
-	string ParamsToString() const override;
+	InsertionOrderPreservingMap<string> ParamsToString() const override;
 };
 
 } // namespace duckdb
diff --git a/src/include/storage/postgres_insert.hpp b/src/include/storage/postgres_insert.hpp
index a3c7d97..13ea902 100644
--- a/src/include/storage/postgres_insert.hpp
+++ b/src/include/storage/postgres_insert.hpp
@@ -53,7 +53,7 @@ public:
 	}
 
 	string GetName() const override;
-	string ParamsToString() const override;
+	InsertionOrderPreservingMap<string> ParamsToString() const override;
 };
 
 } // namespace duckdb
diff --git a/src/include/storage/postgres_update.hpp b/src/include/storage/postgres_update.hpp
index b5d44d2..417657e 100644
--- a/src/include/storage/postgres_update.hpp
+++ b/src/include/storage/postgres_update.hpp
@@ -46,7 +46,7 @@ public:
 	}
 
 	string GetName() const override;
-	string ParamsToString() const override;
+	InsertionOrderPreservingMap<string> ParamsToString() const override;
 };
 
 } // namespace duckdb
diff --git a/src/storage/postgres_delete.cpp b/src/storage/postgres_delete.cpp
index 0c05ba8..814bd18 100644
--- a/src/storage/postgres_delete.cpp
+++ b/src/storage/postgres_delete.cpp
@@ -110,8 +110,10 @@ string PostgresDelete::GetName() const {
 	return "PG_DELETE";
 }
 
-string PostgresDelete::ParamsToString() const {
-	return table.name;
+InsertionOrderPreservingMap<string> PostgresDelete::ParamsToString() const {
+	InsertionOrderPreservingMap<string> result;
+	result["Table Name"] = table.name;
+	return result;
 }
 
 //===--------------------------------------------------------------------===//
diff --git a/src/storage/postgres_insert.cpp b/src/storage/postgres_insert.cpp
index cccd186..f6d5193 100644
--- a/src/storage/postgres_insert.cpp
+++ b/src/storage/postgres_insert.cpp
@@ -144,8 +144,10 @@ string PostgresInsert::GetName() const {
 	return table ? "PG_INSERT" : "PG_CREATE_TABLE_AS";
 }
 
-string PostgresInsert::ParamsToString() const {
-	return table ? table->name : info->Base().table;
+InsertionOrderPreservingMap<string> PostgresInsert::ParamsToString() const {
+	InsertionOrderPreservingMap<string> result;
+	result["Table Name"] = table ? table->name : info->Base().table;
+	return result;
 }
 
 //===--------------------------------------------------------------------===//
diff --git a/src/storage/postgres_update.cpp b/src/storage/postgres_update.cpp
index a58f2fa..90e7312 100644
--- a/src/storage/postgres_update.cpp
+++ b/src/storage/postgres_update.cpp
@@ -171,8 +171,10 @@ string PostgresUpdate::GetName() const {
 	return "PG_UPDATE";
 }
 
-string PostgresUpdate::ParamsToString() const {
-	return table.name;
+InsertionOrderPreservingMap<string> PostgresUpdate::ParamsToString() const {
+	InsertionOrderPreservingMap<string> result;
+	result["Table Name"] = table.name;
+	return result;
 }
 
 //===--------------------------------------------------------------------===//
