diff --git a/src/postgres_scanner.cpp b/src/postgres_scanner.cpp
index dfee871..a39abc5 100644
--- a/src/postgres_scanner.cpp
+++ b/src/postgres_scanner.cpp
@@ -491,10 +491,12 @@ static idx_t PostgresScanBatchIndex(ClientContext &context, const FunctionData *
 	return local_state.batch_idx;
 }
 
-static string PostgresScanToString(const FunctionData *bind_data_p) {
-	D_ASSERT(bind_data_p);
-	auto &bind_data = bind_data_p->Cast<PostgresBindData>();
-	return bind_data.table_name;
+static InsertionOrderPreservingMap<string> PostgresScanToString(TableFunctionToStringInput &input) {
+	D_ASSERT(input.bind_data);
+	InsertionOrderPreservingMap<string> result;
+	auto &bind_data = input.bind_data->Cast<PostgresBindData>();
+	result["Table"] = bind_data.table_name;
+	return result;
 }
 
 unique_ptr<NodeStatistics> PostgresScanCardinality(ClientContext &context, const FunctionData *bind_data_p) {
