diff --git a/src/include/storage/postgres_optimizer.hpp b/src/include/storage/postgres_optimizer.hpp
index b17f259..1e573f0 100644
--- a/src/include/storage/postgres_optimizer.hpp
+++ b/src/include/storage/postgres_optimizer.hpp
@@ -14,7 +14,7 @@ namespace duckdb {
 
 class PostgresOptimizer {
 public:
-	static void Optimize(ClientContext &context, OptimizerExtensionInfo *info, unique_ptr<LogicalOperator> &plan);
+	static void Optimize(OptimizerExtensionInput &input, unique_ptr<LogicalOperator> &plan);
 };
 
 } // namespace duckdb
diff --git a/src/storage/postgres_optimizer.cpp b/src/storage/postgres_optimizer.cpp
index 83943e9..0c83cdc 100644
--- a/src/storage/postgres_optimizer.cpp
+++ b/src/storage/postgres_optimizer.cpp
@@ -34,8 +34,7 @@ void GatherPostgresScans(LogicalOperator &op, PostgresOperators &result) {
 	}
 }
 
-void PostgresOptimizer::Optimize(ClientContext &context, OptimizerExtensionInfo *info,
-                                 unique_ptr<LogicalOperator> &plan) {
+void PostgresOptimizer::Optimize(OptimizerExtensionInput &input, unique_ptr<LogicalOperator> &plan) {
 	// look at the query plan and check if we can enable streaming query scans
 	PostgresOperators operators;
 	GatherPostgresScans(*plan, operators);
