diff --git a/src/storage/postgres_catalog.cpp b/src/storage/postgres_catalog.cpp
index 160ad2e..deb8f6d 100644
--- a/src/storage/postgres_catalog.cpp
+++ b/src/storage/postgres_catalog.cpp
@@ -181,7 +181,7 @@ optional_ptr<CatalogEntry> PostgresCatalog::CreateSchema(CatalogTransaction tran
 		case OnCreateConflict::REPLACE_ON_CONFLICT: {
 			DropInfo try_drop;
 			try_drop.type = CatalogType::SCHEMA_ENTRY;
-			try_drop.NameMutable() = info.GetQualifiedName().Schema();
+			try_drop.SetName(info.GetQualifiedName().Schema());
 			try_drop.if_not_found = OnEntryNotFound::RETURN_NULL;
 			try_drop.cascade = false;
 			schemas.DropEntry(postgres_transaction, try_drop);
diff --git a/src/storage/postgres_index.cpp b/src/storage/postgres_index.cpp
index 90e17db..7acbb23 100644
--- a/src/storage/postgres_index.cpp
+++ b/src/storage/postgres_index.cpp
@@ -39,7 +39,7 @@ SourceResultType PostgresCreateIndex::GetDataInternal(ExecutionContext &context,
 			drop_info.SetQualifiedName(QualifiedName(drop_info.GetQualifiedName().Catalog(),
 			                                         info->GetQualifiedName().Schema(),
 			                                         drop_info.GetQualifiedName().Name()));
-			drop_info.NameMutable() = info->GetIndexName();
+			drop_info.SetName(info->GetIndexName());
 			schema.DropEntry(context.client, drop_info);
 			break;
 		}
diff --git a/src/storage/postgres_schema_entry.cpp b/src/storage/postgres_schema_entry.cpp
index 5a4f104..d2a19e0 100644
--- a/src/storage/postgres_schema_entry.cpp
+++ b/src/storage/postgres_schema_entry.cpp
@@ -42,7 +42,7 @@ PostgresTransaction &GetPostgresTransaction(CatalogTransaction transaction) {
 void PostgresSchemaEntry::TryDropEntry(ClientContext &context, CatalogType catalog_type, const string &name) {
 	DropInfo info;
 	info.type = catalog_type;
-	info.NameMutable() = Identifier(name);
+	info.SetName(Identifier(name));
 	info.cascade = false;
 	info.if_not_found = OnEntryNotFound::RETURN_NULL;
 	DropEntry(context, info);
