diff --git a/src/storage/postgres_merge_into.cpp b/src/storage/postgres_merge_into.cpp
index 1824f10..756eb40 100644
--- a/src/storage/postgres_merge_into.cpp
+++ b/src/storage/postgres_merge_into.cpp
@@ -116,10 +116,8 @@ PhysicalOperator &PostgresCatalog::PlanMergeInto(ClientContext &context, Physica
 		actions.emplace(entry.first, std::move(planned_actions));
 	}
 
-	auto &result = planner.Make<PhysicalMergeInto>(op.types, std::move(actions), op.row_id_start, op.source_marker,
-	                                               false, op.return_chunk);
-	result.children.push_back(plan);
-	return result;
+	return planner.Make<PhysicalMergeInto>(op.types, plan, std::move(actions), op.row_id_start, op.source_marker,
+	                                       false, op.return_chunk);
 }
 
 } // namespace duckdb
