diff --git a/src/storage/delta_catalog.cpp b/src/storage/delta_catalog.cpp
index ace86df..a3413f5 100644
--- a/src/storage/delta_catalog.cpp
+++ b/src/storage/delta_catalog.cpp
@@ -12,7 +12,7 @@
 namespace duckdb {
 
 idx_t ParseDeltaVersionFromAtClause(const BoundAtClause &at_clause) {
-	if (StringUtil::Lower(at_clause.Unit()) != "version") {
+	if (at_clause.Unit() != "version") {
 		throw InvalidConfigurationException("Delta tables only support at_clause with unit 'version'");
 	}
 	Value version_value = at_clause.GetValue();
