diff --git a/src/storage/delta_catalog.cpp b/src/storage/delta_catalog.cpp
index a3413f5..883d3c8 100644
--- a/src/storage/delta_catalog.cpp
+++ b/src/storage/delta_catalog.cpp
@@ -15,12 +15,12 @@ idx_t ParseDeltaVersionFromAtClause(const BoundAtClause &at_clause) {
 	if (at_clause.Unit() != "version") {
 		throw InvalidConfigurationException("Delta tables only support at_clause with unit 'version'");
 	}
-	Value version_value = at_clause.GetValue();
-	if (!version_value.DefaultTryCastAs(LogicalType::UBIGINT, false)) {
+	auto version_value = at_clause.GetValue().DefaultTryCastAs(LogicalType::UBIGINT);
+	if (!version_value) {
 		throw InvalidInputException("Failed to parse version number '%s' into a valid version",
 		                            at_clause.GetValue().ToString().c_str());
 	}
-	return version_value.GetValue<idx_t>();
+	return version_value->GetValue<idx_t>();
 }
 
 DeltaCatalog::DeltaCatalog(AttachedDatabase &db_p, const string &path, AccessMode access_mode)
