diff --git a/src/include/storage/delta_table_entry.hpp b/src/include/storage/delta_table_entry.hpp
index e92b40e..8a9569e 100644
--- a/src/include/storage/delta_table_entry.hpp
+++ b/src/include/storage/delta_table_entry.hpp
@@ -23,6 +23,8 @@ public:
 	~DeltaTableEntry();
 
 public:
+	const ColumnList &GetColumns() const override;
+
 	unique_ptr<BaseStatistics> GetStatistics(ClientContext &context, column_t column_id) override;
 
 	TableFunction GetScanFunction(ClientContext &context, unique_ptr<FunctionData> &bind_data) override;
@@ -41,6 +43,8 @@ public:
 	shared_ptr<DeltaMultiFileList> snapshot;
 
 protected:
+	ColumnList columns;
+
 	TableFunction GetScanFunctionInternal(ClientContext &context, unique_ptr<FunctionData> &bind_data,
 	                                      optional_ptr<const EntryLookupInfo> lookup_info);
 };
diff --git a/src/storage/delta_table_entry.cpp b/src/storage/delta_table_entry.cpp
index 9fdc2f7..0f63cd1 100644
--- a/src/storage/delta_table_entry.cpp
+++ b/src/storage/delta_table_entry.cpp
@@ -15,12 +15,16 @@
 namespace duckdb {
 
 DeltaTableEntry::DeltaTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, CreateTableInfo &info)
-    : TableCatalogEntry(catalog, schema, info) {
+    : TableCatalogEntry(catalog, schema, info), columns(std::move(info.columns)) {
 	this->internal = false;
 }
 
 DeltaTableEntry::~DeltaTableEntry() = default;
 
+const ColumnList &DeltaTableEntry::GetColumns() const {
+	return columns;
+}
+
 unique_ptr<BaseStatistics> DeltaTableEntry::GetStatistics(ClientContext &context, column_t column_id) {
 	return nullptr;
 }
