diff --git a/src/delta_utils.cpp b/src/delta_utils.cpp
index 8c08714..708dd0a 100644
--- a/src/delta_utils.cpp
+++ b/src/delta_utils.cpp
@@ -3,6 +3,7 @@
 #include <list>
 
 #include "delta_log_types.hpp"
+#include "functions/delta_scan/delta_multi_file_list.hpp"
 #include "duckdb/common/operator/decimal_cast_operators.hpp"
 
 #include "duckdb.hpp"
@@ -851,13 +852,13 @@ KernelUtils::UnpackTransformExpression(const vector<unique_ptr<ParsedExpression>
 }
 
 PredicateVisitor::PredicateVisitor(const vector<DeltaMultiFileColumnDefinition> &columns,
-                                   optional_ptr<const TableFilterSet> filters) {
+                                   optional_ptr<const DeltaTableFilters> filters) {
 	predicate = this;
 	visitor = (uintptr_t(*)(void *, ffi::KernelExpressionVisitorState *)) & VisitPredicate;
 
 	if (filters) {
-		for (auto &filter : filters->filters) {
-			column_filters[columns[filter.first].name] = filter.second.get();
+		for (auto &entry : *filters) {
+			column_filters[columns[entry.first].name] = entry.second.get();
 		}
 	}
 }
diff --git a/src/functions/delta_scan/delta_multi_file_list.cpp b/src/functions/delta_scan/delta_multi_file_list.cpp
index 7d918bd..06483e7 100644
--- a/src/functions/delta_scan/delta_multi_file_list.cpp
+++ b/src/functions/delta_scan/delta_multi_file_list.cpp
@@ -787,20 +787,22 @@ void DeltaMultiFileList::EnsureScanInitialized() const {
 }
 
 unique_ptr<DeltaMultiFileList> DeltaMultiFileList::PushdownInternal(ClientContext &context,
-                                                                    TableFilterSet &new_filters) const {
+                                                                    TableFilterSet &new_filters,
+                                                                    vector<column_t> column_indexes) const {
 	auto filtered_list = make_uniq<DeltaMultiFileList>(context, paths[0].path, version);
 
-	TableFilterSet result_filter_set;
+	DeltaTableFilters result_filter_set;
 
 	// Add pre-existing filters
-	for (auto &entry : table_filters.filters) {
-		result_filter_set.PushFilter(ColumnIndex(entry.first), entry.second->Copy());
+	for (auto &entry : table_filters) {
+		result_filter_set.PushFilter(entry.first, entry.second->Copy());
 	}
 
 	// Add new filters
-	for (auto &entry : new_filters.filters) {
-		if (entry.first < global_columns.size()) {
-			result_filter_set.PushFilter(ColumnIndex(entry.first), entry.second->Copy());
+	for (auto &entry : new_filters) {
+		auto &column_id = column_indexes[entry.GetIndex()];
+		if (column_id < global_columns.size()) {
+			result_filter_set.PushFilter(column_id, entry.Filter().Copy());
 		}
 	}
 
@@ -849,11 +851,11 @@ unique_ptr<MultiFileList> DeltaMultiFileList::ComplexFilterPushdown(ClientContex
 
 	vector<FilterPushdownResult> pushdown_results;
 	auto filter_set = combiner.GenerateTableScanFilters(info.column_indexes, pushdown_results);
-	if (filter_set.filters.empty()) {
+	if (!filter_set.HasFilters()) {
 		return nullptr;
 	}
 
-	auto filtered_list = PushdownInternal(context, filter_set);
+	auto filtered_list = PushdownInternal(context, filter_set, info.column_ids);
 
 	ReportFilterPushdown(context, *filtered_list, info.column_ids, "constant", info);
 
@@ -916,24 +918,24 @@ void DeltaMultiFileList::ReportFilterPushdown(ClientContext &context, DeltaMulti
 
 	// Report the new filters
 	vector<Value> old_filters_value_list;
-	for (auto &f : table_filters.filters) {
-		auto &column_index = f.first;
-		auto &filter = f.second;
-		if (column_index < global_columns.size()) {
-			auto &col_name = global_columns[column_index].name;
-			old_filters_value_list.push_back(filter->ToString(col_name));
+	for (auto &entry : table_filters) {
+		auto &filter = *entry.second;
+		auto column_id = entry.first;
+		if (column_id < global_columns.size()) {
+			auto &col_name = global_columns[column_id].name;
+			old_filters_value_list.push_back(filter.ToString(col_name));
 		}
 	}
 	auto old_filters_value = Value::LIST(LogicalType::VARCHAR, old_filters_value_list);
 
 	// Report the new filters
 	vector<Value> filters_value_list;
-	for (auto &f : new_list.table_filters.filters) {
-		auto &column_index = f.first;
-		auto &filter = f.second;
-		if (column_index < global_columns.size()) {
-			auto &col_name = global_columns[column_index].name;
-			filters_value_list.push_back(filter->ToString(col_name));
+	for (auto &entry : new_list.table_filters) {
+		auto column_id = entry.first;
+		auto &filter = *entry.second;
+		if (column_id < global_columns.size()) {
+			auto &col_name = global_columns[column_id].name;
+			filters_value_list.push_back(filter.ToString(col_name));
 		}
 	}
 	auto filters_value = Value::LIST(LogicalType::VARCHAR, filters_value_list);
@@ -970,24 +972,25 @@ DeltaMultiFileList::DynamicFilterPushdown(ClientContext &context, const MultiFil
 		return nullptr;
 	}
 
-	if (filters.filters.empty()) {
+	if (!filters.HasFilters()) {
 		return nullptr;
 	}
 
 	TableFilterSet filters_copy;
-	for (auto &filter : filters.filters) {
-		auto column_id = column_ids[filter.first];
-		auto previously_pushed_down_filter = this->table_filters.filters.find(column_id);
-		if (previously_pushed_down_filter != this->table_filters.filters.end() &&
-		    filter.second->Equals(*previously_pushed_down_filter->second)) {
+	for (auto &entry : filters) {
+		auto proj_id = entry.GetIndex();
+		auto &filter = entry.Filter();
+		auto column_id = column_ids[proj_id];
+		auto previously_pushed_down_filter = table_filters.TryGetFilterByColumnIndex(column_id);
+		if (previously_pushed_down_filter && filter.Equals(*previously_pushed_down_filter)) {
 			// Skip filters that we already have pushed down
 			continue;
 		}
-		filters_copy.PushFilter(ColumnIndex(column_id), filter.second->Copy());
+		filters_copy.PushFilter(proj_id, filter.Copy());
 	}
 
-	if (!filters_copy.filters.empty()) {
-		auto new_snap = PushdownInternal(context, filters_copy);
+	if (filters_copy.HasFilters()) {
+		auto new_snap = PushdownInternal(context, filters_copy, column_ids);
 		ReportFilterPushdown(context, *new_snap, column_ids, "dynamic", nullptr);
 		return std::move(new_snap);
 	}
diff --git a/src/include/delta_utils.hpp b/src/include/delta_utils.hpp
index 3fdce7f..11acc31 100644
--- a/src/include/delta_utils.hpp
+++ b/src/include/delta_utils.hpp
@@ -11,7 +11,8 @@
 #include "duckdb/parser/expression/conjunction_expression.hpp"
 #include "duckdb/common/error_data.hpp"
 #include "duckdb/parser/expression/comparison_expression.hpp"
-#include <duckdb/planner/filter/null_filter.hpp>
+#include "duckdb/planner/table_filter_set.hpp"
+#include "duckdb/planner/filter/null_filter.hpp"
 #include <iostream>
 
 #include "duckdb/planner/tableref/bound_at_clause.hpp"
@@ -423,15 +424,16 @@ protected:
 };
 
 typedef SharedKernelPointer<ffi::SharedSnapshot, ffi::free_snapshot> SharedKernelSnapshot;
+struct DeltaTableFilters;
 
 class PredicateVisitor : public ffi::EnginePredicate {
 public:
-	PredicateVisitor(const vector<DeltaMultiFileColumnDefinition> &columns, optional_ptr<const TableFilterSet> filters);
+	PredicateVisitor(const vector<DeltaMultiFileColumnDefinition> &columns, optional_ptr<const DeltaTableFilters> filters);
 
 	ErrorData error_data;
 
 private:
-	unordered_map<string, TableFilter *> column_filters;
+	unordered_map<string, optional_ptr<const TableFilter>> column_filters;
 
 	static uintptr_t VisitPredicate(PredicateVisitor *predicate, ffi::KernelExpressionVisitorState *state);
 
diff --git a/src/include/functions/delta_scan/delta_multi_file_list.hpp b/src/include/functions/delta_scan/delta_multi_file_list.hpp
index b196db7..de2623e 100644
--- a/src/include/functions/delta_scan/delta_multi_file_list.hpp
+++ b/src/include/functions/delta_scan/delta_multi_file_list.hpp
@@ -41,6 +41,41 @@ struct DeltaFileMetaData {
 	unique_ptr<vector<unique_ptr<ParsedExpression>>> transform_expression;
 };
 
+struct DeltaTableFilters {
+	using filter_set_t = unordered_map<idx_t, unique_ptr<TableFilter>>;
+	using iterator = filter_set_t::iterator;
+	using const_iterator = filter_set_t::const_iterator;
+public:
+	bool HasFilters() const {
+		return !table_filters.empty();
+	}
+	void PushFilter(column_t column_idx, unique_ptr<TableFilter> table_filter) {
+		table_filters[column_idx] = std::move(table_filter);
+	}
+	optional_ptr<const TableFilter> TryGetFilterByColumnIndex(column_t column_idx) const {
+		auto entry = table_filters.find(column_idx);
+		if (entry == table_filters.end()) {
+			return nullptr;
+		}
+		return entry->second.get();
+	}
+
+	iterator begin() { // NOLINT: match stl API
+		return table_filters.begin();
+	}
+	iterator end() { // NOLINT: match stl API
+		return table_filters.end();
+	}
+	const_iterator begin() const { // NOLINT: match stl API
+		return table_filters.begin();
+	}
+	const_iterator end() const { // NOLINT: match stl API
+		return table_filters.end();
+	}
+private:
+	filter_set_t table_filters;
+};
+
 // Constraint only for internal delta extension use
 // Todo: refactor to use duckdb constraint classes, updating the DuckDB side NotNullConstraint
 class NestedNotNullConstraint {
@@ -73,7 +108,7 @@ public:
 	                                                const vector<column_t> &column_ids,
 	                                                TableFilterSet &filters) const override;
 
-	unique_ptr<DeltaMultiFileList> PushdownInternal(ClientContext &context, TableFilterSet &new_filters) const;
+	unique_ptr<DeltaMultiFileList> PushdownInternal(ClientContext &context, TableFilterSet &new_filters, vector<column_t> column_indexes) const;
 
 	vector<OpenFileInfo> GetAllFiles() const override;
 	FileExpandResult GetExpandResult() const override;
@@ -142,7 +177,7 @@ protected:
 	mutable vector<unique_ptr<DeltaFileMetaData>> metadata;
 
 	mutable vector<OpenFileInfo> resolved_files;
-	mutable TableFilterSet table_filters;
+	mutable DeltaTableFilters table_filters;
 
 	mutable vector<NestedNotNullConstraint> not_null_constraints;
 	mutable bool has_null_constraints_in_arrays = false;
diff --git a/src/storage/delta_insert.cpp b/src/storage/delta_insert.cpp
index de50268..c2bae88 100644
--- a/src/storage/delta_insert.cpp
+++ b/src/storage/delta_insert.cpp
@@ -368,7 +368,6 @@ PhysicalOperator &DeltaCatalog::PlanInsert(ClientContext &context, PhysicalPlanG
 	physical_copy_ref.file_extension = "parquet";
 	physical_copy_ref.overwrite_mode = CopyOverwriteMode::COPY_OVERWRITE_OR_IGNORE;
 	physical_copy_ref.per_thread_output = false;
-	physical_copy_ref.rotate = false;
 	physical_copy_ref.return_type = CopyFunctionReturnType::WRITTEN_FILE_STATISTICS;
 	physical_copy_ref.write_partition_columns = true;
 	physical_copy_ref.children.push_back(*plan);
