diff --git a/src/delta_extension.cpp b/src/delta_extension.cpp
index 2485cd9..568c286 100644
--- a/src/delta_extension.cpp
+++ b/src/delta_extension.cpp
@@ -79,7 +79,7 @@ static void LoadInternal(ExtensionLoader &loader) {
 
 	// Register the "single table" delta catalog (to ATTACH a single delta table)
 	auto &config = DBConfig::GetConfig(loader.GetDatabaseInstance());
-	config.storage_extensions["delta"] = make_uniq<DeltaStorageExtension>();
+	StorageExtension::Register(config, "delta", make_shared_ptr<DeltaStorageExtension>());
 
 	config.AddExtensionOption("delta_scan_explain_files_filtered",
 	                          "Adds the filtered files to the explain output. Warning: this may impact performance of "
diff --git a/src/delta_utils.cpp b/src/delta_utils.cpp
index 3edbbea..bfcede2 100644
--- a/src/delta_utils.cpp
+++ b/src/delta_utils.cpp
@@ -11,6 +11,7 @@
 #include "duckdb/main/database.hpp"
 #include "duckdb/parser/parsed_data/create_scalar_function_info.hpp"
 #include "duckdb/common/exception/conversion_exception.hpp"
+#include "duckdb/parser/expression/columnref_expression.hpp"
 #include "duckdb/parser/expression/conjunction_expression.hpp"
 #include "duckdb/parser/expression/comparison_expression.hpp"
 #include "duckdb/parser/expression/function_expression.hpp"
@@ -1060,8 +1061,8 @@ LogLevel LoggerCallback::GetDuckDBLogLevel(ffi::Level level) {
 		return LogLevel::LOG_DEBUG;
 	case ffi::Level::INFO:
 		return LogLevel::LOG_INFO;
-	case ffi::Level::WARN:
-		return LogLevel::LOG_WARN;
+	case ffi::Level::WARNING:
+		return LogLevel::LOG_WARNING;
 	case ffi::Level::ERROR:
 		return LogLevel::LOG_ERROR;
 	default:
diff --git a/src/functions/delta_scan/delta_multi_file_list.cpp b/src/functions/delta_scan/delta_multi_file_list.cpp
index f94dee5..49bdce7 100644
--- a/src/functions/delta_scan/delta_multi_file_list.cpp
+++ b/src/functions/delta_scan/delta_multi_file_list.cpp
@@ -9,11 +9,13 @@
 #include "duckdb/main/query_profiler.hpp"
 #include "duckdb/main/secret/secret_manager.hpp"
 #include "duckdb/optimizer/filter_combiner.hpp"
+#include "duckdb/parser/expression/columnref_expression.hpp"
 #include "duckdb/parser/expression/function_expression.hpp"
 #include "duckdb/planner/table_filter.hpp"
 #include "duckdb/planner/operator/logical_get.hpp"
 #include "duckdb/parser/constraint.hpp"
 #include "duckdb/parser/constraints/not_null_constraint.hpp"
+#include "duckdb/planner/expression/bound_columnref_expression.hpp"
 
 #include <regex>
 
@@ -524,7 +526,7 @@ void ScanDataCallBack::VisitData(ffi::NullableCvoid engine_context,
 }
 
 DeltaMultiFileList::DeltaMultiFileList(ClientContext &context_p, const string &path, idx_t version_p)
-    : MultiFileList({ToDeltaPath(path)}, FileGlobOptions::ALLOW_EMPTY), version (version_p), context(context_p) {
+    : delta_path(ToDeltaPath(path)), version (version_p), context(context_p) {
 
     Value setting_res;
     auto res = context.TryGetCurrentSetting("variant_legacy_encoding", setting_res);
@@ -536,8 +538,8 @@ DeltaMultiFileList::DeltaMultiFileList(ClientContext &context_p, const string &p
     }
 }
 
-string DeltaMultiFileList::GetPath() const {
-	return GetPaths()[0].path;
+const string &DeltaMultiFileList::GetPath() const {
+	return delta_path;
 }
 
 string DeltaMultiFileList::ToDuckDBPath(const string &raw_path) {
@@ -687,16 +689,16 @@ idx_t DeltaMultiFileList::GetTotalFileCountInternal() const {
 	return resolved_files.size();
 }
 
-OpenFileInfo DeltaMultiFileList::GetFile(idx_t i) {
+OpenFileInfo DeltaMultiFileList::GetFile(idx_t i) const {
 	// TODO: profile this: we should be able to use atomics here to optimize
 	unique_lock<mutex> lck(lock);
 	return GetFileInternal(i);
 }
 
 void DeltaMultiFileList::InitializeSnapshot() const {
-	auto path_slice = KernelUtils::ToDeltaString(paths[0].path);
+	auto path_slice = KernelUtils::ToDeltaString(delta_path);
 
-	auto interface_builder = CreateBuilder(context, paths[0].path);
+	auto interface_builder = CreateBuilder(context, delta_path);
 	extern_engine = TryUnpackKernelResult(ffi::builder_build(interface_builder));
 
 	if (!snapshot) {
@@ -731,7 +733,7 @@ void DeltaMultiFileList::InitializeScan() const {
 	scan = TryUnpackKernelResult(ffi::scan(snapshot_ref.GetPtr(), extern_engine.get(), &visitor));
 
 	if (visitor.error_data.HasError()) {
-		throw IOException("Failed to initialize Scan for Delta table at '%s'. Original error: '%s'", paths[0].path,
+		throw IOException("Failed to initialize Scan for Delta table at '%s'. Original error: '%s'", delta_path,
 		                  visitor.error_data.Message());
 	}
 
@@ -793,7 +795,7 @@ void DeltaMultiFileList::EnsureScanInitialized() const {
 
 unique_ptr<DeltaMultiFileList> DeltaMultiFileList::PushdownInternal(ClientContext &context,
                                                                     TableFilterSet &new_filters) const {
-	auto filtered_list = make_uniq<DeltaMultiFileList>(context, paths[0].path, version);
+	auto filtered_list = make_uniq<DeltaMultiFileList>(context, delta_path, version);
 
 	TableFilterSet result_filter_set;
 
@@ -836,7 +838,7 @@ static DeltaFilterPushdownMode GetDeltaFilterPushdownMode(ClientContext &context
 unique_ptr<MultiFileList> DeltaMultiFileList::ComplexFilterPushdown(ClientContext &context,
                                                                     const MultiFileOptions &options,
                                                                     MultiFilePushdownInfo &info,
-                                                                    vector<unique_ptr<Expression>> &filters) {
+                                                                    vector<unique_ptr<Expression>> &filters) const {
 	auto pushdown_mode = GetDeltaFilterPushdownMode(context, options);
 	if (pushdown_mode == DeltaFilterPushdownMode::NONE || pushdown_mode == DeltaFilterPushdownMode::DYNAMIC_ONLY) {
 		return nullptr;
@@ -1000,7 +1002,7 @@ DeltaMultiFileList::DynamicFilterPushdown(ClientContext &context, const MultiFil
 	return nullptr;
 }
 
-vector<OpenFileInfo> DeltaMultiFileList::GetAllFiles() {
+vector<OpenFileInfo> DeltaMultiFileList::GetAllFiles() const {
 	unique_lock<mutex> lck(lock);
 	idx_t i = resolved_files.size();
 	// TODO: this can probably be improved
@@ -1010,7 +1012,7 @@ vector<OpenFileInfo> DeltaMultiFileList::GetAllFiles() {
 	return resolved_files;
 }
 
-FileExpandResult DeltaMultiFileList::GetExpandResult() {
+FileExpandResult DeltaMultiFileList::GetExpandResult() const {
 	// We avoid exposing the ExpandResult to DuckDB here because we want to materialize the Snapshot as late as
 	// possible: materializing too early (GetExpandResult is called *before* filter pushdown by the Parquet scanner),
 	// will lead into needing to create 2 scans of the snapshot TODO: we need to investigate if this is actually a
@@ -1018,12 +1020,12 @@ FileExpandResult DeltaMultiFileList::GetExpandResult() {
 	return FileExpandResult::MULTIPLE_FILES;
 }
 
-idx_t DeltaMultiFileList::GetTotalFileCount() {
+idx_t DeltaMultiFileList::GetTotalFileCount() const {
 	unique_lock<mutex> lck(lock);
 	return GetTotalFileCountInternal();
 }
 
-unique_ptr<NodeStatistics> DeltaMultiFileList::GetCardinality(ClientContext &context) {
+unique_ptr<NodeStatistics> DeltaMultiFileList::GetCardinality(ClientContext &context) const {
 	// This also ensures all files are expanded
 	auto total_file_count = DeltaMultiFileList::GetTotalFileCount();
 
diff --git a/src/include/delta_kernel_ffi.hpp b/src/include/delta_kernel_ffi.hpp
index 07358d3..b714663 100644
--- a/src/include/delta_kernel_ffi.hpp
+++ b/src/include/delta_kernel_ffi.hpp
@@ -68,7 +68,7 @@ enum class KernelError {
 /// Level::ERROR is the lowest, and Level::TRACE the highest.
 enum class Level {
   ERROR = 0,
-  WARN = 1,
+  WARNING = 1,
   INFO = 2,
 	DEBUGGING = 3,
   TRACE = 4,
diff --git a/src/include/functions/delta_scan/delta_multi_file_list.hpp b/src/include/functions/delta_scan/delta_multi_file_list.hpp
index b12ed7f..7d20329 100644
--- a/src/include/functions/delta_scan/delta_multi_file_list.hpp
+++ b/src/include/functions/delta_scan/delta_multi_file_list.hpp
@@ -56,7 +56,7 @@ class DeltaMultiFileList : public MultiFileList {
 
 public:
 	DeltaMultiFileList(ClientContext &context, const string &path, idx_t version);
-	string GetPath() const;
+	const string &GetPath() const;
 	static string ToDuckDBPath(const string &raw_path);
 	static string ToDeltaPath(const string &raw_path);
 
@@ -65,7 +65,7 @@ public:
 	void Bind(vector<LogicalType> &return_types, vector<string> &names);
 	unique_ptr<MultiFileList> ComplexFilterPushdown(ClientContext &context, const MultiFileOptions &options,
 	                                                MultiFilePushdownInfo &info,
-	                                                vector<unique_ptr<Expression>> &filters) override;
+	                                                vector<unique_ptr<Expression>> &filters) const override;
 
 	unique_ptr<MultiFileList> DynamicFilterPushdown(ClientContext &context, const MultiFileOptions &options,
 	                                                const vector<string> &names, const vector<LogicalType> &types,
@@ -74,10 +74,10 @@ public:
 
 	unique_ptr<DeltaMultiFileList> PushdownInternal(ClientContext &context, TableFilterSet &new_filters) const;
 
-	vector<OpenFileInfo> GetAllFiles() override;
-	FileExpandResult GetExpandResult() override;
-	idx_t GetTotalFileCount() override;
-	unique_ptr<NodeStatistics> GetCardinality(ClientContext &context) override;
+	vector<OpenFileInfo> GetAllFiles() const override;
+	FileExpandResult GetExpandResult() const override;
+	idx_t GetTotalFileCount() const override;
+	unique_ptr<NodeStatistics> GetCardinality(ClientContext &context) const override;
 	DeltaFileMetaData &GetMetaData(idx_t index) const;
 	idx_t GetVersion();
 	vector<string> GetPartitionColumns();
@@ -86,13 +86,13 @@ public:
     vector<NestedNotNullConstraint> GetNestedNotNullConstraints() const;
     bool HasNullConstraintsInArrays() const;
 
-    bool VariantEnabled() {
+    bool VariantEnabled() const {
         return enable_variant;
     }
 
 protected:
 	//! Get the i-th expanded file
-	OpenFileInfo GetFile(idx_t i) override;
+	OpenFileInfo GetFile(idx_t i) const override;
 
 protected:
 	OpenFileInfo GetFileInternal(idx_t i) const;
@@ -127,6 +127,8 @@ protected:
 	mutable mutex lock;
 	mutable idx_t version;
 
+	string delta_path;
+
 	//! Delta Kernel Structures
 	mutable shared_ptr<SharedKernelSnapshot> snapshot;
 
diff --git a/src/include/storage/delta_insert.hpp b/src/include/storage/delta_insert.hpp
index cf74ede..1cc4cb7 100644
--- a/src/include/storage/delta_insert.hpp
+++ b/src/include/storage/delta_insert.hpp
@@ -35,7 +35,7 @@ public:
 
 public:
 	// // Source interface
-	SourceResultType GetData(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const override;
+	SourceResultType GetDataInternal(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const override;
 
 	bool IsSource() const override {
 		return true;
diff --git a/src/storage/delta_insert.cpp b/src/storage/delta_insert.cpp
index ac73e47..2a85ef8 100644
--- a/src/storage/delta_insert.cpp
+++ b/src/storage/delta_insert.cpp
@@ -1,7 +1,5 @@
 #include "storage/delta_insert.hpp"
 
-#include <duckdb/common/sort/partition_state.hpp>
-
 #include "duckdb/catalog/catalog_entry/copy_function_catalog_entry.hpp"
 #include "duckdb/main/client_data.hpp"
 #include "duckdb/planner/operator/logical_copy_to_file.hpp"
@@ -19,6 +17,7 @@
 #include "duckdb/planner/expression/bound_cast_expression.hpp"
 #include "duckdb/planner/expression/bound_reference_expression.hpp"
 #include "functions/delta_scan/delta_multi_file_list.hpp"
+#include "duckdb/catalog/catalog_entry_retriever.hpp"
 
 namespace duckdb {
 
@@ -219,7 +218,7 @@ SinkResultType DeltaInsert::Sink(ExecutionContext &context, DataChunk &chunk, Op
 //===--------------------------------------------------------------------===//
 // GetData
 //===--------------------------------------------------------------------===//
-SourceResultType DeltaInsert::GetData(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const {
+SourceResultType DeltaInsert::GetDataInternal(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const {
     auto &global_state = sink_state->Cast<DeltaInsertGlobalState>();
     auto value = Value::BIGINT(global_state.insert_count);
     chunk.SetCardinality(1);
@@ -290,7 +289,7 @@ PhysicalOperator &DeltaCatalog::PlanInsert(ClientContext &context, PhysicalPlanG
         table_entry = op.table.Cast<DeltaTableEntry>();
     }
 
-    string delta_path =  table_entry->snapshot->GetPaths()[0].path;
+    auto &delta_path =  table_entry->snapshot->GetPath();
 
     // Create Copy Info
     auto info = make_uniq<CopyInfo>();
diff --git a/src/storage/delta_schema_entry.cpp b/src/storage/delta_schema_entry.cpp
index 1a60f23..f87d2f9 100644
--- a/src/storage/delta_schema_entry.cpp
+++ b/src/storage/delta_schema_entry.cpp
@@ -14,7 +14,7 @@
 #include "duckdb/common/unordered_set.hpp"
 #include "duckdb/parser/parsed_expression_iterator.hpp"
 #include "duckdb/planner/tableref/bound_at_clause.hpp"
-
+#include "duckdb/parser/expression/columnref_expression.hpp"
 namespace duckdb {
 
 DeltaSchemaEntry::DeltaSchemaEntry(Catalog &catalog, CreateSchemaInfo &info) : SchemaCatalogEntry(catalog, info) {
diff --git a/src/storage/delta_transaction.cpp b/src/storage/delta_transaction.cpp
index 1a0002c..7a6d377 100644
--- a/src/storage/delta_transaction.cpp
+++ b/src/storage/delta_transaction.cpp
@@ -138,11 +138,11 @@ struct WriteMetaData {
 
     WriteMetaData(DeltaMultiFileList &snapshot, vector<DeltaDataFile> &outstanding_appends) : WriteMetaData() {
         for (const auto &file : outstanding_appends) {
-            auto table_path = snapshot.GetPaths()[0];
+            auto &table_path = snapshot.GetPath();
             auto file_without_double_slash = StringUtil::Replace(file.file_name, "\\", "/");
             // auto file_split = StringUtil::Split(file, "/");
             // auto file_name = file_split[file_split.size()-1];
-            auto file_name = file.file_name.substr(table_path.path.size());
+            auto file_name = file.file_name.substr(table_path.size());
             InsertionOrderPreservingMap<string> partitions = {};
 
             // TODO: probably horribly wrong
@@ -267,7 +267,7 @@ void DeltaTransaction::InitializeTransaction(ClientContext &context) {
     D_ASSERT(table_entry);
 
     // Start the kernel transaction
-    string path =  table_entry->snapshot->GetPaths()[0].path;
+    auto &path =  table_entry->snapshot->GetPath();
     auto path_slice = KernelUtils::ToDeltaString(path);
     auto new_kernel_transaction = table_entry->snapshot->TryUnpackKernelResult(ffi::transaction(path_slice, table_entry->snapshot->extern_engine.get()));
 
