diff --git a/src/catalog/glue_schema_entry.cpp b/src/catalog/glue_schema_entry.cpp
index 95c4462..8c8f308 100644
--- a/src/catalog/glue_schema_entry.cpp
+++ b/src/catalog/glue_schema_entry.cpp
@@ -655,7 +655,10 @@ void GlueSchemaEntry::Alter(CatalogTransaction transaction, AlterInfo &info) {
 	}
 	case AlterTableType::ALTER_COLUMN_TYPE: {
 		auto &change = alter_table.Cast<ChangeColumnTypeInfo>();
-		auto &name = change.column_name.GetIdentifierName();
+		if (change.column_path.size() > 1) {
+			throw NotImplementedException("Changing the type of a nested field is not yet supported");
+		}
+		auto &name = change.column_path[0].GetIdentifierName();
 		if (is_partition_key(name)) {
 			throw CatalogException("Column \"%s\" is a partition key of table \"%s\" and its type can not be "
 			                       "changed",
