diff --git a/src/hnsw/hnsw_index_macros.cpp b/src/hnsw/hnsw_index_macros.cpp
index 7bcb7bb..10ec6e3 100644
--- a/src/hnsw/hnsw_index_macros.cpp
+++ b/src/hnsw/hnsw_index_macros.cpp
@@ -91,7 +91,7 @@ static void RegisterTableMacro(ExtensionLoader &loader, const string &name, cons
 
 	for (auto &param : named_params) {
 		func->parameters.push_back(make_uniq<ColumnRefExpression>(param.first));
-		func->default_parameters[Identifier(param.first)] = make_uniq<ConstantExpression>(Value(param.second));
+		func->default_parameters[Identifier(param.first)] = ConstantExpression::FromValue(Value(param.second));
 	}
 
 	CreateMacroInfo info(CatalogType::TABLE_MACRO_ENTRY);
