diff --git a/src/hnsw/hnsw_optimize_join.cpp b/src/hnsw/hnsw_optimize_join.cpp
index 9b90e97..eff7fe4 100644
--- a/src/hnsw/hnsw_optimize_join.cpp
+++ b/src/hnsw/hnsw_optimize_join.cpp
@@ -453,7 +453,7 @@ bool HNSWIndexJoinOptimizer::TryOptimize(Binder &binder, ClientContext &context,
 
 	auto &delim_get = (*delim_get_ptr)->Cast<LogicalDelimGet>();
 	auto &inner_get = (*inner_get_ptr)->Cast<LogicalGet>();
-	if (inner_get.function.name != "seq_scan") {
+	if (inner_get.function.GetName() != "seq_scan") {
 		return false;
 	}
 
@@ -847,7 +847,7 @@ bool HNSWIndexJoinOptimizer::TryOptimizeArgMin(Binder &binder, ClientContext &co
 	}
 	auto &delim_get = (*delim_get_ptr)->Cast<LogicalDelimGet>();
 	auto &inner_get = (*inner_get_ptr)->Cast<LogicalGet>();
-	if (inner_get.function.name != "seq_scan") {
+	if (inner_get.function.GetName() != "seq_scan") {
 		return false;
 	}
 
@@ -1205,11 +1205,11 @@ bool HNSWIndexJoinOptimizer::TryOptimizeArgMinCte(Binder &binder, ClientContext
 	auto &cp_rhs = cross_product.children[1];
 	unique_ptr<LogicalOperator> *corr_source_ptr = nullptr;
 	unique_ptr<LogicalOperator> *inner_get_ptr = nullptr;
-	if (cp_lhs->type == LogicalOperatorType::LOGICAL_GET && cp_lhs->Cast<LogicalGet>().function.name == "seq_scan") {
+	if (cp_lhs->type == LogicalOperatorType::LOGICAL_GET && cp_lhs->Cast<LogicalGet>().function.GetName() == "seq_scan") {
 		inner_get_ptr = &cp_lhs;
 		corr_source_ptr = &cp_rhs;
 	} else if (cp_rhs->type == LogicalOperatorType::LOGICAL_GET &&
-	           cp_rhs->Cast<LogicalGet>().function.name == "seq_scan") {
+	           cp_rhs->Cast<LogicalGet>().function.GetName() == "seq_scan") {
 		inner_get_ptr = &cp_rhs;
 		corr_source_ptr = &cp_lhs;
 	} else {
diff --git a/src/hnsw/hnsw_optimize_scan.cpp b/src/hnsw/hnsw_optimize_scan.cpp
index c429cd1..88eff3f 100644
--- a/src/hnsw/hnsw_optimize_scan.cpp
+++ b/src/hnsw/hnsw_optimize_scan.cpp
@@ -175,7 +175,7 @@ public:
 
 		auto &get = (*get_ptr)->Cast<LogicalGet>();
 		// Check if the get is a table scan
-		if (get.function.name != "seq_scan") {
+		if (get.function.GetName() != "seq_scan") {
 			return false;
 		}
 
@@ -261,7 +261,7 @@ public:
 		}
 
 		const auto cardinality = get.function.cardinality(context, bind_data.get());
-		get.function = HNSWIndexScanFunction::GetFunction();
+		get.function = BoundTableFunction(HNSWIndexScanFunction::GetFunction());
 		get.has_estimated_cardinality = cardinality->has_estimated_cardinality;
 		get.estimated_cardinality = cardinality->estimated_cardinality;
 		get.bind_data = std::move(bind_data);
@@ -308,7 +308,7 @@ public:
 				auto &child = plan->children[0];
 
 				if (child->children[0]->type == LogicalOperatorType::LOGICAL_GET &&
-				    child->children[0]->Cast<LogicalGet>().function.name == "hnsw_index_scan") {
+				    child->children[0]->Cast<LogicalGet>().function.GetName() == "hnsw_index_scan") {
 					auto &parent_projection = plan->Cast<LogicalProjection>();
 					auto &child_projection = child->Cast<LogicalProjection>();
 
diff --git a/src/hnsw/hnsw_optimize_topk.cpp b/src/hnsw/hnsw_optimize_topk.cpp
index 32f702f..01a0561 100644
--- a/src/hnsw/hnsw_optimize_topk.cpp
+++ b/src/hnsw/hnsw_optimize_topk.cpp
@@ -109,7 +109,7 @@ public:
 		}
 
 		auto &get = (*get_ptr)->Cast<LogicalGet>();
-		if (get.function.name != "seq_scan") {
+		if (get.function.GetName() != "seq_scan") {
 			return false;
 		}
 
@@ -194,7 +194,7 @@ public:
 		}
 
 		// Replace the aggregate with a index scan + projection
-		get.function = HNSWIndexScanFunction::GetFunction();
+		get.function = BoundTableFunction(HNSWIndexScanFunction::GetFunction());
 		const auto cardinality = get.function.cardinality(context, bind_data.get());
 		get.has_estimated_cardinality = cardinality->has_estimated_cardinality;
 		get.estimated_cardinality = cardinality->estimated_cardinality;
