diff --git a/src/hnsw/hnsw_index_macros.cpp b/src/hnsw/hnsw_index_macros.cpp
index c0bf1e9..b3b8f85 100644
--- a/src/hnsw/hnsw_index_macros.cpp
+++ b/src/hnsw/hnsw_index_macros.cpp
@@ -80,7 +80,7 @@ FROM
 static void RegisterTableMacro(ExtensionLoader &loader, const string &name, const string &query,
                                const vector<string> &params, const child_list_t<Value> &named_params) {
 
-	Parser parser;
+	auto parser = Parser::GetBuiltinParser();
 	parser.ParseQuery(query);
 	const auto &stmt = parser.statements.back();
 	auto &node = stmt->Cast<SelectStatement>().node;
