--- a/src/include/hnsw/hnsw_index_physical_create.hpp
+++ b/src/include/hnsw/hnsw_index_physical_create.hpp
@@ -12,6 +12,7 @@
 #include "duckdb/storage/buffer_manager.hpp"
 #include "duckdb/storage/index.hpp"
 #include "duckdb/storage/index_storage_info.hpp"
+#include "duckdb/catalog/catalog_entry/table_catalog_entry.hpp"
 
 namespace duckdb {
 
--- a/src/hnsw/hnsw_index_plan.cpp
+++ b/src/hnsw/hnsw_index_plan.cpp
@@ -10,6 +10,8 @@
 
 #include "hnsw/hnsw_index.hpp"
 #include "hnsw/hnsw_index_physical_create.hpp"
+#include "duckdb/execution/physical_plan_generator.hpp"
+#include "duckdb/main/attached_database.hpp"
 
 namespace duckdb {
 
--- a/src/hnsw/hnsw_optimize_scan.cpp
+++ b/src/hnsw/hnsw_optimize_scan.cpp
@@ -15,6 +15,7 @@
 #include "hnsw/hnsw.hpp"
 #include "hnsw/hnsw_index.hpp"
 #include "hnsw/hnsw_index_scan.hpp"
+#include "duckdb/planner/column_binding_map.hpp"
 
 namespace duckdb {
 
