diff --git a/src/hnsw/hnsw_index_macros.cpp b/src/hnsw/hnsw_index_macros.cpp
index 774b409..226a2f0 100644
--- a/src/hnsw/hnsw_index_macros.cpp
+++ b/src/hnsw/hnsw_index_macros.cpp
@@ -91,6 +91,7 @@ static void RegisterTableMacro(DatabaseInstance &db, const string &name, const s
 	}
 
 	for (auto &param : named_params) {
+		func->parameters.push_back(make_uniq<ColumnRefExpression>(param.first));
 		func->default_parameters[param.first] = make_uniq<ConstantExpression>(param.second);
 	}
 
@@ -113,4 +114,4 @@ void HNSWModule::RegisterMacros(DatabaseInstance &db) {
 	                   {{"metric", Value("l2sq")}});
 }
 
-} // namespace duckdb
\ No newline at end of file
+} // namespace duckdb
