diff --git a/src/hnsw/hnsw_index.cpp b/src/hnsw/hnsw_index.cpp
index 6878607..94359ee 100644
--- a/src/hnsw/hnsw_index.cpp
+++ b/src/hnsw/hnsw_index.cpp
@@ -501,7 +501,7 @@ IndexStorageInfo HNSWIndex::GetStorageInfo(const bool get_buffers) {
        if (!get_buffers) {
                // use the partial block manager to serialize all allocator data
                auto &block_manager = table_io_manager.GetIndexBlockManager();
-               PartialBlockManager partial_block_manager(block_manager, PartialBlockType::FULL_CHECKPOINT);
+               PartialBlockManager partial_block_manager(block_manager, PartialBlockType::FULL_CHECKPOINT, nullptr);
                linked_block_allocator->SerializeBuffers(partial_block_manager);
                partial_block_manager.FlushPartialBlocks();
        } else {