diff --git a/src/hnsw/hnsw_optimize_expr.cpp b/src/hnsw/hnsw_optimize_expr.cpp
index e519d60..8473c7e 100644
--- a/src/hnsw/hnsw_optimize_expr.cpp
+++ b/src/hnsw/hnsw_optimize_expr.cpp
@@ -65,7 +65,7 @@ unique_ptr<Expression> CosineDistanceRule::Apply(LogicalOperator &op, vector<ref
 		}
 
 		changes_made = true;
-		const auto &func = func_entry->functions.GetFunctionByArguments(context, arg_types);
+		const auto &func = *func_entry->functions.GetFunctionByArguments(context, arg_types);
 		return func.Bind(context, std::move(args));
 	}
 	return nullptr;
diff --git a/src/hnsw/hnsw_optimize_topk.cpp b/src/hnsw/hnsw_optimize_topk.cpp
index 79a5e6a..280b399 100644
--- a/src/hnsw/hnsw_optimize_topk.cpp
+++ b/src/hnsw/hnsw_optimize_topk.cpp
@@ -33,7 +33,7 @@ static unique_ptr<Expression> CreateListOrderByExpr(ClientContext &context, uniq
 		return nullptr;
 	}
 
-	const auto &func = func_entry->functions.GetFunctionByOffset(0);
+	const auto &func = *func_entry->functions.GetFunctionByOffset(0);
 	vector<unique_ptr<Expression>> arguments;
 	arguments.push_back(std::move(elem_expr));
 
