diff --git a/test/extension/autoloading_base.test b/test/extension/autoloading_base.test
index 238f1fd..0c137c5 100644
--- a/test/extension/autoloading_base.test
+++ b/test/extension/autoloading_base.test
@@ -33,12 +33,12 @@ set autoinstall_known_extensions=false
 statement error
 SET s3_region='eu-west-1';
 ----
-<REGEX>:.*Catalog Error.*Setting with name "s3_region" is not in the catalog.*
+<REGEX>:.*Catalog Error.*Setting with name s3_region is not in the catalog.*
 
 statement error
 select * from read_json_auto('data/json/example_n.ndjson');
 ----
-<REGEX>:.*Catalog Error.*Table Function with name "read_json_auto" is not in the catalog.*
+<REGEX>:.*Catalog Error.*Table Function with name read_json_auto is not in the catalog.*
 
 statement error
 select * from thistablefunctionwillnotexistfosho();
diff --git a/test/extension/autoloading_current_setting.test b/test/extension/autoloading_current_setting.test
index 8d4b98d..60fd820 100644
--- a/test/extension/autoloading_current_setting.test
+++ b/test/extension/autoloading_current_setting.test
@@ -21,7 +21,7 @@ set autoinstall_known_extensions=false
 statement error
 select current_setting('s3_region');
 ----
-<REGEX>:.*Catalog Error.*Setting with name "s3_region" is not in the catalog.*
+<REGEX>:.*Catalog Error.*Setting with name s3_region is not in the catalog.*
 
 ### Autoloading, but but not autoinstall
 statement ok
diff --git a/test/extension/autoloading_load_only.test b/test/extension/autoloading_load_only.test
index 66f9770..defbfe7 100644
--- a/test/extension/autoloading_load_only.test
+++ b/test/extension/autoloading_load_only.test
@@ -22,7 +22,7 @@ set autoinstall_known_extensions=false
 statement error
 SET s3_region='eu-west-1';
 ----
-<REGEX>:.*Catalog Error.*Setting with name "s3_region" is not in the catalog.*
+<REGEX>:.*Catalog Error.*Setting with name s3_region is not in the catalog.*
 
 ### Autoloading but not autoinstall, while the extension is not installed: still not working
 statement ok
diff --git a/test/extension/autoloading_reset_setting.test b/test/extension/autoloading_reset_setting.test
index 3473e3b..b1a606f 100644
--- a/test/extension/autoloading_reset_setting.test
+++ b/test/extension/autoloading_reset_setting.test
@@ -22,7 +22,7 @@ set autoinstall_known_extensions=false
 statement error
 RESET s3_region;
 ----
-Catalog Error: Setting with name "s3_region" is not in the catalog, but it exists in the httpfs extension.
+Catalog Error: Setting with name s3_region is not in the catalog, but it exists in the httpfs extension.
 
 ### Autoloading, but no auto install
 statement ok
