diff --git a/test/sql/copy/csv/test_12314.test_slow b/test/sql/copy/csv/test_12314.test_slow
--- a/test/sql/copy/csv/test_12314.test_slow
+++ b/test/sql/copy/csv/test_12314.test_slow
@@ -5,5 +5,6 @@
 require httpfs
 
+require notwindows
 
 statement error
 from read_csv('https://github.com/duckdb/duckdb-data/releases/download/v1.0/sample_data_12314.csv.gz',HEADER = 1,  PARALLEL=false);
diff --git a/test/sql/delete/test_issue_1834.test_slow b/test/sql/delete/test_issue_1834.test_slow
--- a/test/sql/delete/test_issue_1834.test_slow
+++ b/test/sql/delete/test_issue_1834.test_slow
@@ -5,5 +5,7 @@
 require httpfs
 
+require notwindows
+
 statement ok
 CREATE TABLE Person_likes_Comment (creationDate timestamp without time zone not null, id bigint not null, likes_Comment bigint not null);
 
