diff --git a/test/sql/secrets/create_secret_binding.test b/test/sql/secrets/create_secret_binding.test
index 7e5a095..5592b5a 100644
--- a/test/sql/secrets/create_secret_binding.test
+++ b/test/sql/secrets/create_secret_binding.test
@@ -114,7 +114,7 @@ CREATE SECRET incorrect_type (
     USE_SSL 'fliepflap'
 )
 ----
-Binder Error: Failed to cast option 'use_ssl' to type 'BOOLEAN': 'Could not convert string 'fliepflap' to BOOL'
+Binder Error: Failed to cast option "use_ssl" to type 'BOOLEAN': 'Could not convert string 'fliepflap' to BOOL'
 
 # Refresh mode values should be normalized before validation and not be used together
 statement error
@@ -140,7 +140,7 @@ CREATE SECRET incorrect_type (
     FLIEPFLAP true
 )
 ----
-Binder Error: Unknown parameter 'fliepflap' for secret type 'r2' with provider 'config'
+Binder Error: Unknown parameter "fliepflap" for secret type "r2" with provider "config"
 
 # Incorrect param for this type, but correct for other
 statement error
@@ -150,7 +150,7 @@ CREATE SECRET incorrect_type (
     ACCOUNT_ID 'my_acount'
 )
 ----
-Binder Error: Unknown parameter 'account_id' for secret type 's3' with provider 'config'
+Binder Error: Unknown parameter "account_id" for secret type "s3" with provider "config"
 
 # Params can only occur once
 statement error
diff --git a/test/sql/secrets/create_secret_name_conflicts.test b/test/sql/secrets/create_secret_name_conflicts.test
index 5f23bbf..8ef9f4a 100644
--- a/test/sql/secrets/create_secret_name_conflicts.test
+++ b/test/sql/secrets/create_secret_name_conflicts.test
@@ -15,7 +15,7 @@ CREATE TEMPORARY SECRET s1 ( TYPE S3 )
 statement error
 CREATE TEMPORARY SECRET s1 ( TYPE S3 )
 ----
-Invalid Input Error: Temporary secret with name 's1' already exists!
+Invalid Input Error: Temporary secret with name "s1" already exists!
 
 statement ok
 CREATE PERSISTENT SECRET s1 ( TYPE S3 )
@@ -23,17 +23,17 @@ CREATE PERSISTENT SECRET s1 ( TYPE S3 )
 statement error
 CREATE PERSISTENT SECRET s1 ( TYPE S3 )
 ----
-Persistent secret with name 's1' already exists in secret storage 'local_file'!
+Persistent secret with name "s1" already exists in secret storage 'local_file'!
 
 statement error
 DROP SECRET s1;
 ----
-Invalid Input Error: Ambiguity found for secret name 's1', secret occurs in multiple storages
+Invalid Input Error: Ambiguity found for secret name "s1", secret occurs in multiple storages
 
 statement error
 DROP SECRET s1 FROM bogus;
 ----
-Invalid Input Error: Unknown storage type found for drop secret: 'bogus'
+Invalid Input Error: Unknown storage type found for drop secret: "bogus"
 
 statement ok
 DROP TEMPORARY SECRET s1;
@@ -42,7 +42,7 @@ DROP TEMPORARY SECRET s1;
 statement error
 DROP TEMPORARY SECRET s1;
 ----
-Invalid Input Error: Failed to remove non-existent secret with name 's1'
+Invalid Input Error: Failed to remove non-existent secret with name "s1"
 
 query II
 SELECT name, storage FROM duckdb_secrets()
@@ -62,7 +62,7 @@ CREATE TEMPORARY SECRET s1 ( TYPE S3 )
 statement error
 DROP SECRET s1;
 ----
-Invalid Input Error: Ambiguity found for secret name 's1', secret occurs in multiple storages
+Invalid Input Error: Ambiguity found for secret name "s1", secret occurs in multiple storages
 
 # Fully specified drop statement this time
 statement ok
diff --git a/test/sql/secrets/create_secret_overwriting.test b/test/sql/secrets/create_secret_overwriting.test
index 55caa6f..684b6ce 100644
--- a/test/sql/secrets/create_secret_overwriting.test
+++ b/test/sql/secrets/create_secret_overwriting.test
@@ -28,7 +28,7 @@ CREATE SECRET my_secret (
     SCOPE 's3://bucket1'
 )
 ----
-Invalid Input Error: Temporary secret with name 'my_secret' already exists!
+Invalid Input Error: Temporary secret with name "my_secret" already exists!
 
 # We should be able to replace the secret though
 statement ok
@@ -58,7 +58,7 @@ my_secret	['s3://bucket2']
 statement error
 DROP SECRET my_secret_does_not_exist;
 ----
-Failed to remove non-existent secret with name 'my_secret_does_not_exist'
+Failed to remove non-existent secret with name "my_secret_does_not_exist"
 
 # Drop one that does exist
 statement ok
diff --git a/test/sql/secrets/create_secret_persistence.test b/test/sql/secrets/create_secret_persistence.test
index 4ffe766..e02dcac 100644
--- a/test/sql/secrets/create_secret_persistence.test
+++ b/test/sql/secrets/create_secret_persistence.test
@@ -63,7 +63,7 @@ CREATE PERSISTENT SECRET my_tmp_secret_3 (
     SCOPE 's3://bucket3_not_used'
 )
 ----
-Invalid Input Error: Persistent secret with name 'my_tmp_secret_3' already exists in secret storage 'local_file'!
+Invalid Input Error: Persistent secret with name "my_tmp_secret_3" already exists in secret storage 'local_file'!
 
 restart
 
@@ -77,7 +77,7 @@ CREATE PERSISTENT SECRET my_tmp_secret_3 (
     SCOPE 's3://bucket3_not_used'
 )
 ----
-Invalid Input Error: Persistent secret with name 'my_tmp_secret_3' already exists in secret storage 'local_file'!
+Invalid Input Error: Persistent secret with name "my_tmp_secret_3" already exists in secret storage 'local_file'!
 
 restart
 
diff --git a/test/sql/secrets/create_secret_r2.test b/test/sql/secrets/create_secret_r2.test
index a9c48be..785acaf 100644
--- a/test/sql/secrets/create_secret_r2.test
+++ b/test/sql/secrets/create_secret_r2.test
@@ -45,7 +45,7 @@ CREATE SECRET (
     SECRET 'my_secret'
 )
 ----
-Binder Error: Unknown parameter 'account_id' for secret type 's3' with default provider 'config'
+Binder Error: Unknown parameter "account_id" for secret type "s3" with default provider "config"
 
 # Account ID is only for R2, trying to set this for GCS will fail
 statement error
@@ -57,7 +57,7 @@ CREATE SECRET (
     SECRET 'my_secret'
 )
 ----
-Binder Error: Unknown parameter 'account_id' for secret type 'gcs' with provider 'config'
+Binder Error: Unknown parameter "account_id" for secret type "gcs" with provider "config"
 
 # Ensure secret lookup works correctly;
 statement ok
