--- a/src/httpfs_extension.cpp
+++ b/src/httpfs_extension.cpp
@@ -29,7 +29,7 @@ static void ClearHTTPFSConnectionCacheFunction(ClientContext &context, TableFunc
 
 static unique_ptr<FunctionData> ClearHTTPFSConnectionCacheBind(ClientContext &, TableFunctionBindInput &,
                                                                vector<LogicalType> &return_types,
-                                                               vector<string> &names) {
+                                                               vector<Identifier> &names) {
 	return_types.emplace_back(LogicalType::BOOLEAN);
 	names.emplace_back("success");
 	return nullptr;
