diff --git a/test/sql/logging/file_system_logging.test b/test/sql/logging/file_system_logging.test
index 9fc4ea8..211de17 100644
--- a/test/sql/logging/file_system_logging.test
+++ b/test/sql/logging/file_system_logging.test
@@ -24,12 +24,11 @@ pragma threads=1
 query IIII
 SELECT scope, type, log_level, regexp_replace(message, '\"path\":.*test.csv"', '"test.csv"')
 FROM duckdb_logs
-WHERE type = 'FileSystem'
+WHERE type = 'FileSystem' AND message NOT LIKE '%"op":"WRITE"%'
 ORDER BY timestamp
 ----
 CONNECTION	FileSystem	TRACE	{"fs":"LocalFileSystem","test.csv","op":"OPEN"}
-CONNECTION	FileSystem	TRACE	{"fs":"LocalFileSystem","test.csv","op":"WRITE","bytes":"4","pos":"0"}
 CONNECTION	FileSystem	TRACE	{"fs":"LocalFileSystem","test.csv","op":"CLOSE"}
 CONNECTION	FileSystem	TRACE	{"fs":"LocalFileSystem","test.csv","op":"OPEN"}
 CONNECTION	FileSystem	TRACE	{"fs":"LocalFileSystem","test.csv","op":"READ","bytes":"4","pos":"0"}
 CONNECTION	FileSystem	TRACE	{"fs":"LocalFileSystem","test.csv","op":"CLOSE"}
@@ -37,2 +36,9 @@ CONNECTION	FileSystem	TRACE	{"fs":"LocalFileSystem","test.csv","op":"CLOSE"}
+query III
+SELECT sum(bytes), min(pos), max(pos + bytes)
+FROM duckdb_logs_parsed('FileSystem')
+WHERE op = 'WRITE' AND parse_filename(path) = 'test.csv'
+----
+4	0	4
+
 statement ok
 CALL truncate_duckdb_logs();
