diff --git a/test/sql/secrets/create_secret_binding.test b/test/sql/secrets/create_secret_binding.test
index 66ae79f..493c758 100644
--- a/test/sql/secrets/create_secret_binding.test
+++ b/test/sql/secrets/create_secret_binding.test
@@ -181,4 +181,4 @@ CREATE SECRET duplicate_param (
         account_id blablabla
 )
 ----
-Binder Error: Duplicate query param found while parsing create secret: 'key_id'
+Parser Error: Unexpected duplicate option "key_id"
