diff --git a/CMakeLists.txt b/CMakeLists.txt
index 141ded6..cc77271 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,6 +48,16 @@ else()
   target_link_libraries(httpfs_loadable_extension duckdb_mbedtls
                         ${OPENSSL_LIBRARIES})
   target_link_libraries(httpfs_extension duckdb_mbedtls ${OPENSSL_LIBRARIES})
+
+  if(APPLE)
+    find_library(COREFOUNDATION_FRAMEWORK CoreFoundation)
+    find_library(SECURITY_FRAMEWORK Security)
+    target_link_libraries(httpfs_loadable_extension
+                          ${COREFOUNDATION_FRAMEWORK}
+                          ${SECURITY_FRAMEWORK})
+    target_link_libraries(httpfs_extension ${COREFOUNDATION_FRAMEWORK}
+                          ${SECURITY_FRAMEWORK})
+  endif()
 
   if(MINGW)
     find_package(ZLIB)
