diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ebc01e0..cd50fb9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,4 @@
-set(HTTPFS_SOURCES hffs.cpp hash_functions.cpp create_secret_functions.cpp
-                   httpfs_extension.cpp)
+set(HTTPFS_SOURCES hffs.cpp create_secret_functions.cpp httpfs_extension.cpp)
 if(NOT EMSCRIPTEN)
   list(APPEND HTTPFS_SOURCES crypto.cpp)
 endif()
diff --git a/src/hash_functions.cpp b/src/hash_functions.cpp
deleted file mode 100644
index c5293bb..0000000
--- a/src/hash_functions.cpp
+++ /dev/null
@@ -1,25 +0,0 @@
-#include "hash_functions.hpp"
-
-namespace duckdb {
-
-void sha256(EncryptionUtil &encryption_util, const_data_ptr_t in, idx_t in_len, hash_bytes &out) {
-	D_ASSERT(CryptoHash::GetDigestSize(CryptoHashFunction::SHA256) == sizeof(hash_bytes));
-	encryption_util.Hash(CryptoHashFunction::SHA256, in, in_len, out);
-}
-
-void hmac256(EncryptionUtil &encryption_util, const std::string &message, const_data_ptr_t secret, idx_t secret_len,
-             hash_bytes &out) {
-	D_ASSERT(CryptoHash::GetDigestSize(CryptoHashFunction::SHA256) == sizeof(hash_bytes));
-	encryption_util.Hmac(CryptoHashFunction::SHA256, secret, secret_len, const_data_ptr_cast(message.data()),
-	                     message.size(), out);
-}
-
-void hmac256(EncryptionUtil &encryption_util, std::string message, hash_bytes secret, hash_bytes &out) {
-	hmac256(encryption_util, message, secret, sizeof(hash_bytes), out);
-}
-
-void hex256(hash_bytes &in, hash_str &out) {
-	CryptoHash::ToHex(in, sizeof(hash_bytes), char_ptr_cast(out));
-}
-
-} // namespace duckdb
diff --git a/src/include/hash_functions.hpp b/src/include/hash_functions.hpp
deleted file mode 100644
index 27fc855..0000000
--- a/src/include/hash_functions.hpp
+++ /dev/null
@@ -1,22 +0,0 @@
-#pragma once
-
-#include "duckdb/common/encryption_state.hpp"
-#include "duckdb/common/helper.hpp"
-
-#include <string>
-
-namespace duckdb {
-
-typedef unsigned char hash_bytes[32];
-typedef unsigned char hash_str[64];
-
-void sha256(EncryptionUtil &encryption_util, const_data_ptr_t in, idx_t in_len, hash_bytes &out);
-
-void hmac256(EncryptionUtil &encryption_util, const std::string &message, const_data_ptr_t secret, idx_t secret_len,
-             hash_bytes &out);
-
-void hmac256(EncryptionUtil &encryption_util, std::string message, hash_bytes secret, hash_bytes &out);
-
-void hex256(hash_bytes &in, hash_str &out);
-
-} // namespace duckdb
