diff --git a/src/s3/s3_request.cpp b/src/s3/s3_request.cpp
index 10802ae..3131312 100644
--- a/src/s3/s3_request.cpp
+++ b/src/s3/s3_request.cpp
@@ -2,11 +2,12 @@
 
 #include "s3/s3fs.hpp"
 #include "create_secret_functions.hpp"
-#include "hash_functions.hpp"
 #include "http/http_state.hpp"
 
 #include "duckdb/common/exception/http_exception.hpp"
+#include "duckdb/common/hash_functions.hpp"
 #include "duckdb/common/helper.hpp"
+#include "duckdb/common/http_util.hpp"
 #include "duckdb/common/string_util.hpp"
 #include "duckdb/common/thread.hpp"
 #include "duckdb/common/types/timestamp.hpp"
@@ -124,25 +125,15 @@ private:
 	}
 
 	string CreateSignature(const string &canonical_request) const {
-		hash_bytes canonical_request_hash;
-		hash_str canonical_request_hash_str;
-		sha256(encryption_util, const_data_ptr_cast(canonical_request.data()), canonical_request.length(),
-		       canonical_request_hash);
-		hex256(canonical_request_hash, canonical_request_hash_str);
-
-		auto string_to_sign = "AWS4-HMAC-SHA256\n" + datetime_now + "\n" + CredentialScope() + "\n" +
-		                      string(const_char_ptr_cast(canonical_request_hash_str), sizeof(hash_str));
-		hash_bytes k_date, k_region, k_service, signing_key, signature;
-		auto sign_key = "AWS4" + auth_params.secret_access_key;
-		hmac256(encryption_util, date_now, const_data_ptr_cast(sign_key.data()), sign_key.length(), k_date);
-		hmac256(encryption_util, auth_params.region, k_date, k_region);
-		hmac256(encryption_util, service, k_region, k_service);
-		hmac256(encryption_util, "aws4_request", k_service, signing_key);
-		hmac256(encryption_util, string_to_sign, signing_key, signature);
-
-		hash_str signature_str;
-		hex256(signature, signature_str);
-		return string(const_char_ptr_cast(signature_str), sizeof(hash_str));
+		SignatureV4Params sig_params;
+		sig_params.canonical_request = canonical_request;
+		sig_params.credential_scope = CredentialScope();
+		sig_params.region = auth_params.region;
+		sig_params.service = service;
+		sig_params.secret_access_key = auth_params.secret_access_key;
+		sig_params.date_now = date_now;
+		sig_params.datetime_now = datetime_now;
+		return HTTPUtil::CreateSignatureV4(encryption_util, sig_params);
 	}
 
 	string CredentialScope() const {
