diff --git a/src/httpfs_connection_caching.cpp b/src/httpfs_connection_caching.cpp
index 210abd4..e2bf406 100644
--- a/src/httpfs_connection_caching.cpp
+++ b/src/httpfs_connection_caching.cpp
@@ -149,7 +149,7 @@ unique_ptr<HTTPResponse> HTTPFSCurlUtil::CachingSendRequest(BaseRequest &request
 	if (!caller_owns_client) {
 		connection_cache.Store(std::move(client));
 	}
-	return std::move(r);
+	return r;
 }
 
 unique_ptr<HTTPResponse> HTTPFSCurlUtil::SendRequest(BaseRequest &request, unique_ptr<HTTPClient> &client) {
diff --git a/src/s3fs.cpp b/src/s3fs.cpp
index 95b4342..c21b5b0 100644
--- a/src/s3fs.cpp
+++ b/src/s3fs.cpp
@@ -359,7 +359,6 @@ void S3FileHandle::Close() {
 }
 
 void S3FileHandle::FinalizeUpload() {
-	auto &s3fs = (S3FileSystem &)file_system;
 	if (flags.OpenForWriting() && multi_part_upload) {
 		multi_part_upload->Finalize();
 	}
