--- a/src/httpfs_extension.cpp
+++ b/src/httpfs_extension.cpp
@@ -1,4 +1,5 @@
 #include "httpfs_extension.hpp"
+#include "duckdb/logging/logger.hpp"
 
 #include "httpfs_client.hpp"
 #include "create_secret_functions.hpp"
--- a/src/create_secret_functions.cpp
+++ b/src/create_secret_functions.cpp
@@ -1,4 +1,5 @@
 #include "create_secret_functions.hpp"
+#include "duckdb/logging/logger.hpp"
 #include "s3fs.hpp"
 #include "duckdb/main/extension/extension_loader.hpp"
 #include "duckdb/common/local_file_system.hpp"
