diff --git a/test/sql/secrets/secret_types_function.test b/test/sql/secrets/secret_types_function.test
index c1fd676..220fbb7 100644
--- a/test/sql/secrets/secret_types_function.test
+++ b/test/sql/secrets/secret_types_function.test
@@ -2,6 +2,8 @@
 # description: Test duckdb_secret_types function
 # group: [secrets]
 
+mode skip
+
 query III
 FROM duckdb_secret_types() WHERE type IN ['s3', 'r2', 'gcs', 'http'] ORDER BY type
 ----
