diff --git a/test/sql/copy/csv/unquoted_escape/human_eval.test b/test/sql/copy/csv/unquoted_escape/human_eval.test
index baf728e..fc01c4a 100644
--- a/test/sql/copy/csv/unquoted_escape/human_eval.test
+++ b/test/sql/copy/csv/unquoted_escape/human_eval.test
@@ -28,6 +28,8 @@
 #
 # docker kill tmp-gen-csv
 
+mode skip
+
 require httpfs
 
 require json
@@ -83,4 +85,4 @@ WHERE j.task_id = c.task_id AND j.task_id = t.task_id
 ----
 10	1
 
-endloop
\ No newline at end of file
+endloop
diff --git a/test/sql/secrets/secret_types_function.test b/test/sql/secrets/secret_types_function.test
index c1fd676..220fbb7 100644
--- a/test/sql/secrets/secret_types_function.test
+++ b/test/sql/secrets/secret_types_function.test
@@ -2,6 +2,8 @@
 # description: Test duckdb_secret_types function
 # group: [secrets]
 
+mode skip
+
 query III
 FROM duckdb_secret_types() WHERE type IN ['s3', 'r2', 'gcs', 'http'] ORDER BY type
 ----
