commit cf27a82878f55aa60ced8ddf4c922e9985e56ddc
Author: dentiny <dentinyhao@gmail.com>
Date:   Wed Mar 25 11:28:03 2026 +0000

    Sync DuckDB core

diff --git a/duckdb b/duckdb
index 775cda2..490e7a8 160000
--- a/duckdb
+++ b/duckdb
@@ -1 +1 @@
-Subproject commit 775cda2b490eafb7630dbe3fe4c6708ec8197021
+Subproject commit 490e7a8f479b607798541d21dccf2815abfa673a
diff --git a/test/sql/json/table/internal_issue_6807.test_slow b/test/sql/json/table/internal_issue_6807.test_slow
index a354a7f..e759862 100644
--- a/test/sql/json/table/internal_issue_6807.test_slow
+++ b/test/sql/json/table/internal_issue_6807.test_slow
@@ -13,6 +13,6 @@ statement ok
 CREATE TABLE T AS FROM 'https://data.gharchive.org/2023-02-08-0.json.gz';
 
 query I
-SELECT count(*) < 20 FROM duckdb_logs_parsed('HTTP') WHERE request.type = 'GET' GROUP BY request.type;
+SELECT count(*) < 60 FROM duckdb_logs_parsed('HTTP') WHERE request.type = 'GET' GROUP BY request.type;
 ----
 true
diff --git a/test/sql/logging/file_system_logging.test b/test/sql/logging/file_system_logging.test
index 6aa2ed0..54babde 100644
--- a/test/sql/logging/file_system_logging.test
+++ b/test/sql/logging/file_system_logging.test
@@ -43,7 +43,6 @@ require httpfs
 statement ok
 FROM 'https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv'
 
-# FIXME: investigate why we call READ twice?
 query IIII
 SELECT scope, type, log_level, regexp_replace(message, '\"path\":.*test.csv"', '"test.csv"')
 FROM duckdb_logs
@@ -52,5 +51,4 @@ ORDER BY timestamp
 ----
 CONNECTION	FileSystem	TRACE	{"fs":"HTTPFileSystem","path":"https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv","op":"OPEN"}
 CONNECTION	FileSystem	TRACE	{"fs":"HTTPFileSystem","path":"https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv","op":"READ","bytes":"1276","pos":"0"}
-CONNECTION	FileSystem	TRACE	{"fs":"HTTPFileSystem","path":"https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv","op":"READ","bytes":"0","pos":"1276"}
 CONNECTION	FileSystem	TRACE	{"fs":"HTTPFileSystem","path":"https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv","op":"CLOSE"}
diff --git a/test/sql/metadata_stats.test b/test/sql/metadata_stats.test
index 27cb0a6..dfe7a26 100644
--- a/test/sql/metadata_stats.test
+++ b/test/sql/metadata_stats.test
@@ -18,4 +18,4 @@ SET force_download=false;
 query II
 explain analyze SELECT id, first_name, last_name, email FROM PARQUET_SCAN('https://raw.githubusercontent.com/duckdb/duckdb/main/data/parquet-testing/userdata1.parquet')
 ----
-analyzed_plan	<REGEX>:.*GET: 2.*
+analyzed_plan	<REGEX>:.*GET: 1.*
