commit c05b6d0c65bac3c3da7edab7e73e88dbcdabe55c
Author: dentiny <dentinyhao@gmail.com>
Date:   Thu Jan 22 09:55:06 2026 +0000

    Uodate expected result based on cached json reader

diff --git a/test/sql/json/table/internal_issue_6807.test_slow b/test/sql/json/table/internal_issue_6807.test_slow
index d2163f6..749dea2 100644
--- a/test/sql/json/table/internal_issue_6807.test_slow
+++ b/test/sql/json/table/internal_issue_6807.test_slow
@@ -15,4 +15,4 @@ CREATE TABLE T AS FROM 'https://data.gharchive.org/2023-02-08-0.json.gz';
 query I
 SELECT count(*) FROM duckdb_logs_parsed('HTTP') WHERE request.type = 'GET' GROUP BY request.type;
 ----
-9
+8
