diff --git a/src/include/s3fs.hpp b/src/include/s3fs.hpp
index d75423e..90071d9 100644
--- a/src/include/s3fs.hpp
+++ b/src/include/s3fs.hpp
@@ -32,7 +32,12 @@ public:
 		auto setting_scope = reader.TryGetSecretKeyOrSetting(secret_key, setting_name, temp_result);
 		if (!temp_result.IsNull() &&
 		    !(setting_scope.GetScope() == SettingScope::GLOBAL && !use_env_variables_for_secret_settings)) {
-			result = temp_result.GetValue<TYPE>();
+		    try {
+				result = temp_result.GetValue<TYPE>();
+			} catch(std::exception &ex) {
+				ErrorData error(ex);
+				error.Throw("Failed to convert value for secret key \"" + secret_key + "\": ");
+			}
 		}
 		return setting_scope;
 	}
diff --git a/test/sql/logging/file_system_logging.test b/test/sql/logging/file_system_logging.test
index 6aa2ed0..54babde 100644
--- a/test/sql/logging/file_system_logging.test
+++ b/test/sql/logging/file_system_logging.test
@@ -43,7 +43,6 @@ require httpfs
 statement ok
 FROM 'https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv'
 
-# FIXME: investigate why we call READ twice?
 query IIII
 SELECT scope, type, log_level, regexp_replace(message, '\"path\":.*test.csv"', '"test.csv"')
 FROM duckdb_logs
@@ -52,5 +51,4 @@ ORDER BY timestamp
 ----
 CONNECTION	FileSystem	TRACE	{"fs":"HTTPFileSystem","path":"https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv","op":"OPEN"}
 CONNECTION	FileSystem	TRACE	{"fs":"HTTPFileSystem","path":"https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv","op":"READ","bytes":"1276","pos":"0"}
-CONNECTION	FileSystem	TRACE	{"fs":"HTTPFileSystem","path":"https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv","op":"READ","bytes":"0","pos":"1276"}
 CONNECTION	FileSystem	TRACE	{"fs":"HTTPFileSystem","path":"https://github.com/duckdb/duckdb/raw/main/data/csv/customer.csv","op":"CLOSE"}
