diff --git a/src/crypto.cpp b/src/crypto.cpp
index 034638e..8a0504a 100644
--- a/src/crypto.cpp
+++ b/src/crypto.cpp
@@ -90,7 +90,7 @@ void AESStateSSL::GenerateRandomData(data_ptr_t data, idx_t len) {
 	}
 }
 
-void AESStateSSL::InitializeIVEncrypt(EncryptionNonce nonce,  const_data_ptr_t key) {
+void AESStateSSL::InitializeIVEncrypt(EncryptionNonce &nonce,  const_data_ptr_t key) {
 	if (1 != EVP_CIPHER_CTX_ctrl(context, EVP_CTRL_GCM_SET_IVLEN, nonce.total_size(), NULL)) {
 		throw InternalException("EVP_CIPHER_CTX_ctrl failed (EVP_CTRL_GCM_SET_IVLEN)");
 	}
@@ -100,7 +100,7 @@ void AESStateSSL::InitializeIVEncrypt(EncryptionNonce nonce,  const_data_ptr_t k
 	}
 }
 
-void AESStateSSL::InitializeIVDecrypt(EncryptionNonce nonce, const_data_ptr_t key) {
+void AESStateSSL::InitializeIVDecrypt(EncryptionNonce &nonce, const_data_ptr_t key) {
 	if (1 != EVP_CIPHER_CTX_ctrl(context, EVP_CTRL_GCM_SET_IVLEN, nonce.total_size(), NULL)) {
 		throw InternalException("EVP_CIPHER_CTX_ctrl failed to set GCM iv len");
 	}
@@ -109,7 +109,7 @@ void AESStateSSL::InitializeIVDecrypt(EncryptionNonce nonce, const_data_ptr_t ke
 	}
 }
 
-void AESStateSSL::InitializeEncryption(EncryptionNonce nonce, const_data_ptr_t key,
+void AESStateSSL::InitializeEncryption(EncryptionNonce &nonce, const_data_ptr_t key,
                                        const_data_ptr_t aad, idx_t aad_len) {
 	mode = EncryptionTypes::ENCRYPT;
 
@@ -117,7 +117,7 @@ void AESStateSSL::InitializeEncryption(EncryptionNonce nonce, const_data_ptr_t k
 		throw InternalException("EncryptInit failed (attempt 1)");
 	}
 
-	InitializeIVEncrypt(std::move(nonce), key);
+	InitializeIVEncrypt(nonce, key);
 
 	int len;
 	if (aad_len > 0) {
@@ -127,7 +127,7 @@ void AESStateSSL::InitializeEncryption(EncryptionNonce nonce, const_data_ptr_t k
 	}
 }
 
-void AESStateSSL::InitializeDecryption(EncryptionNonce nonce, const_data_ptr_t key,
+void AESStateSSL::InitializeDecryption(EncryptionNonce &nonce, const_data_ptr_t key,
                                        const_data_ptr_t aad, idx_t aad_len) {
 	mode = EncryptionTypes::DECRYPT;
 
@@ -135,7 +135,7 @@ void AESStateSSL::InitializeDecryption(EncryptionNonce nonce, const_data_ptr_t k
 		throw InternalException("EVP_DecryptInit_ex failed to set cipher");
 	}
 
-	InitializeIVDecrypt(std::move(nonce), key);
+	InitializeIVDecrypt(nonce, key);
 
 	int len;
 	if (aad_len > 0) {
diff --git a/src/include/crypto.hpp b/src/include/crypto.hpp
index 909e8d8..0322478 100644
--- a/src/include/crypto.hpp
+++ b/src/include/crypto.hpp
@@ -30,9 +30,9 @@ public:
 	~AESStateSSL() override;
 
 public:
-	void InitializeEncryption(EncryptionNonce nonce, const_data_ptr_t key,
+	void InitializeEncryption(EncryptionNonce &nonce, const_data_ptr_t key,
 	                          const_data_ptr_t aad, idx_t aad_len) override;
-	void InitializeDecryption(EncryptionNonce nonce, const_data_ptr_t key,
+	void InitializeDecryption(EncryptionNonce &nonce, const_data_ptr_t key,
 	                          const_data_ptr_t aad, idx_t aad_len) override;
 	size_t Process(const_data_ptr_t in, idx_t in_len, data_ptr_t out, idx_t out_len) override;
 	size_t Finalize(data_ptr_t out, idx_t out_len, data_ptr_t tag, idx_t tag_len) override;
@@ -40,8 +40,8 @@ public:
 
 	const EVP_CIPHER *GetCipher();
 	size_t FinalizeGCM(data_ptr_t out, idx_t out_len, data_ptr_t tag, idx_t tag_len);
-	void InitializeIVEncrypt(EncryptionNonce nonce, const_data_ptr_t key);
-	void InitializeIVDecrypt(EncryptionNonce nonce, const_data_ptr_t key);
+	void InitializeIVEncrypt(EncryptionNonce &nonce, const_data_ptr_t key);
+	void InitializeIVDecrypt(EncryptionNonce &nonce, const_data_ptr_t key);
 
 private:
 	EVP_CIPHER_CTX *context;
