diff --git a/extension/httpfs/create_secret_functions.cpp b/extension/httpfs/create_secret_functions.cpp
index b3984b3..476bc14 100644
--- a/extension/httpfs/create_secret_functions.cpp
+++ b/extension/httpfs/create_secret_functions.cpp
@@ -1,15 +1,15 @@
 #include "create_secret_functions.hpp"
 #include "s3fs.hpp"
-#include "duckdb/main/extension_util.hpp"
+#include "duckdb/main/extension/extension_loader.hpp"
 #include "duckdb/common/local_file_system.hpp"
 
 namespace duckdb {
 
-void CreateS3SecretFunctions::Register(DatabaseInstance &instance) {
-	RegisterCreateSecretFunction(instance, "s3");
-	RegisterCreateSecretFunction(instance, "aws");
-	RegisterCreateSecretFunction(instance, "r2");
-	RegisterCreateSecretFunction(instance, "gcs");
+void CreateS3SecretFunctions::Register(ExtensionLoader &loader) {
+	RegisterCreateSecretFunction(loader, "s3");
+	RegisterCreateSecretFunction(loader, "aws");
+	RegisterCreateSecretFunction(loader, "r2");
+	RegisterCreateSecretFunction(loader, "gcs");
 }
 
 static Value MapToStruct(const Value &map) {
@@ -205,7 +205,7 @@ void CreateS3SecretFunctions::SetBaseNamedParams(CreateSecretFunction &function,
 	}
 }
 
-void CreateS3SecretFunctions::RegisterCreateSecretFunction(DatabaseInstance &instance, string type) {
+void CreateS3SecretFunctions::RegisterCreateSecretFunction(ExtensionLoader &loader, string type) {
 	// Register the new type
 	SecretType secret_type;
 	secret_type.name = type;
@@ -213,31 +213,31 @@ void CreateS3SecretFunctions::RegisterCreateSecretFunction(DatabaseInstance &ins
 	secret_type.default_provider = "config";
 	secret_type.extension = "httpfs";
 
-	ExtensionUtil::RegisterSecretType(instance, secret_type);
+	loader.RegisterSecretType(secret_type);
 
 	CreateSecretFunction from_empty_config_fun2 = {type, "config", CreateS3SecretFromConfig};
 	SetBaseNamedParams(from_empty_config_fun2, type);
-	ExtensionUtil::RegisterFunction(instance, from_empty_config_fun2);
+	loader.RegisterFunction(from_empty_config_fun2);
 }
 
-void CreateBearerTokenFunctions::Register(DatabaseInstance &instance) {
+void CreateBearerTokenFunctions::Register(ExtensionLoader &loader) {
 	// HuggingFace secret
 	SecretType secret_type_hf;
 	secret_type_hf.name = HUGGINGFACE_TYPE;
 	secret_type_hf.deserializer = KeyValueSecret::Deserialize<KeyValueSecret>;
 	secret_type_hf.default_provider = "config";
 	secret_type_hf.extension = "httpfs";
-	ExtensionUtil::RegisterSecretType(instance, secret_type_hf);
+	loader.RegisterSecretType(secret_type_hf);
 
 	// Huggingface config provider
 	CreateSecretFunction hf_config_fun = {HUGGINGFACE_TYPE, "config", CreateBearerSecretFromConfig};
 	hf_config_fun.named_parameters["token"] = LogicalType::VARCHAR;
-	ExtensionUtil::RegisterFunction(instance, hf_config_fun);
+	loader.RegisterFunction(hf_config_fun);
 
 	// Huggingface credential_chain provider
 	CreateSecretFunction hf_cred_fun = {HUGGINGFACE_TYPE, "credential_chain",
 	                                    CreateHuggingFaceSecretFromCredentialChain};
-	ExtensionUtil::RegisterFunction(instance, hf_cred_fun);
+	loader.RegisterFunction(hf_cred_fun);
 }
 
 unique_ptr<BaseSecret> CreateBearerTokenFunctions::CreateSecretFunctionInternal(ClientContext &context,
diff --git a/extension/httpfs/crypto.cpp b/extension/httpfs/crypto.cpp
index 04bd795..3a3e89e 100644
--- a/extension/httpfs/crypto.cpp
+++ b/extension/httpfs/crypto.cpp
@@ -42,11 +42,11 @@ AESStateSSL::~AESStateSSL() {
 	EVP_CIPHER_CTX_free(context);
 }
 
-const EVP_CIPHER *AESStateSSL::GetCipher(const string &key) {
+const EVP_CIPHER *AESStateSSL::GetCipher(idx_t key_len) {
 
 	switch (cipher) {
 	case GCM:
-		switch (key.size()) {
+		switch (key_len) {
 		case 16:
 			return EVP_aes_128_gcm();
 		case 24:
@@ -57,7 +57,7 @@ const EVP_CIPHER *AESStateSSL::GetCipher(const string &key) {
 			throw InternalException("Invalid AES key length");
 		}
 	case CTR:
-		switch (key.size()) {
+		switch (key_len) {
 		case 16:
 			return EVP_aes_128_ctr();
 		case 24:
@@ -78,20 +78,34 @@ void AESStateSSL::GenerateRandomData(data_ptr_t data, idx_t len) {
 	RAND_bytes(data, len);
 }
 
-void AESStateSSL::InitializeEncryption(const_data_ptr_t iv, idx_t iv_len, const string *key) {
+void AESStateSSL::InitializeEncryption(const_data_ptr_t iv, idx_t iv_len, const_data_ptr_t key, idx_t key_len, const_data_ptr_t aad, idx_t aad_len) {
 	mode = ENCRYPT;
 
-	if (1 != EVP_EncryptInit_ex(context, GetCipher(*key), NULL, const_data_ptr_cast(key->data()), iv)) {
+	if (1 != EVP_EncryptInit_ex(context, GetCipher(key_len), NULL, key, iv)) {
 		throw InternalException("EncryptInit failed");
 	}
+	
+	int len;
+	if (aad_len > 0){
+		if (!EVP_DecryptUpdate(context, NULL, &len, aad, aad_len)) {
+    			throw InternalException("Setting Additional Authenticated Data  failed");
+		}
+	}	
 }
 
-void AESStateSSL::InitializeDecryption(const_data_ptr_t iv, idx_t iv_len, const string *key) {
+void AESStateSSL::InitializeDecryption(const_data_ptr_t iv, idx_t iv_len, const_data_ptr_t key, idx_t key_len, const_data_ptr_t aad, idx_t aad_len) {
 	mode = DECRYPT;
 
-	if (1 != EVP_DecryptInit_ex(context, GetCipher(*key), NULL, const_data_ptr_cast(key->data()), iv)) {
+	if (1 != EVP_DecryptInit_ex(context, GetCipher(key_len), NULL, key, iv)) {
 		throw InternalException("DecryptInit failed");
 	}
+
+	int len;
+        if (aad_len > 0){
+                if (!EVP_DecryptUpdate(context, NULL, &len, aad, aad_len)) {
+                        throw InternalException("Setting Additional Authenticated Data  failed");
+                }
+	}
 }
 
 size_t AESStateSSL::Process(const_data_ptr_t in, idx_t in_len, data_ptr_t out, idx_t out_len) {
diff --git a/extension/httpfs/httpfs.cpp b/extension/httpfs/httpfs.cpp
index 686162f..cd43a64 100644
--- a/extension/httpfs/httpfs.cpp
+++ b/extension/httpfs/httpfs.cpp
@@ -7,6 +7,7 @@
 #include "duckdb/common/http_util.hpp"
 #include "duckdb/common/thread.hpp"
 #include "duckdb/common/types/hash.hpp"
+#include "duckdb/common/types/time.hpp"
 #include "duckdb/function/scalar/strftime_format.hpp"
 #include "duckdb/logging/file_system_logger.hpp"
 #include "duckdb/main/client_context.hpp"
@@ -263,19 +264,6 @@ unique_ptr<HTTPResponse> HTTPFileSystem::GetRangeRequest(FileHandle &handle, str
 	return response;
 }
 
-void TimestampToTimeT(timestamp_t timestamp, time_t &result) {
-	auto components = Timestamp::GetComponents(timestamp);
-	struct tm tm {};
-	tm.tm_year = components.year - 1900;
-	tm.tm_mon = components.month - 1;
-	tm.tm_mday = components.day;
-	tm.tm_hour = components.hour;
-	tm.tm_min = components.minute;
-	tm.tm_sec = components.second;
-	tm.tm_isdst = 0;
-	result = mktime(&tm);
-}
-
 HTTPFileHandle::HTTPFileHandle(FileSystem &fs, const OpenFileInfo &file, FileOpenFlags flags,
                                unique_ptr<HTTPParams> params_p)
     : FileHandle(fs, file.path, flags), params(std::move(params_p)), http_params(params->Cast<HTTPFSParams>()),
@@ -286,7 +274,7 @@ HTTPFileHandle::HTTPFileHandle(FileSystem &fs, const OpenFileInfo &file, FileOpe
 		auto &info = file.extended_info->options;
 		auto lm_entry = info.find("last_modified");
 		if (lm_entry != info.end()) {
-			TimestampToTimeT(lm_entry->second.GetValue<timestamp_t>(), last_modified);
+			last_modified = lm_entry->second.GetValue<timestamp_t>();
 		}
 		auto etag_entry = info.find("etag");
 		if (etag_entry != info.end()) {
@@ -466,7 +454,7 @@ int64_t HTTPFileSystem::GetFileSize(FileHandle &handle) {
 	return sfh.length;
 }
 
-time_t HTTPFileSystem::GetLastModifiedTime(FileHandle &handle) {
+timestamp_t HTTPFileSystem::GetLastModifiedTime(FileHandle &handle) {
 	auto &sfh = handle.Cast<HTTPFileHandle>();
 	return sfh.last_modified;
 }
@@ -549,20 +537,14 @@ void HTTPFileHandle::FullDownload(HTTPFileSystem &hfs, bool &should_write_cache)
 	}
 }
 
-bool HTTPFileSystem::TryParseLastModifiedTime(const string &timestamp, time_t &result) {
+bool HTTPFileSystem::TryParseLastModifiedTime(const string &timestamp, timestamp_t &result) {
 	StrpTimeFormat::ParseResult parse_result;
 	if (!StrpTimeFormat::TryParse("%a, %d %h %Y %T %Z", timestamp, parse_result)) {
 		return false;
 	}
-	struct tm tm {};
-	tm.tm_year = parse_result.data[0] - 1900;
-	tm.tm_mon = parse_result.data[1] - 1;
-	tm.tm_mday = parse_result.data[2];
-	tm.tm_hour = parse_result.data[3];
-	tm.tm_min = parse_result.data[4];
-	tm.tm_sec = parse_result.data[5];
-	tm.tm_isdst = 0;
-	result = mktime(&tm);
+	if (!parse_result.TryToTimestamp(result)) {
+		return false;
+	}
 	return true;
 }
 
diff --git a/extension/httpfs/httpfs_extension.cpp b/extension/httpfs/httpfs_extension.cpp
index c9bc985..c800ff2 100644
--- a/extension/httpfs/httpfs_extension.cpp
+++ b/extension/httpfs/httpfs_extension.cpp
@@ -1,5 +1,3 @@
-#define DUCKDB_EXTENSION_MAIN
-
 #include "httpfs_extension.hpp"
 
 #include "create_secret_functions.hpp"
@@ -10,7 +8,8 @@
 
 namespace duckdb {
 
-static void LoadInternal(DatabaseInstance &instance) {
+static void LoadInternal(ExtensionLoader &loader) {
+	auto &instance = loader.GetDatabaseInstance();
 	auto &fs = instance.GetFileSystem();
 
 	fs.RegisterSubSystem(make_uniq<HTTPFileSystem>());
@@ -66,14 +65,14 @@ static void LoadInternal(DatabaseInstance &instance) {
 	auto provider = make_uniq<AWSEnvironmentCredentialsProvider>(config);
 	provider->SetAll();
 
-	CreateS3SecretFunctions::Register(instance);
-	CreateBearerTokenFunctions::Register(instance);
+	CreateS3SecretFunctions::Register(loader);
+	CreateBearerTokenFunctions::Register(loader);
 
 	// set pointer to OpenSSL encryption state
 	config.encryption_util = make_shared_ptr<AESStateSSLFactory>();
 }
-void HttpfsExtension::Load(DuckDB &db) {
-	LoadInternal(*db.instance);
+void HttpfsExtension::Load(ExtensionLoader &loader) {
+	LoadInternal(loader);
 }
 std::string HttpfsExtension::Name() {
 	return "httpfs";
@@ -91,15 +90,8 @@ std::string HttpfsExtension::Version() const {
 
 extern "C" {
 
-DUCKDB_EXTENSION_API void httpfs_init(duckdb::DatabaseInstance &db) {
-	LoadInternal(db);
+DUCKDB_CPP_EXTENSION_ENTRY(httpfs, loader) {
+	duckdb::LoadInternal(loader);
 }
 
-DUCKDB_EXTENSION_API const char *httpfs_version() {
-	return duckdb::DuckDB::LibraryVersion();
-}
 }
-
-#ifndef DUCKDB_EXTENSION_MAIN
-#error DUCKDB_EXTENSION_MAIN not defined
-#endif
diff --git a/extension/httpfs/include/create_secret_functions.hpp b/extension/httpfs/include/create_secret_functions.hpp
index 54b7566..bd3bc4a 100644
--- a/extension/httpfs/include/create_secret_functions.hpp
+++ b/extension/httpfs/include/create_secret_functions.hpp
@@ -8,11 +8,12 @@ struct S3AuthParams;
 class CreateSecretFunction;
 class BaseSecret;
 struct SecretEntry;
+class ExtensionLoader;
 
 struct CreateS3SecretFunctions {
 public:
 	//! Register all CreateSecretFunctions
-	static void Register(DatabaseInstance &instance);
+	static void Register(ExtensionLoader &loader);
 
 	//! Secret refreshing mechanisms
 	static CreateSecretInput GenerateRefreshSecretInfo(const SecretEntry &secret_entry, Value &refresh_info);
@@ -30,7 +31,7 @@ protected:
 	//! Helper function to set named params of secret function
 	static void SetBaseNamedParams(CreateSecretFunction &function, string &type);
 	//! Helper function to create secret types s3/r2/gcs
-	static void RegisterCreateSecretFunction(DatabaseInstance &instance, string type);
+	static void RegisterCreateSecretFunction(ExtensionLoader &loader, string type);
 };
 
 struct CreateBearerTokenFunctions {
@@ -38,7 +39,7 @@ public:
 	static constexpr const char *HUGGINGFACE_TYPE = "huggingface";
 
 	//! Register all CreateSecretFunctions
-	static void Register(DatabaseInstance &instance);
+	static void Register(ExtensionLoader &loader);
 
 protected:
 	//! Internal function to create bearer token
diff --git a/extension/httpfs/include/crypto.hpp b/extension/httpfs/include/crypto.hpp
index f819356..aa6ad70 100644
--- a/extension/httpfs/include/crypto.hpp
+++ b/extension/httpfs/include/crypto.hpp
@@ -29,13 +29,13 @@ public:
 	~AESStateSSL() override;
 
 public:
-	void InitializeEncryption(const_data_ptr_t iv, idx_t iv_len, const std::string *key) override;
-	void InitializeDecryption(const_data_ptr_t iv, idx_t iv_len, const std::string *key) override;
+	void InitializeEncryption(const_data_ptr_t iv, idx_t iv_len, const_data_ptr_t key, idx_t key_len, const_data_ptr_t aad, idx_t aad_len) override;
+	void InitializeDecryption(const_data_ptr_t iv, idx_t iv_len, const_data_ptr_t key, idx_t key_len, const_data_ptr_t aad, idx_t aad_len) override;
 	size_t Process(const_data_ptr_t in, idx_t in_len, data_ptr_t out, idx_t out_len) override;
 	size_t Finalize(data_ptr_t out, idx_t out_len, data_ptr_t tag, idx_t tag_len) override;
 	void GenerateRandomData(data_ptr_t data, idx_t len) override;
 
-	const EVP_CIPHER *GetCipher(const string &key);
+	const EVP_CIPHER *GetCipher(idx_t key_len);
 	size_t FinalizeGCM(data_ptr_t out, idx_t out_len, data_ptr_t tag, idx_t tag_len);
 
 private:
@@ -48,16 +48,16 @@ private:
 
 extern "C" {
 
-class DUCKDB_EXTENSION_API AESStateSSLFactory : public duckdb::EncryptionUtil {
-public:
-	explicit AESStateSSLFactory() {
-	}
+	class DUCKDB_EXTENSION_API AESStateSSLFactory : public duckdb::EncryptionUtil {
+	public:
+		explicit AESStateSSLFactory() {
+		}
 
-	duckdb::shared_ptr<duckdb::EncryptionState> CreateEncryptionState(const std::string *key = nullptr) const override {
-		return duckdb::make_shared_ptr<duckdb::AESStateSSL>();
-	}
+		duckdb::shared_ptr<duckdb::EncryptionState> CreateEncryptionState(duckdb::const_data_ptr_t key = nullptr, duckdb::idx_t key_len = 0) const override {
+			return duckdb::make_shared_ptr<duckdb::AESStateSSL>();
+		}
 
-	~AESStateSSLFactory() override {
-	}
-};
-}
+		~AESStateSSLFactory() override {
+		}
+	};
+}
\ No newline at end of file
diff --git a/extension/httpfs/include/http_metadata_cache.hpp b/extension/httpfs/include/http_metadata_cache.hpp
index 8fc7909..83a05cc 100644
--- a/extension/httpfs/include/http_metadata_cache.hpp
+++ b/extension/httpfs/include/http_metadata_cache.hpp
@@ -17,7 +17,7 @@ namespace duckdb {
 
 struct HTTPMetadataCacheEntry {
 	idx_t length;
-	time_t last_modified;
+	timestamp_t last_modified;
 	string etag;
 };
 
diff --git a/extension/httpfs/include/httpfs.hpp b/extension/httpfs/include/httpfs.hpp
index 62067d4..55c74cc 100644
--- a/extension/httpfs/include/httpfs.hpp
+++ b/extension/httpfs/include/httpfs.hpp
@@ -46,7 +46,7 @@ public:
 	// File handle info
 	FileOpenFlags flags;
 	idx_t length;
-	time_t last_modified;
+	timestamp_t last_modified;
 	string etag;
 	bool force_full_download;
 	bool initialized = false;
@@ -92,7 +92,7 @@ private:
 
 class HTTPFileSystem : public FileSystem {
 public:
-	static bool TryParseLastModifiedTime(const string &timestamp, time_t &result);
+	static bool TryParseLastModifiedTime(const string &timestamp, timestamp_t &result);
 
 	vector<OpenFileInfo> Glob(const string &path, FileOpener *opener = nullptr) override {
 		return {path}; // FIXME
@@ -121,7 +121,7 @@ public:
 	int64_t Write(FileHandle &handle, void *buffer, int64_t nr_bytes) override;
 	void FileSync(FileHandle &handle) override;
 	int64_t GetFileSize(FileHandle &handle) override;
-	time_t GetLastModifiedTime(FileHandle &handle) override;
+	timestamp_t GetLastModifiedTime(FileHandle &handle) override;
 	string GetVersionTag(FileHandle &handle) override;
 	bool FileExists(const string &filename, optional_ptr<FileOpener> opener) override;
 	void Seek(FileHandle &handle, idx_t location) override;
diff --git a/extension/httpfs/include/httpfs_extension.hpp b/extension/httpfs/include/httpfs_extension.hpp
index 3c4f3a1..eeca2c9 100644
--- a/extension/httpfs/include/httpfs_extension.hpp
+++ b/extension/httpfs/include/httpfs_extension.hpp
@@ -6,7 +6,7 @@ namespace duckdb {
 
 class HttpfsExtension : public Extension {
 public:
-	void Load(DuckDB &db) override;
+	void Load(ExtensionLoader &loader) override;
 	std::string Name() override;
 	std::string Version() const override;
 };
