diff --git a/src/httpfs_curl_client.cpp b/src/httpfs_curl_client.cpp
index 6de3c85dad1..a7e4637a924 100644
--- a/src/httpfs_curl_client.cpp
+++ b/src/httpfs_curl_client.cpp
@@ -115,6 +115,11 @@ static idx_t httpfs_client_count = 0;
 class HTTPFSCurlClient : public HTTPClient {
 public:
 	HTTPFSCurlClient(HTTPFSParams &http_params, const string &proto_host_port) {
+		// FIXME: proto_host_port is not used
+		Initialize(http_params);
+	}
+	void Initialize(HTTPParams &http_p) override {
+		HTTPFSParams &http_params = (HTTPFSParams&)http_p;
 		auto bearer_token = "";
 		if (!http_params.bearer_token.empty()) {
 			bearer_token = http_params.bearer_token.c_str();
diff --git a/src/httpfs_httplib_client.cpp b/src/httpfs_httplib_client.cpp
index 239a11248c7..cf1b854075a 100644
--- a/src/httpfs_httplib_client.cpp
+++ b/src/httpfs_httplib_client.cpp
@@ -9,6 +9,10 @@ class HTTPFSClient : public HTTPClient {
 public:
 	HTTPFSClient(HTTPFSParams &http_params, const string &proto_host_port) {
 		client = make_uniq<duckdb_httplib_openssl::Client>(proto_host_port);
+		Initialize(http_params);
+	}
+	void Initialize(HTTPParams &http_p) override {
+		HTTPFSParams &http_params = (HTTPFSParams&)http_p;
 		client->set_follow_location(http_params.follow_location);
 		client->set_keep_alive(http_params.keep_alive);
 		if (!http_params.ca_cert_file.empty()) {
