diff --git a/src/s3fs.cpp b/src/s3fs.cpp
index 0034c0e..833063d 100644
--- a/src/s3fs.cpp
+++ b/src/s3fs.cpp
@@ -1137,7 +1137,7 @@ private:
 };
 
 S3GlobResult::S3GlobResult(S3FileSystem &fs, const string &glob_pattern_p, optional_ptr<FileOpener> opener)
-    : glob_pattern(glob_pattern_p), opener(opener) {
+    : LazyMultiFileList(FileOpener::TryGetClientContext(opener)), glob_pattern(glob_pattern_p), opener(opener) {
 	if (!opener) {
 		throw InternalException("Cannot S3 Glob without FileOpener");
 	}
