From f78af8f2b40705c30924598f6b0cbc6ce279adf2 Mon Sep 17 00:00:00 2001
From: Mytherin <mark.raasveldt@gmail.com>
Date: Thu, 28 May 2026 11:57:55 +0200
Subject: [PATCH 2/3] relassert_thread_safety_lock_guard.patch

---
 src/http_state.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/http_state.cpp b/src/http_state.cpp
index 1b09114..0ac79cf 100644
--- a/src/http_state.cpp
+++ b/src/http_state.cpp
@@ -6,7 +6,7 @@ namespace duckdb {
 CachedFileHandle::CachedFileHandle(shared_ptr<CachedFile> &file_p) {
 	// If the file was not yet initialized, we need to grab a lock.
 	if (!file_p->initialized) {
-		lock = make_uniq<lock_guard<mutex>>(file_p->lock);
+		lock = unique_ptr<lock_guard<mutex>>(new lock_guard<mutex>(file_p->lock));
 	}
 	file = file_p;
 }
-- 
2.39.5 (Apple Git-154)

