diff --git a/src/statement_generator.cpp b/src/statement_generator.cpp
index 4c750ce..add178e 100644
--- a/src/statement_generator.cpp
+++ b/src/statement_generator.cpp
@@ -1187,12 +1187,12 @@ unique_ptr<ParsedExpression> StatementGenerator::GeneratePositionalReference() {
 
 unique_ptr<ParsedExpression> StatementGenerator::GenerateCase() {
 	auto case_stmt = make_uniq<CaseExpression>();
-	case_stmt->else_expr = GenerateExpression();
+	case_stmt->ElseMutable() = GenerateExpression();
 	while (true) {
 		CaseCheck check;
 		check.then_expr = GenerateExpression();
 		check.when_expr = GenerateExpression();
-		case_stmt->case_checks.push_back(std::move(check));
+		case_stmt->CaseChecksMutable().push_back(std::move(check));
 		if (RandomPercentage(50)) {
 			break;
 		}
diff --git a/src/statement_simplifier.cpp b/src/statement_simplifier.cpp
index 3b46e19..a31ecf7 100644
--- a/src/statement_simplifier.cpp
+++ b/src/statement_simplifier.cpp
@@ -309,12 +309,12 @@ void StatementSimplifier::SimplifyExpression(duckdb::unique_ptr<ParsedExpression
 	}
 	case ExpressionClass::CASE: {
 		auto &op = expr->Cast<CaseExpression>();
-		SimplifyChildExpression(expr, op.else_expr);
-		for (auto &case_check : op.case_checks) {
+		SimplifyChildExpression(expr, op.ElseMutable());
+		for (auto &case_check : op.CaseChecksMutable()) {
 			SimplifyChildExpression(expr, case_check.then_expr);
 			SimplifyChildExpression(expr, case_check.when_expr);
 		}
-		SimplifyList(op.case_checks, false);
+		SimplifyList(op.CaseChecksMutable(), false);
 		break;
 	}
 	case ExpressionClass::CAST: {
