diff --git a/src/statement_simplifier.cpp b/src/statement_simplifier.cpp
index a31ecf7..efbecfe 100644
--- a/src/statement_simplifier.cpp
+++ b/src/statement_simplifier.cpp
@@ -319,12 +319,12 @@ void StatementSimplifier::SimplifyExpression(duckdb::unique_ptr<ParsedExpression
 	}
 	case ExpressionClass::CAST: {
 		auto &cast = expr->Cast<CastExpression>();
-		SimplifyChildExpression(expr, cast.child);
+		SimplifyChildExpression(expr, cast.ChildMutable());
 		break;
 	}
 	case ExpressionClass::COLLATE: {
 		auto &collate = expr->Cast<CollateExpression>();
-		SimplifyChildExpression(expr, collate.child);
+		SimplifyChildExpression(expr, collate.ChildMutable());
 		break;
 	}
 	case ExpressionClass::SUBQUERY: {
