diff --git a/src/statement_generator.cpp b/src/statement_generator.cpp
index 257728b..84d44d5 100644
--- a/src/statement_generator.cpp
+++ b/src/statement_generator.cpp
@@ -402,7 +402,8 @@ unique_ptr<QueryNode> StatementGenerator::GenerateQueryNode() {
 			result->modifiers.push_back(GenerateOrderBy());
 		}
 		if (RandomPercentage(50)) {
-			auto limit_percent_modifier = make_uniq<LimitPercentModifier>();
+			auto limit_percent_modifier = make_uniq<LimitModifier>();
+			limit_percent_modifier->limit_type = LimitValueType::PERCENTAGE;
 			if (RandomPercentage(30)) {
 				limit_percent_modifier->limit = GenerateExpression();
 			} else if (RandomPercentage(30)) {
