diff --git a/src/statement_simplifier.cpp b/src/statement_simplifier.cpp
index 52b9f0d..a783b9d 100644
--- a/src/statement_simplifier.cpp
+++ b/src/statement_simplifier.cpp
@@ -293,7 +293,9 @@ void StatementSimplifier::SimplifyExpression(duckdb::unique_ptr<ParsedExpression
 	}
 	case ExpressionClass::FUNCTION: {
 		auto &func = expr->Cast<FunctionExpression>();
-		SimplifyExpressionList(expr, func.GetChildrenMutable());
+		for (auto &child : func.GetArgumentsMutable()) {
+			SimplifyChildExpression(expr, child.GetExpressionMutable());
+		}
 		SimplifyEnum(func.DistinctMutable(), false);
 		SimplifyOptionalExpression(func.FilterMutable());
 		SimplifyOptional(func.OrderByMutable());
