diff --git a/src/statement_generator.cpp b/src/statement_generator.cpp
index 84d44d5..2b38556 100644
--- a/src/statement_generator.cpp
+++ b/src/statement_generator.cpp
@@ -981,7 +981,11 @@ unique_ptr<ParsedExpression> StatementGenerator::GenerateWindowFunction(optional
 	}
 	WindowChecker checker(*this);
 	auto result = make_uniq<WindowExpression>(SYSTEM_CATALOG, DEFAULT_SCHEMA, std::move(name));
-	result->GetChildrenMutable() = GenerateChildren(min_parameters, max_parameters);
+	auto params =  GenerateChildren(min_parameters, max_parameters);
+	auto &args = result->GetArgumentsMutable();
+	for (auto &param : params) {
+		args.push_back(std::move(param));
+	}
 	while (RandomPercentage(50)) {
 		result->PartitionsMutable().push_back(GenerateExpression());
 	}
diff --git a/src/statement_simplifier.cpp b/src/statement_simplifier.cpp
index a783b9d..1bb44cd 100644
--- a/src/statement_simplifier.cpp
+++ b/src/statement_simplifier.cpp
@@ -356,7 +356,9 @@ void StatementSimplifier::SimplifyExpression(duckdb::unique_ptr<ParsedExpression
 	}
 	case ExpressionClass::WINDOW: {
 		auto &window = expr->Cast<WindowExpression>();
-		SimplifyExpressionList(expr, window.GetChildrenMutable());
+		for (auto &child : window.GetArgumentsMutable()) {
+			SimplifyChildExpression(expr, child.GetExpressionMutable());
+		}
 		SimplifyExpressionList(expr, window.PartitionsMutable());
 		SimplifyList(window.OrderByMutable());
 		SimplifyChildExpression(expr, window.FilterMutable());
