diff --git a/src/statement_generator.cpp b/src/statement_generator.cpp
index b767b91..8015e65 100644
--- a/src/statement_generator.cpp
+++ b/src/statement_generator.cpp
@@ -249,7 +249,7 @@ unique_ptr<CreateInfo> StatementGenerator::GenerateCreateInfo() {
 	switch (RandomValue(4)) {
 	case 0: {
 		auto info = make_uniq<CreateTypeInfo>();
-		info->name = Identifier(RandomString(5));
+		info->SetTypeName(Identifier(RandomString(5)));
 		idx_t num_enums = RandomValue(10);
 		auto Enum_Vector = Vector(LogicalType::VARCHAR, num_enums);
 		for (idx_t i = 0; i < num_enums; i++) {
@@ -260,9 +260,9 @@ unique_ptr<CreateInfo> StatementGenerator::GenerateCreateInfo() {
 	}
 	case 1: {
 		auto info = make_uniq<CreateTableInfo>();
-		info->catalog = Identifier::InvalidCatalog();
-		info->schema = Identifier::DefaultSchema();
-		info->table = Identifier(GenerateTableIdentifier());
+		info->CatalogMutable() = Identifier::InvalidCatalog();
+		info->SchemaMutable() = Identifier::DefaultSchema();
+		info->SetTableName(Identifier(GenerateTableIdentifier()));
 		if (RandomPercentage(50)) {
 			info->query = GenerateSelect();
 		} else {
@@ -276,14 +276,14 @@ unique_ptr<CreateInfo> StatementGenerator::GenerateCreateInfo() {
 	}
 	case 2: {
 		auto info = make_uniq<CreateSchemaInfo>();
-		info->catalog = Identifier::InvalidCatalog();
+		info->CatalogMutable() = Identifier::InvalidCatalog();
 		info->on_conflict = OnCreateConflict::REPLACE_ON_CONFLICT;
-		info->schema = Identifier(GenerateSchemaIdentifier());
+		info->SchemaMutable() = Identifier(GenerateSchemaIdentifier());
 		return std::move(info);
 	}
 	case 3: {
 		auto info = make_uniq<CreateViewInfo>();
-		info->view_name = Identifier(GenerateViewIdentifier());
+		info->SetViewName(Identifier(GenerateViewIdentifier()));
 		info->query = GenerateSelect();
 		// TODO: add support for aliases in the view.
 		return std::move(info);
