diff --git a/src/statement_generator.cpp b/src/statement_generator.cpp
index 8015e65..c419987 100644
--- a/src/statement_generator.cpp
+++ b/src/statement_generator.cpp
@@ -189,7 +189,7 @@ unique_ptr<DeleteStatement> StatementGenerator::GenerateDelete() {
 		auto &entry = entry_ref.get();
 		if (entry.type == CatalogType::TABLE_ENTRY) {
 			auto result = make_uniq<BaseTableRef>();
-			result->table_name = entry.name;
+			result->TableMutable() = entry.name;
 			delete_statement->node->table = std::move(result);
 		} else {
 			delete_statement->node->table = GenerateTableRef();
@@ -482,7 +482,7 @@ unique_ptr<TableRef> StatementGenerator::GenerateBaseTableRef() {
 		result->column_name_alias.emplace_back(GenerateIdentifier());
 	}
 	result->alias = Identifier(GenerateTableIdentifier());
-	result->table_name = entry.name;
+	result->TableMutable() = entry.name;
 	return std::move(result);
 }
 
@@ -1308,7 +1308,7 @@ string StatementGenerator::GenerateTestAllTypes(SimpleFunction &base_function) {
 		children.push_back(std::move(argument));
 	}
 	auto from_clause = make_uniq<BaseTableRef>();
-	from_clause->table_name = "all_types";
+	from_clause->TableMutable() = "all_types";
 	node->from_table = std::move(from_clause);
 
 	auto function_expr = make_uniq<FunctionExpression>(base_function.GetName(), std::move(children));
@@ -1360,7 +1360,7 @@ string StatementGenerator::GenerateCast(const LogicalType &target, const string
 	auto node = make_uniq<SelectNode>();
 
 	auto from_clause = make_uniq<BaseTableRef>();
-	from_clause->table_name = "all_types";
+	from_clause->TableMutable() = "all_types";
 	node->from_table = std::move(from_clause);
 
 	unique_ptr<ParsedExpression> source;
