--- a/src/sqlsmith_extension.cpp
+++ b/src/sqlsmith_extension.cpp
@@ -28,7 +28,7 @@ struct SQLSmithFunctionData : public TableFunctionData {
 };
 
 static duckdb::unique_ptr<FunctionData> SQLSmithBind(ClientContext &context, TableFunctionBindInput &input,
-                                                     vector<LogicalType> &return_types, vector<string> &names) {
+                                                     vector<LogicalType> &return_types, vector<Identifier> &names) {
 	auto result = make_uniq<SQLSmithFunctionData>();
 	for (auto &kv : input.named_parameters) {
 		if (kv.first == "seed") {
@@ -87,7 +87,7 @@ struct ReduceSQLFunctionData : public TableFunctionData {
 };
 
 static duckdb::unique_ptr<FunctionData> ReduceSQLBind(ClientContext &context, TableFunctionBindInput &input,
-                                                      vector<LogicalType> &return_types, vector<string> &names) {
+                                                      vector<LogicalType> &return_types, vector<Identifier> &names) {
 	return_types.emplace_back(LogicalType::VARCHAR);
 	names.emplace_back("sql");
 
@@ -130,7 +130,7 @@ struct FuzzyDuckFunctionData : public TableFunctionData {
 };
 
 static duckdb::unique_ptr<FunctionData> FuzzyDuckBind(ClientContext &context, TableFunctionBindInput &input,
-                                                      vector<LogicalType> &return_types, vector<string> &names) {
+                                                      vector<LogicalType> &return_types, vector<Identifier> &names) {
 	auto result = make_uniq<FuzzyDuckFunctionData>(context);
 	for (auto &kv : input.named_parameters) {
 		if (kv.first == "seed") {
