diff --git a/src/statement_generator.cpp b/src/statement_generator.cpp
index 8a975db..4116420 100644
--- a/src/statement_generator.cpp
+++ b/src/statement_generator.cpp
@@ -566,7 +566,7 @@ unique_ptr<TableRef> StatementGenerator::GenerateTableFunctionRef() {
 		table_function_ref = &generator_context->table_functions[random_val];
 	}
 	auto &entry = table_function_ref->get().Cast<TableFunctionCatalogEntry>();
-	auto table_function = entry.functions.GetFunctionByOffset(RandomValue(entry.functions.Size()));
+	auto table_function = *entry.functions.GetFunctionByOffset(RandomValue(entry.functions.Size()));
 
 	auto result = make_uniq<TableFunctionRef>();
 	vector<unique_ptr<ParsedExpression>> children;
@@ -765,7 +765,7 @@ unique_ptr<ParsedExpression> StatementGenerator::GenerateFunction() {
 	case CatalogType::SCALAR_FUNCTION_ENTRY: {
 		auto &scalar_entry = function.Cast<ScalarFunctionCatalogEntry>();
 		auto offset = RandomValue(scalar_entry.functions.Size());
-		auto actual_function = scalar_entry.functions.GetFunctionByOffset(offset);
+		auto actual_function = *scalar_entry.functions.GetFunctionByOffset(offset);
 		name = scalar_entry.name.GetIdentifierName();
 		for (auto &arg : actual_function.GetSignature().GetParameters()) {
 			arguments.push_back(arg.GetType());
@@ -780,7 +780,7 @@ unique_ptr<ParsedExpression> StatementGenerator::GenerateFunction() {
 	case CatalogType::AGGREGATE_FUNCTION_ENTRY: {
 		auto &aggregate_entry = function.Cast<AggregateFunctionCatalogEntry>();
 		auto actual_function =
-		    aggregate_entry.functions.GetFunctionByOffset(RandomValue(aggregate_entry.functions.Size()));
+		    *aggregate_entry.functions.GetFunctionByOffset(RandomValue(aggregate_entry.functions.Size()));
 
 		name = aggregate_entry.name.GetIdentifierName();
 		min_parameters = actual_function.GetSignature().GetParameterCount();
@@ -1379,7 +1379,7 @@ string StatementGenerator::GenerateCast(const LogicalType &target, const string
 
 void StatementGenerator::GenerateAllScalar(ScalarFunctionCatalogEntry &scalar_function, vector<string> &result) {
 	for (idx_t offset = 0; offset < scalar_function.functions.Size(); offset++) {
-		auto function = scalar_function.functions.GetFunctionByOffset(offset);
+		auto function = *scalar_function.functions.GetFunctionByOffset(offset);
 
 		result.push_back(GenerateTestAllTypes(function));
 		result.push_back(GenerateTestVectorTypes(function));
@@ -1389,7 +1389,7 @@ void StatementGenerator::GenerateAllScalar(ScalarFunctionCatalogEntry &scalar_fu
 void StatementGenerator::GenerateAllAggregate(AggregateFunctionCatalogEntry &aggregate_function,
                                               vector<string> &result) {
 	for (idx_t offset = 0; offset < aggregate_function.functions.Size(); offset++) {
-		auto function = aggregate_function.functions.GetFunctionByOffset(offset);
+		auto function = *aggregate_function.functions.GetFunctionByOffset(offset);
 
 		result.push_back(GenerateTestAllTypes(function));
 		result.push_back(GenerateTestVectorTypes(function));
