diff --git a/src/statement_generator.cpp b/src/statement_generator.cpp
index 4116420..7835141 100644
--- a/src/statement_generator.cpp
+++ b/src/statement_generator.cpp
@@ -165,10 +165,10 @@ unique_ptr<DetachStatement> StatementGenerator::GenerateDetach() {
 
 // generate USE statement
 unique_ptr<SetStatement> StatementGenerator::GenerateSet() {
-	auto name_expr = make_uniq<ConstantExpression>(GenerateDataBaseName());
+	auto name_expr = ConstantExpression::FromValue(GenerateDataBaseName());
 	if (RandomPercentage(90)) {
 		auto name = GetRandomAttachedDataBase();
-		name_expr = make_uniq<ConstantExpression>(Value(name));
+		name_expr = ConstantExpression::FromValue(Value(name));
 	}
 	auto set = make_uniq<SetVariableStatement>("schema", std::move(name_expr), SetScope::AUTOMATIC);
 	return unique_ptr_cast<duckdb::SetVariableStatement, duckdb::SetStatement>(std::move(set));
@@ -722,7 +722,7 @@ Value StatementGenerator::GenerateConstantValue() {
 }
 
 unique_ptr<ParsedExpression> StatementGenerator::GenerateConstant() {
-	return make_uniq<ConstantExpression>(GenerateConstantValue());
+	return ConstantExpression::FromValue(GenerateConstantValue());
 }
 
 LogicalType StatementGenerator::GenerateLogicalType() {
@@ -1305,7 +1305,7 @@ string StatementGenerator::GenerateTestAllTypes(SimpleFunction &base_function) {
 			}
 		}
 		if (!argument) {
-			argument = make_uniq<ConstantExpression>(Value(param.GetType()));
+			argument = ConstantExpression::FromValue(Value(param.GetType()));
 		}
 		children.push_back(std::move(argument));
 	}
@@ -1335,11 +1335,11 @@ string StatementGenerator::GenerateTestVectorTypes(SimpleFunction &base_function
 			string argument_name = "c" + to_string(column_aliases.size() + 1);
 			column_aliases.emplace_back(argument_name);
 			argument = make_uniq<ColumnRefExpression>(Identifier(std::move(argument_name)));
-			auto constant_expr = make_uniq<ConstantExpression>(Value());
+			auto constant_expr = ConstantExpression::Null();
 			auto cast = make_uniq<CastExpression>(param.GetType(), std::move(constant_expr));
 			test_vector_types.push_back(std::move(cast));
 		} else {
-			argument = make_uniq<ConstantExpression>(Value(param.GetType()));
+			argument = ConstantExpression::FromValue(Value(param.GetType()));
 		}
 		children.push_back(std::move(argument));
 	}
diff --git a/src/statement_simplifier.cpp b/src/statement_simplifier.cpp
index 1bb44cd..396a6ef 100644
--- a/src/statement_simplifier.cpp
+++ b/src/statement_simplifier.cpp
@@ -283,7 +283,7 @@ void StatementSimplifier::SimplifyExpression(duckdb::unique_ptr<ParsedExpression
 	default:
 		break;
 	}
-	duckdb::unique_ptr<ParsedExpression> constant = make_uniq<ConstantExpression>(Value());
+	duckdb::unique_ptr<ParsedExpression> constant = ConstantExpression::Null();
 	SimplifyReplace(expr, constant);
 	switch (expr_class) {
 	case ExpressionClass::CONJUNCTION: {
