diff --git a/src/third_party/sqlsmith/duckdb.cc b/src/third_party/sqlsmith/duckdb.cc
index 87e04d5..3e0efb3 100644
--- a/src/third_party/sqlsmith/duckdb.cc
+++ b/src/third_party/sqlsmith/duckdb.cc
@@ -35,9 +35,10 @@ schema_duckdb::schema_duckdb(duckdb::DatabaseInstance &database, bool no_catalog
 	if (result->HasError()) {
 		result->ThrowError();
 	}
+	auto table_rows = result->Collection().GetRows();
 	for (size_t i = 0; i < result->RowCount(); i++) {
-		auto type = StringValue::Get(result->GetValue(0, i));
-		auto name = StringValue::Get(result->GetValue(2, i));
+		auto type = StringValue::Get(table_rows.GetValue(0, i));
+		auto name = StringValue::Get(table_rows.GetValue(2, i));
 		bool view = type == "view";
 		table tab(name, "main", !view, !view);
 		tables.push_back(tab);
@@ -56,9 +57,10 @@ schema_duckdb::schema_duckdb(duckdb::DatabaseInstance &database, bool no_catalog
 		if (result->HasError()) {
 			result->ThrowError();
 		}
+		auto column_rows = result->Collection().GetRows();
 		for (size_t i = 0; i < result->RowCount(); i++) {
-			auto name = StringValue::Get(result->GetValue(1, i));
-			auto type = StringValue::Get(result->GetValue(2, i));
+			auto name = StringValue::Get(column_rows.GetValue(1, i));
+			auto type = StringValue::Get(column_rows.GetValue(2, i));
 			column c(name, sqltype::get(type));
 			t->columns().push_back(c);
 		}
