diff --git a/src/statement_generator.cpp b/src/statement_generator.cpp
index 72e8f5b..8eafec0 100644
--- a/src/statement_generator.cpp
+++ b/src/statement_generator.cpp
@@ -18,6 +18,7 @@
 #include "duckdb/parser/statement/attach_statement.hpp"
 #include "duckdb/parser/statement/create_statement.hpp"
 #include "duckdb/parser/statement/delete_statement.hpp"
+#include "duckdb/parser/query_node/delete_query_node.hpp"
 #include "duckdb/parser/statement/detach_statement.hpp"
 #include "duckdb/parser/statement/insert_statement.hpp"
 #include "duckdb/parser/statement/multi_statement.hpp"
@@ -189,12 +190,12 @@ unique_ptr<DeleteStatement> StatementGenerator::GenerateDelete() {
 		if (entry.type == CatalogType::TABLE_ENTRY) {
 			auto result = make_uniq<BaseTableRef>();
 			result->table_name = entry.name;
-			delete_statement->table = std::move(result);
+			delete_statement->node->table = std::move(result);
 		} else {
-			delete_statement->table = GenerateTableRef();
+			delete_statement->node->table = GenerateTableRef();
 		}
 	} else {
-		delete_statement->table = GenerateTableRef();
+		delete_statement->node->table = GenerateTableRef();
 	}

 	return delete_statement;
@@ -302,7 +303,7 @@ void StatementGenerator::GenerateCTEs(QueryNode &node) {
 	}
 	while (RandomPercentage(20)) {
 		auto cte = make_uniq<CommonTableExpressionInfo>();
-		cte->query = unique_ptr_cast<SQLStatement, SelectStatement>(GenerateSelect());
+		cte->query_node = std::move(unique_ptr_cast<SQLStatement, SelectStatement>(GenerateSelect())->node);
 		for (idx_t i = 0; i < 1 + RandomValue(10); i++) {
 			cte->aliases.push_back(GenerateIdentifier());
 		}
@@ -957,9 +958,12 @@ unique_ptr<ParsedExpression> StatementGenerator::GenerateWindowFunction(optional
 		case ExpressionType::WINDOW_NTILE:
 		case ExpressionType::WINDOW_FIRST_VALUE:
 		case ExpressionType::WINDOW_LAST_VALUE:
+			min_parameters = 1;
+			break;
 		case ExpressionType::WINDOW_LEAD:
 		case ExpressionType::WINDOW_LAG:
 			min_parameters = 1;
+			min_parameters = 3;
 			break;
 		case ExpressionType::WINDOW_NTH_VALUE:
 			min_parameters = 2;
@@ -970,7 +974,7 @@ unique_ptr<ParsedExpression> StatementGenerator::GenerateWindowFunction(optional
 		max_parameters = min_parameters;
 	}
 	WindowChecker checker(*this);
-	auto result = make_uniq<WindowExpression>(type, INVALID_CATALOG, INVALID_SCHEMA, std::move(name));
+	auto result = make_uniq<WindowExpression>(SYSTEM_CATALOG, DEFAULT_SCHEMA, std::move(name));
 	result->children = GenerateChildren(min_parameters, max_parameters);
 	while (RandomPercentage(50)) {
 		result->partitions.push_back(GenerateExpression());
@@ -1014,15 +1018,6 @@ unique_ptr<ParsedExpression> StatementGenerator::GenerateWindowFunction(optional
 	default:
 		break;
 	}
-	switch (type) {
-	case ExpressionType::WINDOW_LEAD:
-	case ExpressionType::WINDOW_LAG:
-		result->offset_expr = RandomExpression(30);
-		result->default_expr = RandomExpression(30);
-		break;
-	default:
-		break;
-	}
 	return std::move(result);
 }

diff --git a/src/statement_simplifier.cpp b/src/statement_simplifier.cpp
index 4602928..3e56aca 100644
--- a/src/statement_simplifier.cpp
+++ b/src/statement_simplifier.cpp
@@ -8,8 +8,11 @@
 #include "duckdb/parser/expression/list.hpp"
 #include "duckdb/parser/statement/delete_statement.hpp"
 #include "duckdb/parser/statement/insert_statement.hpp"
+#include "duckdb/parser/query_node/insert_query_node.hpp"
 #include "duckdb/parser/statement/prepare_statement.hpp"
 #include "duckdb/parser/statement/update_statement.hpp"
+#include "duckdb/parser/query_node/update_query_node.hpp"
+#include "duckdb/parser/query_node/delete_query_node.hpp"
 #include "duckdb/parser/statement/select_statement.hpp"
 #endif

@@ -206,7 +209,7 @@ void StatementSimplifier::Simplify(CommonTableExpressionMap &cte) {
 	SimplifyMap(cte.map);
 	for (auto &cte_child : cte.map) {
 		// simplify individual ctes
-		Simplify(cte_child.second->query->node);
+		Simplify(cte_child.second->query_node);
 	}
 }

@@ -357,8 +360,6 @@ void StatementSimplifier::SimplifyExpression(duckdb::unique_ptr<ParsedExpression
 		SimplifyChildExpression(expr, window.filter_expr);
 		SimplifyChildExpression(expr, window.start_expr);
 		SimplifyChildExpression(expr, window.end_expr);
-		SimplifyChildExpression(expr, window.offset_expr);
-		SimplifyChildExpression(expr, window.default_expr);
 		SimplifyEnum(window.ignore_nulls, false);
 		SimplifyEnum(window.distinct, false);
 		SimplifyEnum(window.start, WindowBoundary::UNBOUNDED_PRECEDING);
@@ -395,17 +396,19 @@ void StatementSimplifier::Simplify(SelectStatement &stmt) {
 }

 void StatementSimplifier::Simplify(InsertStatement &stmt) {
-	Simplify(stmt.cte_map);
-	Simplify(*stmt.select_statement);
-	SimplifyList(stmt.returning_list);
+	Simplify(stmt.node->cte_map);
+	if (stmt.node->select_statement) {
+		Simplify(*stmt.node->select_statement);
+	}
+	SimplifyList(stmt.node->returning_list);
 }

 void StatementSimplifier::Simplify(DeleteStatement &stmt) {
-	Simplify(stmt.cte_map);
-	SimplifyOptional(stmt.condition);
-	SimplifyExpression(stmt.condition);
-	SimplifyList(stmt.using_clauses);
-	SimplifyList(stmt.returning_list);
+	Simplify(stmt.node->cte_map);
+	SimplifyOptional(stmt.node->condition);
+	SimplifyExpression(stmt.node->condition);
+	SimplifyList(stmt.node->using_clauses);
+	SimplifyList(stmt.node->returning_list);
 }

 void StatementSimplifier::Simplify(UpdateSetInfo &info) {
@@ -432,11 +435,11 @@ void StatementSimplifier::Simplify(PrepareStatement &stmt) {
 }

 void StatementSimplifier::Simplify(UpdateStatement &stmt) {
-	Simplify(stmt.cte_map);
-	SimplifyOptional(stmt.from_table);
-	D_ASSERT(stmt.set_info);
-	Simplify(*stmt.set_info);
-	SimplifyList(stmt.returning_list);
+	Simplify(stmt.node->cte_map);
+	SimplifyOptional(stmt.node->from_table);
+	D_ASSERT(stmt.node->set_info);
+	Simplify(*stmt.node->set_info);
+	SimplifyList(stmt.node->returning_list);
 }

 void StatementSimplifier::Simplify(SQLStatement &stmt) {
