diff --git a/test/sql/sql_reduce.test b/test/sql/sql_reduce.test
index 428ae92..af0530f 100644
--- a/test/sql/sql_reduce.test
+++ b/test/sql/sql_reduce.test
@@ -48,7 +48,7 @@ INSERT INTO tbl (VALUES (1))
 INSERT INTO tbl (VALUES (1, 2))
 INSERT INTO tbl (VALUES (2))
 INSERT INTO tbl SELECT *
-INSERT INTO tbl SELECT NULL FROM (VALUES (1, 2))
+INSERT INTO tbl SELECT NULL FROM (VALUES (1, 2)) AS valueslist
 INSERT INTO tbl SELECT NULL FROM (VALUES (1, 2)) AS valueslist
 
 query I
