diff --git a/src/azure_blob_filesystem.cpp b/src/azure_blob_filesystem.cpp
index 6ba3a0e..d8c79a3 100644
--- a/src/azure_blob_filesystem.cpp
+++ b/src/azure_blob_filesystem.cpp
@@ -12,7 +12,7 @@
 #include "duckdb/main/secret/secret_manager.hpp"
 #include "duckdb/function/scalar/string_common.hpp"
 #include "duckdb/function/scalar_function.hpp"
-#include "duckdb/main/extension_util.hpp"
+#include "duckdb/main/extension/extension_loader.hpp"
 #include "duckdb/main/client_data.hpp"
 #include "duckdb/parser/parsed_data/create_scalar_function_info.hpp"
 #include <azure/storage/blobs.hpp>
diff --git a/src/azure_extension.cpp b/src/azure_extension.cpp
index 5ce0b8d..d0758e2 100644
--- a/src/azure_extension.cpp
+++ b/src/azure_extension.cpp
@@ -1,5 +1,3 @@
-#define DUCKDB_EXTENSION_MAIN
-
 #include "azure_extension.hpp"
 #include "azure_blob_filesystem.hpp"
 #include "azure_dfs_filesystem.hpp"
@@ -7,14 +5,15 @@
 
 namespace duckdb {
 
-static void LoadInternal(DatabaseInstance &instance) {
+static void LoadInternal(ExtensionLoader &loader) {
 	// Load filesystem
+	auto &instance = loader.GetDatabaseInstance();
 	auto &fs = instance.GetFileSystem();
 	fs.RegisterSubSystem(make_uniq<AzureBlobStorageFileSystem>());
 	fs.RegisterSubSystem(make_uniq<AzureDfsStorageFileSystem>());
 
 	// Load Secret functions
-	CreateAzureSecretFunctions::Register(instance);
+	CreateAzureSecretFunctions::Register(loader);
 
 	// Load extension config
 	auto &config = DBConfig::GetConfig(instance);
@@ -77,8 +76,8 @@ static void LoadInternal(DatabaseInstance &instance) {
 	                          Value(nullptr));
 }
 
-void AzureExtension::Load(DuckDB &db) {
-	LoadInternal(*db.instance);
+void AzureExtension::Load(ExtensionLoader &loader) {
+	LoadInternal(loader);
 }
 std::string AzureExtension::Name() {
 	return "azure";
@@ -88,15 +87,8 @@ std::string AzureExtension::Name() {
 
 extern "C" {
 
-DUCKDB_EXTENSION_API void azure_init(duckdb::DatabaseInstance &db) {
-	LoadInternal(db);
-}
-
-DUCKDB_EXTENSION_API const char *azure_version() {
-	return duckdb::DuckDB::LibraryVersion();
-}
+DUCKDB_CPP_EXTENSION_ENTRY(azure, loader) {
+	duckdb::LoadInternal(loader);
 }
 
-#ifndef DUCKDB_EXTENSION_MAIN
-#error DUCKDB_EXTENSION_MAIN not defined
-#endif
+}
\ No newline at end of file
diff --git a/src/azure_secret.cpp b/src/azure_secret.cpp
index e9f08f0..7b1919a 100644
--- a/src/azure_secret.cpp
+++ b/src/azure_secret.cpp
@@ -2,7 +2,7 @@
 #include "azure_dfs_filesystem.hpp"
 #include "duckdb/common/types.hpp"
 #include "duckdb/common/unique_ptr.hpp"
-#include "duckdb/main/extension_util.hpp"
+#include "duckdb/main/extension/extension_loader.hpp"
 #include "duckdb/main/secret/secret.hpp"
 #include <azure/identity/azure_cli_credential.hpp>
 #include <azure/identity/chained_token_credential.hpp>
@@ -148,7 +148,7 @@ static void RegisterCommonSecretParameters(CreateSecretFunction &function) {
 	function.named_parameters["proxy_password"] = LogicalType::VARCHAR;
 }
 
-void CreateAzureSecretFunctions::Register(DatabaseInstance &instance) {
+void CreateAzureSecretFunctions::Register(ExtensionLoader &loader) {
 	string type = "azure";
 
 	// Register the new type
@@ -156,19 +156,19 @@ void CreateAzureSecretFunctions::Register(DatabaseInstance &instance) {
 	secret_type.name = type;
 	secret_type.deserializer = KeyValueSecret::Deserialize<KeyValueSecret>;
 	secret_type.default_provider = "config";
-	ExtensionUtil::RegisterSecretType(instance, secret_type);
+	loader.RegisterSecretType(secret_type);
 
 	// Register the connection string secret provider
 	CreateSecretFunction connection_string_function = {type, "config", CreateAzureSecretFromConfig};
 	connection_string_function.named_parameters["connection_string"] = LogicalType::VARCHAR;
 	RegisterCommonSecretParameters(connection_string_function);
-	ExtensionUtil::RegisterFunction(instance, connection_string_function);
+	loader.RegisterFunction(connection_string_function);
 
 	// Register the credential_chain secret provider
 	CreateSecretFunction cred_chain_function = {type, "credential_chain", CreateAzureSecretFromCredentialChain};
 	cred_chain_function.named_parameters["chain"] = LogicalType::VARCHAR;
 	RegisterCommonSecretParameters(cred_chain_function);
-	ExtensionUtil::RegisterFunction(instance, cred_chain_function);
+	loader.RegisterFunction(cred_chain_function);
 
 	// Register the service_principal secret provider
 	CreateSecretFunction service_principal_function = {type, "service_principal",
@@ -178,13 +178,13 @@ void CreateAzureSecretFunctions::Register(DatabaseInstance &instance) {
 	service_principal_function.named_parameters["client_secret"] = LogicalType::VARCHAR;
 	service_principal_function.named_parameters["client_certificate_path"] = LogicalType::VARCHAR;
 	RegisterCommonSecretParameters(service_principal_function);
-	ExtensionUtil::RegisterFunction(instance, service_principal_function);
+	loader.RegisterFunction(service_principal_function);
 
 	// Register the access_token secret provider
 	CreateSecretFunction access_token_function = {type, "access_token", CreateAzureSecretFromAccessToken};
 	access_token_function.named_parameters["access_token"] = LogicalType::VARCHAR;
 	RegisterCommonSecretParameters(access_token_function);
-	ExtensionUtil::RegisterFunction(instance, access_token_function);
+	loader.RegisterFunction(access_token_function);
 }
 
 } // namespace duckdb
diff --git a/src/include/azure_extension.hpp b/src/include/azure_extension.hpp
index c9be37e..1a8404f 100644
--- a/src/include/azure_extension.hpp
+++ b/src/include/azure_extension.hpp
@@ -5,7 +5,7 @@
 namespace duckdb {
 class AzureExtension : public Extension {
 public:
-	void Load(DuckDB &db) override;
+	void Load(ExtensionLoader &loader) override;
 	std::string Name() override;
 };
 
diff --git a/src/include/azure_secret.hpp b/src/include/azure_secret.hpp
index 3dc39e9..6b81816 100644
--- a/src/include/azure_secret.hpp
+++ b/src/include/azure_secret.hpp
@@ -18,7 +18,7 @@ class CreateSecretFunction;
 struct CreateAzureSecretFunctions {
 public:
 	//! Register all CreateSecretFunctions
-	static void Register(DatabaseInstance &instance);
+	static void Register(ExtensionLoader &instance);
 };
 
 } // namespace duckdb
