diff --git a/src/planning/snapshot/iceberg_snapshot_lookup.cpp b/src/planning/snapshot/iceberg_snapshot_lookup.cpp
index 8b7dd413..49852872 100644
--- a/src/planning/snapshot/iceberg_snapshot_lookup.cpp
+++ b/src/planning/snapshot/iceberg_snapshot_lookup.cpp
@@ -13,13 +13,14 @@ IcebergSnapshotLookup IcebergSnapshotLookup::FromAtClause(optional_ptr<BoundAtCl
 	if (value.IsNull()) {
 		throw InvalidInputException("NULL values can not be used as the 'unit' of a time travel clause");
 	}
-	if (StringUtil::CIEquals(unit, "version")) {
+	if (unit == "version") {
 		return FromSnapshotId(value.DefaultCastAs(LogicalType::BIGINT).GetValue<int64_t>());
-	} else if (StringUtil::CIEquals(unit, "timestamp")) {
+	} else if (unit == "timestamp") {
 		return FromTimestamp(value.DefaultCastAs(LogicalType::TIMESTAMP_MS).GetValue<timestamp_ms_t>());
 	} else {
 		throw InvalidInputException(
-		    "Unit '%s' for time travel is not valid, supported options are 'version' and 'timestamp'", unit);
+		    "Unit '%s' for time travel is not valid, supported options are 'version' and 'timestamp'",
+		    unit.GetIdentifierName());
 	}
 }
 
