diff --git a/src/catalog/rest/catalog_entry/table/iceberg_table_schema_version.cpp b/src/catalog/rest/catalog_entry/table/iceberg_table_schema_version.cpp
index e7fd8d5b..0ef33864 100644
--- a/src/catalog/rest/catalog_entry/table/iceberg_table_schema_version.cpp
+++ b/src/catalog/rest/catalog_entry/table/iceberg_table_schema_version.cpp
@@ -32,10 +32,15 @@ constexpr column_t IcebergMultiFileReader::COLUMN_IDENTIFIER_LAST_SEQUENCE_NUMBE
 IcebergTableSchemaVersion::IcebergTableSchemaVersion(IcebergTable &table_info, Catalog &catalog,
                                                      SchemaCatalogEntry &schema, CreateTableInfo &info,
                                                      optional_idx schema_id)
-    : TableCatalogEntry(catalog, schema, info), table_info(table_info), schema_id(schema_id) {
+    : TableCatalogEntry(catalog, schema, info), columns(std::move(info.columns)), table_info(table_info),
+      schema_id(schema_id) {
 	this->internal = false;
 }
 
+const ColumnList &IcebergTableSchemaVersion::GetColumns() const {
+	return columns;
+}
+
 unique_ptr<BaseStatistics> IcebergTableSchemaVersion::GetStatistics(ClientContext &context, column_t column_id) {
 	return nullptr;
 }
diff --git a/src/include/catalog/rest/catalog_entry/table/iceberg_table_schema_version.hpp b/src/include/catalog/rest/catalog_entry/table/iceberg_table_schema_version.hpp
index 00d14a7b..d6fd93dd 100644
--- a/src/include/catalog/rest/catalog_entry/table/iceberg_table_schema_version.hpp
+++ b/src/include/catalog/rest/catalog_entry/table/iceberg_table_schema_version.hpp
@@ -19,6 +19,8 @@ public:
 	vector<column_t> GetRowIdColumns() const override;
 
 public:
+	const ColumnList &GetColumns() const override;
+
 	unique_ptr<BaseStatistics> GetStatistics(ClientContext &context, column_t column_id) override;
 	void PrepareIcebergScanFromEntry(ClientContext &context) const;
 	TableFunction GetScanFunction(ClientContext &context, unique_ptr<FunctionData> &bind_data) override;
@@ -34,6 +36,9 @@ public:
 	void BindUpdateConstraints(Binder &binder, LogicalGet &get, LogicalProjection &proj, LogicalUpdate &update,
 	                           ClientContext &context) override;
 
+protected:
+	ColumnList columns;
+
 public:
 	IcebergTable &table_info;
 	//! 'schema_id' used to create the entry, or invalid if this is a dummy
