diff --git a/src/catalog/rest/api/iceberg_type.cpp b/src/catalog/rest/api/iceberg_type.cpp
index f8aa32f..603b453 100644
--- a/src/catalog/rest/api/iceberg_type.cpp
+++ b/src/catalog/rest/api/iceberg_type.cpp
@@ -2,7 +2,7 @@
 #include "common/iceberg_constants.hpp"
 
 #include "duckdb/common/string_util.hpp"
-#include "duckdb/common/extra_type_info.hpp"
+#include "duckdb/common/logical_type_info.hpp"
 #include "duckdb/common/types.hpp"
 #include "duckdb/common/types/blob.hpp"
 #include "duckdb/parser/column_definition.hpp"
