diff --git a/src/function/scan/iceberg_scan.cpp b/src/function/scan/iceberg_scan.cpp
index 8a6d9263e..0cd186c9a 100644
--- a/src/function/scan/iceberg_scan.cpp
+++ b/src/function/scan/iceberg_scan.cpp
@@ -64,14 +64,14 @@ static unique_ptr<FunctionData> IcebergScanDeserialize(Deserializer &deserialize
 	throw NotImplementedException("IcebergScan deserialization not implemented");
 }
 
-BindInfo IcebergBindInfo(const optional_ptr<FunctionData> bind_data) {
+optional_ptr<TableCatalogEntry> IcebergGetTableEntry(optional_ptr<const FunctionData> bind_data) {
 	auto &multi_file_data = bind_data->Cast<MultiFileBindData>();
 	auto &file_list = multi_file_data.file_list->Cast<IcebergMultiFileList>();
 	auto table = file_list.GetTable();
 	if (!table) {
-		return BindInfo(ScanType::EXTERNAL);
+		return nullptr;
 	}
-	return BindInfo(*table);
+	return table.get();
 }
 
 static void IcebergSetScanOrder(unique_ptr<RowGroupOrderOptions> order_options, optional_ptr<FunctionData> bind_data) {
@@ -110,7 +110,7 @@ TableFunctionSet IcebergFunctions::GetIcebergScanFunction(ExtensionLoader &loade
 
 		function.statistics = nullptr;
 		function.table_scan_progress = nullptr;
-		function.get_bind_info = IcebergBindInfo;
+		function.get_table_entry = IcebergGetTableEntry;
 		function.get_virtual_columns = IcebergVirtualColumns;
 		function.get_partition_stats = IcebergMultiFileReader::IcebergGetPartitionStats;
 		function.set_scan_order = IcebergSetScanOrder;
