diff --git a/src/include/execution/operator/iceberg_insert.hpp b/src/include/execution/operator/iceberg_insert.hpp
index 49078028..e52dbe72 100644
--- a/src/include/execution/operator/iceberg_insert.hpp
+++ b/src/include/execution/operator/iceberg_insert.hpp
@@ -32,7 +32,7 @@ public:
 	const IcebergTableSchema &schema;
 	string data_path;
 	//! Set of (key, value) options
-	case_insensitive_map_t<vector<Value>> options;
+	identifier_map_t<vector<Value>> options;
 	//! Partition specification for the table (if partitioned)
 	optional_ptr<const IcebergPartitionSpec> partition_spec;
 	//! Table index for logical plan generation (used when generating partition expressions)
diff --git a/src/include/planning/iceberg_multi_file_reader.hpp b/src/include/planning/iceberg_multi_file_reader.hpp
index 9094aa8a..d3d3d9cf 100644
--- a/src/include/planning/iceberg_multi_file_reader.hpp
+++ b/src/include/planning/iceberg_multi_file_reader.hpp
@@ -73,7 +73,8 @@ public:
 	                          const IcebergMultiFileList &multi_file_list,
 	                          const BoundIcebergManifestEntry &manifest_entry,
 	                          const vector<MultiFileColumnDefinition> &local_columns);
-	bool ParseOption(const string &key, const Value &val, MultiFileOptions &options, ClientContext &context) override;
+	bool ParseOption(const Identifier &key, const Value &val, MultiFileOptions &options,
+	                 ClientContext &context) override;
 
 	MultiFileReaderVirtualColumnBinding
 	GetVirtualColumnExpression(ClientContext &context, MultiFileReaderData &reader_data,
diff --git a/src/planning/iceberg_multi_file_reader.cpp b/src/planning/iceberg_multi_file_reader.cpp
index 5b4d56cb..ee2d295a 100644
--- a/src/planning/iceberg_multi_file_reader.cpp
+++ b/src/planning/iceberg_multi_file_reader.cpp
@@ -442,24 +442,23 @@ void IcebergMultiFileReader::FinalizeChunk(ClientContext &context, const MultiFi
 	output_chunk.Flatten();
 }
 
-bool IcebergMultiFileReader::ParseOption(const string &key, const Value &val, MultiFileOptions &options,
+bool IcebergMultiFileReader::ParseOption(const Identifier &key, const Value &val, MultiFileOptions &options,
                                          ClientContext &context) {
-	auto loption = StringUtil::Lower(key);
 	auto &snapshot_lookup = this->options.snapshot_lookup;
 
-	if (loption == "allow_moved_paths") {
+	if (key == "allow_moved_paths") {
 		this->options.allow_moved_paths = BooleanValue::Get(val);
 		return true;
 	}
-	if (loption == "metadata_compression_codec") {
+	if (key == "metadata_compression_codec") {
 		this->options.metadata_compression_codec = StringValue::Get(val);
 		return true;
 	}
-	if (loption == "version") {
+	if (key == "version") {
 		this->options.table_version = StringValue::Get(val);
 		return true;
 	}
-	if (loption == "version_name_format") {
+	if (key == "version_name_format") {
 		auto value = StringValue::Get(val);
 		auto string_substitutions = IcebergUtils::CountOccurrences(value, "%s");
 		if (string_substitutions != 2) {
@@ -469,14 +468,14 @@ bool IcebergMultiFileReader::ParseOption(const string &key, const Value &val, Mu
 		this->options.version_name_format = value;
 		return true;
 	}
-	if (loption == "snapshot_from_id") {
+	if (key == "snapshot_from_id") {
 		if (snapshot_lookup->GetSource() != SnapshotSource::LATEST) {
 			throw InvalidInputException("Can't use 'snapshot_from_id' in combination with 'snapshot_from_timestamp'");
 		}
 		snapshot_lookup.emplace(IcebergSnapshotLookup::FromSnapshotId(val.GetValue<uint64_t>()));
 		return true;
 	}
-	if (loption == "snapshot_from_timestamp") {
+	if (key == "snapshot_from_timestamp") {
 		if (snapshot_lookup->GetSource() != SnapshotSource::LATEST) {
 			throw InvalidInputException("Can't use 'snapshot_from_id' in combination with 'snapshot_from_timestamp'");
 		}
