diff --git a/src/storage/iceberg_table_information.cpp b/src/storage/iceberg_table_information.cpp
index f61ba8b4..b42a18b9 100644
--- a/src/storage/iceberg_table_information.cpp
+++ b/src/storage/iceberg_table_information.cpp
@@ -341,23 +341,23 @@ void IcebergTableInformation::SetPartitionedBy(IcebergTransaction &transaction,
 			column_name = colref.ColumnNames().back();
 		} else if (key->GetExpressionType() == ExpressionType::FUNCTION) {
 			auto &funcexpr = key->Cast<FunctionExpression>();
-			transform_name = funcexpr.function_name;
-			if (funcexpr.children.empty()) {
+			transform_name = funcexpr.FunctionName();
+			if (funcexpr.GetChildren().empty()) {
 				throw NotImplementedException("Unrecognized transform ('%s')", transform_name);
 			} else if (!IcebergTransform::TransformFunctionSupported(transform_name)) {
 				throw NotImplementedException("Unrecognized transform ('%s')", transform_name);
-			} else if (funcexpr.children[0]->GetExpressionType() != ExpressionType::COLUMN_REF) {
+			} else if (funcexpr.GetChildren()[0]->GetExpressionType() != ExpressionType::COLUMN_REF) {
 				throw NotImplementedException("Transforms are only supported on column references, not %s",
-				                              EnumUtil::ToChars(funcexpr.children[0]->GetExpressionType()));
+				                              EnumUtil::ToChars(funcexpr.GetChildren()[0]->GetExpressionType()));
 			}
-			auto &colref = funcexpr.children[0]->Cast<ColumnRefExpression>();
+			auto &colref = funcexpr.GetChildren()[0]->Cast<ColumnRefExpression>();
 			column_name = colref.ColumnNames().back();
 			if (transform_name == "bucket" || transform_name == "truncate") {
-				if (funcexpr.children.size() < 2) {
+				if (funcexpr.GetChildren().size() < 2) {
 					throw InvalidInputException("%s requires a numeric argument, e.g. %s(col, 16)", transform_name,
 					                            transform_name);
 				}
-				auto &param_expr = *funcexpr.children[1];
+				auto &param_expr = *funcexpr.GetChildren()[1];
 				if (param_expr.GetExpressionType() != ExpressionType::VALUE_CONSTANT) {
 					throw InvalidInputException("%s second argument must be a constant integer", transform_name);
 				}
