diff --git a/test/sql/local/partitioning/foldable_expression_filter.test b/test/sql/local/partitioning/foldable_expression_filter.test
index 2f7c4eef70..06fc8bcbd6 100644
--- a/test/sql/local/partitioning/foldable_expression_filter.test
+++ b/test/sql/local/partitioning/foldable_expression_filter.test
@@ -15,6 +15,10 @@ set enable_logging=true
 statement ok
 set logging_level='debug'
 
+#FIXME: logging types are not registered to extensions
+statement ok
+LOAD iceberg;
+
 statement ok
 CALL enable_logging('Iceberg');
 
diff --git a/test/sql/local/partitioning/in_filter.test b/test/sql/local/partitioning/in_filter.test
index 8b492a2672..9b4be1fb2e 100644
--- a/test/sql/local/partitioning/in_filter.test
+++ b/test/sql/local/partitioning/in_filter.test
@@ -15,6 +15,10 @@ set enable_logging=true
 statement ok
 set logging_level='debug'
 
+#FIXME: logging types are not registered to extensions
+statement ok
+LOAD iceberg;
+
 statement ok
 CALL enable_logging('Iceberg');
 
