diff --git a/src/iceberg_extension.cpp b/src/iceberg_extension.cpp
index 4c3b4d5e..ae40ef9c 100644
--- a/src/iceberg_extension.cpp
+++ b/src/iceberg_extension.cpp
@@ -79,8 +79,7 @@ static void LoadInternal(ExtensionLoader &loader) {
 
 	auto &log_manager = instance.GetLogManager();
 	log_manager.RegisterLogType(make_uniq<IcebergLogType>());
-
-	config.storage_extensions["iceberg"] = make_uniq<IRCStorageExtension>();
+	StorageExtension::Register(config, "iceberg", make_shared_ptr<IRCStorageExtension>());
 }
 
 void IcebergExtension::Load(ExtensionLoader &loader) {
diff --git a/src/iceberg_functions/iceberg_multi_file_list.cpp b/src/iceberg_functions/iceberg_multi_file_list.cpp
index 04c55e63..68bbd788 100644
--- a/src/iceberg_functions/iceberg_multi_file_list.cpp
+++ b/src/iceberg_functions/iceberg_multi_file_list.cpp
@@ -24,7 +24,7 @@ namespace duckdb {
 
 IcebergMultiFileList::IcebergMultiFileList(ClientContext &context_p, shared_ptr<IcebergScanInfo> scan_info,
                                            const string &path, const IcebergOptions &options)
-    : MultiFileList(vector<OpenFileInfo> {}, FileGlobOptions::ALLOW_EMPTY), context(context_p),
+    : context(context_p),
       fs(FileSystem::GetFileSystem(context)), scan_info(scan_info), path(path), table(nullptr), lock(),
       options(options) {
 }
@@ -192,7 +192,7 @@ IcebergMultiFileList::DynamicFilterPushdown(ClientContext &context, const MultiF
 unique_ptr<MultiFileList> IcebergMultiFileList::ComplexFilterPushdown(ClientContext &context,
                                                                       const MultiFileOptions &options,
                                                                       MultiFilePushdownInfo &info,
-                                                                      vector<unique_ptr<Expression>> &filters) {
+                                                                      vector<unique_ptr<Expression>> &filters) const {
 	if (filters.empty()) {
 		return nullptr;
 	}
@@ -211,7 +211,7 @@ unique_ptr<MultiFileList> IcebergMultiFileList::ComplexFilterPushdown(ClientCont
 	return PushdownInternal(context, filter_set);
 }
 
-vector<OpenFileInfo> IcebergMultiFileList::GetAllFiles() {
+vector<OpenFileInfo> IcebergMultiFileList::GetAllFiles() const {
 	vector<OpenFileInfo> file_list;
 	//! Lock is required because it reads the 'data_files' vector
 	lock_guard<mutex> guard(lock);
@@ -221,7 +221,7 @@ vector<OpenFileInfo> IcebergMultiFileList::GetAllFiles() {
 	return file_list;
 }
 
-FileExpandResult IcebergMultiFileList::GetExpandResult() {
+FileExpandResult IcebergMultiFileList::GetExpandResult() const {
 	// GetFileInternal(1) will ensure files with index 0 and index 1 are expanded if they are available
 	lock_guard<mutex> guard(lock);
 	GetFileInternal(1, guard);
@@ -235,7 +235,7 @@ FileExpandResult IcebergMultiFileList::GetExpandResult() {
 	return FileExpandResult::NO_FILES;
 }
 
-idx_t IcebergMultiFileList::GetTotalFileCount() {
+idx_t IcebergMultiFileList::GetTotalFileCount() const {
 	// FIXME: the 'added_files_count' + the 'existing_files_count'
 	// in the Manifest List should give us this information without scanning the manifest list
 	lock_guard<mutex> guard(lock);
@@ -247,7 +247,7 @@ idx_t IcebergMultiFileList::GetTotalFileCount() {
 	return data_files.size();
 }
 
-unique_ptr<NodeStatistics> IcebergMultiFileList::GetCardinality(ClientContext &context) {
+unique_ptr<NodeStatistics> IcebergMultiFileList::GetCardinality(ClientContext &context) const {
 	idx_t cardinality = 0;
 
 	if (GetMetadata().iceberg_version == 1) {
@@ -436,7 +436,7 @@ bool IcebergMultiFileList::FileMatchesFilter(const IcebergManifestEntry &file) c
 	return true;
 }
 
-optional_ptr<const IcebergManifestEntry> IcebergMultiFileList::GetDataFile(idx_t file_id, lock_guard<mutex> &guard) {
+optional_ptr<const IcebergManifestEntry> IcebergMultiFileList::GetDataFile(idx_t file_id, lock_guard<mutex> &guard) const {
 	if (file_id < data_files.size()) {
 		//! Have we already scanned this data file and returned it? If so, return it
 		return data_files[file_id];
@@ -507,7 +507,7 @@ optional_ptr<const IcebergManifestEntry> IcebergMultiFileList::GetDataFile(idx_t
 	return data_files[file_id];
 }
 
-OpenFileInfo IcebergMultiFileList::GetFileInternal(idx_t file_id, lock_guard<mutex> &guard) {
+OpenFileInfo IcebergMultiFileList::GetFileInternal(idx_t file_id, lock_guard<mutex> &guard) const {
 	if (!initialized) {
 		InitializeFiles(guard);
 	}
@@ -543,12 +543,12 @@ OpenFileInfo IcebergMultiFileList::GetFileInternal(idx_t file_id, lock_guard<mut
 	return res;
 }
 
-OpenFileInfo IcebergMultiFileList::GetFile(idx_t file_id) {
+OpenFileInfo IcebergMultiFileList::GetFile(idx_t file_id) const {
 	lock_guard<mutex> guard(lock);
 	return GetFileInternal(file_id, guard);
 }
 
-bool IcebergMultiFileList::ManifestMatchesFilter(const IcebergManifestListEntry &manifest) {
+bool IcebergMultiFileList::ManifestMatchesFilter(const IcebergManifestListEntry &manifest) const {
 	auto spec_id = manifest.partition_spec_id;
 	auto &metadata = GetMetadata();
 
@@ -606,7 +606,7 @@ bool IcebergMultiFileList::ManifestMatchesFilter(const IcebergManifestListEntry
 	return true;
 }
 
-void IcebergMultiFileList::InitializeFiles(lock_guard<mutex> &guard) {
+void IcebergMultiFileList::InitializeFiles(lock_guard<mutex> &guard) const {
 	if (initialized) {
 		return;
 	}
diff --git a/src/iceberg_logging.cpp b/src/iceberg_logging.cpp
index f7b3fa02..6bedb2f6 100644
--- a/src/iceberg_logging.cpp
+++ b/src/iceberg_logging.cpp
@@ -4,7 +4,7 @@ namespace duckdb {
 
 constexpr LogLevel IcebergLogType::LEVEL;
 
-IcebergLogType::IcebergLogType() : LogType(NAME, LEVEL, GetLogType()) {
+IcebergLogType::IcebergLogType() : LogType(NAME, LEVEL) {
 }
 
 } // namespace duckdb
diff --git a/src/include/iceberg_multi_file_list.hpp b/src/include/iceberg_multi_file_list.hpp
index 62c19443..44e49545 100644
--- a/src/include/iceberg_multi_file_list.hpp
+++ b/src/include/iceberg_multi_file_list.hpp
@@ -63,25 +63,25 @@ public:
 	                                                TableFilterSet &filters) const override;
 	unique_ptr<MultiFileList> ComplexFilterPushdown(ClientContext &context, const MultiFileOptions &options,
 	                                                MultiFilePushdownInfo &info,
-	                                                vector<unique_ptr<Expression>> &filters) override;
-	vector<OpenFileInfo> GetAllFiles() override;
-	FileExpandResult GetExpandResult() override;
-	idx_t GetTotalFileCount() override;
-	unique_ptr<NodeStatistics> GetCardinality(ClientContext &context) override;
+	                                                vector<unique_ptr<Expression>> &filters) const override;
+	vector<OpenFileInfo> GetAllFiles() const override;
+	FileExpandResult GetExpandResult() const override;
+	idx_t GetTotalFileCount() const override;
+	unique_ptr<NodeStatistics> GetCardinality(ClientContext &context) const override;
 
 protected:
 	//! Get the i-th expanded file
-	OpenFileInfo GetFile(idx_t i) override;
-	OpenFileInfo GetFileInternal(idx_t i, lock_guard<mutex> &guard);
+	OpenFileInfo GetFile(idx_t i) const override;
+	OpenFileInfo GetFileInternal(idx_t i, lock_guard<mutex> &guard) const;
 
 protected:
-	bool ManifestMatchesFilter(const IcebergManifestListEntry &manifest);
+	bool ManifestMatchesFilter(const IcebergManifestListEntry &manifest) const;
 	bool FileMatchesFilter(const IcebergManifestEntry &file) const;
 	// TODO: How to guarantee we only call this after the filter pushdown?
-	void InitializeFiles(lock_guard<mutex> &guard);
+	void InitializeFiles(lock_guard<mutex> &guard) const;
 
 	//! NOTE: this requires the lock because it modifies the 'data_files' vector, potentially invalidating references
-	optional_ptr<const IcebergManifestEntry> GetDataFile(idx_t file_id, lock_guard<mutex> &guard);
+	optional_ptr<const IcebergManifestEntry> GetDataFile(idx_t file_id, lock_guard<mutex> &guard) const;
 
 	optional_ptr<const TableFilter> GetFilterForColumnIndex(const TableFilterSet &filter_set,
 	                                                        const ColumnIndex &column_index) const;
@@ -100,23 +100,23 @@ public:
 	vector<LogicalType> types;
 	TableFilterSet table_filters;
 
-	unique_ptr<manifest_file::ManifestFileReader> data_manifest_reader;
-	unique_ptr<manifest_file::ManifestFileReader> delete_manifest_reader;
+	mutable unique_ptr<manifest_file::ManifestFileReader> data_manifest_reader;
+	mutable unique_ptr<manifest_file::ManifestFileReader> delete_manifest_reader;
 
-	vector<IcebergManifestEntry> data_files;
-	vector<IcebergManifestListEntry> data_manifests;
-	vector<IcebergManifestListEntry> delete_manifests;
-	vector<reference<IcebergManifestFile>> transaction_data_manifests;
-	vector<reference<IcebergManifestFile>> transaction_delete_manifests;
-	idx_t transaction_data_idx = 0;
+	mutable vector<IcebergManifestEntry> data_files;
+	mutable vector<IcebergManifestListEntry> data_manifests;
+	mutable vector<IcebergManifestListEntry> delete_manifests;
+	mutable vector<reference<IcebergManifestFile>> transaction_data_manifests;
+	mutable vector<reference<IcebergManifestFile>> transaction_delete_manifests;
+	mutable idx_t transaction_data_idx = 0;
 	idx_t transaction_delete_idx = 0;
 
-	vector<IcebergManifestListEntry>::iterator current_data_manifest;
+	mutable vector<IcebergManifestListEntry>::iterator current_data_manifest;
 	mutable vector<IcebergManifestListEntry>::iterator current_delete_manifest;
 	mutable vector<reference<IcebergManifestFile>>::iterator current_transaction_delete_manifest;
 	//! The data files of the manifest file that we last scanned
-	idx_t data_file_idx = 0;
-	vector<IcebergManifestEntry> current_data_files;
+	mutable idx_t data_file_idx = 0;
+	mutable vector<IcebergManifestEntry> current_data_files;
 
 	//! For each file that has a delete file, the state for processing that/those delete file(s)
 	mutable case_insensitive_map_t<unique_ptr<DeleteFilter>> positional_delete_data;
@@ -124,7 +124,7 @@ public:
 	mutable map<sequence_number_t, unique_ptr<IcebergEqualityDeleteData>> equality_delete_data;
 	mutable mutex delete_lock;
 
-	bool initialized = false;
+	mutable bool initialized = false;
 	const IcebergOptions &options;
 };
 
