diff --git a/duckdb b/duckdb
index 35fde53..2f131ae 160000
--- a/duckdb
+++ b/duckdb
@@ -1 +1 @@
-Subproject commit 35fde53437d93697136953ef4e23d89bf35018b2
+Subproject commit 2f131ae80ad7b1c10cdab2e132d4d7ac87e4d75f
diff --git a/src/common/iceberg.cpp b/src/common/iceberg.cpp
index 9f43afd..ebea592 100644
--- a/src/common/iceberg.cpp
+++ b/src/common/iceberg.cpp
@@ -138,7 +138,7 @@ IcebergSnapshot IcebergSnapshot::ParseSnapShot(yyjson_val *snapshot) {
 
 idx_t IcebergSnapshot::GetTableVersion(string &path, FileSystem &fs) {
 	auto meta_path = fs.JoinPath(path, "metadata");
-	auto version_file_path = FileSystem::JoinPath(meta_path, "version-hint.text");
+	auto version_file_path = fs.JoinPath(meta_path, "version-hint.text");
 	auto version_file_content = IcebergUtils::FileToString(version_file_path, fs);
 
 	try {
