diff --git a/src/common/iceberg.cpp b/src/common/iceberg.cpp
index 3c15105..face972 100644
--- a/src/common/iceberg.cpp
+++ b/src/common/iceberg.cpp
@@ -127,7 +127,7 @@ unique_ptr<SnapshotParseInfo> IcebergSnapshot::GetParseInfo(const string &path,
 	parse_info->doc = doc;
 	parse_info->document = std::move(metadata_json);
 
-	return std::move(parse_info);
+	return parse_info;
 }
 
 IcebergSnapshot IcebergSnapshot::GetLatestSnapshot(const string &path, FileSystem &fs) {
diff --git a/src/common/utils.cpp b/src/common/utils.cpp
index c0272bf..d2b01d9 100644
--- a/src/common/utils.cpp
+++ b/src/common/utils.cpp
@@ -4,8 +4,7 @@
 namespace duckdb {
 
 string IcebergUtils::FileToString(const string &path, FileSystem &fs) {
-	auto handle =
-	    fs.OpenFile(path, FileFlags::FILE_FLAGS_READ, FileSystem::DEFAULT_LOCK, FileSystem::DEFAULT_COMPRESSION);
+	auto handle = fs.OpenFile(path, FileFlags::FILE_FLAGS_READ);
 	auto file_size = handle->GetFileSize();
 	string ret_val(file_size, ' ');
 	handle->Read((char *)ret_val.c_str(), file_size);
diff --git a/src/iceberg_extension.cpp b/src/iceberg_extension.cpp
index 0e59fdc..895b79d 100644
--- a/src/iceberg_extension.cpp
+++ b/src/iceberg_extension.cpp
@@ -16,12 +16,6 @@
 namespace duckdb {
 
 static void LoadInternal(DatabaseInstance &instance) {
-	Connection con(instance);
-	con.BeginTransaction();
-	auto &context = *con.context;
-
-	auto &catalog = Catalog::GetSystemCatalog(*con.context);
-
 	// Iceberg Table Functions
 	for (auto &fun : IcebergFunctions::GetTableFunctions()) {
 		ExtensionUtil::RegisterFunction(instance, fun);
@@ -31,8 +25,6 @@ static void LoadInternal(DatabaseInstance &instance) {
 	for (auto &fun : IcebergFunctions::GetScalarFunctions()) {
 		ExtensionUtil::RegisterFunction(instance, fun);
 	}
-
-	con.Commit();
 }
 
 void IcebergExtension::Load(DuckDB &db) {
diff --git a/src/iceberg_functions/iceberg_snapshots.cpp b/src/iceberg_functions/iceberg_snapshots.cpp
index 6d6ef57..4b67a6d 100644
--- a/src/iceberg_functions/iceberg_snapshots.cpp
+++ b/src/iceberg_functions/iceberg_snapshots.cpp
@@ -93,7 +93,7 @@ TableFunctionSet IcebergFunctions::GetIcebergSnapshotsFunction() {
 	TableFunction table_function({LogicalType::VARCHAR}, IcebergSnapshotsFunction, IcebergSnapshotsBind,
 	                             IcebergSnapshotGlobalTableFunctionState::Init);
 	function_set.AddFunction(table_function);
-	return std::move(function_set);
+	return function_set;
 }
 
 } // namespace duckdb
