diff --git a/src/storage/quack_catalog.cpp b/src/storage/quack_catalog.cpp
index 11772c6..1269a2e 100644
--- a/src/storage/quack_catalog.cpp
+++ b/src/storage/quack_catalog.cpp
@@ -221,13 +221,13 @@ unique_ptr<TableRef> QuackCatalog::RemoteExecute(ClientContext &context, const s
 
 unique_ptr<TableRef> QuackCatalog::CreateRemoteQueryRef(const string &sql, bool refresh_catalog) {
 	vector<unique_ptr<ParsedExpression>> args;
-	args.push_back(make_uniq<ConstantExpression>(Value(GetName())));
-	args.push_back(make_uniq<ConstantExpression>(Value(sql)));
-	auto use_transaction = make_uniq<ConstantExpression>(Value::BOOLEAN(true));
+	args.push_back(ConstantExpression::FromValue(Value(GetName())));
+	args.push_back(ConstantExpression::FromValue(Value(sql)));
+	auto use_transaction = ConstantExpression::FromValue(Value::BOOLEAN(true));
 	use_transaction->SetAlias("use_transaction");
 	args.push_back(std::move(use_transaction));
 	if (refresh_catalog) {
-		auto refresh = make_uniq<ConstantExpression>(Value::BOOLEAN(true));
+		auto refresh = ConstantExpression::FromValue(Value::BOOLEAN(true));
 		refresh->SetAlias("refresh_catalog");
 		args.push_back(std::move(refresh));
 	}
