diff --git a/src/storage/quack_table.cpp b/src/storage/quack_table.cpp
index a992de8..2274b1c 100644
--- a/src/storage/quack_table.cpp
+++ b/src/storage/quack_table.cpp
@@ -6,6 +6,7 @@
 #include "duckdb/parser/statement/create_statement.hpp"
 #include "duckdb/parser/parsed_data/alter_info.hpp"
 #include "duckdb/planner/parsed_data/bound_create_table_info.hpp"
+#include "duckdb/planner/binder.hpp"
 #include "duckdb/parser/parsed_data/create_view_info.hpp"
 #include "storage/quack_view.hpp"
 
