diff --git a/src/quack_server.cpp b/src/quack_server.cpp
--- a/src/quack_server.cpp
+++ b/src/quack_server.cpp
@@ -73,7 +73,7 @@
 	return true;
 }
 
-static string GetSettingString(DatabaseInstance &db, const string &setting_name) {
+static string GetSettingString(DatabaseInstance &db, const Identifier &setting_name) {
 	Value setting_val;
 	auto &config = DBConfig::GetConfig(db);
 
diff --git a/src/quack_extension.cpp b/src/quack_extension.cpp
--- a/src/quack_extension.cpp
+++ b/src/quack_extension.cpp
@@ -93,7 +93,7 @@
 		if (kv.second.IsNull()) {
 			continue;
 		}
-		db_config.SetOptionByName("whoami_" + kv.first, kv.second);
+		db_config.SetOptionByName(Identifier("whoami_" + kv.first), kv.second);
 	}
 	return_types.emplace_back(LogicalType::BOOLEAN);
 	names.emplace_back("ok");
