diff --git a/src/storage/quack_schema.cpp b/src/storage/quack_schema.cpp
index e3d13bc..2ba4542 100644
--- a/src/storage/quack_schema.cpp
+++ b/src/storage/quack_schema.cpp
@@ -2,6 +2,7 @@
 #include "storage/quack_schema.hpp"
 #include "storage/quack_table.hpp"
 #include "quack_client.hpp"
+#include "duckdb/parser/parser.hpp"
 #include "duckdb/parser/parsed_data/create_table_info.hpp"
 #include "duckdb/parser/parsed_data/create_view_info.hpp"
 #include "duckdb/parser/parsed_data/drop_info.hpp"
@@ -205,7 +206,8 @@ optional_ptr<CatalogEntry> QuackSchemaCatalogEntry::CreateView(CatalogTransactio
 
 	// locally, override the query with a remote procedure call to ensure the view is evaluated remotely
 	info.sql = QuackViewCatalogEntry::CreateViewSQL(ParentCatalog().GetName().GetIdentifierName(), remote_name);
-	info.query = CreateViewInfo::ParseSelect(info.sql);
+	auto parser = Parser::GetBuiltinParser();
+	info.query = CreateViewInfo::ParseSelect(parser, info.sql);
 
 	auto quack_entry = make_uniq<QuackViewCatalogEntry>(catalog, *this, info);
 	return tables->CreateEntry(std::move(quack_entry), info.on_conflict);
diff --git a/src/storage/quack_table.cpp b/src/storage/quack_table.cpp
index 2274b1c..fe9e7a6 100644
--- a/src/storage/quack_table.cpp
+++ b/src/storage/quack_table.cpp
@@ -13,7 +13,7 @@
 namespace duckdb {
 
 unique_ptr<CreateInfo> ParseCreateTable(const string &sql) {
-	Parser parser;
+	auto parser = Parser::GetBuiltinParser();
 	parser.ParseQuery(sql);
 	if (parser.statements.size() != 1 || parser.statements[0]->type != StatementType::CREATE_STATEMENT) {
 		throw BinderException(
@@ -66,7 +66,8 @@ QuackTableSet::QuackTableSet(ClientContext &context, QuackSchemaCatalogEntry &pa
 			CreateViewInfo info(schema, view_name);
 			info.sql = QuackViewCatalogEntry::CreateViewSQL(catalog.GetName().GetIdentifierName(),
 			                                                parent.GetRemoteName(view_name));
-			info.query = CreateViewInfo::ParseSelect(info.sql);
+			auto parser = Parser::GetBuiltinParser();
+			info.query = CreateViewInfo::ParseSelect(parser, info.sql);
 
 			// bind to resolve the types
 			auto view = make_uniq<QuackViewCatalogEntry>(catalog, parent, info);
