diff --git a/src/quack_scan.cpp b/src/quack_scan.cpp
index c95cbdc..53e6edb 100644
--- a/src/quack_scan.cpp
+++ b/src/quack_scan.cpp
@@ -464,12 +464,9 @@ unique_ptr<FunctionData> QuackScanDeserialize(Deserializer &deserializer, BoundT
 	throw NotImplementedException("Quack scans cannot be deserialized (yet?)");
 }
 
-BindInfo QuackScanGetBindInfo(const optional_ptr<FunctionData> bind_data_p) {
-	auto &bind_data = bind_data_p->CastNoConst<QuackScanBindData>();
-	if (bind_data.table_entry) {
-		return BindInfo(*bind_data.table_entry);
-	}
-	return BindInfo(ScanType::EXTERNAL);
+optional_ptr<TableCatalogEntry> QuackScanGetTableEntry(optional_ptr<const FunctionData> bind_data_p) {
+	auto &bind_data = bind_data_p->Cast<QuackScanBindData>();
+	return bind_data.table_entry;
 }
 
 TableFunction QuackScanFunction::GetFunction() {
@@ -490,7 +487,7 @@ TableFunction QuackScanFunction::GetFunction() {
 	fun.to_string = QuackScanToString;
 	fun.serialize = QuackScanSerialize;
 	fun.deserialize = QuackScanDeserialize;
-	fun.get_bind_info = QuackScanGetBindInfo;
+	fun.get_table_entry = QuackScanGetTableEntry;
 	// fun.filter_pushdown = true;
 	// fun.filter_prune = true;
 	return fun;
@@ -510,7 +507,7 @@ TableFunction QuackScanByNameFunction::GetFunction() {
 	fun.to_string = QuackScanToString;
 	fun.serialize = QuackScanSerialize;
 	fun.deserialize = QuackScanDeserialize;
-	fun.get_bind_info = QuackScanGetBindInfo;
+	fun.get_table_entry = QuackScanGetTableEntry;
 	// fun.filter_pushdown = true;
 	// fun.filter_prune = true;
 	return fun;
