--- a/src/quack_clear_cache.cpp
+++ b/src/quack_clear_cache.cpp
@@ -14,7 +14,7 @@ struct ClearCacheFunctionData : public TableFunctionData {
 };
 
 unique_ptr<FunctionData> ClearCacheBind(ClientContext &, TableFunctionBindInput &, vector<LogicalType> &return_types,
-                                        vector<string> &names) {
+                                        vector<Identifier> &names) {
 	return_types.emplace_back(LogicalType::BOOLEAN);
 	names.emplace_back("Success");
 	return make_uniq<ClearCacheFunctionData>();
--- a/src/quack_extension.cpp
+++ b/src/quack_extension.cpp
@@ -87,7 +87,7 @@ static void QuackIdentifyFun(ClientContext &, TableFunctionInput &, DataChunk &)
 }
 
 static unique_ptr<FunctionData> QuackIdentifyBind(ClientContext &ctx, TableFunctionBindInput &input,
-                                                  vector<LogicalType> &return_types, vector<string> &names) {
+                                                  vector<LogicalType> &return_types, vector<Identifier> &names) {
 	auto &db_config = DBConfig::GetConfig(ctx);
 	for (auto &kv : input.named_parameters) {
 		if (kv.second.IsNull()) {
--- a/src/quack_scan.cpp
+++ b/src/quack_scan.cpp
@@ -13,7 +13,7 @@
 namespace duckdb {
 
 static unique_ptr<FunctionData> QuackScanBind(ClientContext &context, TableFunctionBindInput &input,
-                                              vector<LogicalType> &return_types, vector<string> &names) {
+                                              vector<LogicalType> &return_types, vector<Identifier> &names) {
 	// Set logging to be pretty verbose (everything except message payloads)
 	if (input.inputs.empty()) {
 		throw InternalException("No input to quack scan?");
@@ -50,7 +50,7 @@ static unique_ptr<FunctionData> QuackScanBind(ClientContext &context, TableFunct
 	    context, make_uniq<PrepareRequestMessage>(client_connection.ConnectionId(), query));
 
 	return_types = bind_response->Types();
-	names = bind_response->Names();
+	names = StringsToIdentifiers(bind_response->Names());
 
 	bind_data->results = std::move(bind_response->MutableResults());
 	bind_data->needs_more_fetch = bind_response->NeedsMoreFetch();
@@ -78,7 +78,7 @@ QuackCatalog &GetQuackCatalog(ClientContext &context, Value &catalog_name) {
 }
 
 static unique_ptr<FunctionData> QuackScanBindCatalogName(ClientContext &context, TableFunctionBindInput &input,
-                                                         vector<LogicalType> &return_types, vector<string> &names) {
+                                                         vector<LogicalType> &return_types, vector<Identifier> &names) {
 	if (input.inputs[0].IsNull() || input.inputs[1].IsNull()) {
 		throw BinderException("catalog_name and query parameters cannot be NULL");
 	}
@@ -96,7 +96,7 @@ static unique_ptr<FunctionData> QuackScanBindCatalogName(ClientContext &context,
 	    context, make_uniq<PrepareRequestMessage>(bind_data->client_connection->ConnectionId(), query));
 
 	return_types = bind_response->Types();
-	names = bind_response->Names();
+	names = StringsToIdentifiers(bind_response->Names());
 
 	// new stuff
 	bind_data->results = std::move(bind_response->MutableResults());
--- a/src/quack_start_stop.cpp
+++ b/src/quack_start_stop.cpp
@@ -15,7 +15,7 @@ struct QuackStartStopFunctionData : public TableFunctionData {
 };
 
 static unique_ptr<FunctionData> QuackServeBind(ClientContext &context, TableFunctionBindInput &input,
-                                               vector<LogicalType> &return_types, vector<string> &names) {
+                                               vector<LogicalType> &return_types, vector<Identifier> &names) {
 	auto bind_data = make_uniq<QuackStartStopFunctionData>();
 	string listen_uri;
 	if (input.inputs.empty()) {
@@ -89,7 +89,7 @@ TableFunctionSet QuackServeFunction::GetFunction() {
 }
 
 static unique_ptr<FunctionData> QuackStopBind(ClientContext &context, TableFunctionBindInput &input,
-                                              vector<LogicalType> &return_types, vector<string> &names) {
+                                              vector<LogicalType> &return_types, vector<Identifier> &names) {
 	auto bind_data = make_uniq<QuackStartStopFunctionData>();
 	auto &uri_value = input.inputs[0];
 	if (uri_value.IsNull() || uri_value.GetValue<string>().empty()) {
@@ -127,7 +127,7 @@ struct QuackServerListFunctionData : public TableFunctionData {
 };
 
 static unique_ptr<FunctionData> QuackServerListBind(ClientContext &context, TableFunctionBindInput &input,
-                                                    vector<LogicalType> &return_types, vector<string> &names) {
+                                                    vector<LogicalType> &return_types, vector<Identifier> &names) {
 	return_types.emplace_back(LogicalType::VARCHAR);
 	names.emplace_back("listen_uri");
 	return_types.emplace_back(LogicalType::VARCHAR);
