From 4534a6394215735b56ee6f2caeebd50d81726d7f Mon Sep 17 00:00:00 2001
From: duckdb <x@x>
Date: Fri, 26 Jun 2026 11:34:19 +0200
Subject: [PATCH] zzzzzz-setqualifiedname.patch

---
 src/storage/quack_schema.cpp | 36 ++++++++++++++++++++++++++----------
 1 file changed, 26 insertions(+), 10 deletions(-)

diff --git a/src/storage/quack_schema.cpp b/src/storage/quack_schema.cpp
index c92d91f..122cba1 100644
--- a/src/storage/quack_schema.cpp
+++ b/src/storage/quack_schema.cpp
@@ -20,8 +20,11 @@ void QuackSchemaSet::Reload(ClientContext &context, QuackCatalog &catalog, const
 	Clear();
 	for (auto &row : load_data.schemas->Rows()) {
 		CreateSchemaInfo info;
-		info.CatalogMutable() = Identifier(row.GetValue(0).GetValue<string>());
-		info.SchemaMutable() = Identifier(row.GetValue(1).GetValue<string>());
+		info.SetQualifiedName(QualifiedName(Identifier(row.GetValue(0).GetValue<string>()),
+		                                    info.GetQualifiedName().Schema(), info.GetQualifiedName().Name()));
+		info.SetQualifiedName(QualifiedName(info.GetQualifiedName().Catalog(),
+		                                    Identifier(row.GetValue(1).GetValue<string>()),
+		                                    info.GetQualifiedName().Name()));
 		// TODO this will fail if there are two schemas with the same name in different catalogs :/
 		auto schema = make_uniq<QuackSchemaCatalogEntry>(context, catalog, info, load_data);
 		CreateEntry(std::move(schema), OnCreateConflict::REPLACE_ON_CONFLICT);
@@ -86,8 +89,12 @@ optional_ptr<CatalogEntry> QuackSchemaCatalogEntry::CreateFunction(CatalogTransa
 optional_ptr<CatalogEntry> QuackSchemaCatalogEntry::CreateTable(CatalogTransaction transaction,
                                                                 BoundCreateTableInfo &info) {
 	auto create_table_info = info.Base().Copy();
-	create_table_info->CatalogMutable() = GetInfo()->GetQualifiedName().Catalog();
-	create_table_info->SchemaMutable() = GetInfo()->GetQualifiedName().Schema();
+	create_table_info->SetQualifiedName(QualifiedName(GetInfo()->GetQualifiedName().Catalog(),
+	                                                  create_table_info->GetQualifiedName().Schema(),
+	                                                  create_table_info->GetQualifiedName().Name()));
+	create_table_info->SetQualifiedName(QualifiedName(create_table_info->GetQualifiedName().Catalog(),
+	                                                  GetInfo()->GetQualifiedName().Schema(),
+	                                                  create_table_info->GetQualifiedName().Name()));
 
 	auto &quack_transaction = QuackTransaction::Get(transaction);
 	quack_transaction.Query(create_table_info->ToString());
@@ -97,15 +104,20 @@ optional_ptr<CatalogEntry> QuackSchemaCatalogEntry::CreateTable(CatalogTransacti
 
 optional_ptr<CatalogEntry> QuackSchemaCatalogEntry::CreateView(CatalogTransaction transaction, CreateViewInfo &info) {
 	auto create_view_info = info.Copy();
-	create_view_info->CatalogMutable() = GetInfo()->GetQualifiedName().Catalog();
-	create_view_info->SchemaMutable() = GetInfo()->GetQualifiedName().Schema();
+	create_view_info->SetQualifiedName(QualifiedName(GetInfo()->GetQualifiedName().Catalog(),
+	                                                 create_view_info->GetQualifiedName().Schema(),
+	                                                 create_view_info->GetQualifiedName().Name()));
+	create_view_info->SetQualifiedName(QualifiedName(create_view_info->GetQualifiedName().Catalog(),
+	                                                 GetInfo()->GetQualifiedName().Schema(),
+	                                                 create_view_info->GetQualifiedName().Name()));
 
 	// create the view verbatim in the serer
 	auto &quack_transaction = QuackTransaction::Get(transaction);
 	quack_transaction.Query(create_view_info->ToString());
 
 	// locally, override the query with a remote procedure call to ensure the view is evaluated remotely
-	info.sql = QuackViewCatalogEntry::CreateViewSQL(ParentCatalog().GetName().GetIdentifierName(), name.GetIdentifierName(), info.GetViewName().GetIdentifierName());
+	info.sql = QuackViewCatalogEntry::CreateViewSQL(ParentCatalog().GetName().GetIdentifierName(),
+	                                                name.GetIdentifierName(), info.GetViewName().GetIdentifierName());
 	info.query = CreateViewInfo::ParseSelect(info.sql);
 
 	auto quack_entry = make_uniq<QuackViewCatalogEntry>(catalog, *this, info);
@@ -138,8 +150,11 @@ optional_ptr<CatalogEntry> QuackSchemaCatalogEntry::CreateType(CatalogTransactio
 
 void QuackSchemaCatalogEntry::DropEntry(ClientContext &context, DropInfo &info_p) {
 	auto drop_info = info_p.Copy();
-	drop_info->CatalogMutable() = GetInfo()->GetQualifiedName().Catalog();
-	drop_info->SchemaMutable() = name;
+	drop_info->SetQualifiedName(QualifiedName(GetInfo()->GetQualifiedName().Catalog(),
+	                                          drop_info->GetQualifiedName().Schema(),
+	                                          drop_info->GetQualifiedName().Name()));
+	drop_info->SetQualifiedName(
+	    QualifiedName(drop_info->GetQualifiedName().Catalog(), name, drop_info->GetQualifiedName().Name()));
 	switch (drop_info->type) {
 	case CatalogType::TABLE_ENTRY:
 	case CatalogType::VIEW_ENTRY:
@@ -165,7 +180,8 @@ static const DefaultTableMacro quack_table_macros[] = {
 // 'borrowed' from ducklake
 optional_ptr<CatalogEntry> QuackSchemaCatalogEntry::LoadBuiltInFunction(DefaultTableMacro macro) {
 	string macro_def = macro.macro;
-	macro_def = StringUtil::Replace(macro_def, "{CATALOG}", KeywordHelper::WriteQuoted(catalog.GetName().GetIdentifierName(), '\''));
+	macro_def = StringUtil::Replace(macro_def, "{CATALOG}",
+	                                KeywordHelper::WriteQuoted(catalog.GetName().GetIdentifierName(), '\''));
 	macro_def = StringUtil::Replace(macro_def, "{SCHEMA}", KeywordHelper::WriteQuoted(name.GetIdentifierName(), '\''));
 	macro.macro = macro_def.c_str();
 	auto info = DefaultTableFunctionGenerator::CreateTableMacroInfo(macro);
-- 
2.39.5 (Apple Git-154)

