diff --git a/src/common/parquet_file_scanner.cpp b/src/common/parquet_file_scanner.cpp
index 91f49172..7dc26751 100644
--- a/src/common/parquet_file_scanner.cpp
+++ b/src/common/parquet_file_scanner.cpp
@@ -16,7 +16,7 @@ ParquetFileScanner::ParquetFileScanner(ClientContext &context, const DuckLakeFil
 	auto &instance = DatabaseInstance::GetDatabase(context);
 	ExtensionLoader loader(instance, "ducklake");
 	auto &parquet_scan_entry = loader.GetTableFunction("parquet_scan");
-	parquet_scan = parquet_scan_entry.functions.functions[0];
+	parquet_scan = *parquet_scan_entry.functions.functions[0];
 
 	// Prepare the inputs for the bind
 	vector<Value> children;
diff --git a/src/storage/ducklake_scan.cpp b/src/storage/ducklake_scan.cpp
index 97730353..83c1a911 100644
--- a/src/storage/ducklake_scan.cpp
+++ b/src/storage/ducklake_scan.cpp
@@ -238,7 +238,7 @@ TableFunction DuckLakeFunctions::GetDuckLakeScanFunction(DatabaseInstance &insta
 	auto parquet_entry = loader.TryGetTableFunction("parquet_scan");
 	if (parquet_entry) {
 		auto &parquet_scan = parquet_entry->Cast<TableFunctionCatalogEntry>();
-		function = parquet_scan.functions.GetFunctionByOffset(0);
+		function = *parquet_scan.functions.GetFunctionByOffset(0);
 		function.get_multi_file_reader = DuckLakeMultiFileReader::CreateInstance;
 	}
 
