diff --git a/test/sql/compaction/merge_adjacent_max_files.test b/test/sql/compaction/merge_adjacent_max_files.test
index 36ba241d..244f0ff8 100644
--- a/test/sql/compaction/merge_adjacent_max_files.test
+++ b/test/sql/compaction/merge_adjacent_max_files.test
@@ -6,6 +6,8 @@ require ducklake
 
 require parquet
 
+require notmusl
+
 test-env DUCKLAKE_CONNECTION {TEST_DIR}/{UUID}.db
 
 test-env DATA_PATH {TEST_DIR}
diff --git a/test/sql/concurrent/concurrent_view_bind.test b/test/sql/concurrent/concurrent_view_bind.test
index 219c1b50..5ac442ab 100644
--- a/test/sql/concurrent/concurrent_view_bind.test
+++ b/test/sql/concurrent/concurrent_view_bind.test
@@ -8,6 +8,8 @@ require parquet
 
 require notwindows
 
+require notmusl
+
 statement ok
 ATTACH 'ducklake:{TEST_DIR}/concurrent_view_bind.db' AS ducklake (DATA_PATH '{TEST_DIR}/concurrent_view_bind')
 
