diff --git a/test/sql/compaction/merge_adjacent_max_files.test b/test/sql/compaction/merge_adjacent_max_files.test
index 36ba241d..244f0ff8 100644
--- a/test/sql/compaction/merge_adjacent_max_files.test
+++ b/test/sql/compaction/merge_adjacent_max_files.test
@@ -6,6 +6,8 @@ require ducklake
 
 require parquet
 
+require notmusl
+
 test-env DUCKLAKE_CONNECTION {TEST_DIR}/{UUID}.db
 
 test-env DATA_PATH {TEST_DIR}
