diff --git a/test/sql/alter/many_schema_versions_low_memory.test_slow b/test/sql/alter/many_schema_versions_low_memory.test_slow
index 60d93545..a14ed190 100644
--- a/test/sql/alter/many_schema_versions_low_memory.test_slow
+++ b/test/sql/alter/many_schema_versions_low_memory.test_slow
@@ -4,6 +4,8 @@
 
 require ducklake
 
+require notmusl
+
 require parquet
 
 test-env DUCKLAKE_CONNECTION {TEST_DIR}/{UUID}.db
diff --git a/test/sql/deletion_inlining/test_multiple_files_flush.test_slow b/test/sql/deletion_inlining/test_multiple_files_flush.test_slow
index 3d0e6b28..3b05b185 100644
--- a/test/sql/deletion_inlining/test_multiple_files_flush.test_slow
+++ b/test/sql/deletion_inlining/test_multiple_files_flush.test_slow
@@ -4,6 +4,8 @@
 
 require ducklake
 
+require notmusl
+
 require parquet
 
 test-env DUCKLAKE_CONNECTION {TEST_DIR}/{UUID}.db
diff --git a/test/sql/insert/insert_wide_table_low_memory.test_slow b/test/sql/insert/insert_wide_table_low_memory.test_slow
index a95c2980..157cbd3e 100644
--- a/test/sql/insert/insert_wide_table_low_memory.test_slow
+++ b/test/sql/insert/insert_wide_table_low_memory.test_slow
@@ -4,6 +4,8 @@
 
 require ducklake
 
+require notmusl
+
 require parquet
 
 test-env DUCKLAKE_CONNECTION {TEST_DIR}/{UUID}.db
