diff --git a/src/storage/ducklake_merge_into.cpp b/src/storage/ducklake_merge_into.cpp
index 0b48d5c..76bfdd7 100644
--- a/src/storage/ducklake_merge_into.cpp
+++ b/src/storage/ducklake_merge_into.cpp
@@ -204,7 +204,7 @@ unique_ptr<MergeIntoOperator> DuckLakePlanMergeIntoAction(DuckLakeCatalog &catal
 		}
 		result->expressions = std::move(action.expressions);
 		auto &insert = catalog.PlanInsert(context, planner, insert_op, child_plan);
-		auto &copy = insert.children[0].get();
+		auto &copy = insert.children.getAt(0).get();
 		result->op = planner.Make<DuckLakeMergeInsert>(insert.types, insert, copy);
 		break;
 	}
