diff --git a/test/sql/autoloading/autoload_data_path.test b/test/sql/autoloading/autoload_data_path.test
index 1030fc4e23..5dfa5e2a46 100644
--- a/test/sql/autoloading/autoload_data_path.test
+++ b/test/sql/autoloading/autoload_data_path.test
@@ -6,6 +6,9 @@ require-env LOCAL_EXTENSION_REPO
 
 require ducklake
 
+statement ok
+LOAD ducklake;
+
 statement ok
 set allow_persistent_secrets=false;
 
diff --git a/test/sql/checkpoint/checkpoint_ducklake.test b/test/sql/checkpoint/checkpoint_ducklake.test
index 6be186e38c..6e6f915400 100644
--- a/test/sql/checkpoint/checkpoint_ducklake.test
+++ b/test/sql/checkpoint/checkpoint_ducklake.test
@@ -81,9 +81,13 @@ SELECT COUNT(*) FROM GLOB('${DATA_PATH}/checkpoint_ducklake/main/**')
 statement ok
 CALL ducklake.set_option('delete_older_than', '1 millisecond');
 
+
+# FIXME: this could be solved by autoloading icu as appropriate, instead of "No function matches the given name and argument types '-(TIMESTAMP WITH TIME ZONE, INTERVAL)'."
 statement ok
-CHECKPOINT;
+LOAD icu
 
+statement ok
+CHECKPOINT;
 
 query II
 SELECT data_file_id, table_id FROM __ducklake_metadata_ducklake.ducklake_data_file
@@ -101,4 +105,4 @@ SELECT COUNT(*)
 FROM GLOB('${DATA_PATH}/checkpoint_ducklake/main/**')
 WHERE file LIKE '%bla%';
 ----
-0
\ No newline at end of file
+0
diff --git a/test/sql/checkpoint/checkpoint_updates_interleaved.test b/test/sql/checkpoint/checkpoint_updates_interleaved.test
index 99f359538b..2d566b5900 100644
--- a/test/sql/checkpoint/checkpoint_updates_interleaved.test
+++ b/test/sql/checkpoint/checkpoint_updates_interleaved.test
@@ -36,6 +36,10 @@ BEGIN;
 statement ok
 INSERT INTO test FROM range(10,13)
 
+# FIXME: this could be solved by autoloading icu as appropriate, instead of "No function matches the given name and argument types '-(TIMESTAMP WITH TIME ZONE, INTERVAL)'."
+statement ok
+LOAD icu
+
 statement ok
 CHECKPOINT;
 
diff --git a/test/sql/checkpoint/many_inserts_transaction.test b/test/sql/checkpoint/many_inserts_transaction.test
index 0fddf155c8..d86c98eb4e 100644
--- a/test/sql/checkpoint/many_inserts_transaction.test
+++ b/test/sql/checkpoint/many_inserts_transaction.test
@@ -39,10 +39,14 @@ INSERT INTO integers SELECT 3;
 statement ok
 COMMIT;
 
+# FIXME: this could be solved by autoloading icu as appropriate, instead of "No function matches the given name and argument types '-(TIMESTAMP WITH TIME ZONE, INTERVAL)'."
+statement ok
+LOAD icu
+
 statement ok
 CHECKPOINT;
 
 query IIII
 SELECT MIN(i), MAX(i), COUNT(*), COUNT(i) FROM integers;
 ----
-1	3	13	13
\ No newline at end of file
+1	3	13	13
diff --git a/test/sql/checkpoint/view_checkpoint.test b/test/sql/checkpoint/view_checkpoint.test
index 8da36a1a65..6c3bc8116c 100644
--- a/test/sql/checkpoint/view_checkpoint.test
+++ b/test/sql/checkpoint/view_checkpoint.test
@@ -24,5 +24,9 @@ create table t (a integer, b integer, c integer)
 statement ok
 create view v as select a from t;
 
+# FIXME: this could be solved by autoloading icu as appropriate, instead of "No function matches the given name and argument types '-(TIMESTAMP WITH TIME ZONE, INTERVAL)'."
 statement ok
-CHECKPOINT;
\ No newline at end of file
+LOAD icu
+
+statement ok
+CHECKPOINT;
diff --git a/test/sql/compaction/cleanup_old_files_global_option.test b/test/sql/compaction/cleanup_old_files_global_option.test
index cab6ac5264..2068d02cb3 100644
--- a/test/sql/compaction/cleanup_old_files_global_option.test
+++ b/test/sql/compaction/cleanup_old_files_global_option.test
@@ -37,6 +37,10 @@ CALL ducklake_expire_snapshots('ducklake', dry_run => false, versions => [2])
 statement ok
 CALL ducklake.set_option('delete_older_than', '1 millisecond')
 
+# FIXME: this could be solved by autoloading icu as appropriate, instead of "No function matches the given name and argument types '-(TIMESTAMP WITH TIME ZONE, INTERVAL)'."
+statement ok
+LOAD icu
+
 query I
 SELECT count(*) FROM ducklake_cleanup_old_files('ducklake', dry_run => true)
 ----
diff --git a/test/sql/compaction/compaction_cleanup_global.test b/test/sql/compaction/compaction_cleanup_global.test
index 5792fa1118..6cc29f41b6 100644
--- a/test/sql/compaction/compaction_cleanup_global.test
+++ b/test/sql/compaction/compaction_cleanup_global.test
@@ -33,6 +33,10 @@ CALL ducklake_expire_snapshots('ducklake', dry_run => false, versions => [2])
 statement ok
 CALL ducklake.set_option('delete_older_than', '1 week')
 
+# FIXME: this could be solved by autoloading icu as appropriate, instead of "No function matches the given name and argument types '-(TIMESTAMP WITH TIME ZONE, INTERVAL)'."
+statement ok
+LOAD icu
+
 # This should return 0 since file to be deleted is not that old
 query I
 SELECT count(*) FROM ducklake_cleanup_old_files('ducklake', dry_run => true);
diff --git a/test/sql/compaction/expire_snapshot_global_option.test b/test/sql/compaction/expire_snapshot_global_option.test
index 64e9e98302..3915c9e488 100644
--- a/test/sql/compaction/expire_snapshot_global_option.test
+++ b/test/sql/compaction/expire_snapshot_global_option.test
@@ -27,6 +27,10 @@ INSERT INTO ducklake.test FROM range(10)
 statement ok
 CALL ducklake.set_option('expire_older_than', '1 millisecond')
 
+# FIXME: this could be solved by autoloading icu as appropriate, instead of "No function matches the given name and argument types '-(TIMESTAMP WITH TIME ZONE, INTERVAL)'."
+statement ok
+LOAD icu
+
 query I
 SELECT count(*) FROM ducklake_expire_snapshots('ducklake', dry_run => true)
 ----
@@ -43,4 +47,4 @@ SELECT count(*) FROM ducklake_expire_snapshots('ducklake', dry_run => true)
 query I
 SELECT count(*) FROM ducklake_expire_snapshots('ducklake', dry_run => true, versions => [1])
 ----
-1
\ No newline at end of file
+1
diff --git a/test/sql/remove_orphans/remove_orphaned_files.test b/test/sql/remove_orphans/remove_orphaned_files.test
index 84feb40516..c2a24ea9dc 100644
--- a/test/sql/remove_orphans/remove_orphaned_files.test
+++ b/test/sql/remove_orphans/remove_orphaned_files.test
@@ -54,6 +54,10 @@ SELECT count(*) FROM ducklake_delete_orphaned_files('ducklake', cleanup_all => t
 statement ok
 CALL ducklake.set_option('delete_older_than', '1 millisecond')
 
+# FIXME: this could be solved by autoloading icu as appropriate, instead of "No function matches the given name and argument types '-(TIMESTAMP WITH TIME ZONE, INTERVAL)'."
+statement ok
+LOAD icu
+
 query I
 SELECT count(*) FROM ducklake_delete_orphaned_files('ducklake', dry_run => true);
 ----
