diff --git a/test/sql/initialize/read_only_mode.test b/test/sql/initialize/read_only_mode.test
index cfbcb43631..cfaea5654f 100644
--- a/test/sql/initialize/read_only_mode.test
+++ b/test/sql/initialize/read_only_mode.test
@@ -7,22 +7,22 @@ require ducklake
 require parquet
 
 statement ok
-ATTACH '__TEST_DIR__/ducklake_read_only.db' AS db1
+ATTACH '__TEST_DIR__/read_only_mode.db' AS db1
 
 statement ok
 DETACH db1;
 
 statement error
-ATTACH 'ducklake:__TEST_DIR__/ducklake_read_only.db' AS ducklake (READONLY 1, CREATE_IF_NOT_EXISTS false)
+ATTACH 'ducklake:__TEST_DIR__/read_only_mode.db' AS ducklake (READONLY 1, CREATE_IF_NOT_EXISTS false)
 ----
 creating a new DuckLake is explicitly disabled
 
 statement error
-ATTACH 'ducklake:__TEST_DIR__/ducklake_read_only.db' AS ducklake (READONLY 1)
+ATTACH 'ducklake:__TEST_DIR__/read_only_mode.db' AS ducklake (READONLY 1)
 ----
 creating a new DuckLake is explicitly disabled
 
 statement error
-ATTACH 'ducklake:__TEST_DIR__/ducklake_read_only.db' AS ducklake (READONLY 1, CREATE_IF_NOT_EXISTS true)
+ATTACH 'ducklake:__TEST_DIR__/read_only_mode.db' AS ducklake (READONLY 1, CREATE_IF_NOT_EXISTS true)
 ----
-"CREATE" on database "__ducklake_metadata_ducklake" which is attached in read-only mode!
\ No newline at end of file
+"CREATE" on database "__ducklake_metadata_ducklake" which is attached in read-only mode!
