diff --git a/src/avro_multi_file_info.cpp b/src/avro_multi_file_info.cpp
index e8fcb2f..c217a1c 100644
--- a/src/avro_multi_file_info.cpp
+++ b/src/avro_multi_file_info.cpp
@@ -15,7 +15,7 @@ unique_ptr<BaseFileReaderOptions> AvroMultiFileInfo::InitializeOptions(ClientCon
 }
 
 bool AvroMultiFileInfo::ParseCopyOption(ClientContext &context, const Identifier &key, const vector<Value> &values,
-                                        BaseFileReaderOptions &options_p, vector<string> &expected_names,
+                                        BaseFileReaderOptions &options_p, vector<Identifier> &expected_names,
                                         vector<LogicalType> &expected_types) {
 	// We currently do not have any options for the scanner, so we always return false
 	return false;
diff --git a/src/include/avro_multi_file_info.hpp b/src/include/avro_multi_file_info.hpp
index a9eba17..5527e9d 100644
--- a/src/include/avro_multi_file_info.hpp
+++ b/src/include/avro_multi_file_info.hpp
@@ -20,7 +20,7 @@ struct AvroMultiFileInfo : MultiFileReaderInterface {
 	unique_ptr<BaseFileReaderOptions> InitializeOptions(ClientContext &context,
 	                                                    optional_ptr<TableFunctionInfo> info) override;
 	bool ParseCopyOption(ClientContext &context, const Identifier &key, const vector<Value> &values,
-	                     BaseFileReaderOptions &options, vector<string> &expected_names,
+	                     BaseFileReaderOptions &options, vector<Identifier> &expected_names,
 	                     vector<LogicalType> &expected_types) override;
 
 	bool ParseOption(ClientContext &context, const Identifier &key, const Value &val, MultiFileOptions &file_options,
