diff --git a/src/avro_copy.cpp b/src/avro_copy.cpp
index 91a3526..8b9a6b5 100644
--- a/src/avro_copy.cpp
+++ b/src/avro_copy.cpp
@@ -545,10 +545,10 @@ WriteAvroBindData::WriteAvroBindData(CopyFunctionBindInput &input, const vector<
 		}
 		auto key = option.first;
 		if (option.second.empty()) {
-			unrecognized_options.push_back(StringUtil::Format("key: '%s'", key));
+			unrecognized_options.push_back(StringUtil::Format("key: %s", key));
 		} else {
 			unrecognized_options.push_back(
-			    StringUtil::Format("key: '%s' with value: '%s'", key, option.second[0].ToString()));
+			    StringUtil::Format("key: %s with value: '%s'", key, option.second[0].ToString()));
 		}
 	}
 	if (!unrecognized_options.empty()) {
diff --git a/test/sql/copy/invalid_option.test b/test/sql/copy/invalid_option.test
index ad94908..b97e2ef 100644
--- a/test/sql/copy/invalid_option.test
+++ b/test/sql/copy/invalid_option.test
@@ -11,7 +11,7 @@ COPY (
 	ROOT_NAME 'test'
 );
 ----
-Invalid Configuration Error: The following option(s) are not recognized: key: 'not_recognized'
+Invalid Configuration Error: The following option(s) are not recognized: key: "not_recognized"
 
 statement error
 COPY (
@@ -21,4 +21,4 @@ COPY (
 	ROOT_NAME 'test'
 );
 ----
-Invalid Configuration Error: The following option(s) are not recognized: key: 'not_recognized' with value: 'with_value'
+Invalid Configuration Error: The following option(s) are not recognized: key: "not_recognized" with value: 'with_value'
