diff --git a/src/avro_copy.cpp b/src/avro_copy.cpp
index ee47234..890bda2 100644
--- a/src/avro_copy.cpp
+++ b/src/avro_copy.cpp
@@ -675,7 +675,7 @@ static idx_t PopulateValue(avro_value_t *target, const Value &val) {
 	}
 	case LogicalTypeId::TIME: {
 		auto date = val.GetValueUnsafe<dtime_t>();
-		avro_value_set_long(target, date.micros);
+		avro_value_set_long(target, date.value);
 		return sizeof(int64_t);
 	}
 	case LogicalTypeId::TIMESTAMP:
