diff --git a/test/sql/copy/test_decimal_types.test b/test/sql/copy/test_decimal_types.test
index 801d2d0..d51d8d0 100644
--- a/test/sql/copy/test_decimal_types.test
+++ b/test/sql/copy/test_decimal_types.test
@@ -151,7 +151,7 @@ COPY (
     from values (-1.3::DECIMAL(10000,1)), (NULL) t(decimal_col)
 ) TO '{TEST_DIR}/decimal_type.avro';
 ----
-<REGEX>:.*DECIMAL type width must be between 1 and 38.*
+<REGEX>:.*Type "DECIMAL" does not accept type parameters.*
 
 statement ok
 COPY(
