diff --git a/test/sql/copy/nullable.test b/test/sql/copy/nullable.test
index d2535ac..ddbcf43 100644
--- a/test/sql/copy/nullable.test
+++ b/test/sql/copy/nullable.test
@@ -38,12 +38,11 @@ COPY (
 ----
 <REGEX>:.*Cannot insert NULL to non-nullable field.*
 
-# No rows
-# FIXME: there should be clean up logic in the copy
-query I
+# as the copy failed, the file shouldn't exist
+statement error
 select * from read_avro('__TEST_DIR__/field_ids_error.avro');
 ----
-
+IO Error: No files found that match the pattern "__TEST_DIR__/field_ids_error.avro"
 
 # cannot cast foo to BOOL
 statement error
