commit 378944acf4e26a98eec0a8b5be1680cbc662cfc6
Author: dentiny <dentinyhao@gmail.com>
Date:   Sun Mar 15 10:48:02 2026 +0000

    Adapt caching filesystem interface change

diff --git a/src/avro_reader.cpp b/src/avro_reader.cpp
index 5317c43..876bb8b 100644
--- a/src/avro_reader.cpp
+++ b/src/avro_reader.cpp
@@ -123,13 +123,17 @@ AvroReader::AvroReader(ClientContext &context, OpenFileInfo file) : BaseFileRead
 
 	auto caching_file_handle = caching_file_system.OpenFile(this->file, FileOpenFlags::FILE_FLAGS_READ);
 	auto total_size = caching_file_handle->GetFileSize();
-	data_ptr_t data = nullptr;
 
-	buf_handle = caching_file_handle->Read(data, total_size);
-	auto buffer_data = buf_handle.Ptr();
+	buf_handle_group = caching_file_handle->Read(total_size);
+	data_ptr_t buffer_data;
+	if (buf_handle_group.GetHandles().size() == 1) {
+		buffer_data = buf_handle_group.Ptr();
+	} else {
+		buf_data = Allocator::DefaultAllocator().Allocate(total_size);
+		buf_handle_group.CopyTo(buf_data.get(), total_size);
+		buffer_data = buf_data.get();
+	}
 
-	D_ASSERT(buf_handle.IsValid());
-	D_ASSERT(buffer_data == data);
 	auto avro_reader = avro_reader_memory(const_char_ptr_cast(buffer_data), total_size);
 
 	if (avro_reader_reader(avro_reader, &reader)) {
diff --git a/src/include/avro_reader.hpp b/src/include/avro_reader.hpp
index ebb941a..af5b41b 100644
--- a/src/include/avro_reader.hpp
+++ b/src/include/avro_reader.hpp
@@ -1,8 +1,10 @@
 #pragma once
 
+#include "duckdb/common/allocator.hpp"
 #include "duckdb/common/helper.hpp"
 #include "avro_type.hpp"
 #include "duckdb/common/multi_file/base_file_reader.hpp"
+#include "duckdb/storage/file_buffer_handle_group.hpp"
 
 namespace duckdb {
 
@@ -32,7 +34,8 @@ public:
 	avro_value_t value;
 	unique_ptr<Vector> read_vec;
 
-	BufferHandle buf_handle;
+	FileBufferHandleGroup buf_handle_group;
+	AllocatedData buf_data;
 	AvroType avro_type;
 	LogicalType duckdb_type;
 };
diff --git a/test/sql/external_file_cache.test b/test/sql/external_file_cache.test
index 1d4a43a..b6aa7fe 100644
--- a/test/sql/external_file_cache.test
+++ b/test/sql/external_file_cache.test
@@ -31,8 +31,8 @@ FROM read_avro('test/bigdata.avro') ORDER BY c1 LIMIT 10
 statement ok
 from read_avro('test/userdata1.avro');
 
-query IIII
-from duckdb_external_file_cache() order by path;
+query II
+select path, sum(nr_bytes) from duckdb_external_file_cache() group by path order by path;
 ----
-test/bigdata.avro	17647257	0	true
-test/userdata1.avro	93561	0	true
\ No newline at end of file
+test/bigdata.avro	17647257
+test/userdata1.avro	93561
