diff --git a/src/avro_multi_file_info.cpp b/src/avro_multi_file_info.cpp
index 02dc660..d67d744 100644
--- a/src/avro_multi_file_info.cpp
+++ b/src/avro_multi_file_info.cpp
@@ -4,7 +4,7 @@
 namespace duckdb {
 
 unique_ptr<MultiFileReaderInterface>
-AvroMultiFileInfo::InitializeInterface(ClientContext &context, MultiFileReader &reader, MultiFileList &file_list) {
+AvroMultiFileInfo::CreateInterface(ClientContext &context) {
 	return make_uniq<AvroMultiFileInfo>();
 }
 
diff --git a/src/include/avro_multi_file_info.hpp b/src/include/avro_multi_file_info.hpp
index 5202c8b..dac2c2c 100644
--- a/src/include/avro_multi_file_info.hpp
+++ b/src/include/avro_multi_file_info.hpp
@@ -16,8 +16,7 @@ namespace duckdb {
 class AvroFileReaderOptions : public BaseFileReaderOptions {};
 
 struct AvroMultiFileInfo : MultiFileReaderInterface {
-	static unique_ptr<MultiFileReaderInterface> InitializeInterface(ClientContext &context, MultiFileReader &reader,
-	                                                                MultiFileList &file_list);
+	static unique_ptr<MultiFileReaderInterface> CreateInterface(ClientContext &context);
 
 	unique_ptr<BaseFileReaderOptions> InitializeOptions(ClientContext &context,
 	                                                    optional_ptr<TableFunctionInfo> info) override;
