diff --git a/src/avro_reader.cpp b/src/avro_reader.cpp
index 5317c43..b1ef246 100644
--- a/src/avro_reader.cpp
+++ b/src/avro_reader.cpp
@@ -1,3 +1,8 @@
+#include "duckdb/common/vector/flat_vector.hpp"
+#include "duckdb/common/vector/list_vector.hpp"
+#include "duckdb/common/vector/map_vector.hpp"
+#include "duckdb/common/vector/string_vector.hpp"
+#include "duckdb/common/vector/union_vector.hpp"
 #include "avro_reader.hpp"
 #include "utf8proc_wrapper.hpp"
 #include "duckdb/storage/caching_file_system.hpp"
