commit 2f2636db2d05b5015d22876c51a0e69ec0b34e05
Author: dentiny <dentinyhao@gmail.com>
Date:   Sun Mar 15 10:48:02 2026 +0000

    Adapt caching filesystem interface change

diff --git a/src/avro_reader.cpp b/src/avro_reader.cpp
index 5317c43..f02ed8a 100644
--- a/src/avro_reader.cpp
+++ b/src/avro_reader.cpp
@@ -1,6 +1,5 @@
 #include "avro_reader.hpp"
 #include "utf8proc_wrapper.hpp"
-#include "duckdb/storage/caching_file_system.hpp"
 #include "duckdb/common/file_system.hpp"
 #include "duckdb/common/multi_file/multi_file_data.hpp"
 
@@ -119,18 +118,16 @@ static AvroType TransformSchema(avro_schema_t &avro_schema, unordered_set<string
 }
 
 AvroReader::AvroReader(ClientContext &context, OpenFileInfo file) : BaseFileReader(file) {
-	auto caching_file_system = CachingFileSystem::Get(context);
+	auto &fs = FileSystem::GetFileSystem(context);
+	FileOpenFlags flags = FileFlags::FILE_FLAGS_READ;
+	flags.SetCachingMode(CachingMode::CACHE_REMOTE_ONLY);
+	auto file_handle = fs.OpenFile(this->file.path, flags);
+	auto total_size = fs.GetFileSize(*file_handle);
 
-	auto caching_file_handle = caching_file_system.OpenFile(this->file, FileOpenFlags::FILE_FLAGS_READ);
-	auto total_size = caching_file_handle->GetFileSize();
-	data_ptr_t data = nullptr;
+	buf_data = Allocator::DefaultAllocator().Allocate(total_size);
+	fs.Read(*file_handle, buf_data.get(), total_size, 0);
 
-	buf_handle = caching_file_handle->Read(data, total_size);
-	auto buffer_data = buf_handle.Ptr();
-
-	D_ASSERT(buf_handle.IsValid());
-	D_ASSERT(buffer_data == data);
-	auto avro_reader = avro_reader_memory(const_char_ptr_cast(buffer_data), total_size);
+	auto avro_reader = avro_reader_memory(const_char_ptr_cast(buf_data.get()), total_size);
 
 	if (avro_reader_reader(avro_reader, &reader)) {
 		throw InvalidInputException(avro_strerror());
diff --git a/src/include/avro_reader.hpp b/src/include/avro_reader.hpp
index ebb941a..436fd81 100644
--- a/src/include/avro_reader.hpp
+++ b/src/include/avro_reader.hpp
@@ -1,5 +1,6 @@
 #pragma once
 
+#include "duckdb/common/allocator.hpp"
 #include "duckdb/common/helper.hpp"
 #include "avro_type.hpp"
 #include "duckdb/common/multi_file/base_file_reader.hpp"
@@ -32,7 +33,7 @@ public:
 	avro_value_t value;
 	unique_ptr<Vector> read_vec;
 
-	BufferHandle buf_handle;
+	AllocatedData buf_data;
 	AvroType avro_type;
 	LogicalType duckdb_type;
 };
diff --git a/test/sql/external_file_cache.test b/test/sql/external_file_cache.test
index 1d4a43a..b6aa7fe 100644
--- a/test/sql/external_file_cache.test
+++ b/test/sql/external_file_cache.test
@@ -31,8 +31,8 @@ FROM read_avro('test/bigdata.avro') ORDER BY c1 LIMIT 10
 statement ok
 from read_avro('test/userdata1.avro');
 
-query IIII
-from duckdb_external_file_cache() order by path;
+query II
+select path, sum(nr_bytes) from duckdb_external_file_cache() group by path order by path;
 ----
-test/bigdata.avro	17647257	0	true
-test/userdata1.avro	93561	0	true
\ No newline at end of file
+test/bigdata.avro	17647257
+test/userdata1.avro	93561
