diff --git a/src/avro_reader.cpp b/src/avro_reader.cpp
index 5317c43..28d6a31 100644
--- a/src/avro_reader.cpp
+++ b/src/avro_reader.cpp
@@ -1,3 +1,9 @@
+#include "duckdb/common/vector/flat_vector.hpp"
+#include "duckdb/common/vector/list_vector.hpp"
+#include "duckdb/common/vector/map_vector.hpp"
+#include "duckdb/common/vector/struct_vector.hpp"
+#include "duckdb/common/vector/string_vector.hpp"
+#include "duckdb/common/vector/union_vector.hpp"
 #include "avro_reader.hpp"
 #include "utf8proc_wrapper.hpp"
 #include "duckdb/storage/caching_file_system.hpp"
@@ -254,7 +260,7 @@ static void TransformValue(avro_value *avro_val, const AvroType &avro_type, Vect
 				throw InvalidInputException(avro_strerror());
 			}
 			TransformValue(&child_value, avro_type.children[child_idx].second,
-			               *StructVector::GetEntries(target)[child_idx], out_idx);
+			               StructVector::GetEntries(target)[child_idx], out_idx);
 		}
 		break;
 	}
@@ -284,7 +290,7 @@ static void TransformValue(avro_value *avro_val, const AvroType &avro_type, Vect
 			// orrrrrrrrrrrrr
 			for (idx_t child_idx = 1; child_idx < StructVector::GetEntries(target).size(); child_idx++) {
 				if (child_idx != duckdb_child_index + 1) { // duckdb child index is bigger because of the tag
-					FlatVector::SetNull(*StructVector::GetEntries(target)[child_idx], out_idx, true);
+					FlatVector::SetNull(StructVector::GetEntries(target)[child_idx], out_idx, true);
 				}
 			}
 
@@ -391,7 +397,7 @@ void AvroReader::Read(DataChunk &output) {
 				continue; // to be filled in later
 			}
 			output.data[col_idx].Reference(
-			    *StructVector::GetEntries(*read_vec)[column_indexes[col_idx].GetPrimaryIndex()]);
+			    StructVector::GetEntries(*read_vec)[column_indexes[col_idx].GetPrimaryIndex()]);
 		}
 	} else {
 		output.data[column_indexes[0].GetPrimaryIndex()].Reference(*read_vec);
