diff --git a/test/sql/test_ipv6_inet_nested.test b/test/sql/test_ipv6_inet_nested.test
index 007a502..0395130 100644
--- a/test/sql/test_ipv6_inet_nested.test
+++ b/test/sql/test_ipv6_inet_nested.test
@@ -15,7 +15,7 @@ SELECT {'a': INET '2345:425::5673:23b5/64'}, NULL::STRUCT(a INET), {'a': INET '2
 query III
 SELECT {'a': INET '2345:425::5673:23b5/64'}::STRUCT(a VARCHAR), NULL::STRUCT(a INET)::STRUCT(a VARCHAR), {'a': INET '2266:25::12:0:ad12/96'}::STRUCT(a VARCHAR)
 ----
-{'a': 2345:425::5673:23b5/64}	NULL	{'a': 2266:25::12:0:ad12/96}
+{'a': '2345:425::5673:23b5/64'}	NULL	{'a': '2266:25::12:0:ad12/96'}
 
 query I
 SELECT [INET '2345:425::5673:23b5/64', NULL, INET '2266:25::12:0:ad12/96']::VARCHAR[]
