diff --git a/test/sql/test_inet_storage.test b/test/sql/test_inet_storage.test
index 6679439..7de7620 100644
--- a/test/sql/test_inet_storage.test
+++ b/test/sql/test_inet_storage.test
@@ -5,7 +5,7 @@
 require inet
 
 # FIXME: Make inet properly autoloadable
-require no_extension_autoloading
+require no_extension_autoloading "FIXME: to be reviewed whether this can be lifted"
 
 # load the DB from disk
 load __TEST_DIR__/store_inet.db
diff --git a/test/sql/test_ipv6_inet_storage.test b/test/sql/test_ipv6_inet_storage.test
index 5be1ff2..7dc1930 100644
--- a/test/sql/test_ipv6_inet_storage.test
+++ b/test/sql/test_ipv6_inet_storage.test
@@ -25,7 +25,7 @@ CREATE VIEW iview AS SELECT INET '::1'
 restart
 
 #FIXME: INET needs to be explicitly autoloaded on restart
-require no_extension_autoloading
+require no_extension_autoloading "FIXME: INET needs to be explicitly autoloaded on restart"
 
 query IIIIII
 DESCRIBE tbl
