diff --git a/src/arrow_scan_ipc.cpp b/src/arrow_scan_ipc.cpp
index dc6e8a7..7d5b2ff 100644
--- a/src/arrow_scan_ipc.cpp
+++ b/src/arrow_scan_ipc.cpp
@@ -53,7 +53,7 @@ unique_ptr <FunctionData> ArrowIPCTableFunction::ArrowScanBind(ClientContext &co
 
     // TODO Everything below this is identical to the bind in duckdb/src/function/table/arrow.cpp
     auto &data = *res;
-    stream_factory_get_schema(stream_factory_ptr, data.schema_root);
+    stream_factory_get_schema((ArrowArrayStream *) stream_factory_ptr, data.schema_root.arrow_schema);
     for (idx_t col_idx = 0; col_idx < (idx_t) data.schema_root.arrow_schema.n_children; col_idx++) {
         auto &schema = *data.schema_root.arrow_schema.children[col_idx];
         if (!schema.release) {
@@ -75,7 +75,7 @@ unique_ptr <FunctionData> ArrowIPCTableFunction::ArrowScanBind(ClientContext &co
         }
         names.push_back(name);
     }
-    ArrowTableFunction::RenameArrowColumns(names);
+    QueryResult::DeduplicateColumns(names);
     return std::move(res);
 }
 
