diff --git a/src/arrow_scan_ipc.cpp b/src/arrow_scan_ipc.cpp
index a60d255..6dd725a 100644
--- a/src/arrow_scan_ipc.cpp
+++ b/src/arrow_scan_ipc.cpp
@@ -15,7 +15,6 @@ TableFunction ArrowIPCTableFunction::GetFunction() {
       ArrowTableFunction::ArrowScanInitLocal);
 
   scan_arrow_ipc_func.cardinality = ArrowTableFunction::ArrowScanCardinality;
-  scan_arrow_ipc_func.get_batch_index = nullptr; // TODO implement
   scan_arrow_ipc_func.projection_pushdown = true;
   scan_arrow_ipc_func.filter_pushdown = false;
 
