diff --git a/test/sql/test_except.test b/test/sql/test_except.test
index b10a59d..7e1e992 100644
--- a/test/sql/test_except.test
+++ b/test/sql/test_except.test
@@ -4,6 +4,8 @@
 
 require substrait
 
+mode skip
+
 statement ok
 PRAGMA enable_verification
 
diff --git a/test/sql/test_intersect.test b/test/sql/test_intersect.test
index 6ca7bba..bef0866 100644
--- a/test/sql/test_intersect.test
+++ b/test/sql/test_intersect.test
@@ -4,6 +4,8 @@
 
 require substrait
 
+mode skip
+
 statement ok
 PRAGMA enable_verification
 
