diff --git a/src/from_substrait.cpp b/src/from_substrait.cpp
index abea04e..033ad1d 100644
--- a/src/from_substrait.cpp
+++ b/src/from_substrait.cpp
@@ -24,7 +24,6 @@
 #include "substrait/plan.pb.h"
 #include "google/protobuf/util/json_util.h"
 #include "duckdb/main/client_data.hpp"
-#include "duckdb/common/http_state.hpp"
 
 namespace duckdb {
 const std::unordered_map<std::string, std::string> SubstraitToDuckDB::function_names_remap = {
@@ -66,9 +65,6 @@ std::string SubstraitToDuckDB::RemoveExtension(std::string &function_name) {
 }
 
 SubstraitToDuckDB::SubstraitToDuckDB(Connection &con_p, const string &serialized, bool json) : con(con_p) {
-	auto http_state = HTTPState::TryGetState(*con_p.context);
-	http_state->Reset();
-
 	if (!json) {
 		if (!plan.ParseFromString(serialized)) {
 			throw std::runtime_error("Was not possible to convert binary into Substrait plan");
