diff --git a/lib/cmake/duckdb.cmake b/lib/cmake/duckdb.cmake
index cb0e60d6..d7a4851d 100644
--- a/lib/cmake/duckdb.cmake
+++ b/lib/cmake/duckdb.cmake
@@ -15,6 +15,10 @@ endif()
 set(DUCKDB_CXX_FLAGS "${DUCKDB_CXX_FLAGS} -Wno-unqualified-std-cast-call -DDUCKDB_DEBUG_NO_SAFETY")
 message("DUCKDB_CXX_FLAGS=${DUCKDB_CXX_FLAGS}")
 
+set(DUCKDB_EXTENSIONS "fts;excel;json")
+# Escape semicolons in DUCKDB_EXTENSIONS before passing to ExternalProject_Add
+string(REPLACE ";" "$<SEMICOLON>" DUCKDB_EXTENSIONS_PACKED "${DUCKDB_EXTENSIONS}")
+
 ExternalProject_Add(
   duckdb_ep
   SOURCE_DIR "${DUCKDB_CORE_DIR}"
@@ -29,13 +33,10 @@ ExternalProject_Add(
              -DCMAKE_MODULE_PATH=${CMAKE_MODULE_PATH}
              -DCMAKE_BUILD_TYPE=${DUCKDB_BUILD_TYPE}
              -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
-             -DBUILD_PARQUET_EXTENSION=TRUE
-             -DBUILD_FTS_EXTENSION=TRUE
-             -DBUILD_EXCEL_EXTENSION=TRUE
-             -DBUILD_JSON_EXTENSION=TRUE
+             -DBUILD_EXTENSIONS=${DUCKDB_EXTENSIONS_PACKED}
+             -DSKIP_EXTENSIONS=jemalloc
              -DBUILD_SHELL=FALSE
              -DBUILD_UNITTESTS=FALSE
-             -DBUILD_JEMALLOC_EXTENSION=FALSE
              -DDISABLE_BUILTIN_EXTENSIONS=TRUE
   BUILD_BYPRODUCTS
     <INSTALL_DIR>/lib/libduckdb_re2.a
