diff --git a/lib/src/extensions/excel_extension.cc b/lib/src/extensions/excel_extension.cc
index 8972de26..ce63c65d 100644
--- a/lib/src/extensions/excel_extension.cc
+++ b/lib/src/extensions/excel_extension.cc
@@ -1,5 +1,5 @@
 #include "duckdb/web/extensions/excel_extension.h"
 
-#include "excel-extension.hpp"
+#include "excel_extension.hpp"
 
-extern "C" void duckdb_web_excel_init(duckdb::DuckDB* db) { db->LoadExtension<duckdb::EXCELExtension>(); }
+extern "C" void duckdb_web_excel_init(duckdb::DuckDB* db) { db->LoadExtension<duckdb::ExcelExtension>(); }
diff --git a/lib/src/extensions/fts_extension.cc b/lib/src/extensions/fts_extension.cc
index e8b58572..d5fff211 100644
--- a/lib/src/extensions/fts_extension.cc
+++ b/lib/src/extensions/fts_extension.cc
@@ -1,5 +1,5 @@
 #include "duckdb/web/extensions/fts_extension.h"
 
-#include "fts-extension.hpp"
+#include "fts_extension.hpp"
 
-extern "C" void duckdb_web_fts_init(duckdb::DuckDB* db) { db->LoadExtension<duckdb::FTSExtension>(); }
+extern "C" void duckdb_web_fts_init(duckdb::DuckDB* db) { db->LoadExtension<duckdb::FtsExtension>(); }
diff --git a/lib/src/extensions/json_extension.cc b/lib/src/extensions/json_extension.cc
index 4723e41f..f143e04f 100644
--- a/lib/src/extensions/json_extension.cc
+++ b/lib/src/extensions/json_extension.cc
@@ -1,5 +1,5 @@
 #include "duckdb/web/extensions/json_extension.h"
 
-#include "json-extension.hpp"
+#include "json_extension.hpp"
 
-extern "C" void duckdb_web_json_init(duckdb::DuckDB* db) { db->LoadExtension<duckdb::JSONExtension>(); }
+extern "C" void duckdb_web_json_init(duckdb::DuckDB* db) { db->LoadExtension<duckdb::JsonExtension>(); }
diff --git a/lib/src/extensions/parquet_extension.cc b/lib/src/extensions/parquet_extension.cc
index f3189e68..bb862ff4 100644
--- a/lib/src/extensions/parquet_extension.cc
+++ b/lib/src/extensions/parquet_extension.cc
@@ -1,5 +1,5 @@
 #include "duckdb/web/extensions/parquet_extension.h"
 
-#include "parquet-extension.hpp"
+#include "parquet_extension.hpp"
 
 extern "C" void duckdb_web_parquet_init(duckdb::DuckDB* db) { db->LoadExtension<duckdb::ParquetExtension>(); }
