diff -Naur a/src/backend/nodes/makefuncs.c b/src/backend/nodes/makefuncs.c
--- a/src/backend/nodes/makefuncs.c	2016-05-09 13:50:23.000000000 -0700
+++ b/src/backend/nodes/makefuncs.c	2016-06-25 22:30:41.000000000 -0700
@@ -518,6 +518,7 @@
 	res->defname = name;
 	res->arg = arg;
 	res->defaction = DEFELEM_UNSPEC;
+	res->location = -1;
 
 	return res;
 }
@@ -528,7 +529,7 @@
  */
 DefElem *
 makeDefElemExtended(char *nameSpace, char *name, Node *arg,
-					DefElemAction defaction)
+					DefElemAction defaction, int location)
 {
 	DefElem    *res = makeNode(DefElem);
 
@@ -536,6 +537,7 @@
 	res->defname = name;
 	res->arg = arg;
 	res->defaction = defaction;
+	res->location = location;
 
 	return res;
 }
diff -Naur a/src/backend/parser/gram.y b/src/backend/parser/gram.y
--- a/src/backend/parser/gram.y	2016-06-25 23:25:54.000000000 -0700
+++ b/src/backend/parser/gram.y	2016-06-25 23:18:24.000000000 -0700
@@ -500,7 +500,7 @@
 %type <str>		createdb_opt_name
 %type <node>	var_value zone_value
 %type <node>	auth_ident RoleSpec opt_granted_by
-%type <node>  Iconst_or_Normalized Sconst_or_Normalized NonReservedWord_or_Sconst_or_Normalized
+%type <node>  Iconst_or_Normalized Sconst_or_Normalized Flat_Sconst_or_Normalized NonReservedWord_or_Sconst_or_Normalized
 
 %type <keyword> unreserved_keyword type_func_name_keyword
 %type <keyword> col_name_keyword reserved_keyword
@@ -923,24 +923,21 @@
 		;
 
 AlterOptRoleElem:
-			PASSWORD Sconst
+			PASSWORD Flat_Sconst_or_Normalized
 				{
-					$$ = makeDefElem("password",
-									 (Node *)makeString($2));
+					$$ = makeDefElemExtended(NULL, "password", $2, DEFELEM_UNSPEC, @2);
 				}
 			| PASSWORD NULL_P
 				{
 					$$ = makeDefElem("password", NULL);
 				}
-			| ENCRYPTED PASSWORD Sconst
+			| ENCRYPTED PASSWORD Flat_Sconst_or_Normalized
 				{
-					$$ = makeDefElem("encryptedPassword",
-									 (Node *)makeString($3));
+					$$ = makeDefElemExtended(NULL, "encryptedPassword", $3, DEFELEM_UNSPEC, @3);
 				}
-			| UNENCRYPTED PASSWORD Sconst
+			| UNENCRYPTED PASSWORD Flat_Sconst_or_Normalized
 				{
-					$$ = makeDefElem("unencryptedPassword",
-									 (Node *)makeString($3));
+					$$ = makeDefElemExtended(NULL, "unencryptedPassword", $3, DEFELEM_UNSPEC, @3);
 				}
 			| INHERIT
 				{
@@ -2447,11 +2444,11 @@
 			| ColLabel '.' ColLabel '=' def_arg
 				{
 					$$ = makeDefElemExtended($1, $3, (Node *) $5,
-											 DEFELEM_UNSPEC);
+											 DEFELEM_UNSPEC, @1);
 				}
 			| ColLabel '.' ColLabel
 				{
-					$$ = makeDefElemExtended($1, $3, NULL, DEFELEM_UNSPEC);
+					$$ = makeDefElemExtended($1, $3, NULL, DEFELEM_UNSPEC, @1);
 				}
 		;
 
@@ -4286,7 +4283,7 @@
 				}
 			| DROP generic_option_name
 				{
-					$$ = makeDefElemExtended(NULL, $2, NULL, DEFELEM_DROP);
+					$$ = makeDefElemExtended(NULL, $2, NULL, DEFELEM_DROP, @2);
 				}
 		;
 
@@ -13590,6 +13587,10 @@
 			| '?'																												{ $$ = makeParamRef( 0, @1); }
 		;
 
+Flat_Sconst_or_Normalized: Sconst  																	 { $$ = (Node *) makeString($1); }
+			| '?'																												{ $$ = makeParamRef( 0, @1); }
+		;
+
 NonReservedWord_or_Sconst_or_Normalized: NonReservedWord_or_Sconst	{ $$ = makeStringConst($1, @1); }
 			| '?'																												{ $$ = makeParamRef( 0, @1); }
 		;
diff -Naur a/src/include/nodes/makefuncs.h b/src/include/nodes/makefuncs.h
--- a/src/include/nodes/makefuncs.h	2016-05-09 13:50:23.000000000 -0700
+++ b/src/include/nodes/makefuncs.h	2016-06-25 22:42:45.000000000 -0700
@@ -79,7 +79,7 @@
 
 extern DefElem *makeDefElem(char *name, Node *arg);
 extern DefElem *makeDefElemExtended(char *nameSpace, char *name, Node *arg,
-					DefElemAction defaction);
+					DefElemAction defaction, int location);
 
 extern GroupingSet *makeGroupingSet(GroupingSetKind kind, List *content, int location);
 
diff -Naur a/src/include/nodes/parsenodes.h b/src/include/nodes/parsenodes.h
--- a/src/include/nodes/parsenodes.h	2016-05-09 13:50:23.000000000 -0700
+++ b/src/include/nodes/parsenodes.h	2016-06-25 22:28:35.000000000 -0700
@@ -660,6 +660,7 @@
 	char	   *defname;
 	Node	   *arg;			/* a (Value *) or a (TypeName *) */
 	DefElemAction defaction;	/* unspecified action, or SET/ADD/DROP */
+	int			location;		/* parse location, or -1 if none/unknown */
 } DefElem;
 
 /*
