diff --git a/src/excel/include/xlsx/parsers/worksheet_parser.hpp b/src/excel/include/xlsx/parsers/worksheet_parser.hpp
index 0b13361..eb631d2 100644
--- a/src/excel/include/xlsx/parsers/worksheet_parser.hpp
+++ b/src/excel/include/xlsx/parsers/worksheet_parser.hpp
@@ -509,7 +509,7 @@ inline void SheetParser::OnCell(const XLSXCellPos &pos, XLSXCellType type, vecto
 	auto &vec = chunk.data[pos.col - range.beg.col];
 
 	// Push the cell data to our chunk
-	const auto ptr = FlatVector::GetData<string_t>(vec);
+	auto ptr = FlatVector::GetDataMutable<string_t>(vec);
 
 	if (type == XLSXCellType::SHARED_STRING) {
 		// Push a null to the buffer so that the string is null-terminated
diff --git a/src/excel/xlsx/copy_xlsx.cpp b/src/excel/xlsx/copy_xlsx.cpp
index 63b052c..de7bccf 100644
--- a/src/excel/xlsx/copy_xlsx.cpp
+++ b/src/excel/xlsx/copy_xlsx.cpp
@@ -7,6 +7,7 @@
 #include "duckdb/planner/expression/bound_reference_expression.hpp"
 #include "xlsx/read_xlsx.hpp"
 #include "xlsx/xlsx_writer.hpp"
+#include "duckdb/execution/expression_executor.hpp"
 
 namespace duckdb {
 
diff --git a/src/excel/xlsx/read_xlsx.cpp b/src/excel/xlsx/read_xlsx.cpp
index 685f23c..342c5b8 100644
--- a/src/excel/xlsx/read_xlsx.cpp
+++ b/src/excel/xlsx/read_xlsx.cpp
@@ -444,7 +444,7 @@ static unique_ptr<FunctionData> Bind(ClientContext &context, TableFunctionBindIn
 
 	// Glob here so that we auto load any required extension filesystems
 	auto &fs = FileSystem::GetFileSystem(context);
-	auto files = fs.GlobFiles(file_path, context, FileGlobOptions::ALLOW_EMPTY);
+	auto files = fs.GlobFiles(file_path, FileGlobOptions::ALLOW_EMPTY);
 	if (files.empty()) {
 		// Use our own error message to be less confusing
 		throw IOException("Cannot open file \"%s\": No such file or directory", file_path);
diff --git a/src/excel/xlsx/read_xlsx.cpp b/src/excel/xlsx/read_xlsx.cpp
index 342c5b8..a634e65 100644
--- a/src/excel/xlsx/read_xlsx.cpp
+++ b/src/excel/xlsx/read_xlsx.cpp
@@ -708,7 +708,6 @@ static void Execute(ClientContext &context, TableFunctionInput &data, DataChunk
 			TryCastFromString(gstate, options.ignore_errors, col_idx, context, target_col);
 		}
 	}
-	output.SetCapacity(row_count);
 	output.SetCardinality(row_count);
 
 	output.Verify();
