diff --git a/src/excel/xlsx/read_xlsx.cpp b/src/excel/xlsx/read_xlsx.cpp
index a634e65..4273aed 100644
--- a/src/excel/xlsx/read_xlsx.cpp
+++ b/src/excel/xlsx/read_xlsx.cpp
@@ -691,6 +691,7 @@ static void Execute(ClientContext &context, TableFunctionInput &data, DataChunk
 		if (source_type == target_type) {
 			// If the types are the same, reference the column
 			target_col.Reference(source_col);
+			FlatVector::SetSize(target_col, count_t(row_count));
 			continue;
 		}
 
@@ -707,6 +708,7 @@ static void Execute(ClientContext &context, TableFunctionInput &data, DataChunk
 			// Cast the from string to the target type
 			TryCastFromString(gstate, options.ignore_errors, col_idx, context, target_col);
 		}
+		FlatVector::SetSize(target_col, count_t(row_count));
 	}
 	output.SetCardinality(row_count);
 
