diff --git a/src/excel/xlsx/copy_xlsx.cpp b/src/excel/xlsx/copy_xlsx.cpp
index de7bccf..b666efa 100644
--- a/src/excel/xlsx/copy_xlsx.cpp
+++ b/src/excel/xlsx/copy_xlsx.cpp
@@ -48,32 +48,29 @@ static void DateToExcelNumberFunction(DataChunk &args, ExpressionState &state, V
 	});
 }
 
-static unique_ptr<Expression> TimestampConversionExpr(unique_ptr<Expression> ref) {
+static unique_ptr<Expression> TimestampConversionExpr(ClientContext &context, unique_ptr<Expression> ref) {
 	vector<unique_ptr<Expression>> children;
 	children.push_back(std::move(ref));
 
 	ScalarFunction sfunc("timestamp_to_excel_number", {LogicalType::TIMESTAMP}, LogicalType::DOUBLE,
 	                     TimestampToExcelNumberFunction);
-	auto func = make_uniq<BoundFunctionExpression>(LogicalType::DOUBLE, sfunc, std::move(children), nullptr);
-	return std::move(func);
+	return sfunc.Bind(context, std::move(children));
 }
 
-static unique_ptr<Expression> TimeConversionExpr(unique_ptr<Expression> ref) {
+static unique_ptr<Expression> TimeConversionExpr(ClientContext &context, unique_ptr<Expression> ref) {
 	vector<unique_ptr<Expression>> children;
 	children.push_back(std::move(ref));
 
 	ScalarFunction sfunc("time_to_excel_number", {LogicalType::TIME}, LogicalType::DOUBLE, TimeToExcelNumberFunction);
-	auto func = make_uniq<BoundFunctionExpression>(LogicalType::DOUBLE, sfunc, std::move(children), nullptr);
-	return std::move(func);
+	return sfunc.Bind(context, std::move(children));
 }
 
-static unique_ptr<Expression> DateConversionExpr(unique_ptr<Expression> ref) {
+static unique_ptr<Expression> DateConversionExpr(ClientContext &context, unique_ptr<Expression> ref) {
 	vector<unique_ptr<Expression>> children;
 	children.push_back(std::move(ref));
 
 	ScalarFunction sfunc("date_to_excel_number", {LogicalType::DATE}, LogicalType::DOUBLE, DateToExcelNumberFunction);
-	auto func = make_uniq<BoundFunctionExpression>(LogicalType::DOUBLE, sfunc, std::move(children), nullptr);
-	return std::move(func);
+	return sfunc.Bind(context, std::move(children));
 }
 
 //------------------------------------------------------------------------------
@@ -186,15 +183,15 @@ struct GlobalWriteXLSXData final : public GlobalFunctionData {
 			case LogicalTypeId::TIMESTAMP_NS:
 				expr = BoundCastExpression::AddCastToType(context, std::move(expr), LogicalType::TIMESTAMP);
 			case LogicalTypeId::TIMESTAMP: // Fall through
-				expr = TimestampConversionExpr(std::move(expr));
+				expr = TimestampConversionExpr(context, std::move(expr));
 				break;
 			case LogicalTypeId::TIME_TZ:
 				expr = BoundCastExpression::AddCastToType(context, std::move(expr), LogicalType::TIME);
 			case LogicalTypeId::TIME: // Fall through
-				expr = TimeConversionExpr(std::move(expr));
+				expr = TimeConversionExpr(context, std::move(expr));
 				break;
 			case LogicalTypeId::DATE:
-				expr = DateConversionExpr(std::move(expr));
+				expr = DateConversionExpr(context, std::move(expr));
 				break;
 			case LogicalTypeId::BOOLEAN:
 				// Convert booleans to numbers first, then number to varchar
