diff --git a/src/excel/xlsx/copy_xlsx.cpp b/src/excel/xlsx/copy_xlsx.cpp
index de7bccf..aa05eeb 100644
--- a/src/excel/xlsx/copy_xlsx.cpp
+++ b/src/excel/xlsx/copy_xlsx.cpp
@@ -147,7 +147,7 @@ static void ParseCopyToOptions(const unique_ptr<WriteXLSXData> &data,
 	}
 }

-static unique_ptr<FunctionData> Bind(ClientContext &context, CopyFunctionBindInput &input, const vector<string> &names,
+static unique_ptr<FunctionData> Bind(ClientContext &context, CopyFunctionBindInput &input, const vector<Identifier> &names,
                                      const vector<LogicalType> &sql_types) {
 	auto data = make_uniq<WriteXLSXData>();

@@ -155,7 +155,7 @@ static unique_ptr<FunctionData> Bind(ClientContext &context, CopyFunctionBindInp
 	ParseCopyToOptions(data, input.info.options);

 	data->column_types = sql_types;
-	data->column_names = names;
+	data->column_names = IdentifiersToStrings(names);
 	data->file_path = input.info.file_path;

 	return std::move(data);
@@ -373,9 +373,9 @@ static void SetBooleanValue(named_parameter_map_t &params, const string &key, co
 		    value.type().id() == LogicalTypeId::DECIMAL) {
 			throw BinderException("'%s' options expects a boolean value (e.g. TRUE or 1)", key);
 		}
-		params[key] = value.DefaultCastAs(LogicalType::BOOLEAN);
+		params[Identifier(key)] = value.DefaultCastAs(LogicalType::BOOLEAN);
 	} else {
-		params[key] = Value::BOOLEAN(true);
+		params[Identifier(key)] = Value::BOOLEAN(true);
 	}
 }

@@ -390,7 +390,7 @@ static void SetVarcharValue(named_parameter_map_t &params, const string &key, co
 	if (val.back().IsNull()) {
 		throw BinderException(error_msg, key);
 	}
-	params[key] = val.back();
+	params[Identifier(key)] = val.back();
 }

 static void ParseCopyFromOptions(XLSXReadData &data, const case_insensitive_map_t<vector<Value>> &options) {
