diff --git a/src/excel/xlsx/copy_xlsx.cpp b/src/excel/xlsx/copy_xlsx.cpp
index aa05eeb..1152e87 100644
--- a/src/excel/xlsx/copy_xlsx.cpp
+++ b/src/excel/xlsx/copy_xlsx.cpp
@@ -33,7 +33,7 @@ static void TimeToExcelNumberFunction(DataChunk &args, ExpressionState &state, V
 	const auto count = args.size();
 	UnaryExecutor::Execute<dtime_t, double>(args.data[0], result, count, [&](dtime_t time) {
 		// 1.0 is a full day;
-		return static_cast<double>(time.micros) / Interval::MICROS_PER_DAY;
+		return static_cast<double>(time.value) / Interval::MICROS_PER_DAY;
 	});
 }
 
@@ -486,4 +486,4 @@ void WriteXLSX::Register(ExtensionLoader &loader) {
 	loader.RegisterFunction(info);
 }
 
-} // namespace duckdb
\ No newline at end of file
+} // namespace duckdb
