--- a/src/excel/xlsx/read_xlsx.cpp
+++ b/src/excel/xlsx/read_xlsx.cpp
@@ -437,7 +437,7 @@ void ReadXLSX::ResolveSheet(const unique_ptr<XLSXReadData> &result, ZipFileReade
 // Bind
 //-------------------------------------------------------------------
 static unique_ptr<FunctionData> Bind(ClientContext &context, TableFunctionBindInput &input,
-                                     vector<LogicalType> &return_types, vector<string> &names) {
+                                     vector<LogicalType> &return_types, vector<Identifier> &names) {
 	auto result = make_uniq<XLSXReadData>();
 	// Get the file name
 	const auto file_path = StringValue::Get(input.inputs[0]);
@@ -463,10 +463,11 @@ static unique_ptr<FunctionData> Bind(ClientContext &context, TableFunctionBindIn
 	ReadXLSX::ResolveSheet(result, archive);
 
 	return_types = result->return_types;
-	names = result->column_names;
 
 	// Clean and normalize names if requested
-	CleanColumnNames(names, result->options.normalize_names);
+	auto column_names = result->column_names;
+	CleanColumnNames(column_names, result->options.normalize_names);
+	names = StringsToIdentifiers(column_names);
 
 	// Deduplicate column names
 	QueryResult::DeduplicateColumns(names);
