diff --git a/src/excel/xlsx/copy_xlsx.cpp b/src/excel/xlsx/copy_xlsx.cpp
index 1152e87..324c9ba 100644
--- a/src/excel/xlsx/copy_xlsx.cpp
+++ b/src/excel/xlsx/copy_xlsx.cpp
@@ -87,7 +87,7 @@ struct WriteXLSXData final : TableFunctionData {
 };
 
 static void ParseCopyToOptions(const unique_ptr<WriteXLSXData> &data,
-                               const case_insensitive_map_t<vector<Value>> &options) {
+                               const identifier_map_t<vector<Value>> &options) {
 	// Find the header options
 	const auto header_opt = options.find("header");
 	if (header_opt != options.end()) {
@@ -356,7 +356,7 @@ CopyFunctionExecutionMode ExecutionMode(bool preserve_insertion_order, bool supp
 //------------------------------------------------------------------------------
 // Copy From
 //------------------------------------------------------------------------------
-static void SetBooleanValue(named_parameter_map_t &params, const string &key, const vector<Value> &val) {
+static void SetBooleanValue(named_parameter_map_t &params, const Identifier &key, const vector<Value> &val) {
 	static constexpr auto error_msg = "'%s' option must be standalone or a BOOLEAN value";
 	if (val.size() > 1) {
 		throw BinderException(error_msg, key);
@@ -370,13 +370,13 @@ static void SetBooleanValue(named_parameter_map_t &params, const string &key, co
 		    value.type().id() == LogicalTypeId::DECIMAL) {
 			throw BinderException("'%s' options expects a boolean value (e.g. TRUE or 1)", key);
 		}
-		params[Identifier(key)] = value.DefaultCastAs(LogicalType::BOOLEAN);
+		params[key] = value.DefaultCastAs(LogicalType::BOOLEAN);
 	} else {
-		params[Identifier(key)] = Value::BOOLEAN(true);
+		params[key] = Value::BOOLEAN(true);
 	}
 }
 
-static void SetVarcharValue(named_parameter_map_t &params, const string &key, const vector<Value> &val) {
+static void SetVarcharValue(named_parameter_map_t &params, const Identifier &key, const vector<Value> &val) {
 	static constexpr auto error_msg = "'%s' option must be a single VARCHAR value";
 	if (val.size() != 1) {
 		throw BinderException(error_msg, key);
@@ -387,16 +387,16 @@ static void SetVarcharValue(named_parameter_map_t &params, const string &key, co
 	if (val.back().IsNull()) {
 		throw BinderException(error_msg, key);
 	}
-	params[Identifier(key)] = val.back();
+	params[key] = val.back();
 }
 
-static void ParseCopyFromOptions(XLSXReadData &data, const case_insensitive_map_t<vector<Value>> &options) {
+static void ParseCopyFromOptions(XLSXReadData &data, const identifier_map_t<vector<Value>> &options) {
 
 	// Just make it really easy for us, extract everything into a named parameter map
 	named_parameter_map_t named_parameters;
 
 	for (auto &kv : options) {
-		auto key = StringUtil::Lower(kv.first);
+		auto &key = kv.first;
 		auto &val = kv.second;
 		if (key == "sheet") {
 			SetVarcharValue(named_parameters, key, val);
