diff --git a/src/excel/xlsx/copy_xlsx.cpp b/src/excel/xlsx/copy_xlsx.cpp
index 324c9ba..7d2415a 100644
--- a/src/excel/xlsx/copy_xlsx.cpp
+++ b/src/excel/xlsx/copy_xlsx.cpp
@@ -95,14 +95,14 @@ static void ParseCopyToOptions(const unique_ptr<WriteXLSXData> &data,
 			throw BinderException("Header option must be a single boolean value");
 		}
 		string error_msg;
-		Value bool_val;
-		if (!header_opt->second.back().DefaultTryCastAs(LogicalType::BOOLEAN, bool_val, &error_msg)) {
+		auto bool_val = header_opt->second.back().DefaultTryCastAs(LogicalType::BOOLEAN, &error_msg);
+		if (!bool_val) {
 			throw BinderException("Header option must be a single boolean value");
 		}
-		if (bool_val.IsNull()) {
+		if (bool_val->IsNull()) {
 			throw BinderException("Header option must be a single boolean value");
 		}
-		data->header = BooleanValue::Get(bool_val);
+		data->header = BooleanValue::Get(*bool_val);
 	} else {
 		data->header = false;
 	}
@@ -131,14 +131,14 @@ static void ParseCopyToOptions(const unique_ptr<WriteXLSXData> &data,
 			throw BinderException("Sheet row limit option must be a single integer value");
 		}
 		string error_msg;
-		Value int_val;
-		if (!sheet_row_limit_opt->second.back().DefaultTryCastAs(LogicalType::INTEGER, int_val, &error_msg)) {
+		auto int_val = sheet_row_limit_opt->second.back().DefaultTryCastAs(LogicalType::INTEGER, &error_msg);
+		if (!int_val) {
 			throw BinderException("Sheet row limit option must be a single integer value");
 		}
-		if (int_val.IsNull()) {
+		if (int_val->IsNull()) {
 			throw BinderException("Sheet row limit option must be a single integer value");
 		}
-		data->sheet_row_limit = IntegerValue::Get(int_val);
+		data->sheet_row_limit = IntegerValue::Get(*int_val);
 	} else {
 		data->sheet_row_limit = XLSX_MAX_CELL_ROWS;
 	}
