--- a/src/excel/xlsx/copy_xlsx.cpp
+++ b/src/excel/xlsx/copy_xlsx.cpp
@@ -419,7 +419,7 @@
 	ReadXLSX::ParseOptions(data.options, named_parameters);
 }
 
-static unique_ptr<FunctionData> CopyFromBind(ClientContext &context, CopyFromFunctionBindInput &input, vector<string> &expected_names,
+static unique_ptr<FunctionData> CopyFromBind(ClientContext &context, CopyFromFunctionBindInput &input, vector<Identifier> &expected_names,
                                              vector<LogicalType> &expected_types) {
 
 	auto result = make_uniq<XLSXReadData>();
@@ -438,14 +438,14 @@
 			if (col_idx > 0) {
 				extended_error += ", ";
 			}
-			extended_error += expected_names[col_idx] + " " + expected_types[col_idx].ToString();
+			extended_error += expected_names[col_idx].GetIdentifierName() + " " + expected_types[col_idx].ToString();
 		}
 		extended_error += "\nXLSX schema: ";
 		for (idx_t col_idx = 0; col_idx < result->return_types.size(); col_idx++) {
 			if (col_idx > 0) {
 				extended_error += ", ";
 			}
-			extended_error += result->column_names[col_idx] + " " + result->return_types[col_idx].ToString();
+			extended_error += result->column_names[col_idx].GetIdentifierName() + " " + result->return_types[col_idx].ToString();
 		}
 		extended_error += "\n\nPossible solutions:";
 		extended_error += "\n* Manually specify which columns to insert using \"INSERT INTO tbl SELECT ... "
--- a/src/excel/include/xlsx/read_xlsx.hpp
+++ b/src/excel/include/xlsx/read_xlsx.hpp
@@ -34,7 +34,7 @@
 
 	vector<LogicalType> return_types;
 	vector<XLSXCellType> source_types;
-	vector<string> column_names;
+	vector<Identifier> column_names;
 
 	XLSXReadOptions options;
 	XLSXStyleSheet style_sheet;
--- a/src/excel/xlsx/read_xlsx.cpp
+++ b/src/excel/xlsx/read_xlsx.cpp
@@ -111,14 +111,11 @@
 	return col_name_cleaned;
 }
 
-static void CleanColumnNames(vector<string> &names, bool normalize) {
+static void CleanColumnNames(vector<Identifier> &names, bool normalize) {
 	for (auto &name : names) {
 		// normalize names or at least trim whitespace
-		if (normalize) {
-			name = NormalizeColumnName(name);
-		} else {
-			name = TrimWhitespace(name);
-		}
+		auto &raw_name = name.GetIdentifierName();
+		name = Identifier(normalize ? NormalizeColumnName(raw_name) : TrimWhitespace(raw_name));
 	}
 }
 
@@ -409,7 +406,7 @@
 
 	// Set the return names
 	for (auto &cell : header_cells) {
-		result->column_names.push_back(cell.data);
+		result->column_names.emplace_back(cell.data);
 	}
 
 	// Convert excel types to duckdb types
@@ -465,9 +462,8 @@
 	return_types = result->return_types;
 
 	// Clean and normalize names if requested
-	auto column_names = result->column_names;
-	CleanColumnNames(column_names, result->options.normalize_names);
-	names = StringsToIdentifiers(column_names);
+	names = result->column_names;
+	CleanColumnNames(names, result->options.normalize_names);
 
 	// Deduplicate column names
 	QueryResult::DeduplicateColumns(names);
