diff --git a/src/excel/xlsx/read_xlsx.cpp b/src/excel/xlsx/read_xlsx.cpp
index a893d29..21c2887 100644
--- a/src/excel/xlsx/read_xlsx.cpp
+++ b/src/excel/xlsx/read_xlsx.cpp
@@ -739,7 +739,7 @@ static unique_ptr<TableRef> XLSXReplacementScan(ClientContext &context, Replacem
 
 	auto result = make_uniq<TableFunctionRef>();
 	vector<unique_ptr<ParsedExpression>> children;
-	children.push_back(make_uniq<ConstantExpression>(Value(table_name)));
+	children.push_back(ConstantExpression::FromValue(Value(table_name)));
 	result->function = make_uniq_base<ParsedExpression, FunctionExpression>("read_xlsx", std::move(children));
 
 	return std::move(result);
