diff --git a/src/excel/include/xlsx/read_xlsx.hpp b/src/excel/include/xlsx/read_xlsx.hpp
index 1f2e461..95a356b 100644
--- a/src/excel/include/xlsx/read_xlsx.hpp
+++ b/src/excel/include/xlsx/read_xlsx.hpp
@@ -44,7 +44,7 @@ class ZipFileReader;
 
 struct ReadXLSX {
 	// options and file path need to be resolved already
-	static void ParseOptions(XLSXReadOptions &options, const named_parameter_map_t &input);
+	static void ParseOptions(XLSXReadOptions &options, const named_argument_map_t &input);
 	static void ResolveSheet(const unique_ptr<XLSXReadData> &result, ZipFileReader &archive);
 
 	static void Register(ExtensionLoader &loader);
diff --git a/src/excel/xlsx/copy_xlsx.cpp b/src/excel/xlsx/copy_xlsx.cpp
index 842e639..fd38551 100644
--- a/src/excel/xlsx/copy_xlsx.cpp
+++ b/src/excel/xlsx/copy_xlsx.cpp
@@ -356,7 +356,7 @@ CopyFunctionExecutionMode ExecutionMode(bool preserve_insertion_order, bool supp
 //------------------------------------------------------------------------------
 // Copy From
 //------------------------------------------------------------------------------
-static void SetBooleanValue(named_parameter_map_t &params, const Identifier &key, const vector<Value> &val) {
+static void SetBooleanValue(named_argument_map_t &params, const Identifier &key, const vector<Value> &val) {
 	static constexpr auto error_msg = "'%s' option must be standalone or a BOOLEAN value";
 	if (val.size() > 1) {
 		throw BinderException(error_msg, key);
@@ -376,7 +376,7 @@ static void SetBooleanValue(named_parameter_map_t &params, const Identifier &key
 	}
 }
 
-static void SetVarcharValue(named_parameter_map_t &params, const Identifier &key, const vector<Value> &val) {
+static void SetVarcharValue(named_argument_map_t &params, const Identifier &key, const vector<Value> &val) {
 	static constexpr auto error_msg = "'%s' option must be a single VARCHAR value";
 	if (val.size() != 1) {
 		throw BinderException(error_msg, key);
@@ -393,7 +393,7 @@ static void SetVarcharValue(named_parameter_map_t &params, const Identifier &key
 static void ParseCopyFromOptions(XLSXReadData &data, const identifier_map_t<vector<Value>> &options) {
 
 	// Just make it really easy for us, extract everything into a named parameter map
-	named_parameter_map_t named_parameters;
+	named_argument_map_t named_parameters;
 
 	for (auto &kv : options) {
 		auto &key = kv.first;
diff --git a/src/excel/xlsx/read_xlsx.cpp b/src/excel/xlsx/read_xlsx.cpp
index 21c2887..e2432b2 100644
--- a/src/excel/xlsx/read_xlsx.cpp
+++ b/src/excel/xlsx/read_xlsx.cpp
@@ -277,7 +277,7 @@ static void ResolveColumnNames(vector<XLSXCell> &header_cells, ZipFileReader &ar
 	}
 }
 
-void ReadXLSX::ParseOptions(XLSXReadOptions &options, const named_parameter_map_t &input) {
+void ReadXLSX::ParseOptions(XLSXReadOptions &options, const named_argument_map_t &input) {
 
 	// Check which sheet to use, default to the primary sheet
 	const auto sheet_opt = input.find("sheet");
@@ -750,20 +750,22 @@ static unique_ptr<TableRef> XLSXReplacementScan(ClientContext &context, Replacem
 //-------------------------------------------------------------------
 TableFunction ReadXLSX::GetFunction() {
 
-	TableFunction read_xlsx("read_xlsx", {LogicalType::VARCHAR}, Execute, Bind);
+	FunctionSignature signature;
+	signature.AddParameter("file_path", LogicalType::VARCHAR).WithTypedKwargs("options", [&](TypedKwargs &options) {
+		options.Add("header", LogicalType::BOOLEAN)
+		    .Add("all_varchar", LogicalType::BOOLEAN)
+		    .Add("ignore_errors", LogicalType::BOOLEAN)
+		    .Add("range", LogicalType::VARCHAR)
+		    .Add("sheet", LogicalType::VARCHAR)
+		    .Add("stop_at_empty", LogicalType::BOOLEAN)
+		    .Add("empty_as_varchar", LogicalType::BOOLEAN)
+		    .Add("normalize_names", LogicalType::BOOLEAN);
+	});
+
+	TableFunction read_xlsx("read_xlsx", std::move(signature), Execute, Bind);
 	read_xlsx.init_global = InitGlobal;
 	read_xlsx.table_scan_progress = Progress;
 
-	// Parameters
-	read_xlsx.named_parameters["header"] = LogicalType::BOOLEAN;
-	read_xlsx.named_parameters["all_varchar"] = LogicalType::BOOLEAN;
-	read_xlsx.named_parameters["ignore_errors"] = LogicalType::BOOLEAN;
-	read_xlsx.named_parameters["range"] = LogicalType::VARCHAR;
-	read_xlsx.named_parameters["sheet"] = LogicalType::VARCHAR;
-	read_xlsx.named_parameters["stop_at_empty"] = LogicalType::BOOLEAN;
-	read_xlsx.named_parameters["empty_as_varchar"] = LogicalType::BOOLEAN;
-	read_xlsx.named_parameters["normalize_names"] = LogicalType::BOOLEAN;
-
 	return read_xlsx;
 }
 
