diff --git a/src/include/utils/helpers.hpp b/src/include/utils/helpers.hpp
index 54b01eb..f79874a 100644
--- a/src/include/utils/helpers.hpp
+++ b/src/include/utils/helpers.hpp
@@ -28,7 +28,7 @@ struct RunOnceTableFunctionState : GlobalTableFunctionState {
 namespace internal {
 
 unique_ptr<FunctionData> SingleBoolResultBind(ClientContext &, TableFunctionBindInput &, vector<LogicalType> &out_types,
-                                              vector<std::string> &out_names);
+                                              vector<Identifier> &out_names);
 
 bool ShouldRun(TableFunctionInput &input);
 
diff --git a/src/utils/helpers.cpp b/src/utils/helpers.cpp
index 7b1571d..7f16055 100644
--- a/src/utils/helpers.cpp
+++ b/src/utils/helpers.cpp
@@ -15,7 +15,7 @@ bool ShouldRun(TableFunctionInput &input) {
 }
 
 unique_ptr<FunctionData> SingleBoolResultBind(ClientContext &, TableFunctionBindInput &input,
-                                              vector<LogicalType> &out_types, vector<std::string> &out_names) {
+                                              vector<LogicalType> &out_types, vector<Identifier> &out_names) {
 	out_names.emplace_back("result");
 	out_types.emplace_back(LogicalType::BOOLEAN);
 	return make_uniq<ParamsTableFunctionData>(input.inputs);
