diff --git a/src/include/utils/misc.hpp b/src/include/utils/misc.hpp
index db8ecc9..fe4ff45 100644
--- a/src/include/utils/misc.hpp
+++ b/src/include/utils/misc.hpp
@@ -10,7 +10,7 @@ void UpdateTxStateFromTxType(const TransactionType type, TestDrivenTransactionSt
 void UpdateTxStateAfterStatement(Connection &con, TestDrivenTransactionState &state, bool rollback_if_none = false);
 
 void DetachAllDatabases(ClientContext &);
-void UseDBAndDetachOthers(Connection &, const std::string &, bool);
+void UseDBAndDetachOthers(Connection &, const Identifier &, bool);
 
 std::string UUIDToString(const hugeint_t &);
 
diff --git a/src/include/utils/serialization_helpers.hpp b/src/include/utils/serialization_helpers.hpp
index 7b2390a..404f2d3 100644
--- a/src/include/utils/serialization_helpers.hpp
+++ b/src/include/utils/serialization_helpers.hpp
@@ -56,7 +56,7 @@ public:
 	bool can_parse_query = true;
 	bool should_skip_query = false;
 	bool can_deserialize_plan = true;
-	string load_db_name;
+	Identifier load_db_name;
 	uint32_t nb_statements;
 	uint32_t query_idx;
 	hugeint_t uuid;
diff --git a/src/utils/misc.cpp b/src/utils/misc.cpp
index 87f610c..6f65b3f 100644
--- a/src/utils/misc.cpp
+++ b/src/utils/misc.cpp
@@ -11,7 +11,7 @@
 
 namespace duckdb {
 
-void UseDBAndDetachOthers(Connection &con, const std::string &db_name, bool need_tx) {
+void UseDBAndDetachOthers(Connection &con, const Identifier &db_name, bool need_tx) {
 	auto &db_manager = DatabaseManager::Get(*con.context);
 
 	if (need_tx) {
diff --git a/src/utils/serialization_helpers.cpp b/src/utils/serialization_helpers.cpp
index e2bd360..a7eae5f 100644
--- a/src/utils/serialization_helpers.cpp
+++ b/src/utils/serialization_helpers.cpp
@@ -147,7 +147,7 @@ SQLLogicQuery::SQLLogicQuery(const string &_query, const uint32_t _query_idx, co
 
 	it = _flags.find("load_db");
 	if (it != _flags.end()) {
-		load_db_name = it->second;
+		load_db_name = Identifier(it->second);
 	}
 
 	// skip_query: query should be skipped entirely
