diff --git a/src/serialize_queries_plans.cpp b/src/serialize_queries_plans.cpp
index 426d740..2980f5c 100644
--- a/src/serialize_queries_plans.cpp
+++ b/src/serialize_queries_plans.cpp
@@ -69,7 +69,7 @@ bool SerializeQueriesPlansFromFile(ClientContext &context, const vector<Value> &
 	TestDrivenTransactionState test_driven_transaction_state = TestDrivenTransactionState::NONE;
 	// Now run each query and serialize the plan to the output file.
 	for (auto &slq : queries) {
-		Parser parser;
+		Parser parser(*con.context);
 		try {
 			parser.ParseQuery(slq.query);
 			slq.nb_statements = parser.statements.size();
