diff --git a/src/execute_all_from_file.cpp b/src/execute_all_from_file.cpp
index f0c5748..67fb26a 100644
--- a/src/execute_all_from_file.cpp
+++ b/src/execute_all_from_file.cpp
@@ -106,7 +106,7 @@ bool ExecuteAllPlansFromFile(ClientContext &context, const vector<Value> &params
 					deserializer.End();
 
 					plan->ResolveOperatorTypes();
-					auto result = con.context->Query(make_uniq<LogicalPlanStatement>(std::move(plan)), false);
+					auto result = con.Query(make_uniq<LogicalPlanStatement>(std::move(plan)));
 					LOG_DEBUG("Executed statement #" << statement_idx << " of query #" << i << ": "
 					                                 << ErrorOrSuccess(*result));
 					SerializeResult(*slq, con, serializer, *result);
@@ -126,7 +126,7 @@ bool ExecuteAllPlansFromFile(ClientContext &context, const vector<Value> &params
 					LOG_DEBUG("Beginning new transaction for query #" << slq->query_idx);
 					con.BeginTransaction();
 				}
-				auto result = con.context->Query(slq->query, false);
+				auto result = con.Query(slq->query);
 				SerializeResult(*slq, con, serializer, *result);
 				LOG_DEBUG("Executed query #" << i << ": " << ErrorOrSuccess(*result));
 			} catch (const Exception &e) {
diff --git a/src/serialize_queries_plans.cpp b/src/serialize_queries_plans.cpp
index 2980f5c..93dac78 100644
--- a/src/serialize_queries_plans.cpp
+++ b/src/serialize_queries_plans.cpp
@@ -81,7 +81,7 @@ bool SerializeQueriesPlansFromFile(ClientContext &context, const vector<Value> &
 			slq.can_deserialize_plan = false;
 			slq.can_parse_query = false;
 
-			auto result = con.context->Query(slq.query, false);
+			auto result = con.Query(slq.query);
 			SerializeAndUpdateState(serializer, state, slq, std::move(result));
 			continue;
 		}
@@ -259,9 +259,9 @@ static void SerializeQueryStatements(Connection &con, BinarySerializer &serializ
 				// If we can deserialize the plan, we execute it
 				type = planner.plan->type;
 				planner.plan->ResolveOperatorTypes();
-				result = con.context->Query(make_uniq<LogicalPlanStatement>(std::move(planner.plan)), false);
+				result = con.Query(make_uniq<LogicalPlanStatement>(std::move(planner.plan)));
 			} else {
-				result = con.context->Query(query, false);
+				result = con.Query(query);
 			}
 
 			if (result->HasError() && slq.ExpectSuccess()) {
